%% Generated by Sphinx.
\def\sphinxdocclass{report}
\documentclass[letterpaper,10pt,english]{sphinxmanual}
\ifdefined\pdfpxdimen
   \let\sphinxpxdimen\pdfpxdimen\else\newdimen\sphinxpxdimen
\fi \sphinxpxdimen=.75bp\relax

\PassOptionsToPackage{warn}{textcomp}
\usepackage[utf8]{inputenc}
\ifdefined\DeclareUnicodeCharacter
% support both utf8 and utf8x syntaxes
\edef\sphinxdqmaybe{\ifdefined\DeclareUnicodeCharacterAsOptional\string"\fi}
  \DeclareUnicodeCharacter{\sphinxdqmaybe00A0}{\nobreakspace}
  \DeclareUnicodeCharacter{\sphinxdqmaybe2500}{\sphinxunichar{2500}}
  \DeclareUnicodeCharacter{\sphinxdqmaybe2502}{\sphinxunichar{2502}}
  \DeclareUnicodeCharacter{\sphinxdqmaybe2514}{\sphinxunichar{2514}}
  \DeclareUnicodeCharacter{\sphinxdqmaybe251C}{\sphinxunichar{251C}}
  \DeclareUnicodeCharacter{\sphinxdqmaybe2572}{\textbackslash}
\fi
\usepackage{cmap}
\usepackage[T1]{fontenc}
\usepackage{amsmath,amssymb,amstext}
\usepackage{babel}
\usepackage{times}
\usepackage[Bjarne]{fncychap}
\usepackage{sphinx}

\fvset{fontsize=\small}
\usepackage{geometry}

% Include hyperref last.
\usepackage{hyperref}
% Fix anchor placement for figures with captions.
\usepackage{hypcap}% it must be loaded after hyperref.
% Set up styles of URL: it should be placed after hyperref.
\urlstyle{same}
\addto\captionsenglish{\renewcommand{\contentsname}{Contents:}}

\addto\captionsenglish{\renewcommand{\figurename}{Fig.\@ }}
\makeatletter
\def\fnum@figure{\figurename\thefigure{}}
\makeatother
\addto\captionsenglish{\renewcommand{\tablename}{Table }}
\makeatletter
\def\fnum@table{\tablename\thetable{}}
\makeatother
\addto\captionsenglish{\renewcommand{\literalblockname}{Listing}}

\addto\captionsenglish{\renewcommand{\literalblockcontinuedname}{continued from previous page}}
\addto\captionsenglish{\renewcommand{\literalblockcontinuesname}{continues on next page}}
\addto\captionsenglish{\renewcommand{\sphinxnonalphabeticalgroupname}{Non-alphabetical}}
\addto\captionsenglish{\renewcommand{\sphinxsymbolsname}{Symbols}}
\addto\captionsenglish{\renewcommand{\sphinxnumbersname}{Numbers}}

\addto\extrasenglish{\def\pageautorefname{page}}

\setcounter{tocdepth}{1}



\title{PyMethylProcess Documentation}
\date{Apr 09, 2019}
\release{0.1}
\author{Joshua Levy}
\newcommand{\sphinxlogo}{\vbox{}}
\renewcommand{\releasename}{Release}
\makeindex
\begin{document}

\pagestyle{empty}
\sphinxmaketitle
\pagestyle{plain}
\sphinxtableofcontents
\pagestyle{normal}
\phantomsection\label{\detokenize{index::doc}}


\sphinxurl{https://github.com/Christensen-Lab-Dartmouth/PyMethylProcess}

To get started, download pymethylprocess using Docker (joshualevy44/pymethylprocess) or PIP (pymethylprocess) and run pymethyl-install\_r\_dependencies.

See README.md in Github repository for more install directions and for example scripts for running the pipeline (not all datasets may be available on GEO at this time).

There is both an API and CLI available for use. Examples for CLI usage can be found in ./example\_scripts.

\noindent{\hspace*{\fill}\sphinxincludegraphics[width=480.00000\sphinxpxdimen,height=360.00000\sphinxpxdimen]{{pymethylprocess_overview}.jpeg}\hspace*{\fill}}

\noindent{\hspace*{\fill}\sphinxincludegraphics[width=480.00000\sphinxpxdimen,height=360.00000\sphinxpxdimen]{{pipeline-download}.jpeg}\hspace*{\fill}}

\noindent{\hspace*{\fill}\sphinxincludegraphics[width=480.00000\sphinxpxdimen,height=360.00000\sphinxpxdimen]{{pipeline-format}.jpeg}\hspace*{\fill}}

\noindent{\hspace*{\fill}\sphinxincludegraphics[width=480.00000\sphinxpxdimen,height=360.00000\sphinxpxdimen]{{pipeline-preprocess}.jpeg}\hspace*{\fill}}

\noindent{\hspace*{\fill}\sphinxincludegraphics[width=480.00000\sphinxpxdimen,height=360.00000\sphinxpxdimen]{{pipeline-visualize}.jpeg}\hspace*{\fill}}

\noindent{\hspace*{\fill}\sphinxincludegraphics[width=480.00000\sphinxpxdimen,height=360.00000\sphinxpxdimen]{{pipeline-train-test-split}.jpeg}\hspace*{\fill}}
\phantomsection\label{\detokenize{index:module-pymethylprocess.PreProcessDataTypes}}\index{pymethylprocess.PreProcessDataTypes (module)@\spxentry{pymethylprocess.PreProcessDataTypes}\spxextra{module}}

\chapter{PreProcessDataTypes.py}
\label{\detokenize{index:preprocessdatatypes-py}}
Contains datatypes core to downloading IDATs, preprocessing IDATs and samplesheets.
\index{PreProcessIDAT (class in pymethylprocess.PreProcessDataTypes)@\spxentry{PreProcessIDAT}\spxextra{class in pymethylprocess.PreProcessDataTypes}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.PreProcessDataTypes.PreProcessIDAT}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pymethylprocess.PreProcessDataTypes.}}\sphinxbfcode{\sphinxupquote{PreProcessIDAT}}}{\emph{idat\_dir}, \emph{minfi=None}, \emph{enmix=None}, \emph{base=None}, \emph{meffil=None}}{}
Class that will preprocess IDATs using R pipelines.
\begin{description}
\item[{idat\_dir}] \leavevmode
Location of idats or samplesheet csv.

\item[{minfi}] \leavevmode
Rpy2 importr minfi library, default to None will load through rpy2

\item[{enmix}] \leavevmode
Rpy2 importr enmix library, default to None will load through rpy2

\item[{base}] \leavevmode
Rpy2 importr base library, default to None will load through rpy2

\item[{meffil}] \leavevmode
Rpy2 importr meffil library, default to None will load through rpy2

\end{description}
\index{export\_csv() (pymethylprocess.PreProcessDataTypes.PreProcessIDAT method)@\spxentry{export\_csv()}\spxextra{pymethylprocess.PreProcessDataTypes.PreProcessIDAT method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.PreProcessDataTypes.PreProcessIDAT.export_csv}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{export\_csv}}}{\emph{output\_dir}}{}
Export pheno and beta dataframes to CSVs
\begin{description}
\item[{output\_dir}] \leavevmode
Where to store csvs.

\end{description}

\end{fulllineitems}

\index{export\_pickle() (pymethylprocess.PreProcessDataTypes.PreProcessIDAT method)@\spxentry{export\_pickle()}\spxextra{pymethylprocess.PreProcessDataTypes.PreProcessIDAT method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.PreProcessDataTypes.PreProcessIDAT.export_pickle}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{export\_pickle}}}{\emph{output\_pickle}, \emph{disease=''}}{}
Export pheno and beta dataframes to pickle, stored in python dict that can be loaded into MethylationArray
\begin{description}
\item[{output\_pickle}] \leavevmode
Where to store MethylationArray.

\item[{disease}] \leavevmode
Custom naming scheme for data.

\end{description}

\end{fulllineitems}

\index{export\_sql() (pymethylprocess.PreProcessDataTypes.PreProcessIDAT method)@\spxentry{export\_sql()}\spxextra{pymethylprocess.PreProcessDataTypes.PreProcessIDAT method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.PreProcessDataTypes.PreProcessIDAT.export_sql}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{export\_sql}}}{\emph{output\_db}, \emph{disease=''}}{}
Export pheno and beta dataframes to SQL
\begin{description}
\item[{output\_db}] \leavevmode
Where to store data, sqlite db.

\item[{disease}] \leavevmode
Custom naming scheme for data.

\end{description}

\end{fulllineitems}

\index{extract\_manifest() (pymethylprocess.PreProcessDataTypes.PreProcessIDAT method)@\spxentry{extract\_manifest()}\spxextra{pymethylprocess.PreProcessDataTypes.PreProcessIDAT method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.PreProcessDataTypes.PreProcessIDAT.extract_manifest}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{extract\_manifest}}}{}{}
Get manifest from RGSet.

\end{fulllineitems}

\index{extract\_pheno\_data() (pymethylprocess.PreProcessDataTypes.PreProcessIDAT method)@\spxentry{extract\_pheno\_data()}\spxextra{pymethylprocess.PreProcessDataTypes.PreProcessIDAT method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.PreProcessDataTypes.PreProcessIDAT.extract_pheno_data}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{extract\_pheno\_data}}}{\emph{methylset=False}}{}
Extract pheno data from MSet or RGSet, minfi.
\begin{description}
\item[{methylset}] \leavevmode
If MSet has beenn created, set to True, else extract from original RGSet.

\end{description}

\end{fulllineitems}

\index{filter\_beta() (pymethylprocess.PreProcessDataTypes.PreProcessIDAT method)@\spxentry{filter\_beta()}\spxextra{pymethylprocess.PreProcessDataTypes.PreProcessIDAT method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.PreProcessDataTypes.PreProcessIDAT.filter_beta}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{filter\_beta}}}{}{}
After creating beta, filter out outliers.

\end{fulllineitems}

\index{get\_beta() (pymethylprocess.PreProcessDataTypes.PreProcessIDAT method)@\spxentry{get\_beta()}\spxextra{pymethylprocess.PreProcessDataTypes.PreProcessIDAT method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.PreProcessDataTypes.PreProcessIDAT.get_beta}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_beta}}}{}{}
Get beta value matrix from minfi after finding RSet.

\end{fulllineitems}

\index{get\_meth() (pymethylprocess.PreProcessDataTypes.PreProcessIDAT method)@\spxentry{get\_meth()}\spxextra{pymethylprocess.PreProcessDataTypes.PreProcessIDAT method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.PreProcessDataTypes.PreProcessIDAT.get_meth}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_meth}}}{}{}
Get methylation intensity matrix from MSet

\end{fulllineitems}

\index{get\_unmeth() (pymethylprocess.PreProcessDataTypes.PreProcessIDAT method)@\spxentry{get\_unmeth()}\spxextra{pymethylprocess.PreProcessDataTypes.PreProcessIDAT method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.PreProcessDataTypes.PreProcessIDAT.get_unmeth}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_unmeth}}}{}{}
Get unmethylated intensity matrix from MSet

\end{fulllineitems}

\index{load\_idats() (pymethylprocess.PreProcessDataTypes.PreProcessIDAT method)@\spxentry{load\_idats()}\spxextra{pymethylprocess.PreProcessDataTypes.PreProcessIDAT method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.PreProcessDataTypes.PreProcessIDAT.load_idats}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_idats}}}{}{}
For minfi pipeline, load IDATs from specified idat\_dir.

\end{fulllineitems}

\index{move\_jpg() (pymethylprocess.PreProcessDataTypes.PreProcessIDAT method)@\spxentry{move\_jpg()}\spxextra{pymethylprocess.PreProcessDataTypes.PreProcessIDAT method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.PreProcessDataTypes.PreProcessIDAT.move_jpg}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{move\_jpg}}}{}{}
Move jpeg files from current working directory to the idat directory.

\end{fulllineitems}

\index{output\_pheno\_beta() (pymethylprocess.PreProcessDataTypes.PreProcessIDAT method)@\spxentry{output\_pheno\_beta()}\spxextra{pymethylprocess.PreProcessDataTypes.PreProcessIDAT method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.PreProcessDataTypes.PreProcessIDAT.output_pheno_beta}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{output\_pheno\_beta}}}{\emph{meffil=False}}{}
Get pheno and beta dataframe objects stored as attributes for input to MethylationArray object.
\begin{description}
\item[{meffil}] \leavevmode
True if ran meffil pipeline.

\end{description}

\end{fulllineitems}

\index{plot\_original\_qc() (pymethylprocess.PreProcessDataTypes.PreProcessIDAT method)@\spxentry{plot\_original\_qc()}\spxextra{pymethylprocess.PreProcessDataTypes.PreProcessIDAT method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.PreProcessDataTypes.PreProcessIDAT.plot_original_qc}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{plot\_original\_qc}}}{\emph{output\_dir}}{}
Plot QC results from ENmix pipeline and possible minfi. Still experimental.
\begin{description}
\item[{output\_dir}] \leavevmode
Where to store plots.

\end{description}

\end{fulllineitems}

\index{plot\_qc\_metrics() (pymethylprocess.PreProcessDataTypes.PreProcessIDAT method)@\spxentry{plot\_qc\_metrics()}\spxextra{pymethylprocess.PreProcessDataTypes.PreProcessIDAT method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.PreProcessDataTypes.PreProcessIDAT.plot_qc_metrics}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{plot\_qc\_metrics}}}{\emph{output\_dir}}{}
Plot QC results from ENmix pipeline and possible minfi. Still experimental.
\begin{description}
\item[{output\_dir}] \leavevmode
Where to store plots.

\end{description}

\end{fulllineitems}

\index{preprocessENmix() (pymethylprocess.PreProcessDataTypes.PreProcessIDAT method)@\spxentry{preprocessENmix()}\spxextra{pymethylprocess.PreProcessDataTypes.PreProcessIDAT method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.PreProcessDataTypes.PreProcessIDAT.preprocessENmix}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{preprocessENmix}}}{\emph{n\_cores=6}}{}
Run ENmix preprocessing pipeline.
\begin{description}
\item[{n\_cores}] \leavevmode
Number of CPUs to use.

\end{description}

\end{fulllineitems}

\index{preprocessMeffil() (pymethylprocess.PreProcessDataTypes.PreProcessIDAT method)@\spxentry{preprocessMeffil()}\spxextra{pymethylprocess.PreProcessDataTypes.PreProcessIDAT method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.PreProcessDataTypes.PreProcessIDAT.preprocessMeffil}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{preprocessMeffil}}}{\emph{n\_cores=6}, \emph{n\_pcs=4}, \emph{qc\_report\_fname='qc/report.html'}, \emph{normalization\_report\_fname='norm/report.html'}, \emph{pc\_plot\_fname='qc/pc\_plot.pdf'}, \emph{useCache=True}, \emph{qc\_only=True}, \emph{qc\_parameters=\{'p.beadnum.cpgs': 0.1}, \emph{'p.beadnum.samples': 0.1}, \emph{'p.detection.cpgs': 0.1}, \emph{'p.detection.samples': 0.1\}}, \emph{rm\_sex=False}}{}
Run meffil preprocessing pipeline with functional normalization.
\begin{description}
\item[{n\_cores}] \leavevmode
Number of CPUs to use.

\item[{n\_pcs}] \leavevmode
Number of principal components to use for functional normalization, set to -1 to autoselect via kneedle algorithm.

\item[{qc\_report\_fname}] \leavevmode
HTML filename to store QC report.

\item[{normalization\_report\_fname}] \leavevmode
HTML filename to store normalization report

\item[{pc\_plot\_fname}] \leavevmode
PDF file to store principal components plot.

\item[{useCache}] \leavevmode
Use saved QC objects instead of running through QC again.

\item[{qc\_only}] \leavevmode
Perform QC, then save and quit before normalization.

\item[{qc\_parameters}] \leavevmode
Python dictionary with parameters for qc.

\item[{rm\_sex}] \leavevmode
Remove non-autosomal cpgs?

\end{description}

\end{fulllineitems}

\index{preprocessNoob() (pymethylprocess.PreProcessDataTypes.PreProcessIDAT method)@\spxentry{preprocessNoob()}\spxextra{pymethylprocess.PreProcessDataTypes.PreProcessIDAT method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.PreProcessDataTypes.PreProcessIDAT.preprocessNoob}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{preprocessNoob}}}{}{}
Run minfi preprocessing with Noob normalization

\end{fulllineitems}

\index{preprocessRAW() (pymethylprocess.PreProcessDataTypes.PreProcessIDAT method)@\spxentry{preprocessRAW()}\spxextra{pymethylprocess.PreProcessDataTypes.PreProcessIDAT method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.PreProcessDataTypes.PreProcessIDAT.preprocessRAW}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{preprocessRAW}}}{}{}
Run minfi preprocessing with RAW normalization

\end{fulllineitems}

\index{preprocess\_enmix\_pipeline() (pymethylprocess.PreProcessDataTypes.PreProcessIDAT method)@\spxentry{preprocess\_enmix\_pipeline()}\spxextra{pymethylprocess.PreProcessDataTypes.PreProcessIDAT method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.PreProcessDataTypes.PreProcessIDAT.preprocess_enmix_pipeline}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{preprocess\_enmix\_pipeline}}}{\emph{n\_cores=6}, \emph{pipeline='enmix'}, \emph{noob=False}, \emph{qc\_only=False}, \emph{use\_cache=False}}{}
Run complete ENmix or minfi preprocessing pipeline.
\begin{description}
\item[{n\_cores}] \leavevmode
Number CPUs.

\item[{pipeline}] \leavevmode
Run enmix or minfi

\item[{noob}] \leavevmode
Noob norm or RAW if minfi running.

\item[{qc\_only}] \leavevmode
Save and quit after only running QC?

\item[{use\_cache}] \leavevmode
Load preexisting RGSet instead of running QC again.

\end{description}

\end{fulllineitems}

\index{return\_beta() (pymethylprocess.PreProcessDataTypes.PreProcessIDAT method)@\spxentry{return\_beta()}\spxextra{pymethylprocess.PreProcessDataTypes.PreProcessIDAT method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.PreProcessDataTypes.PreProcessIDAT.return_beta}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{return\_beta}}}{}{}
Return minfi RSet after having created MSet.

\end{fulllineitems}

\index{to\_methyl\_array() (pymethylprocess.PreProcessDataTypes.PreProcessIDAT method)@\spxentry{to\_methyl\_array()}\spxextra{pymethylprocess.PreProcessDataTypes.PreProcessIDAT method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.PreProcessDataTypes.PreProcessIDAT.to_methyl_array}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{to\_methyl\_array}}}{\emph{disease=''}}{}
Convert results from preprocessing into MethylationArray, and directly return MethylationArray object.
\begin{description}
\item[{disease}] \leavevmode
Custom naming scheme for data.

\end{description}

\end{fulllineitems}


\end{fulllineitems}

\index{PreProcessPhenoData (class in pymethylprocess.PreProcessDataTypes)@\spxentry{PreProcessPhenoData}\spxextra{class in pymethylprocess.PreProcessDataTypes}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.PreProcessDataTypes.PreProcessPhenoData}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pymethylprocess.PreProcessDataTypes.}}\sphinxbfcode{\sphinxupquote{PreProcessPhenoData}}}{\emph{pheno\_sheet}, \emph{idat\_dir}, \emph{header\_line=0}}{}
Class that will manipute phenotype samplesheet before preprocessing of IDATs.
\begin{description}
\item[{pheno\_sheet}] \leavevmode
Location of clinical info csv.

\item[{idat\_dir}] \leavevmode
Location of idats

\item[{header\_line}] \leavevmode
Where to start reading clinical csv

\end{description}
\index{concat() (pymethylprocess.PreProcessDataTypes.PreProcessPhenoData method)@\spxentry{concat()}\spxextra{pymethylprocess.PreProcessDataTypes.PreProcessPhenoData method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.concat}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{concat}}}{\emph{other\_formatted\_sheet}}{}
Concat multiple PreProcessPhenoData objects, concat their dataframes to accept more than one smaplesheet/dataset.
\begin{description}
\item[{other\_formatted\_sheet}] \leavevmode
Other PreProcessPhenoData to concat.

\end{description}

\end{fulllineitems}

\index{export() (pymethylprocess.PreProcessDataTypes.PreProcessPhenoData method)@\spxentry{export()}\spxextra{pymethylprocess.PreProcessDataTypes.PreProcessPhenoData method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.export}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{export}}}{\emph{output\_sheet\_name}}{}
Export pheno data to csv after done with manipulation.
\begin{description}
\item[{output\_sheet\_name}] \leavevmode
Output csv name.

\end{description}

\end{fulllineitems}

\index{format\_custom() (pymethylprocess.PreProcessDataTypes.PreProcessPhenoData method)@\spxentry{format\_custom()}\spxextra{pymethylprocess.PreProcessDataTypes.PreProcessPhenoData method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.format_custom}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{format\_custom}}}{\emph{basename\_col}, \emph{disease\_class\_column}, \emph{include\_columns=\{\}}}{}
Custom format clinical sheet if user supplied idats.
\begin{description}
\item[{basename\_col}] \leavevmode
Column name of sample names.

\item[{disease\_class\_column}] \leavevmode
Disease column of clinical info csv.

\item[{include\_columns}] \leavevmode
Dictionary specifying other columns to include, and new names to assign them to.

\end{description}

\end{fulllineitems}

\index{format\_geo() (pymethylprocess.PreProcessDataTypes.PreProcessPhenoData method)@\spxentry{format\_geo()}\spxextra{pymethylprocess.PreProcessDataTypes.PreProcessPhenoData method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.format_geo}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{format\_geo}}}{\emph{disease\_class\_column='methylation class:ch1'}, \emph{include\_columns=\{\}}}{}
Format clinical sheets if downloaded geo idats.
\begin{description}
\item[{disease\_class\_column}] \leavevmode
Disease column of clinical info csv.

\item[{include\_columns}] \leavevmode
Dictionary specifying other columns to include, and new names to assign them to.

\end{description}

\end{fulllineitems}

\index{format\_tcga() (pymethylprocess.PreProcessDataTypes.PreProcessPhenoData method)@\spxentry{format\_tcga()}\spxextra{pymethylprocess.PreProcessDataTypes.PreProcessPhenoData method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.format_tcga}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{format\_tcga}}}{\emph{mapping\_file='idat\_filename\_case.txt'}}{}
Format clinical sheets if downloaded tcga idats.
\begin{description}
\item[{mapping\_file}] \leavevmode
Maps uuids to proper tcga sample names, should be downloaded with tcga clinical information.

\end{description}

\end{fulllineitems}

\index{get\_categorical\_distribution() (pymethylprocess.PreProcessDataTypes.PreProcessPhenoData method)@\spxentry{get\_categorical\_distribution()}\spxextra{pymethylprocess.PreProcessDataTypes.PreProcessPhenoData method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.get_categorical_distribution}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_categorical\_distribution}}}{\emph{key}, \emph{disease\_only=False}, \emph{subtype\_delimiter='}, \emph{'}}{}
Print categorical distribution, counts for each unique value in phenotype column.
\begin{description}
\item[{key}] \leavevmode
Phenotype Column.

\item[{disease\_only}] \leavevmode
Whether to split phenotype column entries by delimiter.

\item[{subtype\_delimiter}] \leavevmode
Subtype delimiter to split on.

\end{description}

\end{fulllineitems}

\index{merge() (pymethylprocess.PreProcessDataTypes.PreProcessPhenoData method)@\spxentry{merge()}\spxextra{pymethylprocess.PreProcessDataTypes.PreProcessPhenoData method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.merge}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{merge}}}{\emph{other\_formatted\_sheet}, \emph{use\_second\_sheet\_disease=True}, \emph{no\_disease\_merge=False}}{}
Merge multiple PreProcessPhenoData objects, merge their dataframes to accept more than one saplesheet/dataset or add more pheno info.
\begin{description}
\item[{other\_formatted\_sheet}] \leavevmode
Other PreProcessPhenoData to merge.

\item[{use\_second\_sheet\_disease}] \leavevmode
Change disease column to that of second sheet instead of first.

\item[{no\_disease\_merge}] \leavevmode
Keep both disease columns from both sheets.

\end{description}

\end{fulllineitems}

\index{remove\_diseases() (pymethylprocess.PreProcessDataTypes.PreProcessPhenoData method)@\spxentry{remove\_diseases()}\spxextra{pymethylprocess.PreProcessDataTypes.PreProcessPhenoData method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.remove_diseases}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{remove\_diseases}}}{\emph{exclude\_disease\_list}, \emph{low\_count}, \emph{disease\_only}, \emph{subtype\_delimiter}}{}
Remove samples with certain diseases from disease column.
\begin{description}
\item[{exclude\_disease\_list}] \leavevmode
List containing diseases to remove.

\item[{low\_count}] \leavevmode
Remove samples that have less than x disease occurances in column.

\item[{disease\_only}] \leavevmode
Whether to split phenotype column entries by delimiter.

\item[{subtype\_delimiter}] \leavevmode
Subtype delimiter to split on.

\end{description}

\end{fulllineitems}

\index{split\_key() (pymethylprocess.PreProcessDataTypes.PreProcessPhenoData method)@\spxentry{split\_key()}\spxextra{pymethylprocess.PreProcessDataTypes.PreProcessPhenoData method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.split_key}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{split\_key}}}{\emph{key}, \emph{subtype\_delimiter}}{}
Split pheno column by key, with subtype delimiter, eg. entry S1,s2 -\textgreater{} S1 with delimiter “,”.
\begin{description}
\item[{key}] \leavevmode
Pheno column name.

\item[{subtype\_delimiter}] \leavevmode
Subtype delimiter to split on.

\end{description}

\end{fulllineitems}


\end{fulllineitems}

\index{TCGADownloader (class in pymethylprocess.PreProcessDataTypes)@\spxentry{TCGADownloader}\spxextra{class in pymethylprocess.PreProcessDataTypes}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.PreProcessDataTypes.TCGADownloader}}\pysigline{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pymethylprocess.PreProcessDataTypes.}}\sphinxbfcode{\sphinxupquote{TCGADownloader}}}
Downloads TCGA and GEO IDAT and clinical data
\index{download\_clinical() (pymethylprocess.PreProcessDataTypes.TCGADownloader method)@\spxentry{download\_clinical()}\spxextra{pymethylprocess.PreProcessDataTypes.TCGADownloader method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.PreProcessDataTypes.TCGADownloader.download_clinical}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{download\_clinical}}}{\emph{output\_dir}}{}
Download TCGA Clinical Data.
\begin{description}
\item[{output\_dir}] \leavevmode
Where to output clinical data csv.

\end{description}

\end{fulllineitems}

\index{download\_geo() (pymethylprocess.PreProcessDataTypes.TCGADownloader method)@\spxentry{download\_geo()}\spxextra{pymethylprocess.PreProcessDataTypes.TCGADownloader method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.PreProcessDataTypes.TCGADownloader.download_geo}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{download\_geo}}}{\emph{query}, \emph{output\_dir}}{}
Download GEO IDATs.
\begin{description}
\item[{query}] \leavevmode
GEO accession number to query, must be 450k/850k.

\item[{output\_dir}] \leavevmode
Output directory to store idats and clinical information csv

\end{description}

\end{fulllineitems}

\index{download\_tcga() (pymethylprocess.PreProcessDataTypes.TCGADownloader method)@\spxentry{download\_tcga()}\spxextra{pymethylprocess.PreProcessDataTypes.TCGADownloader method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.PreProcessDataTypes.TCGADownloader.download_tcga}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{download\_tcga}}}{\emph{output\_dir}}{}
Download TCGA IDATs.
\begin{description}
\item[{output\_dir}] \leavevmode
Where to output idat files.

\end{description}

\end{fulllineitems}


\end{fulllineitems}

\phantomsection\label{\detokenize{index:module-pymethylprocess.MethylationDataTypes}}\index{pymethylprocess.MethylationDataTypes (module)@\spxentry{pymethylprocess.MethylationDataTypes}\spxextra{module}}

\chapter{MethylationDataTypes.py}
\label{\detokenize{index:methylationdatatypes-py}}
Contains datatypes core to storing beta and phenotype methylation data, and imputation.
\index{ImputerObject (class in pymethylprocess.MethylationDataTypes)@\spxentry{ImputerObject}\spxextra{class in pymethylprocess.MethylationDataTypes}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.MethylationDataTypes.ImputerObject}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pymethylprocess.MethylationDataTypes.}}\sphinxbfcode{\sphinxupquote{ImputerObject}}}{\emph{solver}, \emph{method}, \emph{opts=\{\}}}{}
Class that stores and accesses different types of imputers.
Construct sklearn-like imputer given certain input arguments.
\begin{description}
\item[{solver}] \leavevmode
Library for imputation, eg. sklearn, fancyimpute.

\item[{method}] \leavevmode
Imputation method in library, named.

\item[{opts}] \leavevmode
Additional options to assign to imputer.

\end{description}
\index{return\_imputer() (pymethylprocess.MethylationDataTypes.ImputerObject method)@\spxentry{return\_imputer()}\spxextra{pymethylprocess.MethylationDataTypes.ImputerObject method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.MethylationDataTypes.ImputerObject.return_imputer}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{return\_imputer}}}{}{}
Return initialized sklearn-like imputer.

\end{fulllineitems}


\end{fulllineitems}

\index{MethylationArray (class in pymethylprocess.MethylationDataTypes)@\spxentry{MethylationArray}\spxextra{class in pymethylprocess.MethylationDataTypes}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.MethylationDataTypes.MethylationArray}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pymethylprocess.MethylationDataTypes.}}\sphinxbfcode{\sphinxupquote{MethylationArray}}}{\emph{pheno\_df}, \emph{beta\_df}, \emph{name=''}}{}
Stores beta and phenotype information and performs various operations.
Initialize MethylationArray object by inputting dataframe of phenotypes
and dataframe of beta values with samples as index.
\begin{description}
\item[{pheno\_df}] \leavevmode
Phenotype dataframe (samples x covariates)

\item[{beta\_df}] \leavevmode
Beta Values Dataframe (samples x cpgs)

\end{description}
\index{bin\_column() (pymethylprocess.MethylationDataTypes.MethylationArray method)@\spxentry{bin\_column()}\spxextra{pymethylprocess.MethylationDataTypes.MethylationArray method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.MethylationDataTypes.MethylationArray.bin_column}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{bin\_column}}}{\emph{col}, \emph{n\_bins}}{}
Turn continuous variable/covariate into categorical bins.
Returns name of new column and updates phenotype matrix to reflect this change.
\begin{description}
\item[{col}] \leavevmode
Continuous column of phenotype array to bin.

\item[{n\_bins}] \leavevmode
Number of bins to create.

\end{description}

\end{fulllineitems}

\index{categorical\_breakdown() (pymethylprocess.MethylationDataTypes.MethylationArray method)@\spxentry{categorical\_breakdown()}\spxextra{pymethylprocess.MethylationDataTypes.MethylationArray method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.MethylationDataTypes.MethylationArray.categorical_breakdown}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{categorical\_breakdown}}}{\emph{key}}{}
Print categorical distribution, counts for each unique value in phenotype column.
\begin{description}
\item[{key}] \leavevmode
Phenotype Column.

\end{description}

\end{fulllineitems}

\index{feature\_select() (pymethylprocess.MethylationDataTypes.MethylationArray method)@\spxentry{feature\_select()}\spxextra{pymethylprocess.MethylationDataTypes.MethylationArray method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.MethylationDataTypes.MethylationArray.feature_select}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{feature\_select}}}{\emph{n\_top\_cpgs}, \emph{feature\_selection\_method='mad'}, \emph{metric='correlation'}, \emph{nn=10}}{}
Perform unsupervised feature selection on MethylationArray.
\begin{description}
\item[{n\_top\_cpgs}] \leavevmode
Number of CpGs to retain.

\item[{feature\_selection\_method}] \leavevmode
Method to perform selection.

\item[{metric}] \leavevmode
If considering structural feature selection like SPEC, use this distance metric.

\item[{nn}] \leavevmode
Number of nearest neighbors.

\end{description}

\end{fulllineitems}

\index{from\_pickle() (pymethylprocess.MethylationDataTypes.MethylationArray class method)@\spxentry{from\_pickle()}\spxextra{pymethylprocess.MethylationDataTypes.MethylationArray class method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.MethylationDataTypes.MethylationArray.from_pickle}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{classmethod }}\sphinxbfcode{\sphinxupquote{from\_pickle}}}{\emph{input\_pickle}}{}
Load MethylationArray stored in pickle.

Usage: MethylationArray.from\_pickle({[}input\_pickle{]})
\begin{description}
\item[{input\_pickle}] \leavevmode
Stored MethylationArray pickle.

\end{description}

\end{fulllineitems}

\index{groupby() (pymethylprocess.MethylationDataTypes.MethylationArray method)@\spxentry{groupby()}\spxextra{pymethylprocess.MethylationDataTypes.MethylationArray method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.MethylationDataTypes.MethylationArray.groupby}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{groupby}}}{\emph{key}}{}
Groupby for Methylation Array. Returns generator of methylation arrays grouped by key.
\begin{description}
\item[{preprocess\_sample\_df}] \leavevmode
New phenotype dataframe.

\end{description}

\end{fulllineitems}

\index{impute() (pymethylprocess.MethylationDataTypes.MethylationArray method)@\spxentry{impute()}\spxextra{pymethylprocess.MethylationDataTypes.MethylationArray method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.MethylationDataTypes.MethylationArray.impute}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{impute}}}{\emph{imputer}}{}
Perform imputation on NaN beta vaues. Input imputater returned from ImputerObject.
\begin{description}
\item[{imputer}] \leavevmode
Type of imputer object, in sklearn type interface.

\end{description}

\end{fulllineitems}

\index{merge\_preprocess\_sheet() (pymethylprocess.MethylationDataTypes.MethylationArray method)@\spxentry{merge\_preprocess\_sheet()}\spxextra{pymethylprocess.MethylationDataTypes.MethylationArray method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.MethylationDataTypes.MethylationArray.merge_preprocess_sheet}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{merge\_preprocess\_sheet}}}{\emph{preprocess\_sample\_df}}{}
Feed in another phenotype dataframe that will be merged with existing phenotype array.
\begin{description}
\item[{preprocess\_sample\_df}] \leavevmode
New phenotype dataframe.

\end{description}

\end{fulllineitems}

\index{overwrite\_pheno\_data() (pymethylprocess.MethylationDataTypes.MethylationArray method)@\spxentry{overwrite\_pheno\_data()}\spxextra{pymethylprocess.MethylationDataTypes.MethylationArray method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.MethylationDataTypes.MethylationArray.overwrite_pheno_data}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{overwrite\_pheno\_data}}}{\emph{preprocess\_sample\_df}}{}
Feed in another phenotype dataframe that will overwrite overlapping keys of existing phenotype array.
\begin{description}
\item[{preprocess\_sample\_df}] \leavevmode
New phenotype dataframe.

\end{description}

\end{fulllineitems}

\index{remove\_missingness() (pymethylprocess.MethylationDataTypes.MethylationArray method)@\spxentry{remove\_missingness()}\spxextra{pymethylprocess.MethylationDataTypes.MethylationArray method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.MethylationDataTypes.MethylationArray.remove_missingness}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{remove\_missingness}}}{\emph{cpg\_threshold=None}, \emph{sample\_threshold=None}}{}
Remove samples and CpGs with certain level of missingness..
\begin{description}
\item[{cpg\_threshold}] \leavevmode
If more than fraction of Samples for this CpG are missing, remove cpg.

\item[{sample\_threshold}] \leavevmode
If more than fraction of CpGs for this sample are missing, remove sample.

\end{description}

\end{fulllineitems}

\index{remove\_na\_samples() (pymethylprocess.MethylationDataTypes.MethylationArray method)@\spxentry{remove\_na\_samples()}\spxextra{pymethylprocess.MethylationDataTypes.MethylationArray method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.MethylationDataTypes.MethylationArray.remove_na_samples}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{remove\_na\_samples}}}{\emph{outcome\_cols}}{}
Remove samples of MethylationArray who have missing values in phenotype column.
\begin{description}
\item[{outcome\_cols}] \leavevmode
Phenotype columns, if any rows contain missing values, samples are removed.

\end{description}

\end{fulllineitems}

\index{remove\_whitespace() (pymethylprocess.MethylationDataTypes.MethylationArray method)@\spxentry{remove\_whitespace()}\spxextra{pymethylprocess.MethylationDataTypes.MethylationArray method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.MethylationDataTypes.MethylationArray.remove_whitespace}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{remove\_whitespace}}}{\emph{key}}{}
Remove whitespaces from phenotype column.
\begin{description}
\item[{key}] \leavevmode
Phenotype column.

\end{description}

\end{fulllineitems}

\index{return\_cpgs() (pymethylprocess.MethylationDataTypes.MethylationArray method)@\spxentry{return\_cpgs()}\spxextra{pymethylprocess.MethylationDataTypes.MethylationArray method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.MethylationDataTypes.MethylationArray.return_cpgs}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{return\_cpgs}}}{}{}
Return list of cpgs of MethylationArray

\end{fulllineitems}

\index{return\_idx() (pymethylprocess.MethylationDataTypes.MethylationArray method)@\spxentry{return\_idx()}\spxextra{pymethylprocess.MethylationDataTypes.MethylationArray method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.MethylationDataTypes.MethylationArray.return_idx}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{return\_idx}}}{}{}
Return sample names of MethylationArray.

\end{fulllineitems}

\index{return\_raw\_beta\_array() (pymethylprocess.MethylationDataTypes.MethylationArray method)@\spxentry{return\_raw\_beta\_array()}\spxextra{pymethylprocess.MethylationDataTypes.MethylationArray method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.MethylationDataTypes.MethylationArray.return_raw_beta_array}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{return\_raw\_beta\_array}}}{}{}
Return numpy array of methylation beta vaues.

\end{fulllineitems}

\index{return\_shape() (pymethylprocess.MethylationDataTypes.MethylationArray method)@\spxentry{return\_shape()}\spxextra{pymethylprocess.MethylationDataTypes.MethylationArray method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.MethylationDataTypes.MethylationArray.return_shape}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{return\_shape}}}{}{}
Return dimensionality and number of samples of beta matrix.

\end{fulllineitems}

\index{split\_by\_subtype() (pymethylprocess.MethylationDataTypes.MethylationArray method)@\spxentry{split\_by\_subtype()}\spxextra{pymethylprocess.MethylationDataTypes.MethylationArray method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.MethylationDataTypes.MethylationArray.split_by_subtype}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{split\_by\_subtype}}}{\emph{disease\_only}, \emph{subtype\_delimiter}}{}
Split MethylationArray into generator of MethylationArrays by phenotype column. Much akin to groupby. Only splits from disease column.
\begin{description}
\item[{disease\_only}] \leavevmode
Consider disease superclass.

\item[{subtype\_delimiter}] \leavevmode
How to break up disease column if using disease\_only.

\end{description}

\end{fulllineitems}

\index{split\_key() (pymethylprocess.MethylationDataTypes.MethylationArray method)@\spxentry{split\_key()}\spxextra{pymethylprocess.MethylationDataTypes.MethylationArray method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.MethylationDataTypes.MethylationArray.split_key}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{split\_key}}}{\emph{key}, \emph{subtype\_delimiter}}{}
Manipulate an entire phenotype column, splitting each element up by some delimiter.
\begin{description}
\item[{key}] \leavevmode
Phenotype column.

\item[{subtype\_delimiter}] \leavevmode
How to break up strings in columns. S1,s2 -\textgreater{} S1 for instance.

\end{description}

\end{fulllineitems}

\index{split\_train\_test() (pymethylprocess.MethylationDataTypes.MethylationArray method)@\spxentry{split\_train\_test()}\spxextra{pymethylprocess.MethylationDataTypes.MethylationArray method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.MethylationDataTypes.MethylationArray.split_train_test}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{split\_train\_test}}}{\emph{train\_p=0.8}, \emph{stratified=True}, \emph{disease\_only=False}, \emph{key='disease'}, \emph{subtype\_delimiter='}, \emph{'}, \emph{val\_p=0.0}}{}
Split MethylationArray into training and test sets, with option to stratify by categorical covariate.
\begin{description}
\item[{train\_p}] \leavevmode
Fraction of methylation array to use as training set.

\item[{stratified}] \leavevmode
Whether to stratify by categorical variable.

\item[{disease\_only}] \leavevmode
Consider disease superclass by some delimiter. For instance if disease is S1,s2, superclass would be S1.

\item[{key}] \leavevmode
Column to stratify on.

\item[{subtype\_delimiter}] \leavevmode
How to split disease column into super/subclass.

\item[{val\_p}] \leavevmode
If set greater than 0, will create additional validation set, fraction of which is broken off from training set.

\end{description}

\end{fulllineitems}

\index{subsample() (pymethylprocess.MethylationDataTypes.MethylationArray method)@\spxentry{subsample()}\spxextra{pymethylprocess.MethylationDataTypes.MethylationArray method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.MethylationDataTypes.MethylationArray.subsample}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{subsample}}}{\emph{key='disease'}, \emph{n\_samples=None}, \emph{frac=None}, \emph{categorical=False}}{}
Subsample MethylationArray, make the set randomly smaller.
\begin{description}
\item[{key}] \leavevmode
If stratifying, use this column of pheno array.

\item[{n\_samples}] \leavevmode
Number of samples to consider overall, or per stratum.

\item[{frac}] \leavevmode
Alternative to n\_samples, where x frac of array or stratum is considered.

\item[{categorical}] \leavevmode
Whether to stratify by column.

\end{description}

\end{fulllineitems}

\index{subset\_cpgs() (pymethylprocess.MethylationDataTypes.MethylationArray method)@\spxentry{subset\_cpgs()}\spxextra{pymethylprocess.MethylationDataTypes.MethylationArray method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.MethylationDataTypes.MethylationArray.subset_cpgs}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{subset\_cpgs}}}{\emph{cpgs}}{}
Subset beta matrix by list of Cpgs.
Parameters
———-
cpgs
\begin{quote}

Cpgs to subset by.
\end{quote}

\end{fulllineitems}

\index{subset\_index() (pymethylprocess.MethylationDataTypes.MethylationArray method)@\spxentry{subset\_index()}\spxextra{pymethylprocess.MethylationDataTypes.MethylationArray method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.MethylationDataTypes.MethylationArray.subset_index}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{subset\_index}}}{\emph{index}}{}
Subset MethylationArray by samples.
\begin{description}
\item[{index}] \leavevmode
Sample names to subset by.

\end{description}

\end{fulllineitems}

\index{write\_csvs() (pymethylprocess.MethylationDataTypes.MethylationArray method)@\spxentry{write\_csvs()}\spxextra{pymethylprocess.MethylationDataTypes.MethylationArray method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.MethylationDataTypes.MethylationArray.write_csvs}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{write\_csvs}}}{\emph{output\_dir}}{}
Write phenotype data and beta values to csvs.
\begin{description}
\item[{output\_dir}] \leavevmode
Directory to output csv files.

\end{description}

\end{fulllineitems}

\index{write\_db() (pymethylprocess.MethylationDataTypes.MethylationArray method)@\spxentry{write\_db()}\spxextra{pymethylprocess.MethylationDataTypes.MethylationArray method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.MethylationDataTypes.MethylationArray.write_db}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{write\_db}}}{\emph{conn}, \emph{disease=''}}{}
Store phenotype data and beta values in SQL database.
\begin{description}
\item[{conn}] \leavevmode
SQLite connection.

\item[{disease}] \leavevmode
Create new tables in db that are related to disease state by this name.

\end{description}

\end{fulllineitems}

\index{write\_pickle() (pymethylprocess.MethylationDataTypes.MethylationArray method)@\spxentry{write\_pickle()}\spxextra{pymethylprocess.MethylationDataTypes.MethylationArray method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.MethylationDataTypes.MethylationArray.write_pickle}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{write\_pickle}}}{\emph{output\_pickle}, \emph{disease=''}}{}
Store phenotype data and beta values in pickle file.
Is default file format for storing MethylationArray objects.
\begin{description}
\item[{output\_pickle}] \leavevmode
Pickle file to store MethylationArray data.

\end{description}

\end{fulllineitems}


\end{fulllineitems}

\index{MethylationArrays (class in pymethylprocess.MethylationDataTypes)@\spxentry{MethylationArrays}\spxextra{class in pymethylprocess.MethylationDataTypes}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.MethylationDataTypes.MethylationArrays}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pymethylprocess.MethylationDataTypes.}}\sphinxbfcode{\sphinxupquote{MethylationArrays}}}{\emph{list\_methylation\_arrays}}{}
Literally a list of methylation arrays, with methods operate on these arrays that is memory efficient.
Initialize with list of methylation arrays. Can optionally leave list empty or with one element.
\begin{description}
\item[{list\_methylation\_arrays}] \leavevmode
List of methylation arrays.

\end{description}
\index{combine() (pymethylprocess.MethylationDataTypes.MethylationArrays method)@\spxentry{combine()}\spxextra{pymethylprocess.MethylationDataTypes.MethylationArrays method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.MethylationDataTypes.MethylationArrays.combine}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{combine}}}{\emph{array\_generator=None}}{}
Combine the list of methylation arrays into one array via concatenation of beta matrices and phenotype arrays.
\begin{description}
\item[{array\_generator}] \leavevmode
Generator of additional methylation arrays for computational memory minimization.

\end{description}

\end{fulllineitems}

\index{impute() (pymethylprocess.MethylationDataTypes.MethylationArrays method)@\spxentry{impute()}\spxextra{pymethylprocess.MethylationDataTypes.MethylationArrays method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.MethylationDataTypes.MethylationArrays.impute}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{impute}}}{\emph{imputer}}{}
Impute all methylation arrays.
\begin{description}
\item[{imputer}] \leavevmode
Type of imputation, sklearn-like.

\end{description}

\end{fulllineitems}

\index{write\_dbs() (pymethylprocess.MethylationDataTypes.MethylationArrays method)@\spxentry{write\_dbs()}\spxextra{pymethylprocess.MethylationDataTypes.MethylationArrays method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.MethylationDataTypes.MethylationArrays.write_dbs}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{write\_dbs}}}{\emph{conn}}{}
Write list of methylation arrays to SQL database. Recommend naming MethylationArray.
\begin{description}
\item[{conn}] \leavevmode
SQL connection.

\end{description}

\end{fulllineitems}

\index{write\_pkls() (pymethylprocess.MethylationDataTypes.MethylationArrays method)@\spxentry{write\_pkls()}\spxextra{pymethylprocess.MethylationDataTypes.MethylationArrays method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.MethylationDataTypes.MethylationArrays.write_pkls}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{write\_pkls}}}{\emph{pkl}}{}
Write list of methylation arrays to single pickle. Recommend naming each MethylationArray.
\begin{description}
\item[{pkl}] \leavevmode
Pickle file to write to.

\end{description}

\end{fulllineitems}


\end{fulllineitems}

\index{extract\_pheno\_beta\_df\_from\_folder() (in module pymethylprocess.MethylationDataTypes)@\spxentry{extract\_pheno\_beta\_df\_from\_folder()}\spxextra{in module pymethylprocess.MethylationDataTypes}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.MethylationDataTypes.extract_pheno_beta_df_from_folder}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{pymethylprocess.MethylationDataTypes.}}\sphinxbfcode{\sphinxupquote{extract\_pheno\_beta\_df\_from\_folder}}}{\emph{folder}}{}
Return phenotype and beta dataframes from specified folder with csv.
\begin{description}
\item[{folder}] \leavevmode
Input folder.

\end{description}

\end{fulllineitems}

\index{extract\_pheno\_beta\_df\_from\_pickle\_dict() (in module pymethylprocess.MethylationDataTypes)@\spxentry{extract\_pheno\_beta\_df\_from\_pickle\_dict()}\spxextra{in module pymethylprocess.MethylationDataTypes}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.MethylationDataTypes.extract_pheno_beta_df_from_pickle_dict}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{pymethylprocess.MethylationDataTypes.}}\sphinxbfcode{\sphinxupquote{extract\_pheno\_beta\_df\_from\_pickle\_dict}}}{\emph{input\_dict}, \emph{disease=''}}{}
Return phenotype and beta dataframes from specified dictionary storing MethylationArray python dictionary.
\begin{description}
\item[{input\_dict}] \leavevmode
Python disctionary storing pheno/beta information.

\end{description}

\end{fulllineitems}

\index{extract\_pheno\_beta\_df\_from\_sql() (in module pymethylprocess.MethylationDataTypes)@\spxentry{extract\_pheno\_beta\_df\_from\_sql()}\spxextra{in module pymethylprocess.MethylationDataTypes}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.MethylationDataTypes.extract_pheno_beta_df_from_sql}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{pymethylprocess.MethylationDataTypes.}}\sphinxbfcode{\sphinxupquote{extract\_pheno\_beta\_df\_from\_sql}}}{\emph{conn}, \emph{disease=''}}{}
Return phenotype and beta dataframes from SQL tables storing MethylationArray info.
\begin{description}
\item[{conn}] \leavevmode
SQL connection.

\end{description}

\end{fulllineitems}

\phantomsection\label{\detokenize{index:module-pymethylprocess.meffil_functions}}\index{pymethylprocess.meffil\_functions (module)@\spxentry{pymethylprocess.meffil\_functions}\spxextra{module}}

\chapter{meffil\_functions.py}
\label{\detokenize{index:meffil-functions-py}}
Contains a few R functions that interact with meffil and minfi.
\index{est\_cell\_counts\_IDOL() (in module pymethylprocess.meffil\_functions)@\spxentry{est\_cell\_counts\_IDOL()}\spxextra{in module pymethylprocess.meffil\_functions}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.meffil_functions.est_cell_counts_IDOL}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{pymethylprocess.meffil\_functions.}}\sphinxbfcode{\sphinxupquote{est\_cell\_counts\_IDOL}}}{\emph{rgset}, \emph{library}}{}
Given RGSet object, estimate cell counts for 450k/850k using reference approach via IDOL library.
\begin{description}
\item[{rgset}] \leavevmode
RGSet object stored in python via rpy2

\item[{library}] \leavevmode
What type of CpG library to use.

\end{description}

\end{fulllineitems}

\index{est\_cell\_counts\_meffil() (in module pymethylprocess.meffil\_functions)@\spxentry{est\_cell\_counts\_meffil()}\spxextra{in module pymethylprocess.meffil\_functions}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.meffil_functions.est_cell_counts_meffil}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{pymethylprocess.meffil\_functions.}}\sphinxbfcode{\sphinxupquote{est\_cell\_counts\_meffil}}}{\emph{qc\_list}, \emph{cell\_type\_reference}}{}
Given QCObject list R object, estimate cell counts using reference approach via meffil.
\begin{description}
\item[{qc\_list}] \leavevmode
R list containing qc objects.

\item[{cell\_type\_reference}] \leavevmode
Reference blood/tissue set.

\end{description}

\end{fulllineitems}

\index{est\_cell\_counts\_minfi() (in module pymethylprocess.meffil\_functions)@\spxentry{est\_cell\_counts\_minfi()}\spxextra{in module pymethylprocess.meffil\_functions}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.meffil_functions.est_cell_counts_minfi}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{pymethylprocess.meffil\_functions.}}\sphinxbfcode{\sphinxupquote{est\_cell\_counts\_minfi}}}{\emph{rgset}}{}
Given RGSet object, estimate cell counts using reference approach via minfi.
\begin{description}
\item[{rgset}] \leavevmode
RGSet object stored in python via rpy2

\end{description}

\end{fulllineitems}

\index{load\_detection\_p\_values\_beadnum() (in module pymethylprocess.meffil\_functions)@\spxentry{load\_detection\_p\_values\_beadnum()}\spxextra{in module pymethylprocess.meffil\_functions}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.meffil_functions.load_detection_p_values_beadnum}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{pymethylprocess.meffil\_functions.}}\sphinxbfcode{\sphinxupquote{load\_detection\_p\_values\_beadnum}}}{\emph{qc\_list}, \emph{n\_cores}}{}
Return list of detection p-value matrix and bead number matrix.
\begin{description}
\item[{qc\_list}] \leavevmode
R list containing qc objects.

\item[{n\_cores}] \leavevmode
Number of cores to use in computation.

\end{description}

\end{fulllineitems}

\index{r\_autosomal\_cpgs() (in module pymethylprocess.meffil\_functions)@\spxentry{r\_autosomal\_cpgs()}\spxextra{in module pymethylprocess.meffil\_functions}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.meffil_functions.r_autosomal_cpgs}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{pymethylprocess.meffil\_functions.}}\sphinxbfcode{\sphinxupquote{r\_autosomal\_cpgs}}}{\emph{array\_type='450k'}}{}
Return list of autosomal cpg probes per platform.
\begin{description}
\item[{array\_type}] \leavevmode
450k/850k array?

\end{description}

\end{fulllineitems}

\index{r\_snp\_cpgs() (in module pymethylprocess.meffil\_functions)@\spxentry{r\_snp\_cpgs()}\spxextra{in module pymethylprocess.meffil\_functions}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.meffil_functions.r_snp_cpgs}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{pymethylprocess.meffil\_functions.}}\sphinxbfcode{\sphinxupquote{r\_snp\_cpgs}}}{\emph{array\_type='450k'}}{}
Return list of SNP cpg probes per platform.
\begin{description}
\item[{array\_type}] \leavevmode
450k/850k array?

\end{description}

\end{fulllineitems}

\index{remove\_sex() (in module pymethylprocess.meffil\_functions)@\spxentry{remove\_sex()}\spxextra{in module pymethylprocess.meffil\_functions}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.meffil_functions.remove_sex}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{pymethylprocess.meffil\_functions.}}\sphinxbfcode{\sphinxupquote{remove\_sex}}}{\emph{beta}, \emph{array\_type='450k'}}{}
Remove non-autosomal cpgs from beta matrix.
\begin{description}
\item[{array\_type}] \leavevmode
450k/850k array?

\end{description}

\end{fulllineitems}

\index{set\_missing() (in module pymethylprocess.meffil\_functions)@\spxentry{set\_missing()}\spxextra{in module pymethylprocess.meffil\_functions}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.meffil_functions.set_missing}}\pysiglinewithargsret{\sphinxcode{\sphinxupquote{pymethylprocess.meffil\_functions.}}\sphinxbfcode{\sphinxupquote{set\_missing}}}{\emph{beta}, \emph{pval\_beadnum}, \emph{detection\_val=1e-06}}{}
Set missing beta values to NA, taking into account detection values and bead number thesholds.
\begin{description}
\item[{pval\_beadnum}] \leavevmode
Detection pvalues and number of beads per cpg/samples

\item[{detection\_val}] \leavevmode
If threshold to set site to missingness based on p-value detection.

\end{description}

\end{fulllineitems}

\phantomsection\label{\detokenize{index:module-pymethylprocess.general_machine_learning}}\index{pymethylprocess.general\_machine\_learning (module)@\spxentry{pymethylprocess.general\_machine\_learning}\spxextra{module}}

\chapter{general\_machine\_learning.py}
\label{\detokenize{index:general-machine-learning-py}}
Contains a machine learning class to perform scikit-learn like operations, along with held-out hyperparameter grid search.
\index{MachineLearning (class in pymethylprocess.general\_machine\_learning)@\spxentry{MachineLearning}\spxextra{class in pymethylprocess.general\_machine\_learning}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.general_machine_learning.MachineLearning}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class }}\sphinxcode{\sphinxupquote{pymethylprocess.general\_machine\_learning.}}\sphinxbfcode{\sphinxupquote{MachineLearning}}}{\emph{model}, \emph{options}, \emph{grid=\{\}}, \emph{labelencode=False}, \emph{n\_eval=0}}{}
Machine learning class to run sklearn-like pipeline on MethylationArray data.
Initialize object with scikit-learn model, and optionally supply a hyperparameter search grid.
\begin{description}
\item[{model}] \leavevmode
Scikit-learn-like model, classification, regression, dimensionality reduction, clustering etc.

\item[{options}] \leavevmode
Options to supply model in form of dictionary.

\item[{grid}] \leavevmode
Alternatively, supply search grid to search for bets hyperparameters.

\item[{labelencode}] \leavevmode
T/F encode string labels.

\item[{n\_eval}] \leavevmode
Number of evaluations for randomized grid search, if set to 0, perform exhaustive grid search

\end{description}
\index{assign\_results\_to\_pheno\_col() (pymethylprocess.general\_machine\_learning.MachineLearning method)@\spxentry{assign\_results\_to\_pheno\_col()}\spxextra{pymethylprocess.general\_machine\_learning.MachineLearning method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.general_machine_learning.MachineLearning.assign_results_to_pheno_col}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{assign\_results\_to\_pheno\_col}}}{\emph{methyl\_array}, \emph{new\_col}, \emph{output\_pkl}}{}
Assign results to new phenotype column.
\begin{description}
\item[{methyl\_array}] \leavevmode
MethylationArray.

\item[{new\_col}] \leavevmode
New column name.

\item[{output\_pkl}] \leavevmode
Output pickle to dump MethylationArray to.

\end{description}

\end{fulllineitems}

\index{fit() (pymethylprocess.general\_machine\_learning.MachineLearning method)@\spxentry{fit()}\spxextra{pymethylprocess.general\_machine\_learning.MachineLearning method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.general_machine_learning.MachineLearning.fit}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{fit}}}{\emph{train\_methyl\_array}, \emph{val\_methyl\_array=None}, \emph{outcome\_cols=None}}{}
Fit data to model.
\begin{description}
\item[{train\_methyl\_array}] \leavevmode
Training MethylationArray.

\item[{val\_methyl\_array}] \leavevmode
Validation MethylationArray. Can set to None.

\item[{outcome\_cols}] \leavevmode
Set to none if not needed, but phenotype column to train on, can be multiple.

\end{description}

\end{fulllineitems}

\index{fit\_predict() (pymethylprocess.general\_machine\_learning.MachineLearning method)@\spxentry{fit\_predict()}\spxextra{pymethylprocess.general\_machine\_learning.MachineLearning method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.general_machine_learning.MachineLearning.fit_predict}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{fit\_predict}}}{\emph{train\_methyl\_array}, \emph{outcome\_cols=None}}{}
Fit and predict training data.
\begin{description}
\item[{train\_methyl\_array}] \leavevmode
Training MethylationArray.

\item[{outcome\_cols}] \leavevmode
Set to none if not needed, but phenotype column to train on, can be multiple.

\end{description}

\end{fulllineitems}

\index{fit\_transform() (pymethylprocess.general\_machine\_learning.MachineLearning method)@\spxentry{fit\_transform()}\spxextra{pymethylprocess.general\_machine\_learning.MachineLearning method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.general_machine_learning.MachineLearning.fit_transform}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{fit\_transform}}}{\emph{train\_methyl\_array}, \emph{outcome\_cols=None}}{}
Fit and transform to training data.
\begin{description}
\item[{train\_methyl\_array}] \leavevmode
Training MethylationArray.

\item[{outcome\_cols}] \leavevmode
Set to none if not needed, but phenotype column to train on, can be multiple.

\end{description}

\end{fulllineitems}

\index{predict() (pymethylprocess.general\_machine\_learning.MachineLearning method)@\spxentry{predict()}\spxextra{pymethylprocess.general\_machine\_learning.MachineLearning method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.general_machine_learning.MachineLearning.predict}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{predict}}}{\emph{test\_methyl\_array}}{}
Make new predictions on test methylation array.
\begin{description}
\item[{test\_methyl\_array}] \leavevmode
Testing MethylationArray.

\end{description}

\end{fulllineitems}

\index{return\_outcome\_metric() (pymethylprocess.general\_machine\_learning.MachineLearning method)@\spxentry{return\_outcome\_metric()}\spxextra{pymethylprocess.general\_machine\_learning.MachineLearning method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.general_machine_learning.MachineLearning.return_outcome_metric}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{return\_outcome\_metric}}}{\emph{methyl\_array}, \emph{outcome\_cols}, \emph{metric}, \emph{run\_bootstrap=False}}{}
Supply metric to evaluate results.
\begin{description}
\item[{methyl\_array}] \leavevmode
MethylationArray to evaluate.

\item[{outcome\_cols}] \leavevmode
Outcome phenotype columns.

\item[{metric}] \leavevmode
Sklearn evaluation metric.

\item[{run\_bootstrap}] \leavevmode
Make 95\% CI from 1k bootstraps.

\end{description}

\end{fulllineitems}

\index{store\_results() (pymethylprocess.general\_machine\_learning.MachineLearning method)@\spxentry{store\_results()}\spxextra{pymethylprocess.general\_machine\_learning.MachineLearning method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.general_machine_learning.MachineLearning.store_results}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{store\_results}}}{\emph{output\_pkl}, \emph{results\_dict=\{\}}}{}
Store results in pickle file.
\begin{description}
\item[{output\_pkl}] \leavevmode
Output pickle to dump results to.

\item[{results\_dict}] \leavevmode
Supply own results dict to be dumped.

\end{description}

\end{fulllineitems}

\index{transform() (pymethylprocess.general\_machine\_learning.MachineLearning method)@\spxentry{transform()}\spxextra{pymethylprocess.general\_machine\_learning.MachineLearning method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.general_machine_learning.MachineLearning.transform}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{transform}}}{\emph{test\_methyl\_array}}{}
Transform test methylation array.
\begin{description}
\item[{test\_methyl\_array}] \leavevmode
Testing MethylationArray.

\end{description}

\end{fulllineitems}

\index{transform\_results\_to\_beta() (pymethylprocess.general\_machine\_learning.MachineLearning method)@\spxentry{transform\_results\_to\_beta()}\spxextra{pymethylprocess.general\_machine\_learning.MachineLearning method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:pymethylprocess.general_machine_learning.MachineLearning.transform_results_to_beta}}\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{transform\_results\_to\_beta}}}{\emph{methyl\_array}, \emph{output\_pkl}}{}
Transform beta matrix into reduced beta matrix and store.
\begin{description}
\item[{methyl\_array}] \leavevmode
MethylationArray.

\item[{output\_pkl}] \leavevmode
Output pickle to dump MethylationArray to.

\end{description}

\end{fulllineitems}


\end{fulllineitems}



\chapter{pymethyl-install}
\label{\detokenize{index:pymethyl-install}}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}install \PYG{o}{[}OPTIONS\PYG{o}{]} COMMAND \PYG{o}{[}ARGS\PYG{o}{]}...
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-install command line option@\spxentry{pymethyl-install command line option}!--version@\spxentry{--version}}\index{--version@\spxentry{--version}!pymethyl-install command line option@\spxentry{pymethyl-install command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-install-version}}\pysigline{\sphinxbfcode{\sphinxupquote{-{-}version}}\sphinxcode{\sphinxupquote{}}}
Show the version and exit.

\end{fulllineitems}



\section{change\_gcc\_path}
\label{\detokenize{index:pymethyl-install-change-gcc-path}}
Change GCC and G++ paths if don’t have version 7.2.0. {[}Experimental{]}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}install change\PYGZus{}gcc\PYGZus{}path \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}


\section{install\_bioconductor}
\label{\detokenize{index:pymethyl-install-install-bioconductor}}
Installs bioconductor.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}install install\PYGZus{}bioconductor \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}


\section{install\_custom}
\label{\detokenize{index:pymethyl-install-install-custom}}
Installs bioconductor packages.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}install install\PYGZus{}custom \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-install-install\_custom command line option@\spxentry{pymethyl-install-install\_custom command line option}!-p, --package \textless{}package\textgreater{}@\spxentry{-p, --package \textless{}package\textgreater{}}}\index{-p, --package \textless{}package\textgreater{}@\spxentry{-p, --package \textless{}package\textgreater{}}!pymethyl-install-install\_custom command line option@\spxentry{pymethyl-install-install\_custom command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-install-install-custom-p}}\pysigline{\sphinxbfcode{\sphinxupquote{-p}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}package}}\sphinxcode{\sphinxupquote{~\textless{}package\textgreater{}}}}
Custom packages.  {[}default: ENmix{]}

\end{fulllineitems}

\index{pymethyl-install-install\_custom command line option@\spxentry{pymethyl-install-install\_custom command line option}!-m, --manager@\spxentry{-m, --manager}}\index{-m, --manager@\spxentry{-m, --manager}!pymethyl-install-install\_custom command line option@\spxentry{pymethyl-install-install\_custom command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-install-install-custom-m}}\pysigline{\sphinxbfcode{\sphinxupquote{-m}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}manager}}\sphinxcode{\sphinxupquote{}}}
Use BiocManager (recommended).

\end{fulllineitems}



\section{install\_meffil}
\label{\detokenize{index:pymethyl-install-install-meffil}}
Installs meffil (update!).

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}install install\PYGZus{}meffil \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}


\section{install\_minfi\_others}
\label{\detokenize{index:pymethyl-install-install-minfi-others}}
Installs minfi and other dependencies.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}install install\PYGZus{}minfi\PYGZus{}others \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}


\section{install\_r\_packages}
\label{\detokenize{index:pymethyl-install-install-r-packages}}
Installs r packages.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}install install\PYGZus{}r\PYGZus{}packages \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-install-install\_r\_packages command line option@\spxentry{pymethyl-install-install\_r\_packages command line option}!-p, --package \textless{}package\textgreater{}@\spxentry{-p, --package \textless{}package\textgreater{}}}\index{-p, --package \textless{}package\textgreater{}@\spxentry{-p, --package \textless{}package\textgreater{}}!pymethyl-install-install\_r\_packages command line option@\spxentry{pymethyl-install-install\_r\_packages command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-install-install-r-packages-p}}\pysigline{\sphinxbfcode{\sphinxupquote{-p}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}package}}\sphinxcode{\sphinxupquote{~\textless{}package\textgreater{}}}}
Custom packages.  {[}default: {]}

\end{fulllineitems}



\section{install\_some\_deps}
\label{\detokenize{index:pymethyl-install-install-some-deps}}
Installs bioconductor, minfi, enmix, tcga biolinks, and meffil.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}install install\PYGZus{}some\PYGZus{}deps \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}


\section{install\_tcga\_biolinks}
\label{\detokenize{index:pymethyl-install-install-tcga-biolinks}}
Installs tcga biolinks.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}install install\PYGZus{}tcga\PYGZus{}biolinks \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}


\chapter{pymethyl-visualize}
\label{\detokenize{index:pymethyl-visualize}}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}visualize \PYG{o}{[}OPTIONS\PYG{o}{]} COMMAND \PYG{o}{[}ARGS\PYG{o}{]}...
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-visualize command line option@\spxentry{pymethyl-visualize command line option}!--version@\spxentry{--version}}\index{--version@\spxentry{--version}!pymethyl-visualize command line option@\spxentry{pymethyl-visualize command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-visualize-version}}\pysigline{\sphinxbfcode{\sphinxupquote{-{-}version}}\sphinxcode{\sphinxupquote{}}}
Show the version and exit.

\end{fulllineitems}



\section{plot\_cell\_type\_results}
\label{\detokenize{index:pymethyl-visualize-plot-cell-type-results}}
Plot csv containing cell type results into side by side boxplots.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}visualize plot\PYGZus{}cell\PYGZus{}type\PYGZus{}results \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-visualize-plot\_cell\_type\_results command line option@\spxentry{pymethyl-visualize-plot\_cell\_type\_results command line option}!-i, --input\_csv \textless{}input\_csv\textgreater{}@\spxentry{-i, --input\_csv \textless{}input\_csv\textgreater{}}}\index{-i, --input\_csv \textless{}input\_csv\textgreater{}@\spxentry{-i, --input\_csv \textless{}input\_csv\textgreater{}}!pymethyl-visualize-plot\_cell\_type\_results command line option@\spxentry{pymethyl-visualize-plot\_cell\_type\_results command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-visualize-plot-cell-type-results-i}}\pysigline{\sphinxbfcode{\sphinxupquote{-i}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}input\_csv}}\sphinxcode{\sphinxupquote{~\textless{}input\_csv\textgreater{}}}}
Input csv.  {[}default: cell\_type\_estimates.csv{]}

\end{fulllineitems}

\index{pymethyl-visualize-plot\_cell\_type\_results command line option@\spxentry{pymethyl-visualize-plot\_cell\_type\_results command line option}!-o, --outfilename \textless{}outfilename\textgreater{}@\spxentry{-o, --outfilename \textless{}outfilename\textgreater{}}}\index{-o, --outfilename \textless{}outfilename\textgreater{}@\spxentry{-o, --outfilename \textless{}outfilename\textgreater{}}!pymethyl-visualize-plot\_cell\_type\_results command line option@\spxentry{pymethyl-visualize-plot\_cell\_type\_results command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-visualize-plot-cell-type-results-o}}\pysigline{\sphinxbfcode{\sphinxupquote{-o}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}outfilename}}\sphinxcode{\sphinxupquote{~\textless{}outfilename\textgreater{}}}}
Output png.  {[}default: visualizations/cell\_type\_results.png{]}

\end{fulllineitems}

\index{pymethyl-visualize-plot\_cell\_type\_results command line option@\spxentry{pymethyl-visualize-plot\_cell\_type\_results command line option}!-cols, --plot\_cols \textless{}plot\_cols\textgreater{}@\spxentry{-cols, --plot\_cols \textless{}plot\_cols\textgreater{}}}\index{-cols, --plot\_cols \textless{}plot\_cols\textgreater{}@\spxentry{-cols, --plot\_cols \textless{}plot\_cols\textgreater{}}!pymethyl-visualize-plot\_cell\_type\_results command line option@\spxentry{pymethyl-visualize-plot\_cell\_type\_results command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-visualize-plot-cell-type-results-cols}}\pysigline{\sphinxbfcode{\sphinxupquote{-cols}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}plot\_cols}}\sphinxcode{\sphinxupquote{~\textless{}plot\_cols\textgreater{}}}}
Plot columns.  {[}default: Gran, CD4T, CD8T, Bcell, Mono, NK, gMDSC{]}

\end{fulllineitems}

\index{pymethyl-visualize-plot\_cell\_type\_results command line option@\spxentry{pymethyl-visualize-plot\_cell\_type\_results command line option}!-fs, --font\_scale \textless{}font\_scale\textgreater{}@\spxentry{-fs, --font\_scale \textless{}font\_scale\textgreater{}}}\index{-fs, --font\_scale \textless{}font\_scale\textgreater{}@\spxentry{-fs, --font\_scale \textless{}font\_scale\textgreater{}}!pymethyl-visualize-plot\_cell\_type\_results command line option@\spxentry{pymethyl-visualize-plot\_cell\_type\_results command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-visualize-plot-cell-type-results-fs}}\pysigline{\sphinxbfcode{\sphinxupquote{-fs}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}font\_scale}}\sphinxcode{\sphinxupquote{~\textless{}font\_scale\textgreater{}}}}
Font scaling  {[}default: 1.0{]}

\end{fulllineitems}



\section{plot\_heatmap}
\label{\detokenize{index:pymethyl-visualize-plot-heatmap}}
Plot heatmap from CSV file.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}visualize plot\PYGZus{}heatmap \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-visualize-plot\_heatmap command line option@\spxentry{pymethyl-visualize-plot\_heatmap command line option}!-i, --input\_csv \textless{}input\_csv\textgreater{}@\spxentry{-i, --input\_csv \textless{}input\_csv\textgreater{}}}\index{-i, --input\_csv \textless{}input\_csv\textgreater{}@\spxentry{-i, --input\_csv \textless{}input\_csv\textgreater{}}!pymethyl-visualize-plot\_heatmap command line option@\spxentry{pymethyl-visualize-plot\_heatmap command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-visualize-plot-heatmap-i}}\pysigline{\sphinxbfcode{\sphinxupquote{-i}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}input\_csv}}\sphinxcode{\sphinxupquote{~\textless{}input\_csv\textgreater{}}}}
Input csv.  {[}default: {]}

\end{fulllineitems}

\index{pymethyl-visualize-plot\_heatmap command line option@\spxentry{pymethyl-visualize-plot\_heatmap command line option}!-o, --outfilename \textless{}outfilename\textgreater{}@\spxentry{-o, --outfilename \textless{}outfilename\textgreater{}}}\index{-o, --outfilename \textless{}outfilename\textgreater{}@\spxentry{-o, --outfilename \textless{}outfilename\textgreater{}}!pymethyl-visualize-plot\_heatmap command line option@\spxentry{pymethyl-visualize-plot\_heatmap command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-visualize-plot-heatmap-o}}\pysigline{\sphinxbfcode{\sphinxupquote{-o}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}outfilename}}\sphinxcode{\sphinxupquote{~\textless{}outfilename\textgreater{}}}}
Output png.  {[}default: output.png{]}

\end{fulllineitems}

\index{pymethyl-visualize-plot\_heatmap command line option@\spxentry{pymethyl-visualize-plot\_heatmap command line option}!-idx, --index\_col \textless{}index\_col\textgreater{}@\spxentry{-idx, --index\_col \textless{}index\_col\textgreater{}}}\index{-idx, --index\_col \textless{}index\_col\textgreater{}@\spxentry{-idx, --index\_col \textless{}index\_col\textgreater{}}!pymethyl-visualize-plot\_heatmap command line option@\spxentry{pymethyl-visualize-plot\_heatmap command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-visualize-plot-heatmap-idx}}\pysigline{\sphinxbfcode{\sphinxupquote{-idx}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}index\_col}}\sphinxcode{\sphinxupquote{~\textless{}index\_col\textgreater{}}}}
Index load dataframe  {[}default: 0{]}

\end{fulllineitems}

\index{pymethyl-visualize-plot\_heatmap command line option@\spxentry{pymethyl-visualize-plot\_heatmap command line option}!-fs, --font\_scale \textless{}font\_scale\textgreater{}@\spxentry{-fs, --font\_scale \textless{}font\_scale\textgreater{}}}\index{-fs, --font\_scale \textless{}font\_scale\textgreater{}@\spxentry{-fs, --font\_scale \textless{}font\_scale\textgreater{}}!pymethyl-visualize-plot\_heatmap command line option@\spxentry{pymethyl-visualize-plot\_heatmap command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-visualize-plot-heatmap-fs}}\pysigline{\sphinxbfcode{\sphinxupquote{-fs}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}font\_scale}}\sphinxcode{\sphinxupquote{~\textless{}font\_scale\textgreater{}}}}
Font scaling  {[}default: 1.0{]}

\end{fulllineitems}

\index{pymethyl-visualize-plot\_heatmap command line option@\spxentry{pymethyl-visualize-plot\_heatmap command line option}!-min, --min\_val \textless{}min\_val\textgreater{}@\spxentry{-min, --min\_val \textless{}min\_val\textgreater{}}}\index{-min, --min\_val \textless{}min\_val\textgreater{}@\spxentry{-min, --min\_val \textless{}min\_val\textgreater{}}!pymethyl-visualize-plot\_heatmap command line option@\spxentry{pymethyl-visualize-plot\_heatmap command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-visualize-plot-heatmap-min}}\pysigline{\sphinxbfcode{\sphinxupquote{-min}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}min\_val}}\sphinxcode{\sphinxupquote{~\textless{}min\_val\textgreater{}}}}
Min heat val  {[}default: 0.0{]}

\end{fulllineitems}

\index{pymethyl-visualize-plot\_heatmap command line option@\spxentry{pymethyl-visualize-plot\_heatmap command line option}!-max, --max\_val \textless{}max\_val\textgreater{}@\spxentry{-max, --max\_val \textless{}max\_val\textgreater{}}}\index{-max, --max\_val \textless{}max\_val\textgreater{}@\spxentry{-max, --max\_val \textless{}max\_val\textgreater{}}!pymethyl-visualize-plot\_heatmap command line option@\spxentry{pymethyl-visualize-plot\_heatmap command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-visualize-plot-heatmap-max}}\pysigline{\sphinxbfcode{\sphinxupquote{-max}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}max\_val}}\sphinxcode{\sphinxupquote{~\textless{}max\_val\textgreater{}}}}
Max heat val, if -1, defaults to None  {[}default: 1.0{]}

\end{fulllineitems}

\index{pymethyl-visualize-plot\_heatmap command line option@\spxentry{pymethyl-visualize-plot\_heatmap command line option}!-a, --annot@\spxentry{-a, --annot}}\index{-a, --annot@\spxentry{-a, --annot}!pymethyl-visualize-plot\_heatmap command line option@\spxentry{pymethyl-visualize-plot\_heatmap command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-visualize-plot-heatmap-a}}\pysigline{\sphinxbfcode{\sphinxupquote{-a}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}annot}}\sphinxcode{\sphinxupquote{}}}
Annotate heatmap  {[}default: False{]}

\end{fulllineitems}

\index{pymethyl-visualize-plot\_heatmap command line option@\spxentry{pymethyl-visualize-plot\_heatmap command line option}!-n, --norm@\spxentry{-n, --norm}}\index{-n, --norm@\spxentry{-n, --norm}!pymethyl-visualize-plot\_heatmap command line option@\spxentry{pymethyl-visualize-plot\_heatmap command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-visualize-plot-heatmap-n}}\pysigline{\sphinxbfcode{\sphinxupquote{-n}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}norm}}\sphinxcode{\sphinxupquote{}}}
Normalize matrix data  {[}default: False{]}

\end{fulllineitems}

\index{pymethyl-visualize-plot\_heatmap command line option@\spxentry{pymethyl-visualize-plot\_heatmap command line option}!-c, --cluster@\spxentry{-c, --cluster}}\index{-c, --cluster@\spxentry{-c, --cluster}!pymethyl-visualize-plot\_heatmap command line option@\spxentry{pymethyl-visualize-plot\_heatmap command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-visualize-plot-heatmap-c}}\pysigline{\sphinxbfcode{\sphinxupquote{-c}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}cluster}}\sphinxcode{\sphinxupquote{}}}
Cluster matrix data  {[}default: False{]}

\end{fulllineitems}

\index{pymethyl-visualize-plot\_heatmap command line option@\spxentry{pymethyl-visualize-plot\_heatmap command line option}!-m, --matrix\_type \textless{}matrix\_type\textgreater{}@\spxentry{-m, --matrix\_type \textless{}matrix\_type\textgreater{}}}\index{-m, --matrix\_type \textless{}matrix\_type\textgreater{}@\spxentry{-m, --matrix\_type \textless{}matrix\_type\textgreater{}}!pymethyl-visualize-plot\_heatmap command line option@\spxentry{pymethyl-visualize-plot\_heatmap command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-visualize-plot-heatmap-m}}\pysigline{\sphinxbfcode{\sphinxupquote{-m}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}matrix\_type}}\sphinxcode{\sphinxupquote{~\textless{}matrix\_type\textgreater{}}}}
Type of matrix supplied  {[}default: none{]}

\end{fulllineitems}

\index{pymethyl-visualize-plot\_heatmap command line option@\spxentry{pymethyl-visualize-plot\_heatmap command line option}!-x, --xticks@\spxentry{-x, --xticks}}\index{-x, --xticks@\spxentry{-x, --xticks}!pymethyl-visualize-plot\_heatmap command line option@\spxentry{pymethyl-visualize-plot\_heatmap command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-visualize-plot-heatmap-x}}\pysigline{\sphinxbfcode{\sphinxupquote{-x}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}xticks}}\sphinxcode{\sphinxupquote{}}}
Show x ticks  {[}default: False{]}

\end{fulllineitems}

\index{pymethyl-visualize-plot\_heatmap command line option@\spxentry{pymethyl-visualize-plot\_heatmap command line option}!-y, --yticks@\spxentry{-y, --yticks}}\index{-y, --yticks@\spxentry{-y, --yticks}!pymethyl-visualize-plot\_heatmap command line option@\spxentry{pymethyl-visualize-plot\_heatmap command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-visualize-plot-heatmap-y}}\pysigline{\sphinxbfcode{\sphinxupquote{-y}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}yticks}}\sphinxcode{\sphinxupquote{}}}
Show y ticks  {[}default: False{]}

\end{fulllineitems}

\index{pymethyl-visualize-plot\_heatmap command line option@\spxentry{pymethyl-visualize-plot\_heatmap command line option}!-t, --transpose@\spxentry{-t, --transpose}}\index{-t, --transpose@\spxentry{-t, --transpose}!pymethyl-visualize-plot\_heatmap command line option@\spxentry{pymethyl-visualize-plot\_heatmap command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-visualize-plot-heatmap-t}}\pysigline{\sphinxbfcode{\sphinxupquote{-t}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}transpose}}\sphinxcode{\sphinxupquote{}}}
Transpose matrix data  {[}default: False{]}

\end{fulllineitems}

\index{pymethyl-visualize-plot\_heatmap command line option@\spxentry{pymethyl-visualize-plot\_heatmap command line option}!-col, --color\_column \textless{}color\_column\textgreater{}@\spxentry{-col, --color\_column \textless{}color\_column\textgreater{}}}\index{-col, --color\_column \textless{}color\_column\textgreater{}@\spxentry{-col, --color\_column \textless{}color\_column\textgreater{}}!pymethyl-visualize-plot\_heatmap command line option@\spxentry{pymethyl-visualize-plot\_heatmap command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-visualize-plot-heatmap-col}}\pysigline{\sphinxbfcode{\sphinxupquote{-col}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}color\_column}}\sphinxcode{\sphinxupquote{~\textless{}color\_column\textgreater{}}}}
Color column.  {[}default: color{]}

\end{fulllineitems}



\section{transform\_plot}
\label{\detokenize{index:pymethyl-visualize-transform-plot}}
Dimensionality reduce VAE or original beta values using UMAP and plot using plotly.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}visualize transform\PYGZus{}plot \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-visualize-transform\_plot command line option@\spxentry{pymethyl-visualize-transform\_plot command line option}!-i, --input\_pkl \textless{}input\_pkl\textgreater{}@\spxentry{-i, --input\_pkl \textless{}input\_pkl\textgreater{}}}\index{-i, --input\_pkl \textless{}input\_pkl\textgreater{}@\spxentry{-i, --input\_pkl \textless{}input\_pkl\textgreater{}}!pymethyl-visualize-transform\_plot command line option@\spxentry{pymethyl-visualize-transform\_plot command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-visualize-transform-plot-i}}\pysigline{\sphinxbfcode{\sphinxupquote{-i}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}input\_pkl}}\sphinxcode{\sphinxupquote{~\textless{}input\_pkl\textgreater{}}}}
Input database for beta and phenotype data.  {[}default: ./final\_preprocessed/methyl\_array.pkl{]}

\end{fulllineitems}

\index{pymethyl-visualize-transform\_plot command line option@\spxentry{pymethyl-visualize-transform\_plot command line option}!-c, --column\_of\_interest \textless{}column\_of\_interest\textgreater{}@\spxentry{-c, --column\_of\_interest \textless{}column\_of\_interest\textgreater{}}}\index{-c, --column\_of\_interest \textless{}column\_of\_interest\textgreater{}@\spxentry{-c, --column\_of\_interest \textless{}column\_of\_interest\textgreater{}}!pymethyl-visualize-transform\_plot command line option@\spxentry{pymethyl-visualize-transform\_plot command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-visualize-transform-plot-c}}\pysigline{\sphinxbfcode{\sphinxupquote{-c}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}column\_of\_interest}}\sphinxcode{\sphinxupquote{~\textless{}column\_of\_interest\textgreater{}}}}
Column extract from phenotype data.  {[}default: disease{]}

\end{fulllineitems}

\index{pymethyl-visualize-transform\_plot command line option@\spxentry{pymethyl-visualize-transform\_plot command line option}!-o, --output\_file \textless{}output\_file\textgreater{}@\spxentry{-o, --output\_file \textless{}output\_file\textgreater{}}}\index{-o, --output\_file \textless{}output\_file\textgreater{}@\spxentry{-o, --output\_file \textless{}output\_file\textgreater{}}!pymethyl-visualize-transform\_plot command line option@\spxentry{pymethyl-visualize-transform\_plot command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-visualize-transform-plot-o}}\pysigline{\sphinxbfcode{\sphinxupquote{-o}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}output\_file}}\sphinxcode{\sphinxupquote{~\textless{}output\_file\textgreater{}}}}
Output visualization.  {[}default: ./visualization.html{]}

\end{fulllineitems}

\index{pymethyl-visualize-transform\_plot command line option@\spxentry{pymethyl-visualize-transform\_plot command line option}!-nn, --n\_neighbors \textless{}n\_neighbors\textgreater{}@\spxentry{-nn, --n\_neighbors \textless{}n\_neighbors\textgreater{}}}\index{-nn, --n\_neighbors \textless{}n\_neighbors\textgreater{}@\spxentry{-nn, --n\_neighbors \textless{}n\_neighbors\textgreater{}}!pymethyl-visualize-transform\_plot command line option@\spxentry{pymethyl-visualize-transform\_plot command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-visualize-transform-plot-nn}}\pysigline{\sphinxbfcode{\sphinxupquote{-nn}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}n\_neighbors}}\sphinxcode{\sphinxupquote{~\textless{}n\_neighbors\textgreater{}}}}
Number of neighbors UMAP.  {[}default: 5{]}

\end{fulllineitems}

\index{pymethyl-visualize-transform\_plot command line option@\spxentry{pymethyl-visualize-transform\_plot command line option}!-a, --axes\_off@\spxentry{-a, --axes\_off}}\index{-a, --axes\_off@\spxentry{-a, --axes\_off}!pymethyl-visualize-transform\_plot command line option@\spxentry{pymethyl-visualize-transform\_plot command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-visualize-transform-plot-a}}\pysigline{\sphinxbfcode{\sphinxupquote{-a}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}axes\_off}}\sphinxcode{\sphinxupquote{}}}
Whether to turn axes on or off.

\end{fulllineitems}

\index{pymethyl-visualize-transform\_plot command line option@\spxentry{pymethyl-visualize-transform\_plot command line option}!-s, --supervised@\spxentry{-s, --supervised}}\index{-s, --supervised@\spxentry{-s, --supervised}!pymethyl-visualize-transform\_plot command line option@\spxentry{pymethyl-visualize-transform\_plot command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-visualize-transform-plot-s}}\pysigline{\sphinxbfcode{\sphinxupquote{-s}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}supervised}}\sphinxcode{\sphinxupquote{}}}
Supervise umap embedding.

\end{fulllineitems}

\index{pymethyl-visualize-transform\_plot command line option@\spxentry{pymethyl-visualize-transform\_plot command line option}!-d, --min\_dist \textless{}min\_dist\textgreater{}@\spxentry{-d, --min\_dist \textless{}min\_dist\textgreater{}}}\index{-d, --min\_dist \textless{}min\_dist\textgreater{}@\spxentry{-d, --min\_dist \textless{}min\_dist\textgreater{}}!pymethyl-visualize-transform\_plot command line option@\spxentry{pymethyl-visualize-transform\_plot command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-visualize-transform-plot-d}}\pysigline{\sphinxbfcode{\sphinxupquote{-d}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}min\_dist}}\sphinxcode{\sphinxupquote{~\textless{}min\_dist\textgreater{}}}}
UMAP min distance.  {[}default: 0.1{]}

\end{fulllineitems}

\index{pymethyl-visualize-transform\_plot command line option@\spxentry{pymethyl-visualize-transform\_plot command line option}!-m, --metric \textless{}metric\textgreater{}@\spxentry{-m, --metric \textless{}metric\textgreater{}}}\index{-m, --metric \textless{}metric\textgreater{}@\spxentry{-m, --metric \textless{}metric\textgreater{}}!pymethyl-visualize-transform\_plot command line option@\spxentry{pymethyl-visualize-transform\_plot command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-visualize-transform-plot-m}}\pysigline{\sphinxbfcode{\sphinxupquote{-m}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}metric}}\sphinxcode{\sphinxupquote{~\textless{}metric\textgreater{}}}}
Reduction metric.  {[}default: euclidean{]}

\end{fulllineitems}

\index{pymethyl-visualize-transform\_plot command line option@\spxentry{pymethyl-visualize-transform\_plot command line option}!-cc, --case\_control\_override@\spxentry{-cc, --case\_control\_override}}\index{-cc, --case\_control\_override@\spxentry{-cc, --case\_control\_override}!pymethyl-visualize-transform\_plot command line option@\spxentry{pymethyl-visualize-transform\_plot command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-visualize-transform-plot-cc}}\pysigline{\sphinxbfcode{\sphinxupquote{-cc}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}case\_control\_override}}\sphinxcode{\sphinxupquote{}}}
Add controls from case\_control column and override current disease for classification tasks.  {[}default: False{]}

\end{fulllineitems}



\chapter{pymethyl-preprocess}
\label{\detokenize{index:pymethyl-preprocess}}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}preprocess \PYG{o}{[}OPTIONS\PYG{o}{]} COMMAND \PYG{o}{[}ARGS\PYG{o}{]}...
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-preprocess command line option@\spxentry{pymethyl-preprocess command line option}!--version@\spxentry{--version}}\index{--version@\spxentry{--version}!pymethyl-preprocess command line option@\spxentry{pymethyl-preprocess command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-version}}\pysigline{\sphinxbfcode{\sphinxupquote{-{-}version}}\sphinxcode{\sphinxupquote{}}}
Show the version and exit.

\end{fulllineitems}



\section{batch\_deploy\_preprocess}
\label{\detokenize{index:pymethyl-preprocess-batch-deploy-preprocess}}
Deploy multiple preprocessing jobs in series or parallel.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}preprocess batch\PYGZus{}deploy\PYGZus{}preprocess \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-preprocess-batch\_deploy\_preprocess command line option@\spxentry{pymethyl-preprocess-batch\_deploy\_preprocess command line option}!-n, --n\_cores \textless{}n\_cores\textgreater{}@\spxentry{-n, --n\_cores \textless{}n\_cores\textgreater{}}}\index{-n, --n\_cores \textless{}n\_cores\textgreater{}@\spxentry{-n, --n\_cores \textless{}n\_cores\textgreater{}}!pymethyl-preprocess-batch\_deploy\_preprocess command line option@\spxentry{pymethyl-preprocess-batch\_deploy\_preprocess command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-batch-deploy-preprocess-n}}\pysigline{\sphinxbfcode{\sphinxupquote{-n}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}n\_cores}}\sphinxcode{\sphinxupquote{~\textless{}n\_cores\textgreater{}}}}
Number cores to use for preprocessing.  {[}default: 6{]}

\end{fulllineitems}

\index{pymethyl-preprocess-batch\_deploy\_preprocess command line option@\spxentry{pymethyl-preprocess-batch\_deploy\_preprocess command line option}!-i, --subtype\_output\_dir \textless{}subtype\_output\_dir\textgreater{}@\spxentry{-i, --subtype\_output\_dir \textless{}subtype\_output\_dir\textgreater{}}}\index{-i, --subtype\_output\_dir \textless{}subtype\_output\_dir\textgreater{}@\spxentry{-i, --subtype\_output\_dir \textless{}subtype\_output\_dir\textgreater{}}!pymethyl-preprocess-batch\_deploy\_preprocess command line option@\spxentry{pymethyl-preprocess-batch\_deploy\_preprocess command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-batch-deploy-preprocess-i}}\pysigline{\sphinxbfcode{\sphinxupquote{-i}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}subtype\_output\_dir}}\sphinxcode{\sphinxupquote{~\textless{}subtype\_output\_dir\textgreater{}}}}
Output subtypes pheno csv.  {[}default: ./preprocess\_outputs/{]}

\end{fulllineitems}

\index{pymethyl-preprocess-batch\_deploy\_preprocess command line option@\spxentry{pymethyl-preprocess-batch\_deploy\_preprocess command line option}!-m, --meffil@\spxentry{-m, --meffil}}\index{-m, --meffil@\spxentry{-m, --meffil}!pymethyl-preprocess-batch\_deploy\_preprocess command line option@\spxentry{pymethyl-preprocess-batch\_deploy\_preprocess command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-batch-deploy-preprocess-m}}\pysigline{\sphinxbfcode{\sphinxupquote{-m}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}meffil}}\sphinxcode{\sphinxupquote{}}}
Preprocess using meffil.

\end{fulllineitems}

\index{pymethyl-preprocess-batch\_deploy\_preprocess command line option@\spxentry{pymethyl-preprocess-batch\_deploy\_preprocess command line option}!-t, --torque@\spxentry{-t, --torque}}\index{-t, --torque@\spxentry{-t, --torque}!pymethyl-preprocess-batch\_deploy\_preprocess command line option@\spxentry{pymethyl-preprocess-batch\_deploy\_preprocess command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-batch-deploy-preprocess-t}}\pysigline{\sphinxbfcode{\sphinxupquote{-t}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}torque}}\sphinxcode{\sphinxupquote{}}}
Job submission torque.

\end{fulllineitems}

\index{pymethyl-preprocess-batch\_deploy\_preprocess command line option@\spxentry{pymethyl-preprocess-batch\_deploy\_preprocess command line option}!-r, --run@\spxentry{-r, --run}}\index{-r, --run@\spxentry{-r, --run}!pymethyl-preprocess-batch\_deploy\_preprocess command line option@\spxentry{pymethyl-preprocess-batch\_deploy\_preprocess command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-batch-deploy-preprocess-r}}\pysigline{\sphinxbfcode{\sphinxupquote{-r}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}run}}\sphinxcode{\sphinxupquote{}}}
Actually run local job or just print out command.

\end{fulllineitems}

\index{pymethyl-preprocess-batch\_deploy\_preprocess command line option@\spxentry{pymethyl-preprocess-batch\_deploy\_preprocess command line option}!-s, --series@\spxentry{-s, --series}}\index{-s, --series@\spxentry{-s, --series}!pymethyl-preprocess-batch\_deploy\_preprocess command line option@\spxentry{pymethyl-preprocess-batch\_deploy\_preprocess command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-batch-deploy-preprocess-s}}\pysigline{\sphinxbfcode{\sphinxupquote{-s}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}series}}\sphinxcode{\sphinxupquote{}}}
Run commands in series.

\end{fulllineitems}

\index{pymethyl-preprocess-batch\_deploy\_preprocess command line option@\spxentry{pymethyl-preprocess-batch\_deploy\_preprocess command line option}!-p, --pc\_qc\_parameters\_csv \textless{}pc\_qc\_parameters\_csv\textgreater{}@\spxentry{-p, --pc\_qc\_parameters\_csv \textless{}pc\_qc\_parameters\_csv\textgreater{}}}\index{-p, --pc\_qc\_parameters\_csv \textless{}pc\_qc\_parameters\_csv\textgreater{}@\spxentry{-p, --pc\_qc\_parameters\_csv \textless{}pc\_qc\_parameters\_csv\textgreater{}}!pymethyl-preprocess-batch\_deploy\_preprocess command line option@\spxentry{pymethyl-preprocess-batch\_deploy\_preprocess command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-batch-deploy-preprocess-p}}\pysigline{\sphinxbfcode{\sphinxupquote{-p}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}pc\_qc\_parameters\_csv}}\sphinxcode{\sphinxupquote{~\textless{}pc\_qc\_parameters\_csv\textgreater{}}}}
For meffil, qc parameters and pcs for final qc and functional normalization.  {[}default: ./preprocess\_outputs/pc\_qc\_parameters.csv{]}

\end{fulllineitems}

\index{pymethyl-preprocess-batch\_deploy\_preprocess command line option@\spxentry{pymethyl-preprocess-batch\_deploy\_preprocess command line option}!-u, --use\_cache@\spxentry{-u, --use\_cache}}\index{-u, --use\_cache@\spxentry{-u, --use\_cache}!pymethyl-preprocess-batch\_deploy\_preprocess command line option@\spxentry{pymethyl-preprocess-batch\_deploy\_preprocess command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-batch-deploy-preprocess-u}}\pysigline{\sphinxbfcode{\sphinxupquote{-u}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}use\_cache}}\sphinxcode{\sphinxupquote{}}}
If this is selected, loads qc results rather than running qc again. Only works for meffil selection.

\end{fulllineitems}

\index{pymethyl-preprocess-batch\_deploy\_preprocess command line option@\spxentry{pymethyl-preprocess-batch\_deploy\_preprocess command line option}!-qc, --qc\_only@\spxentry{-qc, --qc\_only}}\index{-qc, --qc\_only@\spxentry{-qc, --qc\_only}!pymethyl-preprocess-batch\_deploy\_preprocess command line option@\spxentry{pymethyl-preprocess-batch\_deploy\_preprocess command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-batch-deploy-preprocess-qc}}\pysigline{\sphinxbfcode{\sphinxupquote{-qc}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}qc\_only}}\sphinxcode{\sphinxupquote{}}}
Only perform QC for meffil pipeline, caches results into rds file for loading again, only works if use\_cache is false.

\end{fulllineitems}

\index{pymethyl-preprocess-batch\_deploy\_preprocess command line option@\spxentry{pymethyl-preprocess-batch\_deploy\_preprocess command line option}!-c, --chunk\_size \textless{}chunk\_size\textgreater{}@\spxentry{-c, --chunk\_size \textless{}chunk\_size\textgreater{}}}\index{-c, --chunk\_size \textless{}chunk\_size\textgreater{}@\spxentry{-c, --chunk\_size \textless{}chunk\_size\textgreater{}}!pymethyl-preprocess-batch\_deploy\_preprocess command line option@\spxentry{pymethyl-preprocess-batch\_deploy\_preprocess command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-batch-deploy-preprocess-c}}\pysigline{\sphinxbfcode{\sphinxupquote{-c}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}chunk\_size}}\sphinxcode{\sphinxupquote{~\textless{}chunk\_size\textgreater{}}}}
If not series, chunk up and run these number of commands at once.. -1 means all commands at once.

\end{fulllineitems}



\section{combine\_methylation\_arrays}
\label{\detokenize{index:pymethyl-preprocess-combine-methylation-arrays}}
If split MethylationArrays by subtype for either preprocessing or imputation, can use to recombine data for downstream step.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}preprocess combine\PYGZus{}methylation\PYGZus{}arrays \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-preprocess-combine\_methylation\_arrays command line option@\spxentry{pymethyl-preprocess-combine\_methylation\_arrays command line option}!-i, --input\_pkls \textless{}input\_pkls\textgreater{}@\spxentry{-i, --input\_pkls \textless{}input\_pkls\textgreater{}}}\index{-i, --input\_pkls \textless{}input\_pkls\textgreater{}@\spxentry{-i, --input\_pkls \textless{}input\_pkls\textgreater{}}!pymethyl-preprocess-combine\_methylation\_arrays command line option@\spxentry{pymethyl-preprocess-combine\_methylation\_arrays command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-combine-methylation-arrays-i}}\pysigline{\sphinxbfcode{\sphinxupquote{-i}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}input\_pkls}}\sphinxcode{\sphinxupquote{~\textless{}input\_pkls\textgreater{}}}}
Input pickles for beta and phenotype data.  {[}default: ./preprocess\_outputs/methyl\_array.pkl{]}

\end{fulllineitems}

\index{pymethyl-preprocess-combine\_methylation\_arrays command line option@\spxentry{pymethyl-preprocess-combine\_methylation\_arrays command line option}!-d, --optional\_input\_pkl\_dir \textless{}optional\_input\_pkl\_dir\textgreater{}@\spxentry{-d, --optional\_input\_pkl\_dir \textless{}optional\_input\_pkl\_dir\textgreater{}}}\index{-d, --optional\_input\_pkl\_dir \textless{}optional\_input\_pkl\_dir\textgreater{}@\spxentry{-d, --optional\_input\_pkl\_dir \textless{}optional\_input\_pkl\_dir\textgreater{}}!pymethyl-preprocess-combine\_methylation\_arrays command line option@\spxentry{pymethyl-preprocess-combine\_methylation\_arrays command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-combine-methylation-arrays-d}}\pysigline{\sphinxbfcode{\sphinxupquote{-d}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}optional\_input\_pkl\_dir}}\sphinxcode{\sphinxupquote{~\textless{}optional\_input\_pkl\_dir\textgreater{}}}}
Auto grab input pkls.  {[}default: {]}

\end{fulllineitems}

\index{pymethyl-preprocess-combine\_methylation\_arrays command line option@\spxentry{pymethyl-preprocess-combine\_methylation\_arrays command line option}!-o, --output\_pkl \textless{}output\_pkl\textgreater{}@\spxentry{-o, --output\_pkl \textless{}output\_pkl\textgreater{}}}\index{-o, --output\_pkl \textless{}output\_pkl\textgreater{}@\spxentry{-o, --output\_pkl \textless{}output\_pkl\textgreater{}}!pymethyl-preprocess-combine\_methylation\_arrays command line option@\spxentry{pymethyl-preprocess-combine\_methylation\_arrays command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-combine-methylation-arrays-o}}\pysigline{\sphinxbfcode{\sphinxupquote{-o}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}output\_pkl}}\sphinxcode{\sphinxupquote{~\textless{}output\_pkl\textgreater{}}}}
Output database for beta and phenotype data.  {[}default: ./combined\_outputs/methyl\_array.pkl{]}

\end{fulllineitems}

\index{pymethyl-preprocess-combine\_methylation\_arrays command line option@\spxentry{pymethyl-preprocess-combine\_methylation\_arrays command line option}!-e, --exclude \textless{}exclude\textgreater{}@\spxentry{-e, --exclude \textless{}exclude\textgreater{}}}\index{-e, --exclude \textless{}exclude\textgreater{}@\spxentry{-e, --exclude \textless{}exclude\textgreater{}}!pymethyl-preprocess-combine\_methylation\_arrays command line option@\spxentry{pymethyl-preprocess-combine\_methylation\_arrays command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-combine-methylation-arrays-e}}\pysigline{\sphinxbfcode{\sphinxupquote{-e}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}exclude}}\sphinxcode{\sphinxupquote{~\textless{}exclude\textgreater{}}}}
If -d selected, these diseases will be excluded from study.  {[}default: {]}

\end{fulllineitems}



\section{concat\_sample\_sheets}
\label{\detokenize{index:pymethyl-preprocess-concat-sample-sheets}}
Concat two sample files for more fields for minfi+ input, adds more samples.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}preprocess concat\PYGZus{}sample\PYGZus{}sheets \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-preprocess-concat\_sample\_sheets command line option@\spxentry{pymethyl-preprocess-concat\_sample\_sheets command line option}!-s1, --sample\_sheet1 \textless{}sample\_sheet1\textgreater{}@\spxentry{-s1, --sample\_sheet1 \textless{}sample\_sheet1\textgreater{}}}\index{-s1, --sample\_sheet1 \textless{}sample\_sheet1\textgreater{}@\spxentry{-s1, --sample\_sheet1 \textless{}sample\_sheet1\textgreater{}}!pymethyl-preprocess-concat\_sample\_sheets command line option@\spxentry{pymethyl-preprocess-concat\_sample\_sheets command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-concat-sample-sheets-s1}}\pysigline{\sphinxbfcode{\sphinxupquote{-s1}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}sample\_sheet1}}\sphinxcode{\sphinxupquote{~\textless{}sample\_sheet1\textgreater{}}}}
Clinical information downloaded from tcga/geo/custom, formatted using create\_sample\_sheet.  {[}default: ./tcga\_idats/clinical\_info1.csv{]}

\end{fulllineitems}

\index{pymethyl-preprocess-concat\_sample\_sheets command line option@\spxentry{pymethyl-preprocess-concat\_sample\_sheets command line option}!-s2, --sample\_sheet2 \textless{}sample\_sheet2\textgreater{}@\spxentry{-s2, --sample\_sheet2 \textless{}sample\_sheet2\textgreater{}}}\index{-s2, --sample\_sheet2 \textless{}sample\_sheet2\textgreater{}@\spxentry{-s2, --sample\_sheet2 \textless{}sample\_sheet2\textgreater{}}!pymethyl-preprocess-concat\_sample\_sheets command line option@\spxentry{pymethyl-preprocess-concat\_sample\_sheets command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-concat-sample-sheets-s2}}\pysigline{\sphinxbfcode{\sphinxupquote{-s2}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}sample\_sheet2}}\sphinxcode{\sphinxupquote{~\textless{}sample\_sheet2\textgreater{}}}}
Clinical information downloaded from tcga/geo/custom, formatted using create\_sample\_sheet.  {[}default: ./tcga\_idats/clinical\_info2.csv{]}

\end{fulllineitems}

\index{pymethyl-preprocess-concat\_sample\_sheets command line option@\spxentry{pymethyl-preprocess-concat\_sample\_sheets command line option}!-os, --output\_sample\_sheet \textless{}output\_sample\_sheet\textgreater{}@\spxentry{-os, --output\_sample\_sheet \textless{}output\_sample\_sheet\textgreater{}}}\index{-os, --output\_sample\_sheet \textless{}output\_sample\_sheet\textgreater{}@\spxentry{-os, --output\_sample\_sheet \textless{}output\_sample\_sheet\textgreater{}}!pymethyl-preprocess-concat\_sample\_sheets command line option@\spxentry{pymethyl-preprocess-concat\_sample\_sheets command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-concat-sample-sheets-os}}\pysigline{\sphinxbfcode{\sphinxupquote{-os}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}output\_sample\_sheet}}\sphinxcode{\sphinxupquote{~\textless{}output\_sample\_sheet\textgreater{}}}}
CSV for minfi input.  {[}default: ./tcga\_idats/minfiSheet.csv{]}

\end{fulllineitems}



\section{create\_sample\_sheet}
\label{\detokenize{index:pymethyl-preprocess-create-sample-sheet}}
Create sample sheet for input to minfi, meffil, or enmix.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}preprocess create\PYGZus{}sample\PYGZus{}sheet \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-preprocess-create\_sample\_sheet command line option@\spxentry{pymethyl-preprocess-create\_sample\_sheet command line option}!-is, --input\_sample\_sheet \textless{}input\_sample\_sheet\textgreater{}@\spxentry{-is, --input\_sample\_sheet \textless{}input\_sample\_sheet\textgreater{}}}\index{-is, --input\_sample\_sheet \textless{}input\_sample\_sheet\textgreater{}@\spxentry{-is, --input\_sample\_sheet \textless{}input\_sample\_sheet\textgreater{}}!pymethyl-preprocess-create\_sample\_sheet command line option@\spxentry{pymethyl-preprocess-create\_sample\_sheet command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-create-sample-sheet-is}}\pysigline{\sphinxbfcode{\sphinxupquote{-is}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}input\_sample\_sheet}}\sphinxcode{\sphinxupquote{~\textless{}input\_sample\_sheet\textgreater{}}}}
Clinical information downloaded from tcga/geo/custom.  {[}default: ./tcga\_idats/clinical\_info.csv{]}

\end{fulllineitems}

\index{pymethyl-preprocess-create\_sample\_sheet command line option@\spxentry{pymethyl-preprocess-create\_sample\_sheet command line option}!-s, --source\_type \textless{}source\_type\textgreater{}@\spxentry{-s, --source\_type \textless{}source\_type\textgreater{}}}\index{-s, --source\_type \textless{}source\_type\textgreater{}@\spxentry{-s, --source\_type \textless{}source\_type\textgreater{}}!pymethyl-preprocess-create\_sample\_sheet command line option@\spxentry{pymethyl-preprocess-create\_sample\_sheet command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-create-sample-sheet-s}}\pysigline{\sphinxbfcode{\sphinxupquote{-s}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}source\_type}}\sphinxcode{\sphinxupquote{~\textless{}source\_type\textgreater{}}}}
Source type of data.  {[}default: tcga{]}

\end{fulllineitems}

\index{pymethyl-preprocess-create\_sample\_sheet command line option@\spxentry{pymethyl-preprocess-create\_sample\_sheet command line option}!-i, --idat\_dir \textless{}idat\_dir\textgreater{}@\spxentry{-i, --idat\_dir \textless{}idat\_dir\textgreater{}}}\index{-i, --idat\_dir \textless{}idat\_dir\textgreater{}@\spxentry{-i, --idat\_dir \textless{}idat\_dir\textgreater{}}!pymethyl-preprocess-create\_sample\_sheet command line option@\spxentry{pymethyl-preprocess-create\_sample\_sheet command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-create-sample-sheet-i}}\pysigline{\sphinxbfcode{\sphinxupquote{-i}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}idat\_dir}}\sphinxcode{\sphinxupquote{~\textless{}idat\_dir\textgreater{}}}}
Idat directory.  {[}default: ./tcga\_idats/{]}

\end{fulllineitems}

\index{pymethyl-preprocess-create\_sample\_sheet command line option@\spxentry{pymethyl-preprocess-create\_sample\_sheet command line option}!-os, --output\_sample\_sheet \textless{}output\_sample\_sheet\textgreater{}@\spxentry{-os, --output\_sample\_sheet \textless{}output\_sample\_sheet\textgreater{}}}\index{-os, --output\_sample\_sheet \textless{}output\_sample\_sheet\textgreater{}@\spxentry{-os, --output\_sample\_sheet \textless{}output\_sample\_sheet\textgreater{}}!pymethyl-preprocess-create\_sample\_sheet command line option@\spxentry{pymethyl-preprocess-create\_sample\_sheet command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-create-sample-sheet-os}}\pysigline{\sphinxbfcode{\sphinxupquote{-os}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}output\_sample\_sheet}}\sphinxcode{\sphinxupquote{~\textless{}output\_sample\_sheet\textgreater{}}}}
CSV for minfi input.  {[}default: ./tcga\_idats/minfiSheet.csv{]}

\end{fulllineitems}

\index{pymethyl-preprocess-create\_sample\_sheet command line option@\spxentry{pymethyl-preprocess-create\_sample\_sheet command line option}!-m, --mapping\_file \textless{}mapping\_file\textgreater{}@\spxentry{-m, --mapping\_file \textless{}mapping\_file\textgreater{}}}\index{-m, --mapping\_file \textless{}mapping\_file\textgreater{}@\spxentry{-m, --mapping\_file \textless{}mapping\_file\textgreater{}}!pymethyl-preprocess-create\_sample\_sheet command line option@\spxentry{pymethyl-preprocess-create\_sample\_sheet command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-create-sample-sheet-m}}\pysigline{\sphinxbfcode{\sphinxupquote{-m}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}mapping\_file}}\sphinxcode{\sphinxupquote{~\textless{}mapping\_file\textgreater{}}}}
Mapping file from uuid to TCGA barcode. Downloaded using download\_tcga.  {[}default: ./idat\_filename\_case.txt{]}

\end{fulllineitems}

\index{pymethyl-preprocess-create\_sample\_sheet command line option@\spxentry{pymethyl-preprocess-create\_sample\_sheet command line option}!-l, --header\_line \textless{}header\_line\textgreater{}@\spxentry{-l, --header\_line \textless{}header\_line\textgreater{}}}\index{-l, --header\_line \textless{}header\_line\textgreater{}@\spxentry{-l, --header\_line \textless{}header\_line\textgreater{}}!pymethyl-preprocess-create\_sample\_sheet command line option@\spxentry{pymethyl-preprocess-create\_sample\_sheet command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-create-sample-sheet-l}}\pysigline{\sphinxbfcode{\sphinxupquote{-l}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}header\_line}}\sphinxcode{\sphinxupquote{~\textless{}header\_line\textgreater{}}}}
Line to begin reading csv/xlsx.  {[}default: 0{]}

\end{fulllineitems}

\index{pymethyl-preprocess-create\_sample\_sheet command line option@\spxentry{pymethyl-preprocess-create\_sample\_sheet command line option}!-d, --disease\_class\_column \textless{}disease\_class\_column\textgreater{}@\spxentry{-d, --disease\_class\_column \textless{}disease\_class\_column\textgreater{}}}\index{-d, --disease\_class\_column \textless{}disease\_class\_column\textgreater{}@\spxentry{-d, --disease\_class\_column \textless{}disease\_class\_column\textgreater{}}!pymethyl-preprocess-create\_sample\_sheet command line option@\spxentry{pymethyl-preprocess-create\_sample\_sheet command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-create-sample-sheet-d}}\pysigline{\sphinxbfcode{\sphinxupquote{-d}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}disease\_class\_column}}\sphinxcode{\sphinxupquote{~\textless{}disease\_class\_column\textgreater{}}}}
Disease classification column, for custom and geo datasets.  {[}default: methylation class:ch1{]}

\end{fulllineitems}

\index{pymethyl-preprocess-create\_sample\_sheet command line option@\spxentry{pymethyl-preprocess-create\_sample\_sheet command line option}!-b, --basename\_col \textless{}basename\_col\textgreater{}@\spxentry{-b, --basename\_col \textless{}basename\_col\textgreater{}}}\index{-b, --basename\_col \textless{}basename\_col\textgreater{}@\spxentry{-b, --basename\_col \textless{}basename\_col\textgreater{}}!pymethyl-preprocess-create\_sample\_sheet command line option@\spxentry{pymethyl-preprocess-create\_sample\_sheet command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-create-sample-sheet-b}}\pysigline{\sphinxbfcode{\sphinxupquote{-b}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}basename\_col}}\sphinxcode{\sphinxupquote{~\textless{}basename\_col\textgreater{}}}}
Basename classification column, for custom datasets.  {[}default: Sentrix ID (.idat){]}

\end{fulllineitems}

\index{pymethyl-preprocess-create\_sample\_sheet command line option@\spxentry{pymethyl-preprocess-create\_sample\_sheet command line option}!-c, --include\_columns\_file \textless{}include\_columns\_file\textgreater{}@\spxentry{-c, --include\_columns\_file \textless{}include\_columns\_file\textgreater{}}}\index{-c, --include\_columns\_file \textless{}include\_columns\_file\textgreater{}@\spxentry{-c, --include\_columns\_file \textless{}include\_columns\_file\textgreater{}}!pymethyl-preprocess-create\_sample\_sheet command line option@\spxentry{pymethyl-preprocess-create\_sample\_sheet command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-create-sample-sheet-c}}\pysigline{\sphinxbfcode{\sphinxupquote{-c}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}include\_columns\_file}}\sphinxcode{\sphinxupquote{~\textless{}include\_columns\_file\textgreater{}}}}
Custom columns file containing columns to keep, separated by n. Add a tab for each line if you wish to rename columns: original\_name t new\_column\_name  {[}default: {]}

\end{fulllineitems}



\section{download\_clinical}
\label{\detokenize{index:pymethyl-preprocess-download-clinical}}
Download all TCGA 450k clinical info.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}preprocess download\PYGZus{}clinical \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-preprocess-download\_clinical command line option@\spxentry{pymethyl-preprocess-download\_clinical command line option}!-o, --output\_dir \textless{}output\_dir\textgreater{}@\spxentry{-o, --output\_dir \textless{}output\_dir\textgreater{}}}\index{-o, --output\_dir \textless{}output\_dir\textgreater{}@\spxentry{-o, --output\_dir \textless{}output\_dir\textgreater{}}!pymethyl-preprocess-download\_clinical command line option@\spxentry{pymethyl-preprocess-download\_clinical command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-download-clinical-o}}\pysigline{\sphinxbfcode{\sphinxupquote{-o}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}output\_dir}}\sphinxcode{\sphinxupquote{~\textless{}output\_dir\textgreater{}}}}
Output directory for exported idats.  {[}default: ./tcga\_idats/{]}

\end{fulllineitems}



\section{download\_geo}
\label{\detokenize{index:pymethyl-preprocess-download-geo}}
Download geo methylation study idats and clinical info.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}preprocess download\PYGZus{}geo \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-preprocess-download\_geo command line option@\spxentry{pymethyl-preprocess-download\_geo command line option}!-g, --geo\_query \textless{}geo\_query\textgreater{}@\spxentry{-g, --geo\_query \textless{}geo\_query\textgreater{}}}\index{-g, --geo\_query \textless{}geo\_query\textgreater{}@\spxentry{-g, --geo\_query \textless{}geo\_query\textgreater{}}!pymethyl-preprocess-download\_geo command line option@\spxentry{pymethyl-preprocess-download\_geo command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-download-geo-g}}\pysigline{\sphinxbfcode{\sphinxupquote{-g}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}geo\_query}}\sphinxcode{\sphinxupquote{~\textless{}geo\_query\textgreater{}}}}
GEO study to query.  {[}default: {]}

\end{fulllineitems}

\index{pymethyl-preprocess-download\_geo command line option@\spxentry{pymethyl-preprocess-download\_geo command line option}!-o, --output\_dir \textless{}output\_dir\textgreater{}@\spxentry{-o, --output\_dir \textless{}output\_dir\textgreater{}}}\index{-o, --output\_dir \textless{}output\_dir\textgreater{}@\spxentry{-o, --output\_dir \textless{}output\_dir\textgreater{}}!pymethyl-preprocess-download\_geo command line option@\spxentry{pymethyl-preprocess-download\_geo command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-download-geo-o}}\pysigline{\sphinxbfcode{\sphinxupquote{-o}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}output\_dir}}\sphinxcode{\sphinxupquote{~\textless{}output\_dir\textgreater{}}}}
Output directory for exported idats.  {[}default: ./geo\_idats/{]}

\end{fulllineitems}



\section{download\_tcga}
\label{\detokenize{index:pymethyl-preprocess-download-tcga}}
Download all tcga 450k data.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}preprocess download\PYGZus{}tcga \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-preprocess-download\_tcga command line option@\spxentry{pymethyl-preprocess-download\_tcga command line option}!-o, --output\_dir \textless{}output\_dir\textgreater{}@\spxentry{-o, --output\_dir \textless{}output\_dir\textgreater{}}}\index{-o, --output\_dir \textless{}output\_dir\textgreater{}@\spxentry{-o, --output\_dir \textless{}output\_dir\textgreater{}}!pymethyl-preprocess-download\_tcga command line option@\spxentry{pymethyl-preprocess-download\_tcga command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-download-tcga-o}}\pysigline{\sphinxbfcode{\sphinxupquote{-o}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}output\_dir}}\sphinxcode{\sphinxupquote{~\textless{}output\_dir\textgreater{}}}}
Output directory for exported idats.  {[}default: ./tcga\_idats/{]}

\end{fulllineitems}



\section{feature\_select}
\label{\detokenize{index:pymethyl-preprocess-feature-select}}
Filter CpGs by taking x top CpGs with highest mean absolute deviation scores or via spectral feature selection.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}preprocess feature\PYGZus{}select \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-preprocess-feature\_select command line option@\spxentry{pymethyl-preprocess-feature\_select command line option}!-i, --input\_pkl \textless{}input\_pkl\textgreater{}@\spxentry{-i, --input\_pkl \textless{}input\_pkl\textgreater{}}}\index{-i, --input\_pkl \textless{}input\_pkl\textgreater{}@\spxentry{-i, --input\_pkl \textless{}input\_pkl\textgreater{}}!pymethyl-preprocess-feature\_select command line option@\spxentry{pymethyl-preprocess-feature\_select command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-feature-select-i}}\pysigline{\sphinxbfcode{\sphinxupquote{-i}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}input\_pkl}}\sphinxcode{\sphinxupquote{~\textless{}input\_pkl\textgreater{}}}}
Input database for beta and phenotype data.  {[}default: ./imputed\_outputs/methyl\_array.pkl{]}

\end{fulllineitems}

\index{pymethyl-preprocess-feature\_select command line option@\spxentry{pymethyl-preprocess-feature\_select command line option}!-o, --output\_pkl \textless{}output\_pkl\textgreater{}@\spxentry{-o, --output\_pkl \textless{}output\_pkl\textgreater{}}}\index{-o, --output\_pkl \textless{}output\_pkl\textgreater{}@\spxentry{-o, --output\_pkl \textless{}output\_pkl\textgreater{}}!pymethyl-preprocess-feature\_select command line option@\spxentry{pymethyl-preprocess-feature\_select command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-feature-select-o}}\pysigline{\sphinxbfcode{\sphinxupquote{-o}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}output\_pkl}}\sphinxcode{\sphinxupquote{~\textless{}output\_pkl\textgreater{}}}}
Output database for beta and phenotype data.  {[}default: ./final\_preprocessed/methyl\_array.pkl{]}

\end{fulllineitems}

\index{pymethyl-preprocess-feature\_select command line option@\spxentry{pymethyl-preprocess-feature\_select command line option}!-n, --n\_top\_cpgs \textless{}n\_top\_cpgs\textgreater{}@\spxentry{-n, --n\_top\_cpgs \textless{}n\_top\_cpgs\textgreater{}}}\index{-n, --n\_top\_cpgs \textless{}n\_top\_cpgs\textgreater{}@\spxentry{-n, --n\_top\_cpgs \textless{}n\_top\_cpgs\textgreater{}}!pymethyl-preprocess-feature\_select command line option@\spxentry{pymethyl-preprocess-feature\_select command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-feature-select-n}}\pysigline{\sphinxbfcode{\sphinxupquote{-n}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}n\_top\_cpgs}}\sphinxcode{\sphinxupquote{~\textless{}n\_top\_cpgs\textgreater{}}}}
Number cpgs to include with highest variance across population.  {[}default: 300000{]}

\end{fulllineitems}

\index{pymethyl-preprocess-feature\_select command line option@\spxentry{pymethyl-preprocess-feature\_select command line option}!-f, --feature\_selection\_method \textless{}feature\_selection\_method\textgreater{}@\spxentry{-f, --feature\_selection\_method \textless{}feature\_selection\_method\textgreater{}}}\index{-f, --feature\_selection\_method \textless{}feature\_selection\_method\textgreater{}@\spxentry{-f, --feature\_selection\_method \textless{}feature\_selection\_method\textgreater{}}!pymethyl-preprocess-feature\_select command line option@\spxentry{pymethyl-preprocess-feature\_select command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-feature-select-f}}\pysigline{\sphinxbfcode{\sphinxupquote{-f}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}feature\_selection\_method}}\sphinxcode{\sphinxupquote{~\textless{}feature\_selection\_method\textgreater{}}}}
\end{fulllineitems}

\index{pymethyl-preprocess-feature\_select command line option@\spxentry{pymethyl-preprocess-feature\_select command line option}!-mm, --metric \textless{}metric\textgreater{}@\spxentry{-mm, --metric \textless{}metric\textgreater{}}}\index{-mm, --metric \textless{}metric\textgreater{}@\spxentry{-mm, --metric \textless{}metric\textgreater{}}!pymethyl-preprocess-feature\_select command line option@\spxentry{pymethyl-preprocess-feature\_select command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-feature-select-mm}}\pysigline{\sphinxbfcode{\sphinxupquote{-mm}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}metric}}\sphinxcode{\sphinxupquote{~\textless{}metric\textgreater{}}}}
\end{fulllineitems}

\index{pymethyl-preprocess-feature\_select command line option@\spxentry{pymethyl-preprocess-feature\_select command line option}!-nn, --n\_neighbors \textless{}n\_neighbors\textgreater{}@\spxentry{-nn, --n\_neighbors \textless{}n\_neighbors\textgreater{}}}\index{-nn, --n\_neighbors \textless{}n\_neighbors\textgreater{}@\spxentry{-nn, --n\_neighbors \textless{}n\_neighbors\textgreater{}}!pymethyl-preprocess-feature\_select command line option@\spxentry{pymethyl-preprocess-feature\_select command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-feature-select-nn}}\pysigline{\sphinxbfcode{\sphinxupquote{-nn}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}n\_neighbors}}\sphinxcode{\sphinxupquote{~\textless{}n\_neighbors\textgreater{}}}}
Number neighbors for feature selection, default enacts rbf kernel.  {[}default: 0{]}

\end{fulllineitems}

\index{pymethyl-preprocess-feature\_select command line option@\spxentry{pymethyl-preprocess-feature\_select command line option}!-m, --mad\_top\_cpgs \textless{}mad\_top\_cpgs\textgreater{}@\spxentry{-m, --mad\_top\_cpgs \textless{}mad\_top\_cpgs\textgreater{}}}\index{-m, --mad\_top\_cpgs \textless{}mad\_top\_cpgs\textgreater{}@\spxentry{-m, --mad\_top\_cpgs \textless{}mad\_top\_cpgs\textgreater{}}!pymethyl-preprocess-feature\_select command line option@\spxentry{pymethyl-preprocess-feature\_select command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-feature-select-m}}\pysigline{\sphinxbfcode{\sphinxupquote{-m}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}mad\_top\_cpgs}}\sphinxcode{\sphinxupquote{~\textless{}mad\_top\_cpgs\textgreater{}}}}
Number cpgs to apply mad filtering first before more sophisticated feature selection. If 0 or primary feature selection is mad, no mad pre-filtering.  {[}default: 0{]}

\end{fulllineitems}



\section{get\_categorical\_distribution}
\label{\detokenize{index:pymethyl-preprocess-get-categorical-distribution}}
Get categorical distribution of columns of sample sheet.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}preprocess get\PYGZus{}categorical\PYGZus{}distribution \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-preprocess-get\_categorical\_distribution command line option@\spxentry{pymethyl-preprocess-get\_categorical\_distribution command line option}!-is, --formatted\_sample\_sheet \textless{}formatted\_sample\_sheet\textgreater{}@\spxentry{-is, --formatted\_sample\_sheet \textless{}formatted\_sample\_sheet\textgreater{}}}\index{-is, --formatted\_sample\_sheet \textless{}formatted\_sample\_sheet\textgreater{}@\spxentry{-is, --formatted\_sample\_sheet \textless{}formatted\_sample\_sheet\textgreater{}}!pymethyl-preprocess-get\_categorical\_distribution command line option@\spxentry{pymethyl-preprocess-get\_categorical\_distribution command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-get-categorical-distribution-is}}\pysigline{\sphinxbfcode{\sphinxupquote{-is}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}formatted\_sample\_sheet}}\sphinxcode{\sphinxupquote{~\textless{}formatted\_sample\_sheet\textgreater{}}}}
Clinical information downloaded from tcga/geo/custom, formatted using create\_sample\_sheet.  {[}default: ./tcga\_idats/minfiSheet.csv{]}

\end{fulllineitems}

\index{pymethyl-preprocess-get\_categorical\_distribution command line option@\spxentry{pymethyl-preprocess-get\_categorical\_distribution command line option}!-k, --key \textless{}key\textgreater{}@\spxentry{-k, --key \textless{}key\textgreater{}}}\index{-k, --key \textless{}key\textgreater{}@\spxentry{-k, --key \textless{}key\textgreater{}}!pymethyl-preprocess-get\_categorical\_distribution command line option@\spxentry{pymethyl-preprocess-get\_categorical\_distribution command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-get-categorical-distribution-k}}\pysigline{\sphinxbfcode{\sphinxupquote{-k}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}key}}\sphinxcode{\sphinxupquote{~\textless{}key\textgreater{}}}}
Column of csv to print statistics for.  {[}default: disease{]}

\end{fulllineitems}

\index{pymethyl-preprocess-get\_categorical\_distribution command line option@\spxentry{pymethyl-preprocess-get\_categorical\_distribution command line option}!-d, --disease\_only@\spxentry{-d, --disease\_only}}\index{-d, --disease\_only@\spxentry{-d, --disease\_only}!pymethyl-preprocess-get\_categorical\_distribution command line option@\spxentry{pymethyl-preprocess-get\_categorical\_distribution command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-get-categorical-distribution-d}}\pysigline{\sphinxbfcode{\sphinxupquote{-d}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}disease\_only}}\sphinxcode{\sphinxupquote{}}}
Only look at disease, or text before subtype\_delimiter.

\end{fulllineitems}

\index{pymethyl-preprocess-get\_categorical\_distribution command line option@\spxentry{pymethyl-preprocess-get\_categorical\_distribution command line option}!-sd, --subtype\_delimiter \textless{}subtype\_delimiter\textgreater{}@\spxentry{-sd, --subtype\_delimiter \textless{}subtype\_delimiter\textgreater{}}}\index{-sd, --subtype\_delimiter \textless{}subtype\_delimiter\textgreater{}@\spxentry{-sd, --subtype\_delimiter \textless{}subtype\_delimiter\textgreater{}}!pymethyl-preprocess-get\_categorical\_distribution command line option@\spxentry{pymethyl-preprocess-get\_categorical\_distribution command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-get-categorical-distribution-sd}}\pysigline{\sphinxbfcode{\sphinxupquote{-sd}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}subtype\_delimiter}}\sphinxcode{\sphinxupquote{~\textless{}subtype\_delimiter\textgreater{}}}}
Delimiter for disease extraction.  {[}default: ,{]}

\end{fulllineitems}



\section{imputation\_pipeline}
\label{\detokenize{index:pymethyl-preprocess-imputation-pipeline}}
Imputation of subtype or no subtype using various imputation methods.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}preprocess imputation\PYGZus{}pipeline \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-preprocess-imputation\_pipeline command line option@\spxentry{pymethyl-preprocess-imputation\_pipeline command line option}!-i, --input\_pkl \textless{}input\_pkl\textgreater{}@\spxentry{-i, --input\_pkl \textless{}input\_pkl\textgreater{}}}\index{-i, --input\_pkl \textless{}input\_pkl\textgreater{}@\spxentry{-i, --input\_pkl \textless{}input\_pkl\textgreater{}}!pymethyl-preprocess-imputation\_pipeline command line option@\spxentry{pymethyl-preprocess-imputation\_pipeline command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-imputation-pipeline-i}}\pysigline{\sphinxbfcode{\sphinxupquote{-i}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}input\_pkl}}\sphinxcode{\sphinxupquote{~\textless{}input\_pkl\textgreater{}}}}
Input database for beta and phenotype data.  {[}default: ./combined\_outputs/methyl\_array.pkl{]}

\end{fulllineitems}

\index{pymethyl-preprocess-imputation\_pipeline command line option@\spxentry{pymethyl-preprocess-imputation\_pipeline command line option}!-ss, --split\_by\_subtype@\spxentry{-ss, --split\_by\_subtype}}\index{-ss, --split\_by\_subtype@\spxentry{-ss, --split\_by\_subtype}!pymethyl-preprocess-imputation\_pipeline command line option@\spxentry{pymethyl-preprocess-imputation\_pipeline command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-imputation-pipeline-ss}}\pysigline{\sphinxbfcode{\sphinxupquote{-ss}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}split\_by\_subtype}}\sphinxcode{\sphinxupquote{}}}
Imputes CpGs by subtype before combining again.

\end{fulllineitems}

\index{pymethyl-preprocess-imputation\_pipeline command line option@\spxentry{pymethyl-preprocess-imputation\_pipeline command line option}!-m, --method \textless{}method\textgreater{}@\spxentry{-m, --method \textless{}method\textgreater{}}}\index{-m, --method \textless{}method\textgreater{}@\spxentry{-m, --method \textless{}method\textgreater{}}!pymethyl-preprocess-imputation\_pipeline command line option@\spxentry{pymethyl-preprocess-imputation\_pipeline command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-imputation-pipeline-m}}\pysigline{\sphinxbfcode{\sphinxupquote{-m}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}method}}\sphinxcode{\sphinxupquote{~\textless{}method\textgreater{}}}}
Method of imputation.  {[}default: KNN{]}

\end{fulllineitems}

\index{pymethyl-preprocess-imputation\_pipeline command line option@\spxentry{pymethyl-preprocess-imputation\_pipeline command line option}!-s, --solver \textless{}solver\textgreater{}@\spxentry{-s, --solver \textless{}solver\textgreater{}}}\index{-s, --solver \textless{}solver\textgreater{}@\spxentry{-s, --solver \textless{}solver\textgreater{}}!pymethyl-preprocess-imputation\_pipeline command line option@\spxentry{pymethyl-preprocess-imputation\_pipeline command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-imputation-pipeline-s}}\pysigline{\sphinxbfcode{\sphinxupquote{-s}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}solver}}\sphinxcode{\sphinxupquote{~\textless{}solver\textgreater{}}}}
Imputation library.  {[}default: fancyimpute{]}

\end{fulllineitems}

\index{pymethyl-preprocess-imputation\_pipeline command line option@\spxentry{pymethyl-preprocess-imputation\_pipeline command line option}!-k, --n\_neighbors \textless{}n\_neighbors\textgreater{}@\spxentry{-k, --n\_neighbors \textless{}n\_neighbors\textgreater{}}}\index{-k, --n\_neighbors \textless{}n\_neighbors\textgreater{}@\spxentry{-k, --n\_neighbors \textless{}n\_neighbors\textgreater{}}!pymethyl-preprocess-imputation\_pipeline command line option@\spxentry{pymethyl-preprocess-imputation\_pipeline command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-imputation-pipeline-k}}\pysigline{\sphinxbfcode{\sphinxupquote{-k}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}n\_neighbors}}\sphinxcode{\sphinxupquote{~\textless{}n\_neighbors\textgreater{}}}}
Number neighbors for imputation if using KNN.  {[}default: 5{]}

\end{fulllineitems}

\index{pymethyl-preprocess-imputation\_pipeline command line option@\spxentry{pymethyl-preprocess-imputation\_pipeline command line option}!-r, --orientation \textless{}orientation\textgreater{}@\spxentry{-r, --orientation \textless{}orientation\textgreater{}}}\index{-r, --orientation \textless{}orientation\textgreater{}@\spxentry{-r, --orientation \textless{}orientation\textgreater{}}!pymethyl-preprocess-imputation\_pipeline command line option@\spxentry{pymethyl-preprocess-imputation\_pipeline command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-imputation-pipeline-r}}\pysigline{\sphinxbfcode{\sphinxupquote{-r}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}orientation}}\sphinxcode{\sphinxupquote{~\textless{}orientation\textgreater{}}}}
Impute CpGs or samples.  {[}default: Samples{]}

\end{fulllineitems}

\index{pymethyl-preprocess-imputation\_pipeline command line option@\spxentry{pymethyl-preprocess-imputation\_pipeline command line option}!-o, --output\_pkl \textless{}output\_pkl\textgreater{}@\spxentry{-o, --output\_pkl \textless{}output\_pkl\textgreater{}}}\index{-o, --output\_pkl \textless{}output\_pkl\textgreater{}@\spxentry{-o, --output\_pkl \textless{}output\_pkl\textgreater{}}!pymethyl-preprocess-imputation\_pipeline command line option@\spxentry{pymethyl-preprocess-imputation\_pipeline command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-imputation-pipeline-o}}\pysigline{\sphinxbfcode{\sphinxupquote{-o}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}output\_pkl}}\sphinxcode{\sphinxupquote{~\textless{}output\_pkl\textgreater{}}}}
Output database for beta and phenotype data.  {[}default: ./imputed\_outputs/methyl\_array.pkl{]}

\end{fulllineitems}

\index{pymethyl-preprocess-imputation\_pipeline command line option@\spxentry{pymethyl-preprocess-imputation\_pipeline command line option}!-n, --n\_top\_cpgs \textless{}n\_top\_cpgs\textgreater{}@\spxentry{-n, --n\_top\_cpgs \textless{}n\_top\_cpgs\textgreater{}}}\index{-n, --n\_top\_cpgs \textless{}n\_top\_cpgs\textgreater{}@\spxentry{-n, --n\_top\_cpgs \textless{}n\_top\_cpgs\textgreater{}}!pymethyl-preprocess-imputation\_pipeline command line option@\spxentry{pymethyl-preprocess-imputation\_pipeline command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-imputation-pipeline-n}}\pysigline{\sphinxbfcode{\sphinxupquote{-n}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}n\_top\_cpgs}}\sphinxcode{\sphinxupquote{~\textless{}n\_top\_cpgs\textgreater{}}}}
Number cpgs to include with highest variance across population. Greater than 0 allows for mad filtering during imputation to skip mad step.  {[}default: 0{]}

\end{fulllineitems}

\index{pymethyl-preprocess-imputation\_pipeline command line option@\spxentry{pymethyl-preprocess-imputation\_pipeline command line option}!-f, --feature\_selection\_method \textless{}feature\_selection\_method\textgreater{}@\spxentry{-f, --feature\_selection\_method \textless{}feature\_selection\_method\textgreater{}}}\index{-f, --feature\_selection\_method \textless{}feature\_selection\_method\textgreater{}@\spxentry{-f, --feature\_selection\_method \textless{}feature\_selection\_method\textgreater{}}!pymethyl-preprocess-imputation\_pipeline command line option@\spxentry{pymethyl-preprocess-imputation\_pipeline command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-imputation-pipeline-f}}\pysigline{\sphinxbfcode{\sphinxupquote{-f}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}feature\_selection\_method}}\sphinxcode{\sphinxupquote{~\textless{}feature\_selection\_method\textgreater{}}}}
\end{fulllineitems}

\index{pymethyl-preprocess-imputation\_pipeline command line option@\spxentry{pymethyl-preprocess-imputation\_pipeline command line option}!-mm, --metric \textless{}metric\textgreater{}@\spxentry{-mm, --metric \textless{}metric\textgreater{}}}\index{-mm, --metric \textless{}metric\textgreater{}@\spxentry{-mm, --metric \textless{}metric\textgreater{}}!pymethyl-preprocess-imputation\_pipeline command line option@\spxentry{pymethyl-preprocess-imputation\_pipeline command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-imputation-pipeline-mm}}\pysigline{\sphinxbfcode{\sphinxupquote{-mm}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}metric}}\sphinxcode{\sphinxupquote{~\textless{}metric\textgreater{}}}}
\end{fulllineitems}

\index{pymethyl-preprocess-imputation\_pipeline command line option@\spxentry{pymethyl-preprocess-imputation\_pipeline command line option}!-nfs, --n\_neighbors\_fs \textless{}n\_neighbors\_fs\textgreater{}@\spxentry{-nfs, --n\_neighbors\_fs \textless{}n\_neighbors\_fs\textgreater{}}}\index{-nfs, --n\_neighbors\_fs \textless{}n\_neighbors\_fs\textgreater{}@\spxentry{-nfs, --n\_neighbors\_fs \textless{}n\_neighbors\_fs\textgreater{}}!pymethyl-preprocess-imputation\_pipeline command line option@\spxentry{pymethyl-preprocess-imputation\_pipeline command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-imputation-pipeline-nfs}}\pysigline{\sphinxbfcode{\sphinxupquote{-nfs}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}n\_neighbors\_fs}}\sphinxcode{\sphinxupquote{~\textless{}n\_neighbors\_fs\textgreater{}}}}
Number neighbors for feature selection, default enacts rbf kernel.  {[}default: 0{]}

\end{fulllineitems}

\index{pymethyl-preprocess-imputation\_pipeline command line option@\spxentry{pymethyl-preprocess-imputation\_pipeline command line option}!-d, --disease\_only@\spxentry{-d, --disease\_only}}\index{-d, --disease\_only@\spxentry{-d, --disease\_only}!pymethyl-preprocess-imputation\_pipeline command line option@\spxentry{pymethyl-preprocess-imputation\_pipeline command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-imputation-pipeline-d}}\pysigline{\sphinxbfcode{\sphinxupquote{-d}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}disease\_only}}\sphinxcode{\sphinxupquote{}}}
Only look at disease, or text before subtype\_delimiter.

\end{fulllineitems}

\index{pymethyl-preprocess-imputation\_pipeline command line option@\spxentry{pymethyl-preprocess-imputation\_pipeline command line option}!-sd, --subtype\_delimiter \textless{}subtype\_delimiter\textgreater{}@\spxentry{-sd, --subtype\_delimiter \textless{}subtype\_delimiter\textgreater{}}}\index{-sd, --subtype\_delimiter \textless{}subtype\_delimiter\textgreater{}@\spxentry{-sd, --subtype\_delimiter \textless{}subtype\_delimiter\textgreater{}}!pymethyl-preprocess-imputation\_pipeline command line option@\spxentry{pymethyl-preprocess-imputation\_pipeline command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-imputation-pipeline-sd}}\pysigline{\sphinxbfcode{\sphinxupquote{-sd}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}subtype\_delimiter}}\sphinxcode{\sphinxupquote{~\textless{}subtype\_delimiter\textgreater{}}}}
Delimiter for disease extraction.  {[}default: ,{]}

\end{fulllineitems}

\index{pymethyl-preprocess-imputation\_pipeline command line option@\spxentry{pymethyl-preprocess-imputation\_pipeline command line option}!-st, --sample\_threshold \textless{}sample\_threshold\textgreater{}@\spxentry{-st, --sample\_threshold \textless{}sample\_threshold\textgreater{}}}\index{-st, --sample\_threshold \textless{}sample\_threshold\textgreater{}@\spxentry{-st, --sample\_threshold \textless{}sample\_threshold\textgreater{}}!pymethyl-preprocess-imputation\_pipeline command line option@\spxentry{pymethyl-preprocess-imputation\_pipeline command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-imputation-pipeline-st}}\pysigline{\sphinxbfcode{\sphinxupquote{-st}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}sample\_threshold}}\sphinxcode{\sphinxupquote{~\textless{}sample\_threshold\textgreater{}}}}
Value between 0 and 1 for NaN removal. If samples has sample\_threshold proportion of cpgs missing, then remove sample. Set to -1 to not remove samples.  {[}default: -1.0{]}

\end{fulllineitems}

\index{pymethyl-preprocess-imputation\_pipeline command line option@\spxentry{pymethyl-preprocess-imputation\_pipeline command line option}!-ct, --cpg\_threshold \textless{}cpg\_threshold\textgreater{}@\spxentry{-ct, --cpg\_threshold \textless{}cpg\_threshold\textgreater{}}}\index{-ct, --cpg\_threshold \textless{}cpg\_threshold\textgreater{}@\spxentry{-ct, --cpg\_threshold \textless{}cpg\_threshold\textgreater{}}!pymethyl-preprocess-imputation\_pipeline command line option@\spxentry{pymethyl-preprocess-imputation\_pipeline command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-imputation-pipeline-ct}}\pysigline{\sphinxbfcode{\sphinxupquote{-ct}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}cpg\_threshold}}\sphinxcode{\sphinxupquote{~\textless{}cpg\_threshold\textgreater{}}}}
Value between 0 and 1 for NaN removal. If cpgs has cpg\_threshold proportion of samples missing, then remove cpg. Set to -1 to not remove samples.  {[}default: -1.0{]}

\end{fulllineitems}



\section{meffil\_encode}
\label{\detokenize{index:pymethyl-preprocess-meffil-encode}}
Reformat file for meffil input.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}preprocess meffil\PYGZus{}encode \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-preprocess-meffil\_encode command line option@\spxentry{pymethyl-preprocess-meffil\_encode command line option}!-is, --input\_sample\_sheet \textless{}input\_sample\_sheet\textgreater{}@\spxentry{-is, --input\_sample\_sheet \textless{}input\_sample\_sheet\textgreater{}}}\index{-is, --input\_sample\_sheet \textless{}input\_sample\_sheet\textgreater{}@\spxentry{-is, --input\_sample\_sheet \textless{}input\_sample\_sheet\textgreater{}}!pymethyl-preprocess-meffil\_encode command line option@\spxentry{pymethyl-preprocess-meffil\_encode command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-meffil-encode-is}}\pysigline{\sphinxbfcode{\sphinxupquote{-is}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}input\_sample\_sheet}}\sphinxcode{\sphinxupquote{~\textless{}input\_sample\_sheet\textgreater{}}}}
CSV for minfi input.  {[}default: ./tcga\_idats/minfiSheet.csv{]}

\end{fulllineitems}

\index{pymethyl-preprocess-meffil\_encode command line option@\spxentry{pymethyl-preprocess-meffil\_encode command line option}!-os, --output\_sample\_sheet \textless{}output\_sample\_sheet\textgreater{}@\spxentry{-os, --output\_sample\_sheet \textless{}output\_sample\_sheet\textgreater{}}}\index{-os, --output\_sample\_sheet \textless{}output\_sample\_sheet\textgreater{}@\spxentry{-os, --output\_sample\_sheet \textless{}output\_sample\_sheet\textgreater{}}!pymethyl-preprocess-meffil\_encode command line option@\spxentry{pymethyl-preprocess-meffil\_encode command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-meffil-encode-os}}\pysigline{\sphinxbfcode{\sphinxupquote{-os}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}output\_sample\_sheet}}\sphinxcode{\sphinxupquote{~\textless{}output\_sample\_sheet\textgreater{}}}}
CSV for minfi input.  {[}default: ./tcga\_idats/minfiSheet.csv{]}

\end{fulllineitems}



\section{merge\_sample\_sheets}
\label{\detokenize{index:pymethyl-preprocess-merge-sample-sheets}}
Merge two sample files for more fields for minfi+ input.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}preprocess merge\PYGZus{}sample\PYGZus{}sheets \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-preprocess-merge\_sample\_sheets command line option@\spxentry{pymethyl-preprocess-merge\_sample\_sheets command line option}!-s1, --sample\_sheet1 \textless{}sample\_sheet1\textgreater{}@\spxentry{-s1, --sample\_sheet1 \textless{}sample\_sheet1\textgreater{}}}\index{-s1, --sample\_sheet1 \textless{}sample\_sheet1\textgreater{}@\spxentry{-s1, --sample\_sheet1 \textless{}sample\_sheet1\textgreater{}}!pymethyl-preprocess-merge\_sample\_sheets command line option@\spxentry{pymethyl-preprocess-merge\_sample\_sheets command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-merge-sample-sheets-s1}}\pysigline{\sphinxbfcode{\sphinxupquote{-s1}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}sample\_sheet1}}\sphinxcode{\sphinxupquote{~\textless{}sample\_sheet1\textgreater{}}}}
Clinical information downloaded from tcga/geo/custom, formatted using create\_sample\_sheet.  {[}default: ./tcga\_idats/clinical\_info1.csv{]}

\end{fulllineitems}

\index{pymethyl-preprocess-merge\_sample\_sheets command line option@\spxentry{pymethyl-preprocess-merge\_sample\_sheets command line option}!-s2, --sample\_sheet2 \textless{}sample\_sheet2\textgreater{}@\spxentry{-s2, --sample\_sheet2 \textless{}sample\_sheet2\textgreater{}}}\index{-s2, --sample\_sheet2 \textless{}sample\_sheet2\textgreater{}@\spxentry{-s2, --sample\_sheet2 \textless{}sample\_sheet2\textgreater{}}!pymethyl-preprocess-merge\_sample\_sheets command line option@\spxentry{pymethyl-preprocess-merge\_sample\_sheets command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-merge-sample-sheets-s2}}\pysigline{\sphinxbfcode{\sphinxupquote{-s2}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}sample\_sheet2}}\sphinxcode{\sphinxupquote{~\textless{}sample\_sheet2\textgreater{}}}}
Clinical information downloaded from tcga/geo/custom, formatted using create\_sample\_sheet.  {[}default: ./tcga\_idats/clinical\_info2.csv{]}

\end{fulllineitems}

\index{pymethyl-preprocess-merge\_sample\_sheets command line option@\spxentry{pymethyl-preprocess-merge\_sample\_sheets command line option}!-os, --output\_sample\_sheet \textless{}output\_sample\_sheet\textgreater{}@\spxentry{-os, --output\_sample\_sheet \textless{}output\_sample\_sheet\textgreater{}}}\index{-os, --output\_sample\_sheet \textless{}output\_sample\_sheet\textgreater{}@\spxentry{-os, --output\_sample\_sheet \textless{}output\_sample\_sheet\textgreater{}}!pymethyl-preprocess-merge\_sample\_sheets command line option@\spxentry{pymethyl-preprocess-merge\_sample\_sheets command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-merge-sample-sheets-os}}\pysigline{\sphinxbfcode{\sphinxupquote{-os}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}output\_sample\_sheet}}\sphinxcode{\sphinxupquote{~\textless{}output\_sample\_sheet\textgreater{}}}}
CSV for minfi input.  {[}default: ./tcga\_idats/minfiSheet.csv{]}

\end{fulllineitems}

\index{pymethyl-preprocess-merge\_sample\_sheets command line option@\spxentry{pymethyl-preprocess-merge\_sample\_sheets command line option}!-d, --second\_sheet\_disease@\spxentry{-d, --second\_sheet\_disease}}\index{-d, --second\_sheet\_disease@\spxentry{-d, --second\_sheet\_disease}!pymethyl-preprocess-merge\_sample\_sheets command line option@\spxentry{pymethyl-preprocess-merge\_sample\_sheets command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-merge-sample-sheets-d}}\pysigline{\sphinxbfcode{\sphinxupquote{-d}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}second\_sheet\_disease}}\sphinxcode{\sphinxupquote{}}}
Use second sheet’s disease column.

\end{fulllineitems}

\index{pymethyl-preprocess-merge\_sample\_sheets command line option@\spxentry{pymethyl-preprocess-merge\_sample\_sheets command line option}!-nd, --no\_disease\_merge@\spxentry{-nd, --no\_disease\_merge}}\index{-nd, --no\_disease\_merge@\spxentry{-nd, --no\_disease\_merge}!pymethyl-preprocess-merge\_sample\_sheets command line option@\spxentry{pymethyl-preprocess-merge\_sample\_sheets command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-merge-sample-sheets-nd}}\pysigline{\sphinxbfcode{\sphinxupquote{-nd}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}no\_disease\_merge}}\sphinxcode{\sphinxupquote{}}}
Don’t merge disease columns.

\end{fulllineitems}



\section{na\_report}
\label{\detokenize{index:pymethyl-preprocess-na-report}}
Print proportion of missing values throughout dataset.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}preprocess na\PYGZus{}report \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-preprocess-na\_report command line option@\spxentry{pymethyl-preprocess-na\_report command line option}!-i, --input\_pkl \textless{}input\_pkl\textgreater{}@\spxentry{-i, --input\_pkl \textless{}input\_pkl\textgreater{}}}\index{-i, --input\_pkl \textless{}input\_pkl\textgreater{}@\spxentry{-i, --input\_pkl \textless{}input\_pkl\textgreater{}}!pymethyl-preprocess-na\_report command line option@\spxentry{pymethyl-preprocess-na\_report command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-na-report-i}}\pysigline{\sphinxbfcode{\sphinxupquote{-i}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}input\_pkl}}\sphinxcode{\sphinxupquote{~\textless{}input\_pkl\textgreater{}}}}
Input database for beta and phenotype data.  {[}default: ./preprocess\_outputs/methyl\_array.pkl{]}

\end{fulllineitems}

\index{pymethyl-preprocess-na\_report command line option@\spxentry{pymethyl-preprocess-na\_report command line option}!-o, --output\_dir \textless{}output\_dir\textgreater{}@\spxentry{-o, --output\_dir \textless{}output\_dir\textgreater{}}}\index{-o, --output\_dir \textless{}output\_dir\textgreater{}@\spxentry{-o, --output\_dir \textless{}output\_dir\textgreater{}}!pymethyl-preprocess-na\_report command line option@\spxentry{pymethyl-preprocess-na\_report command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-na-report-o}}\pysigline{\sphinxbfcode{\sphinxupquote{-o}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}output\_dir}}\sphinxcode{\sphinxupquote{~\textless{}output\_dir\textgreater{}}}}
Output database for na report.  {[}default: ./na\_report/{]}

\end{fulllineitems}

\index{pymethyl-preprocess-na\_report command line option@\spxentry{pymethyl-preprocess-na\_report command line option}!-r, --head\_directory@\spxentry{-r, --head\_directory}}\index{-r, --head\_directory@\spxentry{-r, --head\_directory}!pymethyl-preprocess-na\_report command line option@\spxentry{pymethyl-preprocess-na\_report command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-na-report-r}}\pysigline{\sphinxbfcode{\sphinxupquote{-r}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}head\_directory}}\sphinxcode{\sphinxupquote{}}}
-i option becomes directory, and searches there for multiple input pickles.

\end{fulllineitems}



\section{preprocess\_pipeline}
\label{\detokenize{index:pymethyl-preprocess-preprocess-pipeline}}
Perform preprocessing of idats using enmix or meffil.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}preprocess preprocess\PYGZus{}pipeline \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-preprocess-preprocess\_pipeline command line option@\spxentry{pymethyl-preprocess-preprocess\_pipeline command line option}!-i, --idat\_dir \textless{}idat\_dir\textgreater{}@\spxentry{-i, --idat\_dir \textless{}idat\_dir\textgreater{}}}\index{-i, --idat\_dir \textless{}idat\_dir\textgreater{}@\spxentry{-i, --idat\_dir \textless{}idat\_dir\textgreater{}}!pymethyl-preprocess-preprocess\_pipeline command line option@\spxentry{pymethyl-preprocess-preprocess\_pipeline command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-preprocess-pipeline-i}}\pysigline{\sphinxbfcode{\sphinxupquote{-i}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}idat\_dir}}\sphinxcode{\sphinxupquote{~\textless{}idat\_dir\textgreater{}}}}
Idat dir for one sample sheet, alternatively can be your phenotype sample sheet.  {[}default: ./tcga\_idats/{]}

\end{fulllineitems}

\index{pymethyl-preprocess-preprocess\_pipeline command line option@\spxentry{pymethyl-preprocess-preprocess\_pipeline command line option}!-n, --n\_cores \textless{}n\_cores\textgreater{}@\spxentry{-n, --n\_cores \textless{}n\_cores\textgreater{}}}\index{-n, --n\_cores \textless{}n\_cores\textgreater{}@\spxentry{-n, --n\_cores \textless{}n\_cores\textgreater{}}!pymethyl-preprocess-preprocess\_pipeline command line option@\spxentry{pymethyl-preprocess-preprocess\_pipeline command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-preprocess-pipeline-n}}\pysigline{\sphinxbfcode{\sphinxupquote{-n}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}n\_cores}}\sphinxcode{\sphinxupquote{~\textless{}n\_cores\textgreater{}}}}
Number cores to use for preprocessing.  {[}default: 6{]}

\end{fulllineitems}

\index{pymethyl-preprocess-preprocess\_pipeline command line option@\spxentry{pymethyl-preprocess-preprocess\_pipeline command line option}!-o, --output\_pkl \textless{}output\_pkl\textgreater{}@\spxentry{-o, --output\_pkl \textless{}output\_pkl\textgreater{}}}\index{-o, --output\_pkl \textless{}output\_pkl\textgreater{}@\spxentry{-o, --output\_pkl \textless{}output\_pkl\textgreater{}}!pymethyl-preprocess-preprocess\_pipeline command line option@\spxentry{pymethyl-preprocess-preprocess\_pipeline command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-preprocess-pipeline-o}}\pysigline{\sphinxbfcode{\sphinxupquote{-o}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}output\_pkl}}\sphinxcode{\sphinxupquote{~\textless{}output\_pkl\textgreater{}}}}
Output database for beta and phenotype data.  {[}default: ./preprocess\_outputs/methyl\_array.pkl{]}

\end{fulllineitems}

\index{pymethyl-preprocess-preprocess\_pipeline command line option@\spxentry{pymethyl-preprocess-preprocess\_pipeline command line option}!-m, --meffil@\spxentry{-m, --meffil}}\index{-m, --meffil@\spxentry{-m, --meffil}!pymethyl-preprocess-preprocess\_pipeline command line option@\spxentry{pymethyl-preprocess-preprocess\_pipeline command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-preprocess-pipeline-m}}\pysigline{\sphinxbfcode{\sphinxupquote{-m}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}meffil}}\sphinxcode{\sphinxupquote{}}}
Preprocess using meffil.

\end{fulllineitems}

\index{pymethyl-preprocess-preprocess\_pipeline command line option@\spxentry{pymethyl-preprocess-preprocess\_pipeline command line option}!-pc, --n\_pcs \textless{}n\_pcs\textgreater{}@\spxentry{-pc, --n\_pcs \textless{}n\_pcs\textgreater{}}}\index{-pc, --n\_pcs \textless{}n\_pcs\textgreater{}@\spxentry{-pc, --n\_pcs \textless{}n\_pcs\textgreater{}}!pymethyl-preprocess-preprocess\_pipeline command line option@\spxentry{pymethyl-preprocess-preprocess\_pipeline command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-preprocess-pipeline-pc}}\pysigline{\sphinxbfcode{\sphinxupquote{-pc}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}n\_pcs}}\sphinxcode{\sphinxupquote{~\textless{}n\_pcs\textgreater{}}}}
For meffil, number of principal components for functional normalization. If set to -1, then PCs are selected using elbow method.  {[}default: -1{]}

\end{fulllineitems}

\index{pymethyl-preprocess-preprocess\_pipeline command line option@\spxentry{pymethyl-preprocess-preprocess\_pipeline command line option}!-p, --pipeline \textless{}pipeline\textgreater{}@\spxentry{-p, --pipeline \textless{}pipeline\textgreater{}}}\index{-p, --pipeline \textless{}pipeline\textgreater{}@\spxentry{-p, --pipeline \textless{}pipeline\textgreater{}}!pymethyl-preprocess-preprocess\_pipeline command line option@\spxentry{pymethyl-preprocess-preprocess\_pipeline command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-preprocess-pipeline-p}}\pysigline{\sphinxbfcode{\sphinxupquote{-p}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}pipeline}}\sphinxcode{\sphinxupquote{~\textless{}pipeline\textgreater{}}}}
If not meffil, preprocess using minfi or enmix.  {[}default: enmix{]}

\end{fulllineitems}

\index{pymethyl-preprocess-preprocess\_pipeline command line option@\spxentry{pymethyl-preprocess-preprocess\_pipeline command line option}!-noob, --noob\_norm@\spxentry{-noob, --noob\_norm}}\index{-noob, --noob\_norm@\spxentry{-noob, --noob\_norm}!pymethyl-preprocess-preprocess\_pipeline command line option@\spxentry{pymethyl-preprocess-preprocess\_pipeline command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-preprocess-pipeline-noob}}\pysigline{\sphinxbfcode{\sphinxupquote{-noob}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}noob\_norm}}\sphinxcode{\sphinxupquote{}}}
Run noob normalization of minfi selected.

\end{fulllineitems}

\index{pymethyl-preprocess-preprocess\_pipeline command line option@\spxentry{pymethyl-preprocess-preprocess\_pipeline command line option}!-u, --use\_cache@\spxentry{-u, --use\_cache}}\index{-u, --use\_cache@\spxentry{-u, --use\_cache}!pymethyl-preprocess-preprocess\_pipeline command line option@\spxentry{pymethyl-preprocess-preprocess\_pipeline command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-preprocess-pipeline-u}}\pysigline{\sphinxbfcode{\sphinxupquote{-u}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}use\_cache}}\sphinxcode{\sphinxupquote{}}}
If this is selected, loads qc results rather than running qc again and update with new qc parameters. Only works for meffil selection. Minfi and enmix just loads RG Set.

\end{fulllineitems}

\index{pymethyl-preprocess-preprocess\_pipeline command line option@\spxentry{pymethyl-preprocess-preprocess\_pipeline command line option}!-qc, --qc\_only@\spxentry{-qc, --qc\_only}}\index{-qc, --qc\_only@\spxentry{-qc, --qc\_only}!pymethyl-preprocess-preprocess\_pipeline command line option@\spxentry{pymethyl-preprocess-preprocess\_pipeline command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-preprocess-pipeline-qc}}\pysigline{\sphinxbfcode{\sphinxupquote{-qc}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}qc\_only}}\sphinxcode{\sphinxupquote{}}}
Only perform QC for meffil pipeline, caches results into rds file for loading again, only works if use\_cache is false. Minfi and enmix just saves the RGSet before preprocessing.

\end{fulllineitems}

\index{pymethyl-preprocess-preprocess\_pipeline command line option@\spxentry{pymethyl-preprocess-preprocess\_pipeline command line option}!-bns, --p\_beadnum\_samples \textless{}p\_beadnum\_samples\textgreater{}@\spxentry{-bns, --p\_beadnum\_samples \textless{}p\_beadnum\_samples\textgreater{}}}\index{-bns, --p\_beadnum\_samples \textless{}p\_beadnum\_samples\textgreater{}@\spxentry{-bns, --p\_beadnum\_samples \textless{}p\_beadnum\_samples\textgreater{}}!pymethyl-preprocess-preprocess\_pipeline command line option@\spxentry{pymethyl-preprocess-preprocess\_pipeline command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-preprocess-pipeline-bns}}\pysigline{\sphinxbfcode{\sphinxupquote{-bns}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}p\_beadnum\_samples}}\sphinxcode{\sphinxupquote{~\textless{}p\_beadnum\_samples\textgreater{}}}}
From meffil documentation, “fraction of probes that failed the threshold of 3 beads”.  {[}default: 0.05{]}

\end{fulllineitems}

\index{pymethyl-preprocess-preprocess\_pipeline command line option@\spxentry{pymethyl-preprocess-preprocess\_pipeline command line option}!-pds, --p\_detection\_samples \textless{}p\_detection\_samples\textgreater{}@\spxentry{-pds, --p\_detection\_samples \textless{}p\_detection\_samples\textgreater{}}}\index{-pds, --p\_detection\_samples \textless{}p\_detection\_samples\textgreater{}@\spxentry{-pds, --p\_detection\_samples \textless{}p\_detection\_samples\textgreater{}}!pymethyl-preprocess-preprocess\_pipeline command line option@\spxentry{pymethyl-preprocess-preprocess\_pipeline command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-preprocess-pipeline-pds}}\pysigline{\sphinxbfcode{\sphinxupquote{-pds}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}p\_detection\_samples}}\sphinxcode{\sphinxupquote{~\textless{}p\_detection\_samples\textgreater{}}}}
From meffil documentation, “fraction of probes that failed a detection.pvalue threshold of 0.01”.  {[}default: 0.05{]}

\end{fulllineitems}

\index{pymethyl-preprocess-preprocess\_pipeline command line option@\spxentry{pymethyl-preprocess-preprocess\_pipeline command line option}!-bnc, --p\_beadnum\_cpgs \textless{}p\_beadnum\_cpgs\textgreater{}@\spxentry{-bnc, --p\_beadnum\_cpgs \textless{}p\_beadnum\_cpgs\textgreater{}}}\index{-bnc, --p\_beadnum\_cpgs \textless{}p\_beadnum\_cpgs\textgreater{}@\spxentry{-bnc, --p\_beadnum\_cpgs \textless{}p\_beadnum\_cpgs\textgreater{}}!pymethyl-preprocess-preprocess\_pipeline command line option@\spxentry{pymethyl-preprocess-preprocess\_pipeline command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-preprocess-pipeline-bnc}}\pysigline{\sphinxbfcode{\sphinxupquote{-bnc}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}p\_beadnum\_cpgs}}\sphinxcode{\sphinxupquote{~\textless{}p\_beadnum\_cpgs\textgreater{}}}}
From meffil documentation, “fraction of samples that failed the threshold of 3 beads”.  {[}default: 0.05{]}

\end{fulllineitems}

\index{pymethyl-preprocess-preprocess\_pipeline command line option@\spxentry{pymethyl-preprocess-preprocess\_pipeline command line option}!-pdc, --p\_detection\_cpgs \textless{}p\_detection\_cpgs\textgreater{}@\spxentry{-pdc, --p\_detection\_cpgs \textless{}p\_detection\_cpgs\textgreater{}}}\index{-pdc, --p\_detection\_cpgs \textless{}p\_detection\_cpgs\textgreater{}@\spxentry{-pdc, --p\_detection\_cpgs \textless{}p\_detection\_cpgs\textgreater{}}!pymethyl-preprocess-preprocess\_pipeline command line option@\spxentry{pymethyl-preprocess-preprocess\_pipeline command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-preprocess-pipeline-pdc}}\pysigline{\sphinxbfcode{\sphinxupquote{-pdc}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}p\_detection\_cpgs}}\sphinxcode{\sphinxupquote{~\textless{}p\_detection\_cpgs\textgreater{}}}}
From meffil documentation, “fraction of samples that failed a detection.pvalue threshold of 0.01”.  {[}default: 0.05{]}

\end{fulllineitems}

\index{pymethyl-preprocess-preprocess\_pipeline command line option@\spxentry{pymethyl-preprocess-preprocess\_pipeline command line option}!-sc, --sex\_cutoff \textless{}sex\_cutoff\textgreater{}@\spxentry{-sc, --sex\_cutoff \textless{}sex\_cutoff\textgreater{}}}\index{-sc, --sex\_cutoff \textless{}sex\_cutoff\textgreater{}@\spxentry{-sc, --sex\_cutoff \textless{}sex\_cutoff\textgreater{}}!pymethyl-preprocess-preprocess\_pipeline command line option@\spxentry{pymethyl-preprocess-preprocess\_pipeline command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-preprocess-pipeline-sc}}\pysigline{\sphinxbfcode{\sphinxupquote{-sc}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}sex\_cutoff}}\sphinxcode{\sphinxupquote{~\textless{}sex\_cutoff\textgreater{}}}}
From meffil documentation, “difference of total median intensity for Y chromosome probes and X chromosome probes”.  {[}default: -2{]}

\end{fulllineitems}

\index{pymethyl-preprocess-preprocess\_pipeline command line option@\spxentry{pymethyl-preprocess-preprocess\_pipeline command line option}!-sd, --sex\_sd \textless{}sex\_sd\textgreater{}@\spxentry{-sd, --sex\_sd \textless{}sex\_sd\textgreater{}}}\index{-sd, --sex\_sd \textless{}sex\_sd\textgreater{}@\spxentry{-sd, --sex\_sd \textless{}sex\_sd\textgreater{}}!pymethyl-preprocess-preprocess\_pipeline command line option@\spxentry{pymethyl-preprocess-preprocess\_pipeline command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-preprocess-pipeline-sd}}\pysigline{\sphinxbfcode{\sphinxupquote{-sd}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}sex\_sd}}\sphinxcode{\sphinxupquote{~\textless{}sex\_sd\textgreater{}}}}
From meffil documentation, “sex detection outliers if outside this range”.  {[}default: 5{]}

\end{fulllineitems}



\section{remove\_diseases}
\label{\detokenize{index:pymethyl-preprocess-remove-diseases}}
Exclude diseases from study by count number or exclusion list.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}preprocess remove\PYGZus{}diseases \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-preprocess-remove\_diseases command line option@\spxentry{pymethyl-preprocess-remove\_diseases command line option}!-is, --formatted\_sample\_sheet \textless{}formatted\_sample\_sheet\textgreater{}@\spxentry{-is, --formatted\_sample\_sheet \textless{}formatted\_sample\_sheet\textgreater{}}}\index{-is, --formatted\_sample\_sheet \textless{}formatted\_sample\_sheet\textgreater{}@\spxentry{-is, --formatted\_sample\_sheet \textless{}formatted\_sample\_sheet\textgreater{}}!pymethyl-preprocess-remove\_diseases command line option@\spxentry{pymethyl-preprocess-remove\_diseases command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-remove-diseases-is}}\pysigline{\sphinxbfcode{\sphinxupquote{-is}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}formatted\_sample\_sheet}}\sphinxcode{\sphinxupquote{~\textless{}formatted\_sample\_sheet\textgreater{}}}}
Clinical information downloaded from tcga/geo/custom, formatted using create\_sample\_sheet.  {[}default: ./tcga\_idats/clinical\_info.csv{]}

\end{fulllineitems}

\index{pymethyl-preprocess-remove\_diseases command line option@\spxentry{pymethyl-preprocess-remove\_diseases command line option}!-e, --exclude\_disease\_list \textless{}exclude\_disease\_list\textgreater{}@\spxentry{-e, --exclude\_disease\_list \textless{}exclude\_disease\_list\textgreater{}}}\index{-e, --exclude\_disease\_list \textless{}exclude\_disease\_list\textgreater{}@\spxentry{-e, --exclude\_disease\_list \textless{}exclude\_disease\_list\textgreater{}}!pymethyl-preprocess-remove\_diseases command line option@\spxentry{pymethyl-preprocess-remove\_diseases command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-remove-diseases-e}}\pysigline{\sphinxbfcode{\sphinxupquote{-e}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}exclude\_disease\_list}}\sphinxcode{\sphinxupquote{~\textless{}exclude\_disease\_list\textgreater{}}}}
List of conditions to exclude, from disease column, comma delimited.  {[}default: {]}

\end{fulllineitems}

\index{pymethyl-preprocess-remove\_diseases command line option@\spxentry{pymethyl-preprocess-remove\_diseases command line option}!-os, --output\_sheet\_name \textless{}output\_sheet\_name\textgreater{}@\spxentry{-os, --output\_sheet\_name \textless{}output\_sheet\_name\textgreater{}}}\index{-os, --output\_sheet\_name \textless{}output\_sheet\_name\textgreater{}@\spxentry{-os, --output\_sheet\_name \textless{}output\_sheet\_name\textgreater{}}!pymethyl-preprocess-remove\_diseases command line option@\spxentry{pymethyl-preprocess-remove\_diseases command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-remove-diseases-os}}\pysigline{\sphinxbfcode{\sphinxupquote{-os}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}output\_sheet\_name}}\sphinxcode{\sphinxupquote{~\textless{}output\_sheet\_name\textgreater{}}}}
CSV for minfi input.  {[}default: ./tcga\_idats/minfiSheet.csv{]}

\end{fulllineitems}

\index{pymethyl-preprocess-remove\_diseases command line option@\spxentry{pymethyl-preprocess-remove\_diseases command line option}!-l, --low\_count \textless{}low\_count\textgreater{}@\spxentry{-l, --low\_count \textless{}low\_count\textgreater{}}}\index{-l, --low\_count \textless{}low\_count\textgreater{}@\spxentry{-l, --low\_count \textless{}low\_count\textgreater{}}!pymethyl-preprocess-remove\_diseases command line option@\spxentry{pymethyl-preprocess-remove\_diseases command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-remove-diseases-l}}\pysigline{\sphinxbfcode{\sphinxupquote{-l}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}low\_count}}\sphinxcode{\sphinxupquote{~\textless{}low\_count\textgreater{}}}}
Remove diseases if they are below a certain count, default this is not used.  {[}default: 0{]}

\end{fulllineitems}

\index{pymethyl-preprocess-remove\_diseases command line option@\spxentry{pymethyl-preprocess-remove\_diseases command line option}!-d, --disease\_only@\spxentry{-d, --disease\_only}}\index{-d, --disease\_only@\spxentry{-d, --disease\_only}!pymethyl-preprocess-remove\_diseases command line option@\spxentry{pymethyl-preprocess-remove\_diseases command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-remove-diseases-d}}\pysigline{\sphinxbfcode{\sphinxupquote{-d}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}disease\_only}}\sphinxcode{\sphinxupquote{}}}
Only look at disease, or text before subtype\_delimiter.

\end{fulllineitems}

\index{pymethyl-preprocess-remove\_diseases command line option@\spxentry{pymethyl-preprocess-remove\_diseases command line option}!-sd, --subtype\_delimiter \textless{}subtype\_delimiter\textgreater{}@\spxentry{-sd, --subtype\_delimiter \textless{}subtype\_delimiter\textgreater{}}}\index{-sd, --subtype\_delimiter \textless{}subtype\_delimiter\textgreater{}@\spxentry{-sd, --subtype\_delimiter \textless{}subtype\_delimiter\textgreater{}}!pymethyl-preprocess-remove\_diseases command line option@\spxentry{pymethyl-preprocess-remove\_diseases command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-remove-diseases-sd}}\pysigline{\sphinxbfcode{\sphinxupquote{-sd}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}subtype\_delimiter}}\sphinxcode{\sphinxupquote{~\textless{}subtype\_delimiter\textgreater{}}}}
Delimiter for disease extraction.  {[}default: ,{]}

\end{fulllineitems}



\section{split\_preprocess\_input\_by\_subtype}
\label{\detokenize{index:pymethyl-preprocess-split-preprocess-input-by-subtype}}
Split preprocess input samplesheet by disease subtype.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}preprocess split\PYGZus{}preprocess\PYGZus{}input\PYGZus{}by\PYGZus{}subtype \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-preprocess-split\_preprocess\_input\_by\_subtype command line option@\spxentry{pymethyl-preprocess-split\_preprocess\_input\_by\_subtype command line option}!-i, --idat\_csv \textless{}idat\_csv\textgreater{}@\spxentry{-i, --idat\_csv \textless{}idat\_csv\textgreater{}}}\index{-i, --idat\_csv \textless{}idat\_csv\textgreater{}@\spxentry{-i, --idat\_csv \textless{}idat\_csv\textgreater{}}!pymethyl-preprocess-split\_preprocess\_input\_by\_subtype command line option@\spxentry{pymethyl-preprocess-split\_preprocess\_input\_by\_subtype command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-split-preprocess-input-by-subtype-i}}\pysigline{\sphinxbfcode{\sphinxupquote{-i}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}idat\_csv}}\sphinxcode{\sphinxupquote{~\textless{}idat\_csv\textgreater{}}}}
Idat csv for one sample sheet, alternatively can be your phenotype sample sheet.  {[}default: ./tcga\_idats/minfiSheet.csv{]}

\end{fulllineitems}

\index{pymethyl-preprocess-split\_preprocess\_input\_by\_subtype command line option@\spxentry{pymethyl-preprocess-split\_preprocess\_input\_by\_subtype command line option}!-d, --disease\_only@\spxentry{-d, --disease\_only}}\index{-d, --disease\_only@\spxentry{-d, --disease\_only}!pymethyl-preprocess-split\_preprocess\_input\_by\_subtype command line option@\spxentry{pymethyl-preprocess-split\_preprocess\_input\_by\_subtype command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-split-preprocess-input-by-subtype-d}}\pysigline{\sphinxbfcode{\sphinxupquote{-d}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}disease\_only}}\sphinxcode{\sphinxupquote{}}}
Only look at disease, or text before subtype\_delimiter.

\end{fulllineitems}

\index{pymethyl-preprocess-split\_preprocess\_input\_by\_subtype command line option@\spxentry{pymethyl-preprocess-split\_preprocess\_input\_by\_subtype command line option}!-sd, --subtype\_delimiter \textless{}subtype\_delimiter\textgreater{}@\spxentry{-sd, --subtype\_delimiter \textless{}subtype\_delimiter\textgreater{}}}\index{-sd, --subtype\_delimiter \textless{}subtype\_delimiter\textgreater{}@\spxentry{-sd, --subtype\_delimiter \textless{}subtype\_delimiter\textgreater{}}!pymethyl-preprocess-split\_preprocess\_input\_by\_subtype command line option@\spxentry{pymethyl-preprocess-split\_preprocess\_input\_by\_subtype command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-split-preprocess-input-by-subtype-sd}}\pysigline{\sphinxbfcode{\sphinxupquote{-sd}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}subtype\_delimiter}}\sphinxcode{\sphinxupquote{~\textless{}subtype\_delimiter\textgreater{}}}}
Delimiter for disease extraction.  {[}default: ,{]}

\end{fulllineitems}

\index{pymethyl-preprocess-split\_preprocess\_input\_by\_subtype command line option@\spxentry{pymethyl-preprocess-split\_preprocess\_input\_by\_subtype command line option}!-o, --subtype\_output\_dir \textless{}subtype\_output\_dir\textgreater{}@\spxentry{-o, --subtype\_output\_dir \textless{}subtype\_output\_dir\textgreater{}}}\index{-o, --subtype\_output\_dir \textless{}subtype\_output\_dir\textgreater{}@\spxentry{-o, --subtype\_output\_dir \textless{}subtype\_output\_dir\textgreater{}}!pymethyl-preprocess-split\_preprocess\_input\_by\_subtype command line option@\spxentry{pymethyl-preprocess-split\_preprocess\_input\_by\_subtype command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-preprocess-split-preprocess-input-by-subtype-o}}\pysigline{\sphinxbfcode{\sphinxupquote{-o}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}subtype\_output\_dir}}\sphinxcode{\sphinxupquote{~\textless{}subtype\_output\_dir\textgreater{}}}}
Output subtypes pheno csv.  {[}default: ./preprocess\_outputs/{]}

\end{fulllineitems}



\chapter{pymethyl-utils}
\label{\detokenize{index:pymethyl-utils}}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}utils \PYG{o}{[}OPTIONS\PYG{o}{]} COMMAND \PYG{o}{[}ARGS\PYG{o}{]}...
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-utils command line option@\spxentry{pymethyl-utils command line option}!--version@\spxentry{--version}}\index{--version@\spxentry{--version}!pymethyl-utils command line option@\spxentry{pymethyl-utils command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-version}}\pysigline{\sphinxbfcode{\sphinxupquote{-{-}version}}\sphinxcode{\sphinxupquote{}}}
Show the version and exit.

\end{fulllineitems}



\section{backup\_pkl}
\label{\detokenize{index:pymethyl-utils-backup-pkl}}
Copy methylarray pickle to new location to backup.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}utils backup\PYGZus{}pkl \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-utils-backup\_pkl command line option@\spxentry{pymethyl-utils-backup\_pkl command line option}!-i, --input\_pkl \textless{}input\_pkl\textgreater{}@\spxentry{-i, --input\_pkl \textless{}input\_pkl\textgreater{}}}\index{-i, --input\_pkl \textless{}input\_pkl\textgreater{}@\spxentry{-i, --input\_pkl \textless{}input\_pkl\textgreater{}}!pymethyl-utils-backup\_pkl command line option@\spxentry{pymethyl-utils-backup\_pkl command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-backup-pkl-i}}\pysigline{\sphinxbfcode{\sphinxupquote{-i}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}input\_pkl}}\sphinxcode{\sphinxupquote{~\textless{}input\_pkl\textgreater{}}}}
Input database for beta and phenotype data.  {[}default: ./final\_preprocessed/methyl\_array.pkl{]}

\end{fulllineitems}

\index{pymethyl-utils-backup\_pkl command line option@\spxentry{pymethyl-utils-backup\_pkl command line option}!-o, --output\_pkl \textless{}output\_pkl\textgreater{}@\spxentry{-o, --output\_pkl \textless{}output\_pkl\textgreater{}}}\index{-o, --output\_pkl \textless{}output\_pkl\textgreater{}@\spxentry{-o, --output\_pkl \textless{}output\_pkl\textgreater{}}!pymethyl-utils-backup\_pkl command line option@\spxentry{pymethyl-utils-backup\_pkl command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-backup-pkl-o}}\pysigline{\sphinxbfcode{\sphinxupquote{-o}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}output\_pkl}}\sphinxcode{\sphinxupquote{~\textless{}output\_pkl\textgreater{}}}}
Output database for beta and phenotype data.  {[}default: ./backup/methyl\_array.pkl{]}

\end{fulllineitems}



\section{bin\_column}
\label{\detokenize{index:pymethyl-utils-bin-column}}
Convert continuous phenotype column into categorical by binning.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}utils bin\PYGZus{}column \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-utils-bin\_column command line option@\spxentry{pymethyl-utils-bin\_column command line option}!-t, --test\_pkl \textless{}test\_pkl\textgreater{}@\spxentry{-t, --test\_pkl \textless{}test\_pkl\textgreater{}}}\index{-t, --test\_pkl \textless{}test\_pkl\textgreater{}@\spxentry{-t, --test\_pkl \textless{}test\_pkl\textgreater{}}!pymethyl-utils-bin\_column command line option@\spxentry{pymethyl-utils-bin\_column command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-bin-column-t}}\pysigline{\sphinxbfcode{\sphinxupquote{-t}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}test\_pkl}}\sphinxcode{\sphinxupquote{~\textless{}test\_pkl\textgreater{}}}}
Pickle containing testing set.  {[}default: ./train\_val\_test\_sets/test\_methyl\_array.pkl{]}

\end{fulllineitems}

\index{pymethyl-utils-bin\_column command line option@\spxentry{pymethyl-utils-bin\_column command line option}!-c, --col \textless{}col\textgreater{}@\spxentry{-c, --col \textless{}col\textgreater{}}}\index{-c, --col \textless{}col\textgreater{}@\spxentry{-c, --col \textless{}col\textgreater{}}!pymethyl-utils-bin\_column command line option@\spxentry{pymethyl-utils-bin\_column command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-bin-column-c}}\pysigline{\sphinxbfcode{\sphinxupquote{-c}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}col}}\sphinxcode{\sphinxupquote{~\textless{}col\textgreater{}}}}
Column to turn into bins.  {[}default: age{]}

\end{fulllineitems}

\index{pymethyl-utils-bin\_column command line option@\spxentry{pymethyl-utils-bin\_column command line option}!-n, --n\_bins \textless{}n\_bins\textgreater{}@\spxentry{-n, --n\_bins \textless{}n\_bins\textgreater{}}}\index{-n, --n\_bins \textless{}n\_bins\textgreater{}@\spxentry{-n, --n\_bins \textless{}n\_bins\textgreater{}}!pymethyl-utils-bin\_column command line option@\spxentry{pymethyl-utils-bin\_column command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-bin-column-n}}\pysigline{\sphinxbfcode{\sphinxupquote{-n}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}n\_bins}}\sphinxcode{\sphinxupquote{~\textless{}n\_bins\textgreater{}}}}
Number of bins.  {[}default: 10{]}

\end{fulllineitems}

\index{pymethyl-utils-bin\_column command line option@\spxentry{pymethyl-utils-bin\_column command line option}!-ot, --output\_test\_pkl \textless{}output\_test\_pkl\textgreater{}@\spxentry{-ot, --output\_test\_pkl \textless{}output\_test\_pkl\textgreater{}}}\index{-ot, --output\_test\_pkl \textless{}output\_test\_pkl\textgreater{}@\spxentry{-ot, --output\_test\_pkl \textless{}output\_test\_pkl\textgreater{}}!pymethyl-utils-bin\_column command line option@\spxentry{pymethyl-utils-bin\_column command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-bin-column-ot}}\pysigline{\sphinxbfcode{\sphinxupquote{-ot}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}output\_test\_pkl}}\sphinxcode{\sphinxupquote{~\textless{}output\_test\_pkl\textgreater{}}}}
Binned shap pickle for further testing.  {[}default: ./train\_val\_test\_sets/test\_methyl\_array\_shap\_binned.pkl{]}

\end{fulllineitems}



\section{concat\_csv}
\label{\detokenize{index:pymethyl-utils-concat-csv}}
Concatenate two csv files together.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}utils concat\PYGZus{}csv \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-utils-concat\_csv command line option@\spxentry{pymethyl-utils-concat\_csv command line option}!-i1, --input\_csv \textless{}input\_csv\textgreater{}@\spxentry{-i1, --input\_csv \textless{}input\_csv\textgreater{}}}\index{-i1, --input\_csv \textless{}input\_csv\textgreater{}@\spxentry{-i1, --input\_csv \textless{}input\_csv\textgreater{}}!pymethyl-utils-concat\_csv command line option@\spxentry{pymethyl-utils-concat\_csv command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-concat-csv-i1}}\pysigline{\sphinxbfcode{\sphinxupquote{-i1}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}input\_csv}}\sphinxcode{\sphinxupquote{~\textless{}input\_csv\textgreater{}}}}
Beta csv.  {[}default: ./beta1.csv{]}

\end{fulllineitems}

\index{pymethyl-utils-concat\_csv command line option@\spxentry{pymethyl-utils-concat\_csv command line option}!-i2, --input\_csv2 \textless{}input\_csv2\textgreater{}@\spxentry{-i2, --input\_csv2 \textless{}input\_csv2\textgreater{}}}\index{-i2, --input\_csv2 \textless{}input\_csv2\textgreater{}@\spxentry{-i2, --input\_csv2 \textless{}input\_csv2\textgreater{}}!pymethyl-utils-concat\_csv command line option@\spxentry{pymethyl-utils-concat\_csv command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-concat-csv-i2}}\pysigline{\sphinxbfcode{\sphinxupquote{-i2}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}input\_csv2}}\sphinxcode{\sphinxupquote{~\textless{}input\_csv2\textgreater{}}}}
Beta/other csv 2.  {[}default: ./cell\_estimates.csv{]}

\end{fulllineitems}

\index{pymethyl-utils-concat\_csv command line option@\spxentry{pymethyl-utils-concat\_csv command line option}!-o, --output\_csv \textless{}output\_csv\textgreater{}@\spxentry{-o, --output\_csv \textless{}output\_csv\textgreater{}}}\index{-o, --output\_csv \textless{}output\_csv\textgreater{}@\spxentry{-o, --output\_csv \textless{}output\_csv\textgreater{}}!pymethyl-utils-concat\_csv command line option@\spxentry{pymethyl-utils-concat\_csv command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-concat-csv-o}}\pysigline{\sphinxbfcode{\sphinxupquote{-o}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}output\_csv}}\sphinxcode{\sphinxupquote{~\textless{}output\_csv\textgreater{}}}}
Output csv.  {[}default: ./beta.concat.csv{]}

\end{fulllineitems}

\index{pymethyl-utils-concat\_csv command line option@\spxentry{pymethyl-utils-concat\_csv command line option}!-a, --axis \textless{}axis\textgreater{}@\spxentry{-a, --axis \textless{}axis\textgreater{}}}\index{-a, --axis \textless{}axis\textgreater{}@\spxentry{-a, --axis \textless{}axis\textgreater{}}!pymethyl-utils-concat\_csv command line option@\spxentry{pymethyl-utils-concat\_csv command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-concat-csv-a}}\pysigline{\sphinxbfcode{\sphinxupquote{-a}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}axis}}\sphinxcode{\sphinxupquote{~\textless{}axis\textgreater{}}}}
Axis to merge on. Columns are 0, rows are 1.  {[}default: 1{]}

\end{fulllineitems}

\index{pymethyl-utils-concat\_csv command line option@\spxentry{pymethyl-utils-concat\_csv command line option}!-i, --index\_col \textless{}index\_col\textgreater{}@\spxentry{-i, --index\_col \textless{}index\_col\textgreater{}}}\index{-i, --index\_col \textless{}index\_col\textgreater{}@\spxentry{-i, --index\_col \textless{}index\_col\textgreater{}}!pymethyl-utils-concat\_csv command line option@\spxentry{pymethyl-utils-concat\_csv command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-concat-csv-i}}\pysigline{\sphinxbfcode{\sphinxupquote{-i}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}index\_col}}\sphinxcode{\sphinxupquote{~\textless{}index\_col\textgreater{}}}}
Index Column.  {[}default: 0{]}

\end{fulllineitems}



\section{counts}
\label{\detokenize{index:pymethyl-utils-counts}}
Return categorical breakdown of phenotype column.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}utils counts \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-utils-counts command line option@\spxentry{pymethyl-utils-counts command line option}!-i, --input\_pkl \textless{}input\_pkl\textgreater{}@\spxentry{-i, --input\_pkl \textless{}input\_pkl\textgreater{}}}\index{-i, --input\_pkl \textless{}input\_pkl\textgreater{}@\spxentry{-i, --input\_pkl \textless{}input\_pkl\textgreater{}}!pymethyl-utils-counts command line option@\spxentry{pymethyl-utils-counts command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-counts-i}}\pysigline{\sphinxbfcode{\sphinxupquote{-i}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}input\_pkl}}\sphinxcode{\sphinxupquote{~\textless{}input\_pkl\textgreater{}}}}
Input database for beta and phenotype data.  {[}default: ./final\_preprocessed/methyl\_array.pkl{]}

\end{fulllineitems}

\index{pymethyl-utils-counts command line option@\spxentry{pymethyl-utils-counts command line option}!-k, --key \textless{}key\textgreater{}@\spxentry{-k, --key \textless{}key\textgreater{}}}\index{-k, --key \textless{}key\textgreater{}@\spxentry{-k, --key \textless{}key\textgreater{}}!pymethyl-utils-counts command line option@\spxentry{pymethyl-utils-counts command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-counts-k}}\pysigline{\sphinxbfcode{\sphinxupquote{-k}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}key}}\sphinxcode{\sphinxupquote{~\textless{}key\textgreater{}}}}
Key to split on.  {[}default: disease{]}

\end{fulllineitems}



\section{create\_external\_validation\_set}
\label{\detokenize{index:pymethyl-utils-create-external-validation-set}}
Create external validation set containing same CpGs as training set.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}utils create\PYGZus{}external\PYGZus{}validation\PYGZus{}set \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-utils-create\_external\_validation\_set command line option@\spxentry{pymethyl-utils-create\_external\_validation\_set command line option}!-t, --train\_pkl \textless{}train\_pkl\textgreater{}@\spxentry{-t, --train\_pkl \textless{}train\_pkl\textgreater{}}}\index{-t, --train\_pkl \textless{}train\_pkl\textgreater{}@\spxentry{-t, --train\_pkl \textless{}train\_pkl\textgreater{}}!pymethyl-utils-create\_external\_validation\_set command line option@\spxentry{pymethyl-utils-create\_external\_validation\_set command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-create-external-validation-set-t}}\pysigline{\sphinxbfcode{\sphinxupquote{-t}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}train\_pkl}}\sphinxcode{\sphinxupquote{~\textless{}train\_pkl\textgreater{}}}}
Input methyl array.  {[}default: ./train\_val\_test\_sets/train\_methyl\_array.pkl{]}

\end{fulllineitems}

\index{pymethyl-utils-create\_external\_validation\_set command line option@\spxentry{pymethyl-utils-create\_external\_validation\_set command line option}!-q, --query\_pkl \textless{}query\_pkl\textgreater{}@\spxentry{-q, --query\_pkl \textless{}query\_pkl\textgreater{}}}\index{-q, --query\_pkl \textless{}query\_pkl\textgreater{}@\spxentry{-q, --query\_pkl \textless{}query\_pkl\textgreater{}}!pymethyl-utils-create\_external\_validation\_set command line option@\spxentry{pymethyl-utils-create\_external\_validation\_set command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-create-external-validation-set-q}}\pysigline{\sphinxbfcode{\sphinxupquote{-q}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}query\_pkl}}\sphinxcode{\sphinxupquote{~\textless{}query\_pkl\textgreater{}}}}
Input methylation array to add/subtract cpgs to.  {[}default: ./final\_preprocessed/methyl\_array.pkl{]}

\end{fulllineitems}

\index{pymethyl-utils-create\_external\_validation\_set command line option@\spxentry{pymethyl-utils-create\_external\_validation\_set command line option}!-o, --output\_pkl \textless{}output\_pkl\textgreater{}@\spxentry{-o, --output\_pkl \textless{}output\_pkl\textgreater{}}}\index{-o, --output\_pkl \textless{}output\_pkl\textgreater{}@\spxentry{-o, --output\_pkl \textless{}output\_pkl\textgreater{}}!pymethyl-utils-create\_external\_validation\_set command line option@\spxentry{pymethyl-utils-create\_external\_validation\_set command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-create-external-validation-set-o}}\pysigline{\sphinxbfcode{\sphinxupquote{-o}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}output\_pkl}}\sphinxcode{\sphinxupquote{~\textless{}output\_pkl\textgreater{}}}}
Output methyl array external validation.  {[}default: ./external\_validation/methyl\_array.pkl{]}

\end{fulllineitems}

\index{pymethyl-utils-create\_external\_validation\_set command line option@\spxentry{pymethyl-utils-create\_external\_validation\_set command line option}!-c, --cpg\_replace\_method \textless{}cpg\_replace\_method\textgreater{}@\spxentry{-c, --cpg\_replace\_method \textless{}cpg\_replace\_method\textgreater{}}}\index{-c, --cpg\_replace\_method \textless{}cpg\_replace\_method\textgreater{}@\spxentry{-c, --cpg\_replace\_method \textless{}cpg\_replace\_method\textgreater{}}!pymethyl-utils-create\_external\_validation\_set command line option@\spxentry{pymethyl-utils-create\_external\_validation\_set command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-create-external-validation-set-c}}\pysigline{\sphinxbfcode{\sphinxupquote{-c}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}cpg\_replace\_method}}\sphinxcode{\sphinxupquote{~\textless{}cpg\_replace\_method\textgreater{}}}}
What to do for missing CpGs.  {[}default: mid{]}

\end{fulllineitems}



\section{est\_age}
\label{\detokenize{index:pymethyl-utils-est-age}}
Estimate age using cgAgeR

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}utils est\PYGZus{}age \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-utils-est\_age command line option@\spxentry{pymethyl-utils-est\_age command line option}!-i, --input\_pkl \textless{}input\_pkl\textgreater{}@\spxentry{-i, --input\_pkl \textless{}input\_pkl\textgreater{}}}\index{-i, --input\_pkl \textless{}input\_pkl\textgreater{}@\spxentry{-i, --input\_pkl \textless{}input\_pkl\textgreater{}}!pymethyl-utils-est\_age command line option@\spxentry{pymethyl-utils-est\_age command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-est-age-i}}\pysigline{\sphinxbfcode{\sphinxupquote{-i}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}input\_pkl}}\sphinxcode{\sphinxupquote{~\textless{}input\_pkl\textgreater{}}}}
Input methyl array.  {[}default: ./train\_val\_test\_sets/test\_methyl\_array.pkl{]}

\end{fulllineitems}

\index{pymethyl-utils-est\_age command line option@\spxentry{pymethyl-utils-est\_age command line option}!-ac, --age\_column \textless{}age\_column\textgreater{}@\spxentry{-ac, --age\_column \textless{}age\_column\textgreater{}}}\index{-ac, --age\_column \textless{}age\_column\textgreater{}@\spxentry{-ac, --age\_column \textless{}age\_column\textgreater{}}!pymethyl-utils-est\_age command line option@\spxentry{pymethyl-utils-est\_age command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-est-age-ac}}\pysigline{\sphinxbfcode{\sphinxupquote{-ac}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}age\_column}}\sphinxcode{\sphinxupquote{~\textless{}age\_column\textgreater{}}}}
Age column of Methylation Array. Leave blank for no age  {[}default: {]}

\end{fulllineitems}

\index{pymethyl-utils-est\_age command line option@\spxentry{pymethyl-utils-est\_age command line option}!-a, --analyses \textless{}analyses\textgreater{}@\spxentry{-a, --analyses \textless{}analyses\textgreater{}}}\index{-a, --analyses \textless{}analyses\textgreater{}@\spxentry{-a, --analyses \textless{}analyses\textgreater{}}!pymethyl-utils-est\_age command line option@\spxentry{pymethyl-utils-est\_age command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-est-age-a}}\pysigline{\sphinxbfcode{\sphinxupquote{-a}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}analyses}}\sphinxcode{\sphinxupquote{~\textless{}analyses\textgreater{}}}}
Analyses to run

\end{fulllineitems}

\index{pymethyl-utils-est\_age command line option@\spxentry{pymethyl-utils-est\_age command line option}!-o, --output\_csv \textless{}output\_csv\textgreater{}@\spxentry{-o, --output\_csv \textless{}output\_csv\textgreater{}}}\index{-o, --output\_csv \textless{}output\_csv\textgreater{}@\spxentry{-o, --output\_csv \textless{}output\_csv\textgreater{}}!pymethyl-utils-est\_age command line option@\spxentry{pymethyl-utils-est\_age command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-est-age-o}}\pysigline{\sphinxbfcode{\sphinxupquote{-o}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}output\_csv}}\sphinxcode{\sphinxupquote{~\textless{}output\_csv\textgreater{}}}}
Output csv  {[}default: age\_estimation/output\_age\_estimations.csv{]}

\end{fulllineitems}



\section{feature\_select\_train\_val\_test}
\label{\detokenize{index:pymethyl-utils-feature-select-train-val-test}}
Filter CpGs by taking x top CpGs with highest mean absolute deviation scores or via spectral feature selection.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}utils feature\PYGZus{}select\PYGZus{}train\PYGZus{}val\PYGZus{}test \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-utils-feature\_select\_train\_val\_test command line option@\spxentry{pymethyl-utils-feature\_select\_train\_val\_test command line option}!-i, --input\_pkl\_dir \textless{}input\_pkl\_dir\textgreater{}@\spxentry{-i, --input\_pkl\_dir \textless{}input\_pkl\_dir\textgreater{}}}\index{-i, --input\_pkl\_dir \textless{}input\_pkl\_dir\textgreater{}@\spxentry{-i, --input\_pkl\_dir \textless{}input\_pkl\_dir\textgreater{}}!pymethyl-utils-feature\_select\_train\_val\_test command line option@\spxentry{pymethyl-utils-feature\_select\_train\_val\_test command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-feature-select-train-val-test-i}}\pysigline{\sphinxbfcode{\sphinxupquote{-i}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}input\_pkl\_dir}}\sphinxcode{\sphinxupquote{~\textless{}input\_pkl\_dir\textgreater{}}}}
Input database for beta and phenotype data.  {[}default: ./train\_val\_test\_sets/{]}

\end{fulllineitems}

\index{pymethyl-utils-feature\_select\_train\_val\_test command line option@\spxentry{pymethyl-utils-feature\_select\_train\_val\_test command line option}!-o, --output\_dir \textless{}output\_dir\textgreater{}@\spxentry{-o, --output\_dir \textless{}output\_dir\textgreater{}}}\index{-o, --output\_dir \textless{}output\_dir\textgreater{}@\spxentry{-o, --output\_dir \textless{}output\_dir\textgreater{}}!pymethyl-utils-feature\_select\_train\_val\_test command line option@\spxentry{pymethyl-utils-feature\_select\_train\_val\_test command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-feature-select-train-val-test-o}}\pysigline{\sphinxbfcode{\sphinxupquote{-o}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}output\_dir}}\sphinxcode{\sphinxupquote{~\textless{}output\_dir\textgreater{}}}}
Output database for beta and phenotype data.  {[}default: ./train\_val\_test\_sets\_fs/{]}

\end{fulllineitems}

\index{pymethyl-utils-feature\_select\_train\_val\_test command line option@\spxentry{pymethyl-utils-feature\_select\_train\_val\_test command line option}!-n, --n\_top\_cpgs \textless{}n\_top\_cpgs\textgreater{}@\spxentry{-n, --n\_top\_cpgs \textless{}n\_top\_cpgs\textgreater{}}}\index{-n, --n\_top\_cpgs \textless{}n\_top\_cpgs\textgreater{}@\spxentry{-n, --n\_top\_cpgs \textless{}n\_top\_cpgs\textgreater{}}!pymethyl-utils-feature\_select\_train\_val\_test command line option@\spxentry{pymethyl-utils-feature\_select\_train\_val\_test command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-feature-select-train-val-test-n}}\pysigline{\sphinxbfcode{\sphinxupquote{-n}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}n\_top\_cpgs}}\sphinxcode{\sphinxupquote{~\textless{}n\_top\_cpgs\textgreater{}}}}
Number cpgs to include with highest variance across population.  {[}default: 300000{]}

\end{fulllineitems}

\index{pymethyl-utils-feature\_select\_train\_val\_test command line option@\spxentry{pymethyl-utils-feature\_select\_train\_val\_test command line option}!-f, --feature\_selection\_method \textless{}feature\_selection\_method\textgreater{}@\spxentry{-f, --feature\_selection\_method \textless{}feature\_selection\_method\textgreater{}}}\index{-f, --feature\_selection\_method \textless{}feature\_selection\_method\textgreater{}@\spxentry{-f, --feature\_selection\_method \textless{}feature\_selection\_method\textgreater{}}!pymethyl-utils-feature\_select\_train\_val\_test command line option@\spxentry{pymethyl-utils-feature\_select\_train\_val\_test command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-feature-select-train-val-test-f}}\pysigline{\sphinxbfcode{\sphinxupquote{-f}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}feature\_selection\_method}}\sphinxcode{\sphinxupquote{~\textless{}feature\_selection\_method\textgreater{}}}}
\end{fulllineitems}

\index{pymethyl-utils-feature\_select\_train\_val\_test command line option@\spxentry{pymethyl-utils-feature\_select\_train\_val\_test command line option}!-mm, --metric \textless{}metric\textgreater{}@\spxentry{-mm, --metric \textless{}metric\textgreater{}}}\index{-mm, --metric \textless{}metric\textgreater{}@\spxentry{-mm, --metric \textless{}metric\textgreater{}}!pymethyl-utils-feature\_select\_train\_val\_test command line option@\spxentry{pymethyl-utils-feature\_select\_train\_val\_test command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-feature-select-train-val-test-mm}}\pysigline{\sphinxbfcode{\sphinxupquote{-mm}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}metric}}\sphinxcode{\sphinxupquote{~\textless{}metric\textgreater{}}}}
\end{fulllineitems}

\index{pymethyl-utils-feature\_select\_train\_val\_test command line option@\spxentry{pymethyl-utils-feature\_select\_train\_val\_test command line option}!-nn, --n\_neighbors \textless{}n\_neighbors\textgreater{}@\spxentry{-nn, --n\_neighbors \textless{}n\_neighbors\textgreater{}}}\index{-nn, --n\_neighbors \textless{}n\_neighbors\textgreater{}@\spxentry{-nn, --n\_neighbors \textless{}n\_neighbors\textgreater{}}!pymethyl-utils-feature\_select\_train\_val\_test command line option@\spxentry{pymethyl-utils-feature\_select\_train\_val\_test command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-feature-select-train-val-test-nn}}\pysigline{\sphinxbfcode{\sphinxupquote{-nn}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}n\_neighbors}}\sphinxcode{\sphinxupquote{~\textless{}n\_neighbors\textgreater{}}}}
Number neighbors for feature selection, default enacts rbf kernel.  {[}default: 0{]}

\end{fulllineitems}

\index{pymethyl-utils-feature\_select\_train\_val\_test command line option@\spxentry{pymethyl-utils-feature\_select\_train\_val\_test command line option}!-m, --mad\_top\_cpgs \textless{}mad\_top\_cpgs\textgreater{}@\spxentry{-m, --mad\_top\_cpgs \textless{}mad\_top\_cpgs\textgreater{}}}\index{-m, --mad\_top\_cpgs \textless{}mad\_top\_cpgs\textgreater{}@\spxentry{-m, --mad\_top\_cpgs \textless{}mad\_top\_cpgs\textgreater{}}!pymethyl-utils-feature\_select\_train\_val\_test command line option@\spxentry{pymethyl-utils-feature\_select\_train\_val\_test command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-feature-select-train-val-test-m}}\pysigline{\sphinxbfcode{\sphinxupquote{-m}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}mad\_top\_cpgs}}\sphinxcode{\sphinxupquote{~\textless{}mad\_top\_cpgs\textgreater{}}}}
Number cpgs to apply mad filtering first before more sophisticated feature selection. If 0 or primary feature selection is mad, no mad pre-filtering.  {[}default: 0{]}

\end{fulllineitems}



\section{fix\_key}
\label{\detokenize{index:pymethyl-utils-fix-key}}
Format certain column of phenotype array in MethylationArray.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}utils fix\PYGZus{}key \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-utils-fix\_key command line option@\spxentry{pymethyl-utils-fix\_key command line option}!-i, --input\_pkl \textless{}input\_pkl\textgreater{}@\spxentry{-i, --input\_pkl \textless{}input\_pkl\textgreater{}}}\index{-i, --input\_pkl \textless{}input\_pkl\textgreater{}@\spxentry{-i, --input\_pkl \textless{}input\_pkl\textgreater{}}!pymethyl-utils-fix\_key command line option@\spxentry{pymethyl-utils-fix\_key command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-fix-key-i}}\pysigline{\sphinxbfcode{\sphinxupquote{-i}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}input\_pkl}}\sphinxcode{\sphinxupquote{~\textless{}input\_pkl\textgreater{}}}}
Input database for beta and phenotype data.  {[}default: ./final\_preprocessed/methyl\_array.pkl{]}

\end{fulllineitems}

\index{pymethyl-utils-fix\_key command line option@\spxentry{pymethyl-utils-fix\_key command line option}!-k, --key \textless{}key\textgreater{}@\spxentry{-k, --key \textless{}key\textgreater{}}}\index{-k, --key \textless{}key\textgreater{}@\spxentry{-k, --key \textless{}key\textgreater{}}!pymethyl-utils-fix\_key command line option@\spxentry{pymethyl-utils-fix\_key command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-fix-key-k}}\pysigline{\sphinxbfcode{\sphinxupquote{-k}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}key}}\sphinxcode{\sphinxupquote{~\textless{}key\textgreater{}}}}
Key to split on.  {[}default: disease{]}

\end{fulllineitems}

\index{pymethyl-utils-fix\_key command line option@\spxentry{pymethyl-utils-fix\_key command line option}!-d, --disease\_only@\spxentry{-d, --disease\_only}}\index{-d, --disease\_only@\spxentry{-d, --disease\_only}!pymethyl-utils-fix\_key command line option@\spxentry{pymethyl-utils-fix\_key command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-fix-key-d}}\pysigline{\sphinxbfcode{\sphinxupquote{-d}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}disease\_only}}\sphinxcode{\sphinxupquote{}}}
Only look at disease, or text before subtype\_delimiter.

\end{fulllineitems}

\index{pymethyl-utils-fix\_key command line option@\spxentry{pymethyl-utils-fix\_key command line option}!-sd, --subtype\_delimiter \textless{}subtype\_delimiter\textgreater{}@\spxentry{-sd, --subtype\_delimiter \textless{}subtype\_delimiter\textgreater{}}}\index{-sd, --subtype\_delimiter \textless{}subtype\_delimiter\textgreater{}@\spxentry{-sd, --subtype\_delimiter \textless{}subtype\_delimiter\textgreater{}}!pymethyl-utils-fix\_key command line option@\spxentry{pymethyl-utils-fix\_key command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-fix-key-sd}}\pysigline{\sphinxbfcode{\sphinxupquote{-sd}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}subtype\_delimiter}}\sphinxcode{\sphinxupquote{~\textless{}subtype\_delimiter\textgreater{}}}}
Delimiter for disease extraction.  {[}default: ,{]}

\end{fulllineitems}

\index{pymethyl-utils-fix\_key command line option@\spxentry{pymethyl-utils-fix\_key command line option}!-o, --output\_pkl \textless{}output\_pkl\textgreater{}@\spxentry{-o, --output\_pkl \textless{}output\_pkl\textgreater{}}}\index{-o, --output\_pkl \textless{}output\_pkl\textgreater{}@\spxentry{-o, --output\_pkl \textless{}output\_pkl\textgreater{}}!pymethyl-utils-fix\_key command line option@\spxentry{pymethyl-utils-fix\_key command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-fix-key-o}}\pysigline{\sphinxbfcode{\sphinxupquote{-o}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}output\_pkl}}\sphinxcode{\sphinxupquote{~\textless{}output\_pkl\textgreater{}}}}
Input database for beta and phenotype data.  {[}default: ./fixed\_preprocessed/methyl\_array.pkl{]}

\end{fulllineitems}



\section{modify\_pheno\_data}
\label{\detokenize{index:pymethyl-utils-modify-pheno-data}}
Use another spreadsheet to add more descriptive data to methylarray.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}utils modify\PYGZus{}pheno\PYGZus{}data \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-utils-modify\_pheno\_data command line option@\spxentry{pymethyl-utils-modify\_pheno\_data command line option}!-i, --input\_pkl \textless{}input\_pkl\textgreater{}@\spxentry{-i, --input\_pkl \textless{}input\_pkl\textgreater{}}}\index{-i, --input\_pkl \textless{}input\_pkl\textgreater{}@\spxentry{-i, --input\_pkl \textless{}input\_pkl\textgreater{}}!pymethyl-utils-modify\_pheno\_data command line option@\spxentry{pymethyl-utils-modify\_pheno\_data command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-modify-pheno-data-i}}\pysigline{\sphinxbfcode{\sphinxupquote{-i}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}input\_pkl}}\sphinxcode{\sphinxupquote{~\textless{}input\_pkl\textgreater{}}}}
Input database for beta and phenotype data.  {[}default: ./final\_preprocessed/methyl\_array.pkl{]}

\end{fulllineitems}

\index{pymethyl-utils-modify\_pheno\_data command line option@\spxentry{pymethyl-utils-modify\_pheno\_data command line option}!-is, --input\_formatted\_sample\_sheet \textless{}input\_formatted\_sample\_sheet\textgreater{}@\spxentry{-is, --input\_formatted\_sample\_sheet \textless{}input\_formatted\_sample\_sheet\textgreater{}}}\index{-is, --input\_formatted\_sample\_sheet \textless{}input\_formatted\_sample\_sheet\textgreater{}@\spxentry{-is, --input\_formatted\_sample\_sheet \textless{}input\_formatted\_sample\_sheet\textgreater{}}!pymethyl-utils-modify\_pheno\_data command line option@\spxentry{pymethyl-utils-modify\_pheno\_data command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-modify-pheno-data-is}}\pysigline{\sphinxbfcode{\sphinxupquote{-is}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}input\_formatted\_sample\_sheet}}\sphinxcode{\sphinxupquote{~\textless{}input\_formatted\_sample\_sheet\textgreater{}}}}
Information passed through function create\_sample\_sheet, has Basename and disease fields.  {[}default: ./tcga\_idats/minfi\_sheet.csv{]}

\end{fulllineitems}

\index{pymethyl-utils-modify\_pheno\_data command line option@\spxentry{pymethyl-utils-modify\_pheno\_data command line option}!-o, --output\_pkl \textless{}output\_pkl\textgreater{}@\spxentry{-o, --output\_pkl \textless{}output\_pkl\textgreater{}}}\index{-o, --output\_pkl \textless{}output\_pkl\textgreater{}@\spxentry{-o, --output\_pkl \textless{}output\_pkl\textgreater{}}!pymethyl-utils-modify\_pheno\_data command line option@\spxentry{pymethyl-utils-modify\_pheno\_data command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-modify-pheno-data-o}}\pysigline{\sphinxbfcode{\sphinxupquote{-o}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}output\_pkl}}\sphinxcode{\sphinxupquote{~\textless{}output\_pkl\textgreater{}}}}
Output database for beta and phenotype data.  {[}default: ./modified\_processed/methyl\_array.pkl{]}

\end{fulllineitems}



\section{move\_jpg}
\label{\detokenize{index:pymethyl-utils-move-jpg}}
Move preprocessing jpegs to preprocessing output directory.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}utils move\PYGZus{}jpg \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-utils-move\_jpg command line option@\spxentry{pymethyl-utils-move\_jpg command line option}!-i, --input\_dir \textless{}input\_dir\textgreater{}@\spxentry{-i, --input\_dir \textless{}input\_dir\textgreater{}}}\index{-i, --input\_dir \textless{}input\_dir\textgreater{}@\spxentry{-i, --input\_dir \textless{}input\_dir\textgreater{}}!pymethyl-utils-move\_jpg command line option@\spxentry{pymethyl-utils-move\_jpg command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-move-jpg-i}}\pysigline{\sphinxbfcode{\sphinxupquote{-i}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}input\_dir}}\sphinxcode{\sphinxupquote{~\textless{}input\_dir\textgreater{}}}}
Directory containing jpg.  {[}default: ./{]}

\end{fulllineitems}

\index{pymethyl-utils-move\_jpg command line option@\spxentry{pymethyl-utils-move\_jpg command line option}!-o, --output\_dir \textless{}output\_dir\textgreater{}@\spxentry{-o, --output\_dir \textless{}output\_dir\textgreater{}}}\index{-o, --output\_dir \textless{}output\_dir\textgreater{}@\spxentry{-o, --output\_dir \textless{}output\_dir\textgreater{}}!pymethyl-utils-move\_jpg command line option@\spxentry{pymethyl-utils-move\_jpg command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-move-jpg-o}}\pysigline{\sphinxbfcode{\sphinxupquote{-o}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}output\_dir}}\sphinxcode{\sphinxupquote{~\textless{}output\_dir\textgreater{}}}}
Output directory for images.  {[}default: ./preprocess\_output\_images/{]}

\end{fulllineitems}



\section{overwrite\_pheno\_data}
\label{\detokenize{index:pymethyl-utils-overwrite-pheno-data}}
Use another spreadsheet to add more descriptive data to methylarray.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}utils overwrite\PYGZus{}pheno\PYGZus{}data \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-utils-overwrite\_pheno\_data command line option@\spxentry{pymethyl-utils-overwrite\_pheno\_data command line option}!-i, --input\_pkl \textless{}input\_pkl\textgreater{}@\spxentry{-i, --input\_pkl \textless{}input\_pkl\textgreater{}}}\index{-i, --input\_pkl \textless{}input\_pkl\textgreater{}@\spxentry{-i, --input\_pkl \textless{}input\_pkl\textgreater{}}!pymethyl-utils-overwrite\_pheno\_data command line option@\spxentry{pymethyl-utils-overwrite\_pheno\_data command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-overwrite-pheno-data-i}}\pysigline{\sphinxbfcode{\sphinxupquote{-i}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}input\_pkl}}\sphinxcode{\sphinxupquote{~\textless{}input\_pkl\textgreater{}}}}
Input database for beta and phenotype data.  {[}default: ./final\_preprocessed/methyl\_array.pkl{]}

\end{fulllineitems}

\index{pymethyl-utils-overwrite\_pheno\_data command line option@\spxentry{pymethyl-utils-overwrite\_pheno\_data command line option}!-is, --input\_formatted\_sample\_sheet \textless{}input\_formatted\_sample\_sheet\textgreater{}@\spxentry{-is, --input\_formatted\_sample\_sheet \textless{}input\_formatted\_sample\_sheet\textgreater{}}}\index{-is, --input\_formatted\_sample\_sheet \textless{}input\_formatted\_sample\_sheet\textgreater{}@\spxentry{-is, --input\_formatted\_sample\_sheet \textless{}input\_formatted\_sample\_sheet\textgreater{}}!pymethyl-utils-overwrite\_pheno\_data command line option@\spxentry{pymethyl-utils-overwrite\_pheno\_data command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-overwrite-pheno-data-is}}\pysigline{\sphinxbfcode{\sphinxupquote{-is}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}input\_formatted\_sample\_sheet}}\sphinxcode{\sphinxupquote{~\textless{}input\_formatted\_sample\_sheet\textgreater{}}}}
Information passed through function create\_sample\_sheet, has Basename and disease fields.  {[}default: ./tcga\_idats/minfi\_sheet.csv{]}

\end{fulllineitems}

\index{pymethyl-utils-overwrite\_pheno\_data command line option@\spxentry{pymethyl-utils-overwrite\_pheno\_data command line option}!-o, --output\_pkl \textless{}output\_pkl\textgreater{}@\spxentry{-o, --output\_pkl \textless{}output\_pkl\textgreater{}}}\index{-o, --output\_pkl \textless{}output\_pkl\textgreater{}@\spxentry{-o, --output\_pkl \textless{}output\_pkl\textgreater{}}!pymethyl-utils-overwrite\_pheno\_data command line option@\spxentry{pymethyl-utils-overwrite\_pheno\_data command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-overwrite-pheno-data-o}}\pysigline{\sphinxbfcode{\sphinxupquote{-o}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}output\_pkl}}\sphinxcode{\sphinxupquote{~\textless{}output\_pkl\textgreater{}}}}
Output database for beta and phenotype data.  {[}default: ./modified\_processed/methyl\_array.pkl{]}

\end{fulllineitems}

\index{pymethyl-utils-overwrite\_pheno\_data command line option@\spxentry{pymethyl-utils-overwrite\_pheno\_data command line option}!-c, --index\_col \textless{}index\_col\textgreater{}@\spxentry{-c, --index\_col \textless{}index\_col\textgreater{}}}\index{-c, --index\_col \textless{}index\_col\textgreater{}@\spxentry{-c, --index\_col \textless{}index\_col\textgreater{}}!pymethyl-utils-overwrite\_pheno\_data command line option@\spxentry{pymethyl-utils-overwrite\_pheno\_data command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-overwrite-pheno-data-c}}\pysigline{\sphinxbfcode{\sphinxupquote{-c}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}index\_col}}\sphinxcode{\sphinxupquote{~\textless{}index\_col\textgreater{}}}}
Index col when reading csv.  {[}default: 0{]}

\end{fulllineitems}



\section{pkl\_to\_csv}
\label{\detokenize{index:pymethyl-utils-pkl-to-csv}}
Output methylarray pickle to csv.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}utils pkl\PYGZus{}to\PYGZus{}csv \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-utils-pkl\_to\_csv command line option@\spxentry{pymethyl-utils-pkl\_to\_csv command line option}!-i, --input\_pkl \textless{}input\_pkl\textgreater{}@\spxentry{-i, --input\_pkl \textless{}input\_pkl\textgreater{}}}\index{-i, --input\_pkl \textless{}input\_pkl\textgreater{}@\spxentry{-i, --input\_pkl \textless{}input\_pkl\textgreater{}}!pymethyl-utils-pkl\_to\_csv command line option@\spxentry{pymethyl-utils-pkl\_to\_csv command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-pkl-to-csv-i}}\pysigline{\sphinxbfcode{\sphinxupquote{-i}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}input\_pkl}}\sphinxcode{\sphinxupquote{~\textless{}input\_pkl\textgreater{}}}}
Input database for beta and phenotype data.  {[}default: ./final\_preprocessed/methyl\_array.pkl{]}

\end{fulllineitems}

\index{pymethyl-utils-pkl\_to\_csv command line option@\spxentry{pymethyl-utils-pkl\_to\_csv command line option}!-o, --output\_dir \textless{}output\_dir\textgreater{}@\spxentry{-o, --output\_dir \textless{}output\_dir\textgreater{}}}\index{-o, --output\_dir \textless{}output\_dir\textgreater{}@\spxentry{-o, --output\_dir \textless{}output\_dir\textgreater{}}!pymethyl-utils-pkl\_to\_csv command line option@\spxentry{pymethyl-utils-pkl\_to\_csv command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-pkl-to-csv-o}}\pysigline{\sphinxbfcode{\sphinxupquote{-o}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}output\_dir}}\sphinxcode{\sphinxupquote{~\textless{}output\_dir\textgreater{}}}}
Input database for beta and phenotype data.  {[}default: ./final\_preprocessed/{]}

\end{fulllineitems}

\index{pymethyl-utils-pkl\_to\_csv command line option@\spxentry{pymethyl-utils-pkl\_to\_csv command line option}!-c, --col \textless{}col\textgreater{}@\spxentry{-c, --col \textless{}col\textgreater{}}}\index{-c, --col \textless{}col\textgreater{}@\spxentry{-c, --col \textless{}col\textgreater{}}!pymethyl-utils-pkl\_to\_csv command line option@\spxentry{pymethyl-utils-pkl\_to\_csv command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-pkl-to-csv-c}}\pysigline{\sphinxbfcode{\sphinxupquote{-c}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}col}}\sphinxcode{\sphinxupquote{~\textless{}col\textgreater{}}}}
Column to color.  {[}default: {]}

\end{fulllineitems}



\section{print\_number\_sex\_cpgs}
\label{\detokenize{index:pymethyl-utils-print-number-sex-cpgs}}
Print number of non-autosomal CpGs.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}utils print\PYGZus{}number\PYGZus{}sex\PYGZus{}cpgs \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-utils-print\_number\_sex\_cpgs command line option@\spxentry{pymethyl-utils-print\_number\_sex\_cpgs command line option}!-i, --input\_pkl \textless{}input\_pkl\textgreater{}@\spxentry{-i, --input\_pkl \textless{}input\_pkl\textgreater{}}}\index{-i, --input\_pkl \textless{}input\_pkl\textgreater{}@\spxentry{-i, --input\_pkl \textless{}input\_pkl\textgreater{}}!pymethyl-utils-print\_number\_sex\_cpgs command line option@\spxentry{pymethyl-utils-print\_number\_sex\_cpgs command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-print-number-sex-cpgs-i}}\pysigline{\sphinxbfcode{\sphinxupquote{-i}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}input\_pkl}}\sphinxcode{\sphinxupquote{~\textless{}input\_pkl\textgreater{}}}}
Input database for beta and phenotype data.  {[}default: ./final\_preprocessed/methyl\_array.pkl{]}

\end{fulllineitems}

\index{pymethyl-utils-print\_number\_sex\_cpgs command line option@\spxentry{pymethyl-utils-print\_number\_sex\_cpgs command line option}!-a, --array\_type \textless{}array\_type\textgreater{}@\spxentry{-a, --array\_type \textless{}array\_type\textgreater{}}}\index{-a, --array\_type \textless{}array\_type\textgreater{}@\spxentry{-a, --array\_type \textless{}array\_type\textgreater{}}!pymethyl-utils-print\_number\_sex\_cpgs command line option@\spxentry{pymethyl-utils-print\_number\_sex\_cpgs command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-print-number-sex-cpgs-a}}\pysigline{\sphinxbfcode{\sphinxupquote{-a}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}array\_type}}\sphinxcode{\sphinxupquote{~\textless{}array\_type\textgreater{}}}}
Array Type.  {[}default: 450k{]}

\end{fulllineitems}



\section{print\_shape}
\label{\detokenize{index:pymethyl-utils-print-shape}}
Print dimensions of beta matrix.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}utils print\PYGZus{}shape \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-utils-print\_shape command line option@\spxentry{pymethyl-utils-print\_shape command line option}!-i, --input\_pkl \textless{}input\_pkl\textgreater{}@\spxentry{-i, --input\_pkl \textless{}input\_pkl\textgreater{}}}\index{-i, --input\_pkl \textless{}input\_pkl\textgreater{}@\spxentry{-i, --input\_pkl \textless{}input\_pkl\textgreater{}}!pymethyl-utils-print\_shape command line option@\spxentry{pymethyl-utils-print\_shape command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-print-shape-i}}\pysigline{\sphinxbfcode{\sphinxupquote{-i}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}input\_pkl}}\sphinxcode{\sphinxupquote{~\textless{}input\_pkl\textgreater{}}}}
Input database for beta and phenotype data.  {[}default: ./final\_preprocessed/methyl\_array.pkl{]}

\end{fulllineitems}



\section{rate\_regression}
\label{\detokenize{index:pymethyl-utils-rate-regression}}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}utils rate\PYGZus{}regression \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-utils-rate\_regression command line option@\spxentry{pymethyl-utils-rate\_regression command line option}!-i, --input\_csv \textless{}input\_csv\textgreater{}@\spxentry{-i, --input\_csv \textless{}input\_csv\textgreater{}}}\index{-i, --input\_csv \textless{}input\_csv\textgreater{}@\spxentry{-i, --input\_csv \textless{}input\_csv\textgreater{}}!pymethyl-utils-rate\_regression command line option@\spxentry{pymethyl-utils-rate\_regression command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-rate-regression-i}}\pysigline{\sphinxbfcode{\sphinxupquote{-i}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}input\_csv}}\sphinxcode{\sphinxupquote{~\textless{}input\_csv\textgreater{}}}}
Results csv.  {[}default: ./results.csv{]}

\end{fulllineitems}

\index{pymethyl-utils-rate\_regression command line option@\spxentry{pymethyl-utils-rate\_regression command line option}!-c1, --pred\_col \textless{}pred\_col\textgreater{}@\spxentry{-c1, --pred\_col \textless{}pred\_col\textgreater{}}}\index{-c1, --pred\_col \textless{}pred\_col\textgreater{}@\spxentry{-c1, --pred\_col \textless{}pred\_col\textgreater{}}!pymethyl-utils-rate\_regression command line option@\spxentry{pymethyl-utils-rate\_regression command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-rate-regression-c1}}\pysigline{\sphinxbfcode{\sphinxupquote{-c1}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}pred\_col}}\sphinxcode{\sphinxupquote{~\textless{}pred\_col\textgreater{}}}}
Prediction column.  {[}default: y\_pred{]}

\end{fulllineitems}

\index{pymethyl-utils-rate\_regression command line option@\spxentry{pymethyl-utils-rate\_regression command line option}!-c2, --true\_col \textless{}true\_col\textgreater{}@\spxentry{-c2, --true\_col \textless{}true\_col\textgreater{}}}\index{-c2, --true\_col \textless{}true\_col\textgreater{}@\spxentry{-c2, --true\_col \textless{}true\_col\textgreater{}}!pymethyl-utils-rate\_regression command line option@\spxentry{pymethyl-utils-rate\_regression command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-rate-regression-c2}}\pysigline{\sphinxbfcode{\sphinxupquote{-c2}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}true\_col}}\sphinxcode{\sphinxupquote{~\textless{}true\_col\textgreater{}}}}
True Column.  {[}default: y\_true{]}

\end{fulllineitems}



\section{ref\_estimate\_cell\_counts}
\label{\detokenize{index:pymethyl-utils-ref-estimate-cell-counts}}
Reference based cell type estimates.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}utils ref\PYGZus{}estimate\PYGZus{}cell\PYGZus{}counts \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-utils-ref\_estimate\_cell\_counts command line option@\spxentry{pymethyl-utils-ref\_estimate\_cell\_counts command line option}!-ro, --input\_r\_object\_dir \textless{}input\_r\_object\_dir\textgreater{}@\spxentry{-ro, --input\_r\_object\_dir \textless{}input\_r\_object\_dir\textgreater{}}}\index{-ro, --input\_r\_object\_dir \textless{}input\_r\_object\_dir\textgreater{}@\spxentry{-ro, --input\_r\_object\_dir \textless{}input\_r\_object\_dir\textgreater{}}!pymethyl-utils-ref\_estimate\_cell\_counts command line option@\spxentry{pymethyl-utils-ref\_estimate\_cell\_counts command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-ref-estimate-cell-counts-ro}}\pysigline{\sphinxbfcode{\sphinxupquote{-ro}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}input\_r\_object\_dir}}\sphinxcode{\sphinxupquote{~\textless{}input\_r\_object\_dir\textgreater{}}}}
Input directory containing qc data.  {[}default: ./preprocess\_outputs/{]}

\end{fulllineitems}

\index{pymethyl-utils-ref\_estimate\_cell\_counts command line option@\spxentry{pymethyl-utils-ref\_estimate\_cell\_counts command line option}!-a, --algorithm \textless{}algorithm\textgreater{}@\spxentry{-a, --algorithm \textless{}algorithm\textgreater{}}}\index{-a, --algorithm \textless{}algorithm\textgreater{}@\spxentry{-a, --algorithm \textless{}algorithm\textgreater{}}!pymethyl-utils-ref\_estimate\_cell\_counts command line option@\spxentry{pymethyl-utils-ref\_estimate\_cell\_counts command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-ref-estimate-cell-counts-a}}\pysigline{\sphinxbfcode{\sphinxupquote{-a}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}algorithm}}\sphinxcode{\sphinxupquote{~\textless{}algorithm\textgreater{}}}}
Algorithm to run cell type.  {[}default: meffil{]}

\end{fulllineitems}

\index{pymethyl-utils-ref\_estimate\_cell\_counts command line option@\spxentry{pymethyl-utils-ref\_estimate\_cell\_counts command line option}!-ref, --reference \textless{}reference\textgreater{}@\spxentry{-ref, --reference \textless{}reference\textgreater{}}}\index{-ref, --reference \textless{}reference\textgreater{}@\spxentry{-ref, --reference \textless{}reference\textgreater{}}!pymethyl-utils-ref\_estimate\_cell\_counts command line option@\spxentry{pymethyl-utils-ref\_estimate\_cell\_counts command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-ref-estimate-cell-counts-ref}}\pysigline{\sphinxbfcode{\sphinxupquote{-ref}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}reference}}\sphinxcode{\sphinxupquote{~\textless{}reference\textgreater{}}}}
Cell Type Reference.  {[}default: cord blood gse68456{]}

\end{fulllineitems}

\index{pymethyl-utils-ref\_estimate\_cell\_counts command line option@\spxentry{pymethyl-utils-ref\_estimate\_cell\_counts command line option}!-l, --library \textless{}library\textgreater{}@\spxentry{-l, --library \textless{}library\textgreater{}}}\index{-l, --library \textless{}library\textgreater{}@\spxentry{-l, --library \textless{}library\textgreater{}}!pymethyl-utils-ref\_estimate\_cell\_counts command line option@\spxentry{pymethyl-utils-ref\_estimate\_cell\_counts command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-ref-estimate-cell-counts-l}}\pysigline{\sphinxbfcode{\sphinxupquote{-l}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}library}}\sphinxcode{\sphinxupquote{~\textless{}library\textgreater{}}}}
IDOL Library.  {[}default: IDOLOptimizedCpGs450klegacy{]}

\end{fulllineitems}

\index{pymethyl-utils-ref\_estimate\_cell\_counts command line option@\spxentry{pymethyl-utils-ref\_estimate\_cell\_counts command line option}!-o, --output\_csv \textless{}output\_csv\textgreater{}@\spxentry{-o, --output\_csv \textless{}output\_csv\textgreater{}}}\index{-o, --output\_csv \textless{}output\_csv\textgreater{}@\spxentry{-o, --output\_csv \textless{}output\_csv\textgreater{}}!pymethyl-utils-ref\_estimate\_cell\_counts command line option@\spxentry{pymethyl-utils-ref\_estimate\_cell\_counts command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-ref-estimate-cell-counts-o}}\pysigline{\sphinxbfcode{\sphinxupquote{-o}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}output\_csv}}\sphinxcode{\sphinxupquote{~\textless{}output\_csv\textgreater{}}}}
Output cell type estimates.  {[}default: ./added\_cell\_counts/cell\_type\_estimates.csv{]}

\end{fulllineitems}



\section{ref\_free\_cell\_deconv}
\label{\detokenize{index:pymethyl-utils-ref-free-cell-deconv}}
Reference free cell type deconvolution

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}utils ref\PYGZus{}free\PYGZus{}cell\PYGZus{}deconv \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-utils-ref\_free\_cell\_deconv command line option@\spxentry{pymethyl-utils-ref\_free\_cell\_deconv command line option}!-tr, --train\_pkl \textless{}train\_pkl\textgreater{}@\spxentry{-tr, --train\_pkl \textless{}train\_pkl\textgreater{}}}\index{-tr, --train\_pkl \textless{}train\_pkl\textgreater{}@\spxentry{-tr, --train\_pkl \textless{}train\_pkl\textgreater{}}!pymethyl-utils-ref\_free\_cell\_deconv command line option@\spxentry{pymethyl-utils-ref\_free\_cell\_deconv command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-ref-free-cell-deconv-tr}}\pysigline{\sphinxbfcode{\sphinxupquote{-tr}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}train\_pkl}}\sphinxcode{\sphinxupquote{~\textless{}train\_pkl\textgreater{}}}}
Input methyl array.  {[}default: ./train\_val\_test\_sets/train\_methyl\_array.pkl{]}

\end{fulllineitems}

\index{pymethyl-utils-ref\_free\_cell\_deconv command line option@\spxentry{pymethyl-utils-ref\_free\_cell\_deconv command line option}!-te, --test\_pkl \textless{}test\_pkl\textgreater{}@\spxentry{-te, --test\_pkl \textless{}test\_pkl\textgreater{}}}\index{-te, --test\_pkl \textless{}test\_pkl\textgreater{}@\spxentry{-te, --test\_pkl \textless{}test\_pkl\textgreater{}}!pymethyl-utils-ref\_free\_cell\_deconv command line option@\spxentry{pymethyl-utils-ref\_free\_cell\_deconv command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-ref-free-cell-deconv-te}}\pysigline{\sphinxbfcode{\sphinxupquote{-te}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}test\_pkl}}\sphinxcode{\sphinxupquote{~\textless{}test\_pkl\textgreater{}}}}
Input methyl array.  {[}default: ./train\_val\_test\_sets/test\_methyl\_array.pkl{]}

\end{fulllineitems}

\index{pymethyl-utils-ref\_free\_cell\_deconv command line option@\spxentry{pymethyl-utils-ref\_free\_cell\_deconv command line option}!-c, --cell\_type\_columns \textless{}cell\_type\_columns\textgreater{}@\spxentry{-c, --cell\_type\_columns \textless{}cell\_type\_columns\textgreater{}}}\index{-c, --cell\_type\_columns \textless{}cell\_type\_columns\textgreater{}@\spxentry{-c, --cell\_type\_columns \textless{}cell\_type\_columns\textgreater{}}!pymethyl-utils-ref\_free\_cell\_deconv command line option@\spxentry{pymethyl-utils-ref\_free\_cell\_deconv command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-ref-free-cell-deconv-c}}\pysigline{\sphinxbfcode{\sphinxupquote{-c}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}cell\_type\_columns}}\sphinxcode{\sphinxupquote{~\textless{}cell\_type\_columns\textgreater{}}}}
Cell type columns. Leave blank for auto selection, not incorporate reference information.  {[}default: {]}

\end{fulllineitems}

\index{pymethyl-utils-ref\_free\_cell\_deconv command line option@\spxentry{pymethyl-utils-ref\_free\_cell\_deconv command line option}!-k, --n\_cell\_types \textless{}n\_cell\_types\textgreater{}@\spxentry{-k, --n\_cell\_types \textless{}n\_cell\_types\textgreater{}}}\index{-k, --n\_cell\_types \textless{}n\_cell\_types\textgreater{}@\spxentry{-k, --n\_cell\_types \textless{}n\_cell\_types\textgreater{}}!pymethyl-utils-ref\_free\_cell\_deconv command line option@\spxentry{pymethyl-utils-ref\_free\_cell\_deconv command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-ref-free-cell-deconv-k}}\pysigline{\sphinxbfcode{\sphinxupquote{-k}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}n\_cell\_types}}\sphinxcode{\sphinxupquote{~\textless{}n\_cell\_types\textgreater{}}}}
Number of cell types.  {[}default: 7{]}

\end{fulllineitems}

\index{pymethyl-utils-ref\_free\_cell\_deconv command line option@\spxentry{pymethyl-utils-ref\_free\_cell\_deconv command line option}!-a, --analysis \textless{}analysis\textgreater{}@\spxentry{-a, --analysis \textless{}analysis\textgreater{}}}\index{-a, --analysis \textless{}analysis\textgreater{}@\spxentry{-a, --analysis \textless{}analysis\textgreater{}}!pymethyl-utils-ref\_free\_cell\_deconv command line option@\spxentry{pymethyl-utils-ref\_free\_cell\_deconv command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-ref-free-cell-deconv-a}}\pysigline{\sphinxbfcode{\sphinxupquote{-a}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}analysis}}\sphinxcode{\sphinxupquote{~\textless{}analysis\textgreater{}}}}
Analyses to run  {[}default: reffreecellmix{]}

\end{fulllineitems}



\section{remove\_sex}
\label{\detokenize{index:pymethyl-utils-remove-sex}}
Remove non-autosomal CpGs.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}utils remove\PYGZus{}sex \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-utils-remove\_sex command line option@\spxentry{pymethyl-utils-remove\_sex command line option}!-i, --input\_pkl \textless{}input\_pkl\textgreater{}@\spxentry{-i, --input\_pkl \textless{}input\_pkl\textgreater{}}}\index{-i, --input\_pkl \textless{}input\_pkl\textgreater{}@\spxentry{-i, --input\_pkl \textless{}input\_pkl\textgreater{}}!pymethyl-utils-remove\_sex command line option@\spxentry{pymethyl-utils-remove\_sex command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-remove-sex-i}}\pysigline{\sphinxbfcode{\sphinxupquote{-i}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}input\_pkl}}\sphinxcode{\sphinxupquote{~\textless{}input\_pkl\textgreater{}}}}
Input database for beta and phenotype data.  {[}default: ./preprocess\_outputs/methyl\_array.pkl{]}

\end{fulllineitems}

\index{pymethyl-utils-remove\_sex command line option@\spxentry{pymethyl-utils-remove\_sex command line option}!-o, --output\_pkl \textless{}output\_pkl\textgreater{}@\spxentry{-o, --output\_pkl \textless{}output\_pkl\textgreater{}}}\index{-o, --output\_pkl \textless{}output\_pkl\textgreater{}@\spxentry{-o, --output\_pkl \textless{}output\_pkl\textgreater{}}!pymethyl-utils-remove\_sex command line option@\spxentry{pymethyl-utils-remove\_sex command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-remove-sex-o}}\pysigline{\sphinxbfcode{\sphinxupquote{-o}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}output\_pkl}}\sphinxcode{\sphinxupquote{~\textless{}output\_pkl\textgreater{}}}}
Output methyl array autosomal.  {[}default: ./autosomal/methyl\_array.pkl{]}

\end{fulllineitems}

\index{pymethyl-utils-remove\_sex command line option@\spxentry{pymethyl-utils-remove\_sex command line option}!-a, --array\_type \textless{}array\_type\textgreater{}@\spxentry{-a, --array\_type \textless{}array\_type\textgreater{}}}\index{-a, --array\_type \textless{}array\_type\textgreater{}@\spxentry{-a, --array\_type \textless{}array\_type\textgreater{}}!pymethyl-utils-remove\_sex command line option@\spxentry{pymethyl-utils-remove\_sex command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-remove-sex-a}}\pysigline{\sphinxbfcode{\sphinxupquote{-a}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}array\_type}}\sphinxcode{\sphinxupquote{~\textless{}array\_type\textgreater{}}}}
Array Type.  {[}default: 450k{]}

\end{fulllineitems}



\section{remove\_snps}
\label{\detokenize{index:pymethyl-utils-remove-snps}}
Remove SNPs from methylation array.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}utils remove\PYGZus{}snps \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-utils-remove\_snps command line option@\spxentry{pymethyl-utils-remove\_snps command line option}!-i, --input\_pkl \textless{}input\_pkl\textgreater{}@\spxentry{-i, --input\_pkl \textless{}input\_pkl\textgreater{}}}\index{-i, --input\_pkl \textless{}input\_pkl\textgreater{}@\spxentry{-i, --input\_pkl \textless{}input\_pkl\textgreater{}}!pymethyl-utils-remove\_snps command line option@\spxentry{pymethyl-utils-remove\_snps command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-remove-snps-i}}\pysigline{\sphinxbfcode{\sphinxupquote{-i}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}input\_pkl}}\sphinxcode{\sphinxupquote{~\textless{}input\_pkl\textgreater{}}}}
Input database for beta and phenotype data.  {[}default: ./autosomal/methyl\_array.pkl{]}

\end{fulllineitems}

\index{pymethyl-utils-remove\_snps command line option@\spxentry{pymethyl-utils-remove\_snps command line option}!-o, --output\_pkl \textless{}output\_pkl\textgreater{}@\spxentry{-o, --output\_pkl \textless{}output\_pkl\textgreater{}}}\index{-o, --output\_pkl \textless{}output\_pkl\textgreater{}@\spxentry{-o, --output\_pkl \textless{}output\_pkl\textgreater{}}!pymethyl-utils-remove\_snps command line option@\spxentry{pymethyl-utils-remove\_snps command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-remove-snps-o}}\pysigline{\sphinxbfcode{\sphinxupquote{-o}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}output\_pkl}}\sphinxcode{\sphinxupquote{~\textless{}output\_pkl\textgreater{}}}}
Output methyl array autosomal.  {[}default: ./no\_snp/methyl\_array.pkl{]}

\end{fulllineitems}

\index{pymethyl-utils-remove\_snps command line option@\spxentry{pymethyl-utils-remove\_snps command line option}!-a, --array\_type \textless{}array\_type\textgreater{}@\spxentry{-a, --array\_type \textless{}array\_type\textgreater{}}}\index{-a, --array\_type \textless{}array\_type\textgreater{}@\spxentry{-a, --array\_type \textless{}array\_type\textgreater{}}!pymethyl-utils-remove\_snps command line option@\spxentry{pymethyl-utils-remove\_snps command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-remove-snps-a}}\pysigline{\sphinxbfcode{\sphinxupquote{-a}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}array\_type}}\sphinxcode{\sphinxupquote{~\textless{}array\_type\textgreater{}}}}
Array Type.  {[}default: 450k{]}

\end{fulllineitems}



\section{set\_part\_array\_background}
\label{\detokenize{index:pymethyl-utils-set-part-array-background}}
Set subset of CpGs from beta matrix to background values.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}utils set\PYGZus{}part\PYGZus{}array\PYGZus{}background \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-utils-set\_part\_array\_background command line option@\spxentry{pymethyl-utils-set\_part\_array\_background command line option}!-i, --input\_pkl \textless{}input\_pkl\textgreater{}@\spxentry{-i, --input\_pkl \textless{}input\_pkl\textgreater{}}}\index{-i, --input\_pkl \textless{}input\_pkl\textgreater{}@\spxentry{-i, --input\_pkl \textless{}input\_pkl\textgreater{}}!pymethyl-utils-set\_part\_array\_background command line option@\spxentry{pymethyl-utils-set\_part\_array\_background command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-set-part-array-background-i}}\pysigline{\sphinxbfcode{\sphinxupquote{-i}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}input\_pkl}}\sphinxcode{\sphinxupquote{~\textless{}input\_pkl\textgreater{}}}}
Input methyl array.  {[}default: ./final\_preprocessed/methyl\_array.pkl{]}

\end{fulllineitems}

\index{pymethyl-utils-set\_part\_array\_background command line option@\spxentry{pymethyl-utils-set\_part\_array\_background command line option}!-c, --cpg\_pkl \textless{}cpg\_pkl\textgreater{}@\spxentry{-c, --cpg\_pkl \textless{}cpg\_pkl\textgreater{}}}\index{-c, --cpg\_pkl \textless{}cpg\_pkl\textgreater{}@\spxentry{-c, --cpg\_pkl \textless{}cpg\_pkl\textgreater{}}!pymethyl-utils-set\_part\_array\_background command line option@\spxentry{pymethyl-utils-set\_part\_array\_background command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-set-part-array-background-c}}\pysigline{\sphinxbfcode{\sphinxupquote{-c}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}cpg\_pkl}}\sphinxcode{\sphinxupquote{~\textless{}cpg\_pkl\textgreater{}}}}
Pickled numpy array for subsetting.  {[}default: ./subset\_cpgs.pkl{]}

\end{fulllineitems}

\index{pymethyl-utils-set\_part\_array\_background command line option@\spxentry{pymethyl-utils-set\_part\_array\_background command line option}!-o, --output\_pkl \textless{}output\_pkl\textgreater{}@\spxentry{-o, --output\_pkl \textless{}output\_pkl\textgreater{}}}\index{-o, --output\_pkl \textless{}output\_pkl\textgreater{}@\spxentry{-o, --output\_pkl \textless{}output\_pkl\textgreater{}}!pymethyl-utils-set\_part\_array\_background command line option@\spxentry{pymethyl-utils-set\_part\_array\_background command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-set-part-array-background-o}}\pysigline{\sphinxbfcode{\sphinxupquote{-o}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}output\_pkl}}\sphinxcode{\sphinxupquote{~\textless{}output\_pkl\textgreater{}}}}
Output methyl array external validation.  {[}default: ./removal/methyl\_array.pkl{]}

\end{fulllineitems}



\section{stratify}
\label{\detokenize{index:pymethyl-utils-stratify}}
Split methylation array by key and store.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}utils stratify \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-utils-stratify command line option@\spxentry{pymethyl-utils-stratify command line option}!-i, --input\_pkl \textless{}input\_pkl\textgreater{}@\spxentry{-i, --input\_pkl \textless{}input\_pkl\textgreater{}}}\index{-i, --input\_pkl \textless{}input\_pkl\textgreater{}@\spxentry{-i, --input\_pkl \textless{}input\_pkl\textgreater{}}!pymethyl-utils-stratify command line option@\spxentry{pymethyl-utils-stratify command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-stratify-i}}\pysigline{\sphinxbfcode{\sphinxupquote{-i}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}input\_pkl}}\sphinxcode{\sphinxupquote{~\textless{}input\_pkl\textgreater{}}}}
Input database for beta and phenotype data.  {[}default: ./final\_preprocessed/methyl\_array.pkl{]}

\end{fulllineitems}

\index{pymethyl-utils-stratify command line option@\spxentry{pymethyl-utils-stratify command line option}!-k, --key \textless{}key\textgreater{}@\spxentry{-k, --key \textless{}key\textgreater{}}}\index{-k, --key \textless{}key\textgreater{}@\spxentry{-k, --key \textless{}key\textgreater{}}!pymethyl-utils-stratify command line option@\spxentry{pymethyl-utils-stratify command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-stratify-k}}\pysigline{\sphinxbfcode{\sphinxupquote{-k}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}key}}\sphinxcode{\sphinxupquote{~\textless{}key\textgreater{}}}}
Key to split on.  {[}default: disease{]}

\end{fulllineitems}

\index{pymethyl-utils-stratify command line option@\spxentry{pymethyl-utils-stratify command line option}!-o, --output\_dir \textless{}output\_dir\textgreater{}@\spxentry{-o, --output\_dir \textless{}output\_dir\textgreater{}}}\index{-o, --output\_dir \textless{}output\_dir\textgreater{}@\spxentry{-o, --output\_dir \textless{}output\_dir\textgreater{}}!pymethyl-utils-stratify command line option@\spxentry{pymethyl-utils-stratify command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-stratify-o}}\pysigline{\sphinxbfcode{\sphinxupquote{-o}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}output\_dir}}\sphinxcode{\sphinxupquote{~\textless{}output\_dir\textgreater{}}}}
Output directory for stratified.  {[}default: ./stratified/{]}

\end{fulllineitems}



\section{subset\_array}
\label{\detokenize{index:pymethyl-utils-subset-array}}
Only retain certain number of CpGs from methylation array.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}utils subset\PYGZus{}array \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-utils-subset\_array command line option@\spxentry{pymethyl-utils-subset\_array command line option}!-i, --input\_pkl \textless{}input\_pkl\textgreater{}@\spxentry{-i, --input\_pkl \textless{}input\_pkl\textgreater{}}}\index{-i, --input\_pkl \textless{}input\_pkl\textgreater{}@\spxentry{-i, --input\_pkl \textless{}input\_pkl\textgreater{}}!pymethyl-utils-subset\_array command line option@\spxentry{pymethyl-utils-subset\_array command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-subset-array-i}}\pysigline{\sphinxbfcode{\sphinxupquote{-i}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}input\_pkl}}\sphinxcode{\sphinxupquote{~\textless{}input\_pkl\textgreater{}}}}
Input methyl array.  {[}default: ./final\_preprocessed/methyl\_array.pkl{]}

\end{fulllineitems}

\index{pymethyl-utils-subset\_array command line option@\spxentry{pymethyl-utils-subset\_array command line option}!-c, --cpg\_pkl \textless{}cpg\_pkl\textgreater{}@\spxentry{-c, --cpg\_pkl \textless{}cpg\_pkl\textgreater{}}}\index{-c, --cpg\_pkl \textless{}cpg\_pkl\textgreater{}@\spxentry{-c, --cpg\_pkl \textless{}cpg\_pkl\textgreater{}}!pymethyl-utils-subset\_array command line option@\spxentry{pymethyl-utils-subset\_array command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-subset-array-c}}\pysigline{\sphinxbfcode{\sphinxupquote{-c}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}cpg\_pkl}}\sphinxcode{\sphinxupquote{~\textless{}cpg\_pkl\textgreater{}}}}
Pickled numpy array for subsetting.  {[}default: ./subset\_cpgs.pkl{]}

\end{fulllineitems}

\index{pymethyl-utils-subset\_array command line option@\spxentry{pymethyl-utils-subset\_array command line option}!-o, --output\_pkl \textless{}output\_pkl\textgreater{}@\spxentry{-o, --output\_pkl \textless{}output\_pkl\textgreater{}}}\index{-o, --output\_pkl \textless{}output\_pkl\textgreater{}@\spxentry{-o, --output\_pkl \textless{}output\_pkl\textgreater{}}!pymethyl-utils-subset\_array command line option@\spxentry{pymethyl-utils-subset\_array command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-subset-array-o}}\pysigline{\sphinxbfcode{\sphinxupquote{-o}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}output\_pkl}}\sphinxcode{\sphinxupquote{~\textless{}output\_pkl\textgreater{}}}}
Output methyl array external validation.  {[}default: ./subset/methyl\_array.pkl{]}

\end{fulllineitems}



\section{train\_test\_val\_split}
\label{\detokenize{index:pymethyl-utils-train-test-val-split}}
Split methylation array into train, test, val.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}utils train\PYGZus{}test\PYGZus{}val\PYGZus{}split \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-utils-train\_test\_val\_split command line option@\spxentry{pymethyl-utils-train\_test\_val\_split command line option}!-i, --input\_pkl \textless{}input\_pkl\textgreater{}@\spxentry{-i, --input\_pkl \textless{}input\_pkl\textgreater{}}}\index{-i, --input\_pkl \textless{}input\_pkl\textgreater{}@\spxentry{-i, --input\_pkl \textless{}input\_pkl\textgreater{}}!pymethyl-utils-train\_test\_val\_split command line option@\spxentry{pymethyl-utils-train\_test\_val\_split command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-train-test-val-split-i}}\pysigline{\sphinxbfcode{\sphinxupquote{-i}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}input\_pkl}}\sphinxcode{\sphinxupquote{~\textless{}input\_pkl\textgreater{}}}}
Input database for beta and phenotype data.  {[}default: ./final\_preprocessed/methyl\_array.pkl{]}

\end{fulllineitems}

\index{pymethyl-utils-train\_test\_val\_split command line option@\spxentry{pymethyl-utils-train\_test\_val\_split command line option}!-o, --output\_dir \textless{}output\_dir\textgreater{}@\spxentry{-o, --output\_dir \textless{}output\_dir\textgreater{}}}\index{-o, --output\_dir \textless{}output\_dir\textgreater{}@\spxentry{-o, --output\_dir \textless{}output\_dir\textgreater{}}!pymethyl-utils-train\_test\_val\_split command line option@\spxentry{pymethyl-utils-train\_test\_val\_split command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-train-test-val-split-o}}\pysigline{\sphinxbfcode{\sphinxupquote{-o}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}output\_dir}}\sphinxcode{\sphinxupquote{~\textless{}output\_dir\textgreater{}}}}
Output directory for training, testing, and validation sets.  {[}default: ./train\_val\_test\_sets/{]}

\end{fulllineitems}

\index{pymethyl-utils-train\_test\_val\_split command line option@\spxentry{pymethyl-utils-train\_test\_val\_split command line option}!-tp, --train\_percent \textless{}train\_percent\textgreater{}@\spxentry{-tp, --train\_percent \textless{}train\_percent\textgreater{}}}\index{-tp, --train\_percent \textless{}train\_percent\textgreater{}@\spxentry{-tp, --train\_percent \textless{}train\_percent\textgreater{}}!pymethyl-utils-train\_test\_val\_split command line option@\spxentry{pymethyl-utils-train\_test\_val\_split command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-train-test-val-split-tp}}\pysigline{\sphinxbfcode{\sphinxupquote{-tp}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}train\_percent}}\sphinxcode{\sphinxupquote{~\textless{}train\_percent\textgreater{}}}}
Percent data training on.  {[}default: 0.8{]}

\end{fulllineitems}

\index{pymethyl-utils-train\_test\_val\_split command line option@\spxentry{pymethyl-utils-train\_test\_val\_split command line option}!-vp, --val\_percent \textless{}val\_percent\textgreater{}@\spxentry{-vp, --val\_percent \textless{}val\_percent\textgreater{}}}\index{-vp, --val\_percent \textless{}val\_percent\textgreater{}@\spxentry{-vp, --val\_percent \textless{}val\_percent\textgreater{}}!pymethyl-utils-train\_test\_val\_split command line option@\spxentry{pymethyl-utils-train\_test\_val\_split command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-train-test-val-split-vp}}\pysigline{\sphinxbfcode{\sphinxupquote{-vp}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}val\_percent}}\sphinxcode{\sphinxupquote{~\textless{}val\_percent\textgreater{}}}}
Percent of training data that comprises validation set.  {[}default: 0.1{]}

\end{fulllineitems}

\index{pymethyl-utils-train\_test\_val\_split command line option@\spxentry{pymethyl-utils-train\_test\_val\_split command line option}!-cat, --categorical@\spxentry{-cat, --categorical}}\index{-cat, --categorical@\spxentry{-cat, --categorical}!pymethyl-utils-train\_test\_val\_split command line option@\spxentry{pymethyl-utils-train\_test\_val\_split command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-train-test-val-split-cat}}\pysigline{\sphinxbfcode{\sphinxupquote{-cat}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}categorical}}\sphinxcode{\sphinxupquote{}}}
Multi-class prediction.  {[}default: False{]}

\end{fulllineitems}

\index{pymethyl-utils-train\_test\_val\_split command line option@\spxentry{pymethyl-utils-train\_test\_val\_split command line option}!-do, --disease\_only@\spxentry{-do, --disease\_only}}\index{-do, --disease\_only@\spxentry{-do, --disease\_only}!pymethyl-utils-train\_test\_val\_split command line option@\spxentry{pymethyl-utils-train\_test\_val\_split command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-train-test-val-split-do}}\pysigline{\sphinxbfcode{\sphinxupquote{-do}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}disease\_only}}\sphinxcode{\sphinxupquote{}}}
Only look at disease, or text before subtype\_delimiter.

\end{fulllineitems}

\index{pymethyl-utils-train\_test\_val\_split command line option@\spxentry{pymethyl-utils-train\_test\_val\_split command line option}!-k, --key \textless{}key\textgreater{}@\spxentry{-k, --key \textless{}key\textgreater{}}}\index{-k, --key \textless{}key\textgreater{}@\spxentry{-k, --key \textless{}key\textgreater{}}!pymethyl-utils-train\_test\_val\_split command line option@\spxentry{pymethyl-utils-train\_test\_val\_split command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-train-test-val-split-k}}\pysigline{\sphinxbfcode{\sphinxupquote{-k}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}key}}\sphinxcode{\sphinxupquote{~\textless{}key\textgreater{}}}}
Key to split on.  {[}default: disease{]}

\end{fulllineitems}

\index{pymethyl-utils-train\_test\_val\_split command line option@\spxentry{pymethyl-utils-train\_test\_val\_split command line option}!-sd, --subtype\_delimiter \textless{}subtype\_delimiter\textgreater{}@\spxentry{-sd, --subtype\_delimiter \textless{}subtype\_delimiter\textgreater{}}}\index{-sd, --subtype\_delimiter \textless{}subtype\_delimiter\textgreater{}@\spxentry{-sd, --subtype\_delimiter \textless{}subtype\_delimiter\textgreater{}}!pymethyl-utils-train\_test\_val\_split command line option@\spxentry{pymethyl-utils-train\_test\_val\_split command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-train-test-val-split-sd}}\pysigline{\sphinxbfcode{\sphinxupquote{-sd}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}subtype\_delimiter}}\sphinxcode{\sphinxupquote{~\textless{}subtype\_delimiter\textgreater{}}}}
Delimiter for disease extraction.  {[}default: ,{]}

\end{fulllineitems}



\section{write\_cpgs}
\label{\detokenize{index:pymethyl-utils-write-cpgs}}
Write CpGs in methylation array to file.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pymethyl\PYGZhy{}utils write\PYGZus{}cpgs \PYG{o}{[}OPTIONS\PYG{o}{]}
\end{sphinxVerbatim}
\subsubsection*{Options}
\index{pymethyl-utils-write\_cpgs command line option@\spxentry{pymethyl-utils-write\_cpgs command line option}!-i, --input\_pkl \textless{}input\_pkl\textgreater{}@\spxentry{-i, --input\_pkl \textless{}input\_pkl\textgreater{}}}\index{-i, --input\_pkl \textless{}input\_pkl\textgreater{}@\spxentry{-i, --input\_pkl \textless{}input\_pkl\textgreater{}}!pymethyl-utils-write\_cpgs command line option@\spxentry{pymethyl-utils-write\_cpgs command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-write-cpgs-i}}\pysigline{\sphinxbfcode{\sphinxupquote{-i}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}input\_pkl}}\sphinxcode{\sphinxupquote{~\textless{}input\_pkl\textgreater{}}}}
Input methyl array.  {[}default: ./final\_preprocessed/methyl\_array.pkl{]}

\end{fulllineitems}

\index{pymethyl-utils-write\_cpgs command line option@\spxentry{pymethyl-utils-write\_cpgs command line option}!-c, --cpg\_pkl \textless{}cpg\_pkl\textgreater{}@\spxentry{-c, --cpg\_pkl \textless{}cpg\_pkl\textgreater{}}}\index{-c, --cpg\_pkl \textless{}cpg\_pkl\textgreater{}@\spxentry{-c, --cpg\_pkl \textless{}cpg\_pkl\textgreater{}}!pymethyl-utils-write\_cpgs command line option@\spxentry{pymethyl-utils-write\_cpgs command line option}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{index:cmdoption-pymethyl-utils-write-cpgs-c}}\pysigline{\sphinxbfcode{\sphinxupquote{-c}}\sphinxcode{\sphinxupquote{}}\sphinxcode{\sphinxupquote{,~}}\sphinxbfcode{\sphinxupquote{-{-}cpg\_pkl}}\sphinxcode{\sphinxupquote{~\textless{}cpg\_pkl\textgreater{}}}}
Pickled numpy array for subsetting.  {[}default: ./subset\_cpgs.pkl{]}

\end{fulllineitems}



\chapter{Indices and tables}
\label{\detokenize{index:indices-and-tables}}\begin{itemize}
\item {} 
\DUrole{xref,std,std-ref}{genindex}

\item {} 
\DUrole{xref,std,std-ref}{modindex}

\item {} 
\DUrole{xref,std,std-ref}{search}

\end{itemize}


\renewcommand{\indexname}{Python Module Index}
\begin{sphinxtheindex}
\let\bigletter\sphinxstyleindexlettergroup
\bigletter{p}
\item\relax\sphinxstyleindexentry{pymethylprocess.general\_machine\_learning}\sphinxstyleindexpageref{index:\detokenize{module-pymethylprocess.general_machine_learning}}
\item\relax\sphinxstyleindexentry{pymethylprocess.meffil\_functions}\sphinxstyleindexpageref{index:\detokenize{module-pymethylprocess.meffil_functions}}
\item\relax\sphinxstyleindexentry{pymethylprocess.MethylationDataTypes}\sphinxstyleindexpageref{index:\detokenize{module-pymethylprocess.MethylationDataTypes}}
\item\relax\sphinxstyleindexentry{pymethylprocess.PreProcessDataTypes}\sphinxstyleindexpageref{index:\detokenize{module-pymethylprocess.PreProcessDataTypes}}
\end{sphinxtheindex}

\renewcommand{\indexname}{Index}
\printindex
\end{document}