

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Welcome to PyMethylProcess’s documentation! &mdash; PyMethylProcess 0.1 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script type="text/javascript" src="_static/jquery.js"></script>
        <script type="text/javascript" src="_static/underscore.js"></script>
        <script type="text/javascript" src="_static/doctools.js"></script>
        <script type="text/javascript" src="_static/language_data.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="#" class="icon icon-home"> PyMethylProcess
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <!-- Local TOC -->
              <div class="local-toc"><ul>
<li><a class="reference internal" href="#">Welcome to PyMethylProcess’s documentation!</a><ul>
<li><a class="reference internal" href="#preprocessdatatypes-py">PreProcessDataTypes.py</a><ul>
</ul>
</li>
<li><a class="reference internal" href="#methylationdatatypes-py">MethylationDataTypes.py</a><ul>
</ul>
</li>
<li><a class="reference internal" href="#meffil-functions-py">meffil_functions.py</a></li>
<li><a class="reference internal" href="#general-machine-learning-py">general_machine_learning.py</a></li>
<li><a class="reference internal" href="#pymethyl-install">pymethyl-install</a><ul>
<li><a class="reference internal" href="#pymethyl-install-change-gcc-path">change_gcc_path</a></li>
<li><a class="reference internal" href="#pymethyl-install-install-bioconductor">install_bioconductor</a></li>
<li><a class="reference internal" href="#pymethyl-install-install-custom">install_custom</a></li>
<li><a class="reference internal" href="#pymethyl-install-install-meffil">install_meffil</a></li>
<li><a class="reference internal" href="#pymethyl-install-install-minfi-others">install_minfi_others</a></li>
<li><a class="reference internal" href="#pymethyl-install-install-r-packages">install_r_packages</a></li>
<li><a class="reference internal" href="#pymethyl-install-install-some-deps">install_some_deps</a></li>
<li><a class="reference internal" href="#pymethyl-install-install-tcga-biolinks">install_tcga_biolinks</a></li>
</ul>
</li>
<li><a class="reference internal" href="#pymethyl-visualize">pymethyl-visualize</a><ul>
<li><a class="reference internal" href="#pymethyl-visualize-plot-cell-type-results">plot_cell_type_results</a></li>
<li><a class="reference internal" href="#pymethyl-visualize-plot-heatmap">plot_heatmap</a></li>
<li><a class="reference internal" href="#pymethyl-visualize-transform-plot">transform_plot</a></li>
</ul>
</li>
<li><a class="reference internal" href="#pymethyl-preprocess">pymethyl-preprocess</a><ul>
<li><a class="reference internal" href="#pymethyl-preprocess-batch-deploy-preprocess">batch_deploy_preprocess</a></li>
<li><a class="reference internal" href="#pymethyl-preprocess-combine-methylation-arrays">combine_methylation_arrays</a></li>
<li><a class="reference internal" href="#pymethyl-preprocess-concat-sample-sheets">concat_sample_sheets</a></li>
<li><a class="reference internal" href="#pymethyl-preprocess-create-sample-sheet">create_sample_sheet</a></li>
<li><a class="reference internal" href="#pymethyl-preprocess-download-clinical">download_clinical</a></li>
<li><a class="reference internal" href="#pymethyl-preprocess-download-geo">download_geo</a></li>
<li><a class="reference internal" href="#pymethyl-preprocess-download-tcga">download_tcga</a></li>
<li><a class="reference internal" href="#pymethyl-preprocess-feature-select">feature_select</a></li>
<li><a class="reference internal" href="#pymethyl-preprocess-get-categorical-distribution">get_categorical_distribution</a></li>
<li><a class="reference internal" href="#pymethyl-preprocess-imputation-pipeline">imputation_pipeline</a></li>
<li><a class="reference internal" href="#pymethyl-preprocess-meffil-encode">meffil_encode</a></li>
<li><a class="reference internal" href="#pymethyl-preprocess-merge-sample-sheets">merge_sample_sheets</a></li>
<li><a class="reference internal" href="#pymethyl-preprocess-na-report">na_report</a></li>
<li><a class="reference internal" href="#pymethyl-preprocess-preprocess-pipeline">preprocess_pipeline</a></li>
<li><a class="reference internal" href="#pymethyl-preprocess-remove-diseases">remove_diseases</a></li>
<li><a class="reference internal" href="#pymethyl-preprocess-split-preprocess-input-by-subtype">split_preprocess_input_by_subtype</a></li>
</ul>
</li>
<li><a class="reference internal" href="#pymethyl-utils">pymethyl-utils</a><ul>
<li><a class="reference internal" href="#pymethyl-utils-backup-pkl">backup_pkl</a></li>
<li><a class="reference internal" href="#pymethyl-utils-bin-column">bin_column</a></li>
<li><a class="reference internal" href="#pymethyl-utils-concat-csv">concat_csv</a></li>
<li><a class="reference internal" href="#pymethyl-utils-counts">counts</a></li>
<li><a class="reference internal" href="#pymethyl-utils-create-external-validation-set">create_external_validation_set</a></li>
<li><a class="reference internal" href="#pymethyl-utils-est-age">est_age</a></li>
<li><a class="reference internal" href="#pymethyl-utils-feature-select-train-val-test">feature_select_train_val_test</a></li>
<li><a class="reference internal" href="#pymethyl-utils-fix-key">fix_key</a></li>
<li><a class="reference internal" href="#pymethyl-utils-modify-pheno-data">modify_pheno_data</a></li>
<li><a class="reference internal" href="#pymethyl-utils-move-jpg">move_jpg</a></li>
<li><a class="reference internal" href="#pymethyl-utils-overwrite-pheno-data">overwrite_pheno_data</a></li>
<li><a class="reference internal" href="#pymethyl-utils-pkl-to-csv">pkl_to_csv</a></li>
<li><a class="reference internal" href="#pymethyl-utils-print-number-sex-cpgs">print_number_sex_cpgs</a></li>
<li><a class="reference internal" href="#pymethyl-utils-print-shape">print_shape</a></li>
<li><a class="reference internal" href="#pymethyl-utils-rate-regression">rate_regression</a></li>
<li><a class="reference internal" href="#pymethyl-utils-ref-estimate-cell-counts">ref_estimate_cell_counts</a></li>
<li><a class="reference internal" href="#pymethyl-utils-ref-free-cell-deconv">ref_free_cell_deconv</a></li>
<li><a class="reference internal" href="#pymethyl-utils-remove-sex">remove_sex</a></li>
<li><a class="reference internal" href="#pymethyl-utils-remove-snps">remove_snps</a></li>
<li><a class="reference internal" href="#pymethyl-utils-set-part-array-background">set_part_array_background</a></li>
<li><a class="reference internal" href="#pymethyl-utils-stratify">stratify</a></li>
<li><a class="reference internal" href="#pymethyl-utils-subset-array">subset_array</a></li>
<li><a class="reference internal" href="#pymethyl-utils-train-test-val-split">train_test_val_split</a></li>
<li><a class="reference internal" href="#pymethyl-utils-write-cpgs">write_cpgs</a></li>
</ul>
</li>
</ul>
</li>
<li><a class="reference internal" href="#indices-and-tables">Indices and tables</a></li>
</ul>
</div>
            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="#">PyMethylProcess</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="#">Docs</a> &raquo;</li>
        
      <li>Welcome to PyMethylProcess’s documentation!</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/index.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="welcome-to-pymethylprocess-s-documentation">
<h1>Welcome to PyMethylProcess’s documentation!<a class="headerlink" href="#welcome-to-pymethylprocess-s-documentation" title="Permalink to this headline">¶</a></h1>
<p><a class="reference external" href="https://github.com/Christensen-Lab-Dartmouth/PyMethylProcess">https://github.com/Christensen-Lab-Dartmouth/PyMethylProcess</a></p>
<p>To get started, download pymethylprocess using Docker (joshualevy44/pymethylprocess) or PIP (pymethylprocess) and run pymethyl-install_r_dependencies.</p>
<p>See README.md in Github repository for more install directions and for example scripts for running the pipeline (not all datasets may be available on GEO at this time).</p>
<p>There is both an API and CLI available for use. Examples for CLI usage can be found in ./example_scripts.</p>
<div class="toctree-wrapper compound">
</div>
<a class="reference internal image-reference" href="_images/pymethylprocess_overview.jpeg"><img alt="Download" class="align-center" src="_images/pymethylprocess_overview.jpeg" style="width: 480.0px; height: 360.0px;" /></a>
<a class="reference internal image-reference" href="_images/pipeline-download.jpeg"><img alt="Download" class="align-center" src="_images/pipeline-download.jpeg" style="width: 480.0px; height: 360.0px;" /></a>
<a class="reference internal image-reference" href="_images/pipeline-format.jpeg"><img alt="Format" class="align-center" src="_images/pipeline-format.jpeg" style="width: 480.0px; height: 360.0px;" /></a>
<a class="reference internal image-reference" href="_images/pipeline-preprocess.jpeg"><img alt="Preprocess" class="align-center" src="_images/pipeline-preprocess.jpeg" style="width: 480.0px; height: 360.0px;" /></a>
<a class="reference internal image-reference" href="_images/pipeline-visualize.jpeg"><img alt="Visualize" class="align-center" src="_images/pipeline-visualize.jpeg" style="width: 480.0px; height: 360.0px;" /></a>
<a class="reference internal image-reference" href="_images/pipeline-train-test-split.jpeg"><img alt="TrainTestSplit" class="align-center" src="_images/pipeline-train-test-split.jpeg" style="width: 480.0px; height: 360.0px;" /></a>
<span class="target" id="module-pymethylprocess.PreProcessDataTypes"></span><div class="section" id="preprocessdatatypes-py">
<h2>PreProcessDataTypes.py<a class="headerlink" href="#preprocessdatatypes-py" title="Permalink to this headline">¶</a></h2>
<p>Contains datatypes core to downloading IDATs, preprocessing IDATs and samplesheets.</p>
<dl class="class">
<dt id="pymethylprocess.PreProcessDataTypes.PreProcessIDAT">
<em class="property">class </em><code class="descclassname">pymethylprocess.PreProcessDataTypes.</code><code class="descname">PreProcessIDAT</code><span class="sig-paren">(</span><em>idat_dir</em>, <em>minfi=None</em>, <em>enmix=None</em>, <em>base=None</em>, <em>meffil=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/PreProcessDataTypes.html#PreProcessIDAT"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.PreProcessDataTypes.PreProcessIDAT" title="Permalink to this definition">¶</a></dt>
<dd><p>Class that will preprocess IDATs using R pipelines.</p>
<dl class="docutils">
<dt>idat_dir</dt>
<dd>Location of idats or samplesheet csv.</dd>
<dt>minfi</dt>
<dd>Rpy2 importr minfi library, default to None will load through rpy2</dd>
<dt>enmix</dt>
<dd>Rpy2 importr enmix library, default to None will load through rpy2</dd>
<dt>base</dt>
<dd>Rpy2 importr base library, default to None will load through rpy2</dd>
<dt>meffil</dt>
<dd>Rpy2 importr meffil library, default to None will load through rpy2</dd>
</dl>
<p class="rubric">Methods</p>
<table border="1" class="longtable docutils">
<colgroup>
<col width="10%" />
<col width="90%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td><a class="reference internal" href="#pymethylprocess.PreProcessDataTypes.PreProcessIDAT.export_csv" title="pymethylprocess.PreProcessDataTypes.PreProcessIDAT.export_csv"><code class="xref py py-obj docutils literal notranslate"><span class="pre">export_csv</span></code></a>(self,&nbsp;output_dir)</td>
<td>Export pheno and beta dataframes to CSVs</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#pymethylprocess.PreProcessDataTypes.PreProcessIDAT.export_pickle" title="pymethylprocess.PreProcessDataTypes.PreProcessIDAT.export_pickle"><code class="xref py py-obj docutils literal notranslate"><span class="pre">export_pickle</span></code></a>(self,&nbsp;output_pickle[,&nbsp;disease])</td>
<td>Export pheno and beta dataframes to pickle, stored in python dict that can be loaded into MethylationArray</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#pymethylprocess.PreProcessDataTypes.PreProcessIDAT.export_sql" title="pymethylprocess.PreProcessDataTypes.PreProcessIDAT.export_sql"><code class="xref py py-obj docutils literal notranslate"><span class="pre">export_sql</span></code></a>(self,&nbsp;output_db[,&nbsp;disease])</td>
<td>Export pheno and beta dataframes to SQL</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#pymethylprocess.PreProcessDataTypes.PreProcessIDAT.extract_manifest" title="pymethylprocess.PreProcessDataTypes.PreProcessIDAT.extract_manifest"><code class="xref py py-obj docutils literal notranslate"><span class="pre">extract_manifest</span></code></a>(self)</td>
<td>Get manifest from RGSet.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#pymethylprocess.PreProcessDataTypes.PreProcessIDAT.extract_pheno_data" title="pymethylprocess.PreProcessDataTypes.PreProcessIDAT.extract_pheno_data"><code class="xref py py-obj docutils literal notranslate"><span class="pre">extract_pheno_data</span></code></a>(self[,&nbsp;methylset])</td>
<td>Extract pheno data from MSet or RGSet, minfi.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#pymethylprocess.PreProcessDataTypes.PreProcessIDAT.filter_beta" title="pymethylprocess.PreProcessDataTypes.PreProcessIDAT.filter_beta"><code class="xref py py-obj docutils literal notranslate"><span class="pre">filter_beta</span></code></a>(self)</td>
<td>After creating beta, filter out outliers.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#pymethylprocess.PreProcessDataTypes.PreProcessIDAT.get_beta" title="pymethylprocess.PreProcessDataTypes.PreProcessIDAT.get_beta"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_beta</span></code></a>(self)</td>
<td>Get beta value matrix from minfi after finding RSet.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#pymethylprocess.PreProcessDataTypes.PreProcessIDAT.get_meth" title="pymethylprocess.PreProcessDataTypes.PreProcessIDAT.get_meth"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_meth</span></code></a>(self)</td>
<td>Get methylation intensity matrix from MSet</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#pymethylprocess.PreProcessDataTypes.PreProcessIDAT.get_unmeth" title="pymethylprocess.PreProcessDataTypes.PreProcessIDAT.get_unmeth"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_unmeth</span></code></a>(self)</td>
<td>Get unmethylated intensity matrix from MSet</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#pymethylprocess.PreProcessDataTypes.PreProcessIDAT.load_idats" title="pymethylprocess.PreProcessDataTypes.PreProcessIDAT.load_idats"><code class="xref py py-obj docutils literal notranslate"><span class="pre">load_idats</span></code></a>(self)</td>
<td>For minfi pipeline, load IDATs from specified idat_dir.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#pymethylprocess.PreProcessDataTypes.PreProcessIDAT.move_jpg" title="pymethylprocess.PreProcessDataTypes.PreProcessIDAT.move_jpg"><code class="xref py py-obj docutils literal notranslate"><span class="pre">move_jpg</span></code></a>(self)</td>
<td>Move jpeg files from current working directory to the idat directory.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#pymethylprocess.PreProcessDataTypes.PreProcessIDAT.output_pheno_beta" title="pymethylprocess.PreProcessDataTypes.PreProcessIDAT.output_pheno_beta"><code class="xref py py-obj docutils literal notranslate"><span class="pre">output_pheno_beta</span></code></a>(self[,&nbsp;meffil])</td>
<td>Get pheno and beta dataframe objects stored as attributes for input to MethylationArray object.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#pymethylprocess.PreProcessDataTypes.PreProcessIDAT.plot_original_qc" title="pymethylprocess.PreProcessDataTypes.PreProcessIDAT.plot_original_qc"><code class="xref py py-obj docutils literal notranslate"><span class="pre">plot_original_qc</span></code></a>(self,&nbsp;output_dir)</td>
<td>Plot QC results from ENmix pipeline and possible minfi.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#pymethylprocess.PreProcessDataTypes.PreProcessIDAT.plot_qc_metrics" title="pymethylprocess.PreProcessDataTypes.PreProcessIDAT.plot_qc_metrics"><code class="xref py py-obj docutils literal notranslate"><span class="pre">plot_qc_metrics</span></code></a>(self,&nbsp;output_dir)</td>
<td>Plot QC results from ENmix pipeline and possible minfi.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#pymethylprocess.PreProcessDataTypes.PreProcessIDAT.preprocessENmix" title="pymethylprocess.PreProcessDataTypes.PreProcessIDAT.preprocessENmix"><code class="xref py py-obj docutils literal notranslate"><span class="pre">preprocessENmix</span></code></a>(self[,&nbsp;n_cores])</td>
<td>Run ENmix preprocessing pipeline.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#pymethylprocess.PreProcessDataTypes.PreProcessIDAT.preprocessMeffil" title="pymethylprocess.PreProcessDataTypes.PreProcessIDAT.preprocessMeffil"><code class="xref py py-obj docutils literal notranslate"><span class="pre">preprocessMeffil</span></code></a>(self[,&nbsp;n_cores,&nbsp;n_pcs,&nbsp;…])</td>
<td>Run meffil preprocessing pipeline with functional normalization.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#pymethylprocess.PreProcessDataTypes.PreProcessIDAT.preprocessNoob" title="pymethylprocess.PreProcessDataTypes.PreProcessIDAT.preprocessNoob"><code class="xref py py-obj docutils literal notranslate"><span class="pre">preprocessNoob</span></code></a>(self)</td>
<td>Run minfi preprocessing with Noob normalization</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#pymethylprocess.PreProcessDataTypes.PreProcessIDAT.preprocessRAW" title="pymethylprocess.PreProcessDataTypes.PreProcessIDAT.preprocessRAW"><code class="xref py py-obj docutils literal notranslate"><span class="pre">preprocessRAW</span></code></a>(self)</td>
<td>Run minfi preprocessing with RAW normalization</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#pymethylprocess.PreProcessDataTypes.PreProcessIDAT.preprocess_enmix_pipeline" title="pymethylprocess.PreProcessDataTypes.PreProcessIDAT.preprocess_enmix_pipeline"><code class="xref py py-obj docutils literal notranslate"><span class="pre">preprocess_enmix_pipeline</span></code></a>(self[,&nbsp;n_cores,&nbsp;…])</td>
<td>Run complete ENmix or minfi preprocessing pipeline.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#pymethylprocess.PreProcessDataTypes.PreProcessIDAT.return_beta" title="pymethylprocess.PreProcessDataTypes.PreProcessIDAT.return_beta"><code class="xref py py-obj docutils literal notranslate"><span class="pre">return_beta</span></code></a>(self)</td>
<td>Return minfi RSet after having created MSet.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#pymethylprocess.PreProcessDataTypes.PreProcessIDAT.to_methyl_array" title="pymethylprocess.PreProcessDataTypes.PreProcessIDAT.to_methyl_array"><code class="xref py py-obj docutils literal notranslate"><span class="pre">to_methyl_array</span></code></a>(self[,&nbsp;disease])</td>
<td>Convert results from preprocessing into MethylationArray, and directly return MethylationArray object.</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="pymethylprocess.PreProcessDataTypes.PreProcessIDAT.export_csv">
<code class="descname">export_csv</code><span class="sig-paren">(</span><em>self</em>, <em>output_dir</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/PreProcessDataTypes.html#PreProcessIDAT.export_csv"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.PreProcessDataTypes.PreProcessIDAT.export_csv" title="Permalink to this definition">¶</a></dt>
<dd><p>Export pheno and beta dataframes to CSVs</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>output_dir</strong></dt>
<dd><p class="first last">Where to store csvs.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.PreProcessDataTypes.PreProcessIDAT.export_pickle">
<code class="descname">export_pickle</code><span class="sig-paren">(</span><em>self</em>, <em>output_pickle</em>, <em>disease=''</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/PreProcessDataTypes.html#PreProcessIDAT.export_pickle"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.PreProcessDataTypes.PreProcessIDAT.export_pickle" title="Permalink to this definition">¶</a></dt>
<dd><p>Export pheno and beta dataframes to pickle, stored in python dict that can be loaded into MethylationArray</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>output_pickle</strong></dt>
<dd><p class="first last">Where to store MethylationArray.</p>
</dd>
<dt><strong>disease</strong></dt>
<dd><p class="first last">Custom naming scheme for data.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.PreProcessDataTypes.PreProcessIDAT.export_sql">
<code class="descname">export_sql</code><span class="sig-paren">(</span><em>self</em>, <em>output_db</em>, <em>disease=''</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/PreProcessDataTypes.html#PreProcessIDAT.export_sql"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.PreProcessDataTypes.PreProcessIDAT.export_sql" title="Permalink to this definition">¶</a></dt>
<dd><p>Export pheno and beta dataframes to SQL</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>output_db</strong></dt>
<dd><p class="first last">Where to store data, sqlite db.</p>
</dd>
<dt><strong>disease</strong></dt>
<dd><p class="first last">Custom naming scheme for data.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.PreProcessDataTypes.PreProcessIDAT.extract_manifest">
<code class="descname">extract_manifest</code><span class="sig-paren">(</span><em>self</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/PreProcessDataTypes.html#PreProcessIDAT.extract_manifest"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.PreProcessDataTypes.PreProcessIDAT.extract_manifest" title="Permalink to this definition">¶</a></dt>
<dd><p>Get manifest from RGSet.</p>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.PreProcessDataTypes.PreProcessIDAT.extract_pheno_data">
<code class="descname">extract_pheno_data</code><span class="sig-paren">(</span><em>self</em>, <em>methylset=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/PreProcessDataTypes.html#PreProcessIDAT.extract_pheno_data"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.PreProcessDataTypes.PreProcessIDAT.extract_pheno_data" title="Permalink to this definition">¶</a></dt>
<dd><p>Extract pheno data from MSet or RGSet, minfi.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>methylset</strong></dt>
<dd><p class="first last">If MSet has beenn created, set to True, else extract from original RGSet.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.PreProcessDataTypes.PreProcessIDAT.filter_beta">
<code class="descname">filter_beta</code><span class="sig-paren">(</span><em>self</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/PreProcessDataTypes.html#PreProcessIDAT.filter_beta"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.PreProcessDataTypes.PreProcessIDAT.filter_beta" title="Permalink to this definition">¶</a></dt>
<dd><p>After creating beta, filter out outliers.</p>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.PreProcessDataTypes.PreProcessIDAT.get_beta">
<code class="descname">get_beta</code><span class="sig-paren">(</span><em>self</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/PreProcessDataTypes.html#PreProcessIDAT.get_beta"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.PreProcessDataTypes.PreProcessIDAT.get_beta" title="Permalink to this definition">¶</a></dt>
<dd><p>Get beta value matrix from minfi after finding RSet.</p>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.PreProcessDataTypes.PreProcessIDAT.get_meth">
<code class="descname">get_meth</code><span class="sig-paren">(</span><em>self</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/PreProcessDataTypes.html#PreProcessIDAT.get_meth"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.PreProcessDataTypes.PreProcessIDAT.get_meth" title="Permalink to this definition">¶</a></dt>
<dd><p>Get methylation intensity matrix from MSet</p>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.PreProcessDataTypes.PreProcessIDAT.get_unmeth">
<code class="descname">get_unmeth</code><span class="sig-paren">(</span><em>self</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/PreProcessDataTypes.html#PreProcessIDAT.get_unmeth"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.PreProcessDataTypes.PreProcessIDAT.get_unmeth" title="Permalink to this definition">¶</a></dt>
<dd><p>Get unmethylated intensity matrix from MSet</p>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.PreProcessDataTypes.PreProcessIDAT.load_idats">
<code class="descname">load_idats</code><span class="sig-paren">(</span><em>self</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/PreProcessDataTypes.html#PreProcessIDAT.load_idats"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.PreProcessDataTypes.PreProcessIDAT.load_idats" title="Permalink to this definition">¶</a></dt>
<dd><p>For minfi pipeline, load IDATs from specified idat_dir.</p>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.PreProcessDataTypes.PreProcessIDAT.move_jpg">
<code class="descname">move_jpg</code><span class="sig-paren">(</span><em>self</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/PreProcessDataTypes.html#PreProcessIDAT.move_jpg"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.PreProcessDataTypes.PreProcessIDAT.move_jpg" title="Permalink to this definition">¶</a></dt>
<dd><p>Move jpeg files from current working directory to the idat directory.</p>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.PreProcessDataTypes.PreProcessIDAT.output_pheno_beta">
<code class="descname">output_pheno_beta</code><span class="sig-paren">(</span><em>self</em>, <em>meffil=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/PreProcessDataTypes.html#PreProcessIDAT.output_pheno_beta"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.PreProcessDataTypes.PreProcessIDAT.output_pheno_beta" title="Permalink to this definition">¶</a></dt>
<dd><p>Get pheno and beta dataframe objects stored as attributes for input to MethylationArray object.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>meffil</strong></dt>
<dd><p class="first last">True if ran meffil pipeline.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.PreProcessDataTypes.PreProcessIDAT.plot_original_qc">
<code class="descname">plot_original_qc</code><span class="sig-paren">(</span><em>self</em>, <em>output_dir</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/PreProcessDataTypes.html#PreProcessIDAT.plot_original_qc"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.PreProcessDataTypes.PreProcessIDAT.plot_original_qc" title="Permalink to this definition">¶</a></dt>
<dd><p>Plot QC results from ENmix pipeline and possible minfi. Still experimental.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>output_dir</strong></dt>
<dd><p class="first last">Where to store plots.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.PreProcessDataTypes.PreProcessIDAT.plot_qc_metrics">
<code class="descname">plot_qc_metrics</code><span class="sig-paren">(</span><em>self</em>, <em>output_dir</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/PreProcessDataTypes.html#PreProcessIDAT.plot_qc_metrics"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.PreProcessDataTypes.PreProcessIDAT.plot_qc_metrics" title="Permalink to this definition">¶</a></dt>
<dd><p>Plot QC results from ENmix pipeline and possible minfi. Still experimental.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>output_dir</strong></dt>
<dd><p class="first last">Where to store plots.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.PreProcessDataTypes.PreProcessIDAT.preprocessENmix">
<code class="descname">preprocessENmix</code><span class="sig-paren">(</span><em>self</em>, <em>n_cores=6</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/PreProcessDataTypes.html#PreProcessIDAT.preprocessENmix"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.PreProcessDataTypes.PreProcessIDAT.preprocessENmix" title="Permalink to this definition">¶</a></dt>
<dd><p>Run ENmix preprocessing pipeline.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>n_cores</strong></dt>
<dd><p class="first last">Number of CPUs to use.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.PreProcessDataTypes.PreProcessIDAT.preprocessMeffil">
<code class="descname">preprocessMeffil</code><span class="sig-paren">(</span><em>self</em>, <em>n_cores=6</em>, <em>n_pcs=4</em>, <em>qc_report_fname='qc/report.html'</em>, <em>normalization_report_fname='norm/report.html'</em>, <em>pc_plot_fname='qc/pc_plot.pdf'</em>, <em>useCache=True</em>, <em>qc_only=True</em>, <em>qc_parameters={'p.beadnum.samples': 0.1</em>, <em>'p.detection.samples': 0.1</em>, <em>'p.detection.cpgs': 0.1</em>, <em>'p.beadnum.cpgs': 0.1}</em>, <em>rm_sex=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/PreProcessDataTypes.html#PreProcessIDAT.preprocessMeffil"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.PreProcessDataTypes.PreProcessIDAT.preprocessMeffil" title="Permalink to this definition">¶</a></dt>
<dd><p>Run meffil preprocessing pipeline with functional normalization.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>n_cores</strong></dt>
<dd><p class="first last">Number of CPUs to use.</p>
</dd>
<dt><strong>n_pcs</strong></dt>
<dd><p class="first last">Number of principal components to use for functional normalization, set to -1 to autoselect via kneedle algorithm.</p>
</dd>
<dt><strong>qc_report_fname</strong></dt>
<dd><p class="first last">HTML filename to store QC report.</p>
</dd>
<dt><strong>normalization_report_fname</strong></dt>
<dd><p class="first last">HTML filename to store normalization report</p>
</dd>
<dt><strong>pc_plot_fname</strong></dt>
<dd><p class="first last">PDF file to store principal components plot.</p>
</dd>
<dt><strong>useCache</strong></dt>
<dd><p class="first last">Use saved QC objects instead of running through QC again.</p>
</dd>
<dt><strong>qc_only</strong></dt>
<dd><p class="first last">Perform QC, then save and quit before normalization.</p>
</dd>
<dt><strong>qc_parameters</strong></dt>
<dd><p class="first last">Python dictionary with parameters for qc.</p>
</dd>
<dt><strong>rm_sex</strong></dt>
<dd><p class="first last">Remove non-autosomal cpgs?</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.PreProcessDataTypes.PreProcessIDAT.preprocessNoob">
<code class="descname">preprocessNoob</code><span class="sig-paren">(</span><em>self</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/PreProcessDataTypes.html#PreProcessIDAT.preprocessNoob"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.PreProcessDataTypes.PreProcessIDAT.preprocessNoob" title="Permalink to this definition">¶</a></dt>
<dd><p>Run minfi preprocessing with Noob normalization</p>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.PreProcessDataTypes.PreProcessIDAT.preprocessRAW">
<code class="descname">preprocessRAW</code><span class="sig-paren">(</span><em>self</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/PreProcessDataTypes.html#PreProcessIDAT.preprocessRAW"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.PreProcessDataTypes.PreProcessIDAT.preprocessRAW" title="Permalink to this definition">¶</a></dt>
<dd><p>Run minfi preprocessing with RAW normalization</p>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.PreProcessDataTypes.PreProcessIDAT.preprocess_enmix_pipeline">
<code class="descname">preprocess_enmix_pipeline</code><span class="sig-paren">(</span><em>self</em>, <em>n_cores=6</em>, <em>pipeline='enmix'</em>, <em>noob=False</em>, <em>qc_only=False</em>, <em>use_cache=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/PreProcessDataTypes.html#PreProcessIDAT.preprocess_enmix_pipeline"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.PreProcessDataTypes.PreProcessIDAT.preprocess_enmix_pipeline" title="Permalink to this definition">¶</a></dt>
<dd><p>Run complete ENmix or minfi preprocessing pipeline.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>n_cores</strong></dt>
<dd><p class="first last">Number CPUs.</p>
</dd>
<dt><strong>pipeline</strong></dt>
<dd><p class="first last">Run enmix or minfi</p>
</dd>
<dt><strong>noob</strong></dt>
<dd><p class="first last">Noob norm or RAW if minfi running.</p>
</dd>
<dt><strong>qc_only</strong></dt>
<dd><p class="first last">Save and quit after only running QC?</p>
</dd>
<dt><strong>use_cache</strong></dt>
<dd><p class="first last">Load preexisting RGSet instead of running QC again.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.PreProcessDataTypes.PreProcessIDAT.return_beta">
<code class="descname">return_beta</code><span class="sig-paren">(</span><em>self</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/PreProcessDataTypes.html#PreProcessIDAT.return_beta"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.PreProcessDataTypes.PreProcessIDAT.return_beta" title="Permalink to this definition">¶</a></dt>
<dd><p>Return minfi RSet after having created MSet.</p>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.PreProcessDataTypes.PreProcessIDAT.to_methyl_array">
<code class="descname">to_methyl_array</code><span class="sig-paren">(</span><em>self</em>, <em>disease=''</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/PreProcessDataTypes.html#PreProcessIDAT.to_methyl_array"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.PreProcessDataTypes.PreProcessIDAT.to_methyl_array" title="Permalink to this definition">¶</a></dt>
<dd><p>Convert results from preprocessing into MethylationArray, and directly return MethylationArray object.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>disease</strong></dt>
<dd><p class="first last">Custom naming scheme for data.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="pymethylprocess.PreProcessDataTypes.PreProcessPhenoData">
<em class="property">class </em><code class="descclassname">pymethylprocess.PreProcessDataTypes.</code><code class="descname">PreProcessPhenoData</code><span class="sig-paren">(</span><em>pheno_sheet</em>, <em>idat_dir</em>, <em>header_line=0</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/PreProcessDataTypes.html#PreProcessPhenoData"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.PreProcessDataTypes.PreProcessPhenoData" title="Permalink to this definition">¶</a></dt>
<dd><p>Class that will manipute phenotype samplesheet before preprocessing of IDATs.</p>
<dl class="docutils">
<dt>pheno_sheet</dt>
<dd>Location of clinical info csv.</dd>
<dt>idat_dir</dt>
<dd>Location of idats</dd>
<dt>header_line</dt>
<dd>Where to start reading clinical csv</dd>
</dl>
<p class="rubric">Methods</p>
<table border="1" class="longtable docutils">
<colgroup>
<col width="10%" />
<col width="90%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td><a class="reference internal" href="#pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.concat" title="pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.concat"><code class="xref py py-obj docutils literal notranslate"><span class="pre">concat</span></code></a>(self,&nbsp;other_formatted_sheet)</td>
<td>Concat multiple PreProcessPhenoData objects, concat their dataframes to accept more than one smaplesheet/dataset.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.export" title="pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.export"><code class="xref py py-obj docutils literal notranslate"><span class="pre">export</span></code></a>(self,&nbsp;output_sheet_name)</td>
<td>Export pheno data to csv after done with manipulation.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.format_custom" title="pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.format_custom"><code class="xref py py-obj docutils literal notranslate"><span class="pre">format_custom</span></code></a>(self,&nbsp;basename_col,&nbsp;…[,&nbsp;…])</td>
<td>Custom format clinical sheet if user supplied idats.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.format_geo" title="pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.format_geo"><code class="xref py py-obj docutils literal notranslate"><span class="pre">format_geo</span></code></a>(self[,&nbsp;disease_class_column,&nbsp;…])</td>
<td>Format clinical sheets if downloaded geo idats.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.format_tcga" title="pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.format_tcga"><code class="xref py py-obj docutils literal notranslate"><span class="pre">format_tcga</span></code></a>(self[,&nbsp;mapping_file])</td>
<td>Format clinical sheets if downloaded tcga idats.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.get_categorical_distribution" title="pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.get_categorical_distribution"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_categorical_distribution</span></code></a>(self,&nbsp;key[,&nbsp;…])</td>
<td>Print categorical distribution, counts for each unique value in phenotype column.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.merge" title="pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.merge"><code class="xref py py-obj docutils literal notranslate"><span class="pre">merge</span></code></a>(self,&nbsp;other_formatted_sheet[,&nbsp;…])</td>
<td>Merge multiple PreProcessPhenoData objects, merge their dataframes to accept more than one saplesheet/dataset or add more pheno info.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.remove_diseases" title="pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.remove_diseases"><code class="xref py py-obj docutils literal notranslate"><span class="pre">remove_diseases</span></code></a>(self,&nbsp;exclude_disease_list,&nbsp;…)</td>
<td>Remove samples with certain diseases from disease column.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.split_key" title="pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.split_key"><code class="xref py py-obj docutils literal notranslate"><span class="pre">split_key</span></code></a>(self,&nbsp;key,&nbsp;subtype_delimiter)</td>
<td>Split pheno column by key, with subtype delimiter, eg.</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.concat">
<code class="descname">concat</code><span class="sig-paren">(</span><em>self</em>, <em>other_formatted_sheet</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/PreProcessDataTypes.html#PreProcessPhenoData.concat"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.concat" title="Permalink to this definition">¶</a></dt>
<dd><p>Concat multiple PreProcessPhenoData objects, concat their dataframes to accept more than one smaplesheet/dataset.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>other_formatted_sheet</strong></dt>
<dd><p class="first last">Other PreProcessPhenoData to concat.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.export">
<code class="descname">export</code><span class="sig-paren">(</span><em>self</em>, <em>output_sheet_name</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/PreProcessDataTypes.html#PreProcessPhenoData.export"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.export" title="Permalink to this definition">¶</a></dt>
<dd><p>Export pheno data to csv after done with manipulation.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>output_sheet_name</strong></dt>
<dd><p class="first last">Output csv name.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.format_custom">
<code class="descname">format_custom</code><span class="sig-paren">(</span><em>self</em>, <em>basename_col</em>, <em>disease_class_column</em>, <em>include_columns={}</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/PreProcessDataTypes.html#PreProcessPhenoData.format_custom"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.format_custom" title="Permalink to this definition">¶</a></dt>
<dd><p>Custom format clinical sheet if user supplied idats.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>basename_col</strong></dt>
<dd><p class="first last">Column name of sample names.</p>
</dd>
<dt><strong>disease_class_column</strong></dt>
<dd><p class="first last">Disease column of clinical info csv.</p>
</dd>
<dt><strong>include_columns</strong></dt>
<dd><p class="first last">Dictionary specifying other columns to include, and new names to assign them to.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.format_geo">
<code class="descname">format_geo</code><span class="sig-paren">(</span><em>self</em>, <em>disease_class_column='methylation class:ch1'</em>, <em>include_columns={}</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/PreProcessDataTypes.html#PreProcessPhenoData.format_geo"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.format_geo" title="Permalink to this definition">¶</a></dt>
<dd><p>Format clinical sheets if downloaded geo idats.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>disease_class_column</strong></dt>
<dd><p class="first last">Disease column of clinical info csv.</p>
</dd>
<dt><strong>include_columns</strong></dt>
<dd><p class="first last">Dictionary specifying other columns to include, and new names to assign them to.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.format_tcga">
<code class="descname">format_tcga</code><span class="sig-paren">(</span><em>self</em>, <em>mapping_file='idat_filename_case.txt'</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/PreProcessDataTypes.html#PreProcessPhenoData.format_tcga"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.format_tcga" title="Permalink to this definition">¶</a></dt>
<dd><p>Format clinical sheets if downloaded tcga idats.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>mapping_file</strong></dt>
<dd><p class="first last">Maps uuids to proper tcga sample names, should be downloaded with tcga clinical information.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.get_categorical_distribution">
<code class="descname">get_categorical_distribution</code><span class="sig-paren">(</span><em>self</em>, <em>key</em>, <em>disease_only=False</em>, <em>subtype_delimiter='</em>, <em>'</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/PreProcessDataTypes.html#PreProcessPhenoData.get_categorical_distribution"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.get_categorical_distribution" title="Permalink to this definition">¶</a></dt>
<dd><p>Print categorical distribution, counts for each unique value in phenotype column.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>key</strong></dt>
<dd><p class="first last">Phenotype Column.</p>
</dd>
<dt><strong>disease_only</strong></dt>
<dd><p class="first last">Whether to split phenotype column entries by delimiter.</p>
</dd>
<dt><strong>subtype_delimiter</strong></dt>
<dd><p class="first last">Subtype delimiter to split on.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.merge">
<code class="descname">merge</code><span class="sig-paren">(</span><em>self</em>, <em>other_formatted_sheet</em>, <em>use_second_sheet_disease=True</em>, <em>no_disease_merge=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/PreProcessDataTypes.html#PreProcessPhenoData.merge"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.merge" title="Permalink to this definition">¶</a></dt>
<dd><p>Merge multiple PreProcessPhenoData objects, merge their dataframes to accept more than one saplesheet/dataset or add more pheno info.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>other_formatted_sheet</strong></dt>
<dd><p class="first last">Other PreProcessPhenoData to merge.</p>
</dd>
<dt><strong>use_second_sheet_disease</strong></dt>
<dd><p class="first last">Change disease column to that of second sheet instead of first.</p>
</dd>
<dt><strong>no_disease_merge</strong></dt>
<dd><p class="first last">Keep both disease columns from both sheets.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.remove_diseases">
<code class="descname">remove_diseases</code><span class="sig-paren">(</span><em>self</em>, <em>exclude_disease_list</em>, <em>low_count</em>, <em>disease_only</em>, <em>subtype_delimiter</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/PreProcessDataTypes.html#PreProcessPhenoData.remove_diseases"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.remove_diseases" title="Permalink to this definition">¶</a></dt>
<dd><p>Remove samples with certain diseases from disease column.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>exclude_disease_list</strong></dt>
<dd><p class="first last">List containing diseases to remove.</p>
</dd>
<dt><strong>low_count</strong></dt>
<dd><p class="first last">Remove samples that have less than x disease occurances in column.</p>
</dd>
<dt><strong>disease_only</strong></dt>
<dd><p class="first last">Whether to split phenotype column entries by delimiter.</p>
</dd>
<dt><strong>subtype_delimiter</strong></dt>
<dd><p class="first last">Subtype delimiter to split on.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.split_key">
<code class="descname">split_key</code><span class="sig-paren">(</span><em>self</em>, <em>key</em>, <em>subtype_delimiter</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/PreProcessDataTypes.html#PreProcessPhenoData.split_key"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.PreProcessDataTypes.PreProcessPhenoData.split_key" title="Permalink to this definition">¶</a></dt>
<dd><p>Split pheno column by key, with subtype delimiter, eg. entry S1,s2 -&gt; S1 with delimiter “,”.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>key</strong></dt>
<dd><p class="first last">Pheno column name.</p>
</dd>
<dt><strong>subtype_delimiter</strong></dt>
<dd><p class="first last">Subtype delimiter to split on.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="pymethylprocess.PreProcessDataTypes.TCGADownloader">
<em class="property">class </em><code class="descclassname">pymethylprocess.PreProcessDataTypes.</code><code class="descname">TCGADownloader</code><a class="reference internal" href="_modules/pymethylprocess/PreProcessDataTypes.html#TCGADownloader"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.PreProcessDataTypes.TCGADownloader" title="Permalink to this definition">¶</a></dt>
<dd><p>Downloads TCGA and GEO IDAT and clinical data</p>
<p class="rubric">Methods</p>
<table border="1" class="longtable docutils">
<colgroup>
<col width="10%" />
<col width="90%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td><a class="reference internal" href="#pymethylprocess.PreProcessDataTypes.TCGADownloader.download_clinical" title="pymethylprocess.PreProcessDataTypes.TCGADownloader.download_clinical"><code class="xref py py-obj docutils literal notranslate"><span class="pre">download_clinical</span></code></a>(self,&nbsp;output_dir)</td>
<td>Download TCGA Clinical Data.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#pymethylprocess.PreProcessDataTypes.TCGADownloader.download_geo" title="pymethylprocess.PreProcessDataTypes.TCGADownloader.download_geo"><code class="xref py py-obj docutils literal notranslate"><span class="pre">download_geo</span></code></a>(self,&nbsp;query,&nbsp;output_dir)</td>
<td>Download GEO IDATs.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#pymethylprocess.PreProcessDataTypes.TCGADownloader.download_tcga" title="pymethylprocess.PreProcessDataTypes.TCGADownloader.download_tcga"><code class="xref py py-obj docutils literal notranslate"><span class="pre">download_tcga</span></code></a>(self,&nbsp;output_dir)</td>
<td>Download TCGA IDATs.</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="pymethylprocess.PreProcessDataTypes.TCGADownloader.download_clinical">
<code class="descname">download_clinical</code><span class="sig-paren">(</span><em>self</em>, <em>output_dir</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/PreProcessDataTypes.html#TCGADownloader.download_clinical"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.PreProcessDataTypes.TCGADownloader.download_clinical" title="Permalink to this definition">¶</a></dt>
<dd><p>Download TCGA Clinical Data.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>output_dir</strong></dt>
<dd><p class="first last">Where to output clinical data csv.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.PreProcessDataTypes.TCGADownloader.download_geo">
<code class="descname">download_geo</code><span class="sig-paren">(</span><em>self</em>, <em>query</em>, <em>output_dir</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/PreProcessDataTypes.html#TCGADownloader.download_geo"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.PreProcessDataTypes.TCGADownloader.download_geo" title="Permalink to this definition">¶</a></dt>
<dd><p>Download GEO IDATs.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>query</strong></dt>
<dd><p class="first last">GEO accession number to query, must be 450k/850k.</p>
</dd>
<dt><strong>output_dir</strong></dt>
<dd><p class="first last">Output directory to store idats and clinical information csv</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.PreProcessDataTypes.TCGADownloader.download_tcga">
<code class="descname">download_tcga</code><span class="sig-paren">(</span><em>self</em>, <em>output_dir</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/PreProcessDataTypes.html#TCGADownloader.download_tcga"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.PreProcessDataTypes.TCGADownloader.download_tcga" title="Permalink to this definition">¶</a></dt>
<dd><p>Download TCGA IDATs.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>output_dir</strong></dt>
<dd><p class="first last">Where to output idat files.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

</div>
<span class="target" id="module-pymethylprocess.MethylationDataTypes"></span><div class="section" id="methylationdatatypes-py">
<h2>MethylationDataTypes.py<a class="headerlink" href="#methylationdatatypes-py" title="Permalink to this headline">¶</a></h2>
<p>Contains datatypes core to storing beta and phenotype methylation data, and imputation.</p>
<dl class="class">
<dt id="pymethylprocess.MethylationDataTypes.ImputerObject">
<em class="property">class </em><code class="descclassname">pymethylprocess.MethylationDataTypes.</code><code class="descname">ImputerObject</code><span class="sig-paren">(</span><em>solver</em>, <em>method</em>, <em>opts={}</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/MethylationDataTypes.html#ImputerObject"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.MethylationDataTypes.ImputerObject" title="Permalink to this definition">¶</a></dt>
<dd><p>Class that stores and accesses different types of imputers.
Construct sklearn-like imputer given certain input arguments.</p>
<dl class="docutils">
<dt>solver</dt>
<dd>Library for imputation, eg. sklearn, fancyimpute.</dd>
<dt>method</dt>
<dd>Imputation method in library, named.</dd>
<dt>opts</dt>
<dd>Additional options to assign to imputer.</dd>
</dl>
<p class="rubric">Methods</p>
<table border="1" class="longtable docutils">
<colgroup>
<col width="10%" />
<col width="90%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td><a class="reference internal" href="#pymethylprocess.MethylationDataTypes.ImputerObject.return_imputer" title="pymethylprocess.MethylationDataTypes.ImputerObject.return_imputer"><code class="xref py py-obj docutils literal notranslate"><span class="pre">return_imputer</span></code></a>(self)</td>
<td>Return initialized sklearn-like imputer.</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="pymethylprocess.MethylationDataTypes.ImputerObject.return_imputer">
<code class="descname">return_imputer</code><span class="sig-paren">(</span><em>self</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/MethylationDataTypes.html#ImputerObject.return_imputer"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.MethylationDataTypes.ImputerObject.return_imputer" title="Permalink to this definition">¶</a></dt>
<dd><p>Return initialized sklearn-like imputer.</p>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="pymethylprocess.MethylationDataTypes.MethylationArray">
<em class="property">class </em><code class="descclassname">pymethylprocess.MethylationDataTypes.</code><code class="descname">MethylationArray</code><span class="sig-paren">(</span><em>pheno_df</em>, <em>beta_df</em>, <em>name=''</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/MethylationDataTypes.html#MethylationArray"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.MethylationDataTypes.MethylationArray" title="Permalink to this definition">¶</a></dt>
<dd><p>Stores beta and phenotype information and performs various operations.
Initialize MethylationArray object by inputting dataframe of phenotypes
and dataframe of beta values with samples as index.</p>
<dl class="docutils">
<dt>pheno_df</dt>
<dd>Phenotype dataframe (samples x covariates)</dd>
<dt>beta_df</dt>
<dd>Beta Values Dataframe (samples x cpgs)</dd>
</dl>
<p class="rubric">Methods</p>
<table border="1" class="longtable docutils">
<colgroup>
<col width="10%" />
<col width="90%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td><a class="reference internal" href="#pymethylprocess.MethylationDataTypes.MethylationArray.bin_column" title="pymethylprocess.MethylationDataTypes.MethylationArray.bin_column"><code class="xref py py-obj docutils literal notranslate"><span class="pre">bin_column</span></code></a>(self,&nbsp;col,&nbsp;n_bins)</td>
<td>Turn continuous variable/covariate into categorical bins.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#pymethylprocess.MethylationDataTypes.MethylationArray.categorical_breakdown" title="pymethylprocess.MethylationDataTypes.MethylationArray.categorical_breakdown"><code class="xref py py-obj docutils literal notranslate"><span class="pre">categorical_breakdown</span></code></a>(self,&nbsp;key)</td>
<td>Print categorical distribution, counts for each unique value in phenotype column.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#pymethylprocess.MethylationDataTypes.MethylationArray.feature_select" title="pymethylprocess.MethylationDataTypes.MethylationArray.feature_select"><code class="xref py py-obj docutils literal notranslate"><span class="pre">feature_select</span></code></a>(self,&nbsp;n_top_cpgs[,&nbsp;…])</td>
<td>Perform unsupervised feature selection on MethylationArray.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#pymethylprocess.MethylationDataTypes.MethylationArray.from_pickle" title="pymethylprocess.MethylationDataTypes.MethylationArray.from_pickle"><code class="xref py py-obj docutils literal notranslate"><span class="pre">from_pickle</span></code></a>(input_pickle)</td>
<td>Load MethylationArray stored in pickle.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#pymethylprocess.MethylationDataTypes.MethylationArray.groupby" title="pymethylprocess.MethylationDataTypes.MethylationArray.groupby"><code class="xref py py-obj docutils literal notranslate"><span class="pre">groupby</span></code></a>(self,&nbsp;key)</td>
<td>Groupby for Methylation Array.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#pymethylprocess.MethylationDataTypes.MethylationArray.impute" title="pymethylprocess.MethylationDataTypes.MethylationArray.impute"><code class="xref py py-obj docutils literal notranslate"><span class="pre">impute</span></code></a>(self,&nbsp;imputer)</td>
<td>Perform imputation on NaN beta vaues.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#pymethylprocess.MethylationDataTypes.MethylationArray.merge_preprocess_sheet" title="pymethylprocess.MethylationDataTypes.MethylationArray.merge_preprocess_sheet"><code class="xref py py-obj docutils literal notranslate"><span class="pre">merge_preprocess_sheet</span></code></a>(self,&nbsp;…)</td>
<td>Feed in another phenotype dataframe that will be merged with existing phenotype array.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#pymethylprocess.MethylationDataTypes.MethylationArray.overwrite_pheno_data" title="pymethylprocess.MethylationDataTypes.MethylationArray.overwrite_pheno_data"><code class="xref py py-obj docutils literal notranslate"><span class="pre">overwrite_pheno_data</span></code></a>(self,&nbsp;preprocess_sample_df)</td>
<td>Feed in another phenotype dataframe that will overwrite overlapping keys of existing phenotype array.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#pymethylprocess.MethylationDataTypes.MethylationArray.remove_missingness" title="pymethylprocess.MethylationDataTypes.MethylationArray.remove_missingness"><code class="xref py py-obj docutils literal notranslate"><span class="pre">remove_missingness</span></code></a>(self[,&nbsp;cpg_threshold,&nbsp;…])</td>
<td>Remove samples and CpGs with certain level of missingness..</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#pymethylprocess.MethylationDataTypes.MethylationArray.remove_na_samples" title="pymethylprocess.MethylationDataTypes.MethylationArray.remove_na_samples"><code class="xref py py-obj docutils literal notranslate"><span class="pre">remove_na_samples</span></code></a>(self,&nbsp;outcome_cols)</td>
<td>Remove samples of MethylationArray who have missing values in phenotype column.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#pymethylprocess.MethylationDataTypes.MethylationArray.remove_whitespace" title="pymethylprocess.MethylationDataTypes.MethylationArray.remove_whitespace"><code class="xref py py-obj docutils literal notranslate"><span class="pre">remove_whitespace</span></code></a>(self,&nbsp;key)</td>
<td>Remove whitespaces from phenotype column.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#pymethylprocess.MethylationDataTypes.MethylationArray.return_cpgs" title="pymethylprocess.MethylationDataTypes.MethylationArray.return_cpgs"><code class="xref py py-obj docutils literal notranslate"><span class="pre">return_cpgs</span></code></a>(self)</td>
<td>Return list of cpgs of MethylationArray</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#pymethylprocess.MethylationDataTypes.MethylationArray.return_idx" title="pymethylprocess.MethylationDataTypes.MethylationArray.return_idx"><code class="xref py py-obj docutils literal notranslate"><span class="pre">return_idx</span></code></a>(self)</td>
<td>Return sample names of MethylationArray.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#pymethylprocess.MethylationDataTypes.MethylationArray.return_raw_beta_array" title="pymethylprocess.MethylationDataTypes.MethylationArray.return_raw_beta_array"><code class="xref py py-obj docutils literal notranslate"><span class="pre">return_raw_beta_array</span></code></a>(self)</td>
<td>Return numpy array of methylation beta vaues.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#pymethylprocess.MethylationDataTypes.MethylationArray.return_shape" title="pymethylprocess.MethylationDataTypes.MethylationArray.return_shape"><code class="xref py py-obj docutils literal notranslate"><span class="pre">return_shape</span></code></a>(self)</td>
<td>Return dimensionality and number of samples of beta matrix.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#pymethylprocess.MethylationDataTypes.MethylationArray.split_by_subtype" title="pymethylprocess.MethylationDataTypes.MethylationArray.split_by_subtype"><code class="xref py py-obj docutils literal notranslate"><span class="pre">split_by_subtype</span></code></a>(self,&nbsp;disease_only,&nbsp;…)</td>
<td>Split MethylationArray into generator of MethylationArrays by phenotype column.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#pymethylprocess.MethylationDataTypes.MethylationArray.split_key" title="pymethylprocess.MethylationDataTypes.MethylationArray.split_key"><code class="xref py py-obj docutils literal notranslate"><span class="pre">split_key</span></code></a>(self,&nbsp;key,&nbsp;subtype_delimiter)</td>
<td>Manipulate an entire phenotype column, splitting each element up by some delimiter.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#pymethylprocess.MethylationDataTypes.MethylationArray.split_train_test" title="pymethylprocess.MethylationDataTypes.MethylationArray.split_train_test"><code class="xref py py-obj docutils literal notranslate"><span class="pre">split_train_test</span></code></a>(self[,&nbsp;train_p,&nbsp;…])</td>
<td>Split MethylationArray into training and test sets, with option to stratify by categorical covariate.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#pymethylprocess.MethylationDataTypes.MethylationArray.subsample" title="pymethylprocess.MethylationDataTypes.MethylationArray.subsample"><code class="xref py py-obj docutils literal notranslate"><span class="pre">subsample</span></code></a>(self[,&nbsp;key,&nbsp;n_samples,&nbsp;frac,&nbsp;…])</td>
<td>Subsample MethylationArray, make the set randomly smaller.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#pymethylprocess.MethylationDataTypes.MethylationArray.subset_cpgs" title="pymethylprocess.MethylationDataTypes.MethylationArray.subset_cpgs"><code class="xref py py-obj docutils literal notranslate"><span class="pre">subset_cpgs</span></code></a>(self,&nbsp;cpgs)</td>
<td>Subset beta matrix by list of Cpgs.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#pymethylprocess.MethylationDataTypes.MethylationArray.subset_index" title="pymethylprocess.MethylationDataTypes.MethylationArray.subset_index"><code class="xref py py-obj docutils literal notranslate"><span class="pre">subset_index</span></code></a>(self,&nbsp;index)</td>
<td>Subset MethylationArray by samples.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#pymethylprocess.MethylationDataTypes.MethylationArray.write_csvs" title="pymethylprocess.MethylationDataTypes.MethylationArray.write_csvs"><code class="xref py py-obj docutils literal notranslate"><span class="pre">write_csvs</span></code></a>(self,&nbsp;output_dir)</td>
<td>Write phenotype data and beta values to csvs.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#pymethylprocess.MethylationDataTypes.MethylationArray.write_db" title="pymethylprocess.MethylationDataTypes.MethylationArray.write_db"><code class="xref py py-obj docutils literal notranslate"><span class="pre">write_db</span></code></a>(self,&nbsp;conn[,&nbsp;disease])</td>
<td>Store phenotype data and beta values in SQL database.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#pymethylprocess.MethylationDataTypes.MethylationArray.write_pickle" title="pymethylprocess.MethylationDataTypes.MethylationArray.write_pickle"><code class="xref py py-obj docutils literal notranslate"><span class="pre">write_pickle</span></code></a>(self,&nbsp;output_pickle[,&nbsp;disease])</td>
<td>Store phenotype data and beta values in pickle file.</td>
</tr>
</tbody>
</table>
<table border="1" class="docutils">
<colgroup>
<col width="50%" />
<col width="50%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td><strong>export</strong></td>
<td>&#160;</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="pymethylprocess.MethylationDataTypes.MethylationArray.bin_column">
<code class="descname">bin_column</code><span class="sig-paren">(</span><em>self</em>, <em>col</em>, <em>n_bins</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/MethylationDataTypes.html#MethylationArray.bin_column"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.MethylationDataTypes.MethylationArray.bin_column" title="Permalink to this definition">¶</a></dt>
<dd><p>Turn continuous variable/covariate into categorical bins.
Returns name of new column and updates phenotype matrix to reflect this change.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>col</strong></dt>
<dd><p class="first last">Continuous column of phenotype array to bin.</p>
</dd>
<dt><strong>n_bins</strong></dt>
<dd><p class="first last">Number of bins to create.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.MethylationDataTypes.MethylationArray.categorical_breakdown">
<code class="descname">categorical_breakdown</code><span class="sig-paren">(</span><em>self</em>, <em>key</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/MethylationDataTypes.html#MethylationArray.categorical_breakdown"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.MethylationDataTypes.MethylationArray.categorical_breakdown" title="Permalink to this definition">¶</a></dt>
<dd><p>Print categorical distribution, counts for each unique value in phenotype column.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>key</strong></dt>
<dd><p class="first last">Phenotype Column.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.MethylationDataTypes.MethylationArray.feature_select">
<code class="descname">feature_select</code><span class="sig-paren">(</span><em>self</em>, <em>n_top_cpgs</em>, <em>feature_selection_method='mad'</em>, <em>metric='correlation'</em>, <em>nn=10</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/MethylationDataTypes.html#MethylationArray.feature_select"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.MethylationDataTypes.MethylationArray.feature_select" title="Permalink to this definition">¶</a></dt>
<dd><p>Perform unsupervised feature selection on MethylationArray.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>n_top_cpgs</strong></dt>
<dd><p class="first last">Number of CpGs to retain.</p>
</dd>
<dt><strong>feature_selection_method</strong></dt>
<dd><p class="first last">Method to perform selection.</p>
</dd>
<dt><strong>metric</strong></dt>
<dd><p class="first last">If considering structural feature selection like SPEC, use this distance metric.</p>
</dd>
<dt><strong>nn</strong></dt>
<dd><p class="first last">Number of nearest neighbors.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="classmethod">
<dt id="pymethylprocess.MethylationDataTypes.MethylationArray.from_pickle">
<em class="property">classmethod </em><code class="descname">from_pickle</code><span class="sig-paren">(</span><em>input_pickle</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/MethylationDataTypes.html#MethylationArray.from_pickle"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.MethylationDataTypes.MethylationArray.from_pickle" title="Permalink to this definition">¶</a></dt>
<dd><p>Load MethylationArray stored in pickle.</p>
<p>Usage: MethylationArray.from_pickle([input_pickle])</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>input_pickle</strong></dt>
<dd><p class="first last">Stored MethylationArray pickle.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.MethylationDataTypes.MethylationArray.groupby">
<code class="descname">groupby</code><span class="sig-paren">(</span><em>self</em>, <em>key</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/MethylationDataTypes.html#MethylationArray.groupby"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.MethylationDataTypes.MethylationArray.groupby" title="Permalink to this definition">¶</a></dt>
<dd><p>Groupby for Methylation Array. Returns generator of methylation arrays grouped by key.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>preprocess_sample_df</strong></dt>
<dd><p class="first last">New phenotype dataframe.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.MethylationDataTypes.MethylationArray.impute">
<code class="descname">impute</code><span class="sig-paren">(</span><em>self</em>, <em>imputer</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/MethylationDataTypes.html#MethylationArray.impute"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.MethylationDataTypes.MethylationArray.impute" title="Permalink to this definition">¶</a></dt>
<dd><p>Perform imputation on NaN beta vaues. Input imputater returned from ImputerObject.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>imputer</strong></dt>
<dd><p class="first last">Type of imputer object, in sklearn type interface.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.MethylationDataTypes.MethylationArray.merge_preprocess_sheet">
<code class="descname">merge_preprocess_sheet</code><span class="sig-paren">(</span><em>self</em>, <em>preprocess_sample_df</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/MethylationDataTypes.html#MethylationArray.merge_preprocess_sheet"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.MethylationDataTypes.MethylationArray.merge_preprocess_sheet" title="Permalink to this definition">¶</a></dt>
<dd><p>Feed in another phenotype dataframe that will be merged with existing phenotype array.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>preprocess_sample_df</strong></dt>
<dd><p class="first last">New phenotype dataframe.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.MethylationDataTypes.MethylationArray.overwrite_pheno_data">
<code class="descname">overwrite_pheno_data</code><span class="sig-paren">(</span><em>self</em>, <em>preprocess_sample_df</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/MethylationDataTypes.html#MethylationArray.overwrite_pheno_data"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.MethylationDataTypes.MethylationArray.overwrite_pheno_data" title="Permalink to this definition">¶</a></dt>
<dd><p>Feed in another phenotype dataframe that will overwrite overlapping keys of existing phenotype array.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>preprocess_sample_df</strong></dt>
<dd><p class="first last">New phenotype dataframe.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.MethylationDataTypes.MethylationArray.remove_missingness">
<code class="descname">remove_missingness</code><span class="sig-paren">(</span><em>self</em>, <em>cpg_threshold=None</em>, <em>sample_threshold=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/MethylationDataTypes.html#MethylationArray.remove_missingness"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.MethylationDataTypes.MethylationArray.remove_missingness" title="Permalink to this definition">¶</a></dt>
<dd><p>Remove samples and CpGs with certain level of missingness..</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>cpg_threshold</strong></dt>
<dd><p class="first last">If more than fraction of Samples for this CpG are missing, remove cpg.</p>
</dd>
<dt><strong>sample_threshold</strong></dt>
<dd><p class="first last">If more than fraction of CpGs for this sample are missing, remove sample.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.MethylationDataTypes.MethylationArray.remove_na_samples">
<code class="descname">remove_na_samples</code><span class="sig-paren">(</span><em>self</em>, <em>outcome_cols</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/MethylationDataTypes.html#MethylationArray.remove_na_samples"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.MethylationDataTypes.MethylationArray.remove_na_samples" title="Permalink to this definition">¶</a></dt>
<dd><p>Remove samples of MethylationArray who have missing values in phenotype column.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>outcome_cols</strong></dt>
<dd><p class="first last">Phenotype columns, if any rows contain missing values, samples are removed.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.MethylationDataTypes.MethylationArray.remove_whitespace">
<code class="descname">remove_whitespace</code><span class="sig-paren">(</span><em>self</em>, <em>key</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/MethylationDataTypes.html#MethylationArray.remove_whitespace"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.MethylationDataTypes.MethylationArray.remove_whitespace" title="Permalink to this definition">¶</a></dt>
<dd><p>Remove whitespaces from phenotype column.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>key</strong></dt>
<dd><p class="first last">Phenotype column.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.MethylationDataTypes.MethylationArray.return_cpgs">
<code class="descname">return_cpgs</code><span class="sig-paren">(</span><em>self</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/MethylationDataTypes.html#MethylationArray.return_cpgs"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.MethylationDataTypes.MethylationArray.return_cpgs" title="Permalink to this definition">¶</a></dt>
<dd><p>Return list of cpgs of MethylationArray</p>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.MethylationDataTypes.MethylationArray.return_idx">
<code class="descname">return_idx</code><span class="sig-paren">(</span><em>self</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/MethylationDataTypes.html#MethylationArray.return_idx"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.MethylationDataTypes.MethylationArray.return_idx" title="Permalink to this definition">¶</a></dt>
<dd><p>Return sample names of MethylationArray.</p>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.MethylationDataTypes.MethylationArray.return_raw_beta_array">
<code class="descname">return_raw_beta_array</code><span class="sig-paren">(</span><em>self</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/MethylationDataTypes.html#MethylationArray.return_raw_beta_array"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.MethylationDataTypes.MethylationArray.return_raw_beta_array" title="Permalink to this definition">¶</a></dt>
<dd><p>Return numpy array of methylation beta vaues.</p>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.MethylationDataTypes.MethylationArray.return_shape">
<code class="descname">return_shape</code><span class="sig-paren">(</span><em>self</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/MethylationDataTypes.html#MethylationArray.return_shape"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.MethylationDataTypes.MethylationArray.return_shape" title="Permalink to this definition">¶</a></dt>
<dd><p>Return dimensionality and number of samples of beta matrix.</p>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.MethylationDataTypes.MethylationArray.split_by_subtype">
<code class="descname">split_by_subtype</code><span class="sig-paren">(</span><em>self</em>, <em>disease_only</em>, <em>subtype_delimiter</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/MethylationDataTypes.html#MethylationArray.split_by_subtype"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.MethylationDataTypes.MethylationArray.split_by_subtype" title="Permalink to this definition">¶</a></dt>
<dd><p>Split MethylationArray into generator of MethylationArrays by phenotype column. Much akin to groupby. Only splits from disease column.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>disease_only</strong></dt>
<dd><p class="first last">Consider disease superclass.</p>
</dd>
<dt><strong>subtype_delimiter</strong></dt>
<dd><p class="first last">How to break up disease column if using disease_only.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.MethylationDataTypes.MethylationArray.split_key">
<code class="descname">split_key</code><span class="sig-paren">(</span><em>self</em>, <em>key</em>, <em>subtype_delimiter</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/MethylationDataTypes.html#MethylationArray.split_key"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.MethylationDataTypes.MethylationArray.split_key" title="Permalink to this definition">¶</a></dt>
<dd><p>Manipulate an entire phenotype column, splitting each element up by some delimiter.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>key</strong></dt>
<dd><p class="first last">Phenotype column.</p>
</dd>
<dt><strong>subtype_delimiter</strong></dt>
<dd><p class="first last">How to break up strings in columns. S1,s2 -&gt; S1 for instance.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.MethylationDataTypes.MethylationArray.split_train_test">
<code class="descname">split_train_test</code><span class="sig-paren">(</span><em>self</em>, <em>train_p=0.8</em>, <em>stratified=True</em>, <em>disease_only=False</em>, <em>key='disease'</em>, <em>subtype_delimiter='</em>, <em>'</em>, <em>val_p=0.0</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/MethylationDataTypes.html#MethylationArray.split_train_test"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.MethylationDataTypes.MethylationArray.split_train_test" title="Permalink to this definition">¶</a></dt>
<dd><p>Split MethylationArray into training and test sets, with option to stratify by categorical covariate.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>train_p</strong></dt>
<dd><p class="first last">Fraction of methylation array to use as training set.</p>
</dd>
<dt><strong>stratified</strong></dt>
<dd><p class="first last">Whether to stratify by categorical variable.</p>
</dd>
<dt><strong>disease_only</strong></dt>
<dd><p class="first last">Consider disease superclass by some delimiter. For instance if disease is S1,s2, superclass would be S1.</p>
</dd>
<dt><strong>key</strong></dt>
<dd><p class="first last">Column to stratify on.</p>
</dd>
<dt><strong>subtype_delimiter</strong></dt>
<dd><p class="first last">How to split disease column into super/subclass.</p>
</dd>
<dt><strong>val_p</strong></dt>
<dd><p class="first last">If set greater than 0, will create additional validation set, fraction of which is broken off from training set.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.MethylationDataTypes.MethylationArray.subsample">
<code class="descname">subsample</code><span class="sig-paren">(</span><em>self</em>, <em>key='disease'</em>, <em>n_samples=None</em>, <em>frac=None</em>, <em>categorical=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/MethylationDataTypes.html#MethylationArray.subsample"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.MethylationDataTypes.MethylationArray.subsample" title="Permalink to this definition">¶</a></dt>
<dd><p>Subsample MethylationArray, make the set randomly smaller.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>key</strong></dt>
<dd><p class="first last">If stratifying, use this column of pheno array.</p>
</dd>
<dt><strong>n_samples</strong></dt>
<dd><p class="first last">Number of samples to consider overall, or per stratum.</p>
</dd>
<dt><strong>frac</strong></dt>
<dd><p class="first last">Alternative to n_samples, where x frac of array or stratum is considered.</p>
</dd>
<dt><strong>categorical</strong></dt>
<dd><p class="first last">Whether to stratify by column.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.MethylationDataTypes.MethylationArray.subset_cpgs">
<code class="descname">subset_cpgs</code><span class="sig-paren">(</span><em>self</em>, <em>cpgs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/MethylationDataTypes.html#MethylationArray.subset_cpgs"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.MethylationDataTypes.MethylationArray.subset_cpgs" title="Permalink to this definition">¶</a></dt>
<dd><p>Subset beta matrix by list of Cpgs.
Parameters
———-
cpgs</p>
<blockquote>
<div>Cpgs to subset by.</div></blockquote>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.MethylationDataTypes.MethylationArray.subset_index">
<code class="descname">subset_index</code><span class="sig-paren">(</span><em>self</em>, <em>index</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/MethylationDataTypes.html#MethylationArray.subset_index"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.MethylationDataTypes.MethylationArray.subset_index" title="Permalink to this definition">¶</a></dt>
<dd><p>Subset MethylationArray by samples.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>index</strong></dt>
<dd><p class="first last">Sample names to subset by.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.MethylationDataTypes.MethylationArray.write_csvs">
<code class="descname">write_csvs</code><span class="sig-paren">(</span><em>self</em>, <em>output_dir</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/MethylationDataTypes.html#MethylationArray.write_csvs"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.MethylationDataTypes.MethylationArray.write_csvs" title="Permalink to this definition">¶</a></dt>
<dd><p>Write phenotype data and beta values to csvs.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>output_dir</strong></dt>
<dd><p class="first last">Directory to output csv files.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.MethylationDataTypes.MethylationArray.write_db">
<code class="descname">write_db</code><span class="sig-paren">(</span><em>self</em>, <em>conn</em>, <em>disease=''</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/MethylationDataTypes.html#MethylationArray.write_db"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.MethylationDataTypes.MethylationArray.write_db" title="Permalink to this definition">¶</a></dt>
<dd><p>Store phenotype data and beta values in SQL database.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>conn</strong></dt>
<dd><p class="first last">SQLite connection.</p>
</dd>
<dt><strong>disease</strong></dt>
<dd><p class="first last">Create new tables in db that are related to disease state by this name.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.MethylationDataTypes.MethylationArray.write_pickle">
<code class="descname">write_pickle</code><span class="sig-paren">(</span><em>self</em>, <em>output_pickle</em>, <em>disease=''</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/MethylationDataTypes.html#MethylationArray.write_pickle"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.MethylationDataTypes.MethylationArray.write_pickle" title="Permalink to this definition">¶</a></dt>
<dd><p>Store phenotype data and beta values in pickle file.
Is default file format for storing MethylationArray objects.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>output_pickle</strong></dt>
<dd><p class="first last">Pickle file to store MethylationArray data.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="pymethylprocess.MethylationDataTypes.MethylationArrays">
<em class="property">class </em><code class="descclassname">pymethylprocess.MethylationDataTypes.</code><code class="descname">MethylationArrays</code><span class="sig-paren">(</span><em>list_methylation_arrays</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/MethylationDataTypes.html#MethylationArrays"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.MethylationDataTypes.MethylationArrays" title="Permalink to this definition">¶</a></dt>
<dd><p>Literally a list of methylation arrays, with methods operate on these arrays that is memory efficient.
Initialize with list of methylation arrays. Can optionally leave list empty or with one element.</p>
<dl class="docutils">
<dt>list_methylation_arrays</dt>
<dd>List of methylation arrays.</dd>
</dl>
<p class="rubric">Methods</p>
<table border="1" class="longtable docutils">
<colgroup>
<col width="10%" />
<col width="90%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td><a class="reference internal" href="#pymethylprocess.MethylationDataTypes.MethylationArrays.combine" title="pymethylprocess.MethylationDataTypes.MethylationArrays.combine"><code class="xref py py-obj docutils literal notranslate"><span class="pre">combine</span></code></a>(self[,&nbsp;array_generator])</td>
<td>Combine the list of methylation arrays into one array via concatenation of beta matrices and phenotype arrays.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#pymethylprocess.MethylationDataTypes.MethylationArrays.impute" title="pymethylprocess.MethylationDataTypes.MethylationArrays.impute"><code class="xref py py-obj docutils literal notranslate"><span class="pre">impute</span></code></a>(self,&nbsp;imputer)</td>
<td>Impute all methylation arrays.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#pymethylprocess.MethylationDataTypes.MethylationArrays.write_dbs" title="pymethylprocess.MethylationDataTypes.MethylationArrays.write_dbs"><code class="xref py py-obj docutils literal notranslate"><span class="pre">write_dbs</span></code></a>(self,&nbsp;conn)</td>
<td>Write list of methylation arrays to SQL database.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#pymethylprocess.MethylationDataTypes.MethylationArrays.write_pkls" title="pymethylprocess.MethylationDataTypes.MethylationArrays.write_pkls"><code class="xref py py-obj docutils literal notranslate"><span class="pre">write_pkls</span></code></a>(self,&nbsp;pkl)</td>
<td>Write list of methylation arrays to single pickle.</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="pymethylprocess.MethylationDataTypes.MethylationArrays.combine">
<code class="descname">combine</code><span class="sig-paren">(</span><em>self</em>, <em>array_generator=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/MethylationDataTypes.html#MethylationArrays.combine"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.MethylationDataTypes.MethylationArrays.combine" title="Permalink to this definition">¶</a></dt>
<dd><p>Combine the list of methylation arrays into one array via concatenation of beta matrices and phenotype arrays.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>array_generator</strong></dt>
<dd><p class="first last">Generator of additional methylation arrays for computational memory minimization.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.MethylationDataTypes.MethylationArrays.impute">
<code class="descname">impute</code><span class="sig-paren">(</span><em>self</em>, <em>imputer</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/MethylationDataTypes.html#MethylationArrays.impute"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.MethylationDataTypes.MethylationArrays.impute" title="Permalink to this definition">¶</a></dt>
<dd><p>Impute all methylation arrays.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>imputer</strong></dt>
<dd><p class="first last">Type of imputation, sklearn-like.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.MethylationDataTypes.MethylationArrays.write_dbs">
<code class="descname">write_dbs</code><span class="sig-paren">(</span><em>self</em>, <em>conn</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/MethylationDataTypes.html#MethylationArrays.write_dbs"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.MethylationDataTypes.MethylationArrays.write_dbs" title="Permalink to this definition">¶</a></dt>
<dd><p>Write list of methylation arrays to SQL database. Recommend naming MethylationArray.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>conn</strong></dt>
<dd><p class="first last">SQL connection.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.MethylationDataTypes.MethylationArrays.write_pkls">
<code class="descname">write_pkls</code><span class="sig-paren">(</span><em>self</em>, <em>pkl</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/MethylationDataTypes.html#MethylationArrays.write_pkls"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.MethylationDataTypes.MethylationArrays.write_pkls" title="Permalink to this definition">¶</a></dt>
<dd><p>Write list of methylation arrays to single pickle. Recommend naming each MethylationArray.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>pkl</strong></dt>
<dd><p class="first last">Pickle file to write to.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="function">
<dt id="pymethylprocess.MethylationDataTypes.extract_pheno_beta_df_from_folder">
<code class="descclassname">pymethylprocess.MethylationDataTypes.</code><code class="descname">extract_pheno_beta_df_from_folder</code><span class="sig-paren">(</span><em>folder</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/MethylationDataTypes.html#extract_pheno_beta_df_from_folder"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.MethylationDataTypes.extract_pheno_beta_df_from_folder" title="Permalink to this definition">¶</a></dt>
<dd><p>Return phenotype and beta dataframes from specified folder with csv.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>folder</strong></dt>
<dd><p class="first last">Input folder.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="pymethylprocess.MethylationDataTypes.extract_pheno_beta_df_from_pickle_dict">
<code class="descclassname">pymethylprocess.MethylationDataTypes.</code><code class="descname">extract_pheno_beta_df_from_pickle_dict</code><span class="sig-paren">(</span><em>input_dict</em>, <em>disease=''</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/MethylationDataTypes.html#extract_pheno_beta_df_from_pickle_dict"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.MethylationDataTypes.extract_pheno_beta_df_from_pickle_dict" title="Permalink to this definition">¶</a></dt>
<dd><p>Return phenotype and beta dataframes from specified dictionary storing MethylationArray python dictionary.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>input_dict</strong></dt>
<dd><p class="first last">Python disctionary storing pheno/beta information.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="pymethylprocess.MethylationDataTypes.extract_pheno_beta_df_from_sql">
<code class="descclassname">pymethylprocess.MethylationDataTypes.</code><code class="descname">extract_pheno_beta_df_from_sql</code><span class="sig-paren">(</span><em>conn</em>, <em>disease=''</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/MethylationDataTypes.html#extract_pheno_beta_df_from_sql"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.MethylationDataTypes.extract_pheno_beta_df_from_sql" title="Permalink to this definition">¶</a></dt>
<dd><p>Return phenotype and beta dataframes from SQL tables storing MethylationArray info.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>conn</strong></dt>
<dd><p class="first last">SQL connection.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<span class="target" id="module-pymethylprocess.meffil_functions"></span><div class="section" id="meffil-functions-py">
<h2>meffil_functions.py<a class="headerlink" href="#meffil-functions-py" title="Permalink to this headline">¶</a></h2>
<p>Contains a few R functions that interact with meffil and minfi.</p>
<dl class="function">
<dt id="pymethylprocess.meffil_functions.est_cell_counts_IDOL">
<code class="descclassname">pymethylprocess.meffil_functions.</code><code class="descname">est_cell_counts_IDOL</code><span class="sig-paren">(</span><em>rgset</em>, <em>library</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/meffil_functions.html#est_cell_counts_IDOL"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.meffil_functions.est_cell_counts_IDOL" title="Permalink to this definition">¶</a></dt>
<dd><p>Given RGSet object, estimate cell counts for 450k/850k using reference approach via IDOL library.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>rgset</strong></dt>
<dd><p class="first last">RGSet object stored in python via rpy2</p>
</dd>
<dt><strong>library</strong></dt>
<dd><p class="first last">What type of CpG library to use.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="pymethylprocess.meffil_functions.est_cell_counts_meffil">
<code class="descclassname">pymethylprocess.meffil_functions.</code><code class="descname">est_cell_counts_meffil</code><span class="sig-paren">(</span><em>qc_list</em>, <em>cell_type_reference</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/meffil_functions.html#est_cell_counts_meffil"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.meffil_functions.est_cell_counts_meffil" title="Permalink to this definition">¶</a></dt>
<dd><p>Given QCObject list R object, estimate cell counts using reference approach via meffil.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>qc_list</strong></dt>
<dd><p class="first last">R list containing qc objects.</p>
</dd>
<dt><strong>cell_type_reference</strong></dt>
<dd><p class="first last">Reference blood/tissue set.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="pymethylprocess.meffil_functions.est_cell_counts_minfi">
<code class="descclassname">pymethylprocess.meffil_functions.</code><code class="descname">est_cell_counts_minfi</code><span class="sig-paren">(</span><em>rgset</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/meffil_functions.html#est_cell_counts_minfi"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.meffil_functions.est_cell_counts_minfi" title="Permalink to this definition">¶</a></dt>
<dd><p>Given RGSet object, estimate cell counts using reference approach via minfi.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>rgset</strong></dt>
<dd><p class="first last">RGSet object stored in python via rpy2</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="pymethylprocess.meffil_functions.load_detection_p_values_beadnum">
<code class="descclassname">pymethylprocess.meffil_functions.</code><code class="descname">load_detection_p_values_beadnum</code><span class="sig-paren">(</span><em>qc_list</em>, <em>n_cores</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/meffil_functions.html#load_detection_p_values_beadnum"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.meffil_functions.load_detection_p_values_beadnum" title="Permalink to this definition">¶</a></dt>
<dd><p>Return list of detection p-value matrix and bead number matrix.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>qc_list</strong></dt>
<dd><p class="first last">R list containing qc objects.</p>
</dd>
<dt><strong>n_cores</strong></dt>
<dd><p class="first last">Number of cores to use in computation.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="pymethylprocess.meffil_functions.r_autosomal_cpgs">
<code class="descclassname">pymethylprocess.meffil_functions.</code><code class="descname">r_autosomal_cpgs</code><span class="sig-paren">(</span><em>array_type='450k'</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/meffil_functions.html#r_autosomal_cpgs"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.meffil_functions.r_autosomal_cpgs" title="Permalink to this definition">¶</a></dt>
<dd><p>Return list of autosomal cpg probes per platform.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>array_type</strong></dt>
<dd><p class="first last">450k/850k array?</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="pymethylprocess.meffil_functions.r_snp_cpgs">
<code class="descclassname">pymethylprocess.meffil_functions.</code><code class="descname">r_snp_cpgs</code><span class="sig-paren">(</span><em>array_type='450k'</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/meffil_functions.html#r_snp_cpgs"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.meffil_functions.r_snp_cpgs" title="Permalink to this definition">¶</a></dt>
<dd><p>Return list of SNP cpg probes per platform.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>array_type</strong></dt>
<dd><p class="first last">450k/850k array?</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="pymethylprocess.meffil_functions.remove_sex">
<code class="descclassname">pymethylprocess.meffil_functions.</code><code class="descname">remove_sex</code><span class="sig-paren">(</span><em>beta</em>, <em>array_type='450k'</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/meffil_functions.html#remove_sex"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.meffil_functions.remove_sex" title="Permalink to this definition">¶</a></dt>
<dd><p>Remove non-autosomal cpgs from beta matrix.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>array_type</strong></dt>
<dd><p class="first last">450k/850k array?</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="pymethylprocess.meffil_functions.set_missing">
<code class="descclassname">pymethylprocess.meffil_functions.</code><code class="descname">set_missing</code><span class="sig-paren">(</span><em>beta</em>, <em>pval_beadnum</em>, <em>detection_val=1e-06</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/meffil_functions.html#set_missing"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.meffil_functions.set_missing" title="Permalink to this definition">¶</a></dt>
<dd><p>Set missing beta values to NA, taking into account detection values and bead number thesholds.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>pval_beadnum</strong></dt>
<dd><p class="first last">Detection pvalues and number of beads per cpg/samples</p>
</dd>
<dt><strong>detection_val</strong></dt>
<dd><p class="first last">If threshold to set site to missingness based on p-value detection.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<span class="target" id="module-pymethylprocess.general_machine_learning"></span><div class="section" id="general-machine-learning-py">
<h2>general_machine_learning.py<a class="headerlink" href="#general-machine-learning-py" title="Permalink to this headline">¶</a></h2>
<p>Contains a machine learning class to perform scikit-learn like operations, along with held-out hyperparameter grid search.</p>
<dl class="class">
<dt id="pymethylprocess.general_machine_learning.MachineLearning">
<em class="property">class </em><code class="descclassname">pymethylprocess.general_machine_learning.</code><code class="descname">MachineLearning</code><span class="sig-paren">(</span><em>model</em>, <em>options</em>, <em>grid={}</em>, <em>labelencode=False</em>, <em>n_eval=0</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/general_machine_learning.html#MachineLearning"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.general_machine_learning.MachineLearning" title="Permalink to this definition">¶</a></dt>
<dd><p>Machine learning class to run sklearn-like pipeline on MethylationArray data.
Initialize object with scikit-learn model, and optionally supply a hyperparameter search grid.</p>
<dl class="docutils">
<dt>model</dt>
<dd>Scikit-learn-like model, classification, regression, dimensionality reduction, clustering etc.</dd>
<dt>options</dt>
<dd>Options to supply model in form of dictionary.</dd>
<dt>grid</dt>
<dd>Alternatively, supply search grid to search for bets hyperparameters.</dd>
<dt>labelencode</dt>
<dd>T/F encode string labels.</dd>
<dt>n_eval</dt>
<dd>Number of evaluations for randomized grid search, if set to 0, perform exhaustive grid search</dd>
</dl>
<p class="rubric">Methods</p>
<table border="1" class="longtable docutils">
<colgroup>
<col width="10%" />
<col width="90%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td><a class="reference internal" href="#pymethylprocess.general_machine_learning.MachineLearning.assign_results_to_pheno_col" title="pymethylprocess.general_machine_learning.MachineLearning.assign_results_to_pheno_col"><code class="xref py py-obj docutils literal notranslate"><span class="pre">assign_results_to_pheno_col</span></code></a>(self,&nbsp;…)</td>
<td>Assign results to new phenotype column.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#pymethylprocess.general_machine_learning.MachineLearning.fit" title="pymethylprocess.general_machine_learning.MachineLearning.fit"><code class="xref py py-obj docutils literal notranslate"><span class="pre">fit</span></code></a>(self,&nbsp;train_methyl_array[,&nbsp;…])</td>
<td>Fit data to model.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#pymethylprocess.general_machine_learning.MachineLearning.fit_predict" title="pymethylprocess.general_machine_learning.MachineLearning.fit_predict"><code class="xref py py-obj docutils literal notranslate"><span class="pre">fit_predict</span></code></a>(self,&nbsp;train_methyl_array[,&nbsp;…])</td>
<td>Fit and predict training data.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#pymethylprocess.general_machine_learning.MachineLearning.fit_transform" title="pymethylprocess.general_machine_learning.MachineLearning.fit_transform"><code class="xref py py-obj docutils literal notranslate"><span class="pre">fit_transform</span></code></a>(self,&nbsp;train_methyl_array[,&nbsp;…])</td>
<td>Fit and transform to training data.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#pymethylprocess.general_machine_learning.MachineLearning.predict" title="pymethylprocess.general_machine_learning.MachineLearning.predict"><code class="xref py py-obj docutils literal notranslate"><span class="pre">predict</span></code></a>(self,&nbsp;test_methyl_array)</td>
<td>Make new predictions on test methylation array.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#pymethylprocess.general_machine_learning.MachineLearning.return_outcome_metric" title="pymethylprocess.general_machine_learning.MachineLearning.return_outcome_metric"><code class="xref py py-obj docutils literal notranslate"><span class="pre">return_outcome_metric</span></code></a>(self,&nbsp;methyl_array,&nbsp;…)</td>
<td>Supply metric to evaluate results.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#pymethylprocess.general_machine_learning.MachineLearning.store_results" title="pymethylprocess.general_machine_learning.MachineLearning.store_results"><code class="xref py py-obj docutils literal notranslate"><span class="pre">store_results</span></code></a>(self,&nbsp;output_pkl[,&nbsp;results_dict])</td>
<td>Store results in pickle file.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#pymethylprocess.general_machine_learning.MachineLearning.transform" title="pymethylprocess.general_machine_learning.MachineLearning.transform"><code class="xref py py-obj docutils literal notranslate"><span class="pre">transform</span></code></a>(self,&nbsp;test_methyl_array)</td>
<td>Transform test methylation array.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#pymethylprocess.general_machine_learning.MachineLearning.transform_results_to_beta" title="pymethylprocess.general_machine_learning.MachineLearning.transform_results_to_beta"><code class="xref py py-obj docutils literal notranslate"><span class="pre">transform_results_to_beta</span></code></a>(self,&nbsp;…)</td>
<td>Transform beta matrix into reduced beta matrix and store.</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="pymethylprocess.general_machine_learning.MachineLearning.assign_results_to_pheno_col">
<code class="descname">assign_results_to_pheno_col</code><span class="sig-paren">(</span><em>self</em>, <em>methyl_array</em>, <em>new_col</em>, <em>output_pkl</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/general_machine_learning.html#MachineLearning.assign_results_to_pheno_col"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.general_machine_learning.MachineLearning.assign_results_to_pheno_col" title="Permalink to this definition">¶</a></dt>
<dd><p>Assign results to new phenotype column.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>methyl_array</strong></dt>
<dd><p class="first last">MethylationArray.</p>
</dd>
<dt><strong>new_col</strong></dt>
<dd><p class="first last">New column name.</p>
</dd>
<dt><strong>output_pkl</strong></dt>
<dd><p class="first last">Output pickle to dump MethylationArray to.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.general_machine_learning.MachineLearning.fit">
<code class="descname">fit</code><span class="sig-paren">(</span><em>self</em>, <em>train_methyl_array</em>, <em>val_methyl_array=None</em>, <em>outcome_cols=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/general_machine_learning.html#MachineLearning.fit"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.general_machine_learning.MachineLearning.fit" title="Permalink to this definition">¶</a></dt>
<dd><p>Fit data to model.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>train_methyl_array</strong></dt>
<dd><p class="first last">Training MethylationArray.</p>
</dd>
<dt><strong>val_methyl_array</strong></dt>
<dd><p class="first last">Validation MethylationArray. Can set to None.</p>
</dd>
<dt><strong>outcome_cols</strong></dt>
<dd><p class="first last">Set to none if not needed, but phenotype column to train on, can be multiple.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.general_machine_learning.MachineLearning.fit_predict">
<code class="descname">fit_predict</code><span class="sig-paren">(</span><em>self</em>, <em>train_methyl_array</em>, <em>outcome_cols=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/general_machine_learning.html#MachineLearning.fit_predict"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.general_machine_learning.MachineLearning.fit_predict" title="Permalink to this definition">¶</a></dt>
<dd><p>Fit and predict training data.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>train_methyl_array</strong></dt>
<dd><p class="first last">Training MethylationArray.</p>
</dd>
<dt><strong>outcome_cols</strong></dt>
<dd><p class="first last">Set to none if not needed, but phenotype column to train on, can be multiple.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.general_machine_learning.MachineLearning.fit_transform">
<code class="descname">fit_transform</code><span class="sig-paren">(</span><em>self</em>, <em>train_methyl_array</em>, <em>outcome_cols=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/general_machine_learning.html#MachineLearning.fit_transform"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.general_machine_learning.MachineLearning.fit_transform" title="Permalink to this definition">¶</a></dt>
<dd><p>Fit and transform to training data.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>train_methyl_array</strong></dt>
<dd><p class="first last">Training MethylationArray.</p>
</dd>
<dt><strong>outcome_cols</strong></dt>
<dd><p class="first last">Set to none if not needed, but phenotype column to train on, can be multiple.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.general_machine_learning.MachineLearning.predict">
<code class="descname">predict</code><span class="sig-paren">(</span><em>self</em>, <em>test_methyl_array</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/general_machine_learning.html#MachineLearning.predict"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.general_machine_learning.MachineLearning.predict" title="Permalink to this definition">¶</a></dt>
<dd><p>Make new predictions on test methylation array.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>test_methyl_array</strong></dt>
<dd><p class="first last">Testing MethylationArray.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.general_machine_learning.MachineLearning.return_outcome_metric">
<code class="descname">return_outcome_metric</code><span class="sig-paren">(</span><em>self</em>, <em>methyl_array</em>, <em>outcome_cols</em>, <em>metric</em>, <em>run_bootstrap=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/general_machine_learning.html#MachineLearning.return_outcome_metric"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.general_machine_learning.MachineLearning.return_outcome_metric" title="Permalink to this definition">¶</a></dt>
<dd><p>Supply metric to evaluate results.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>methyl_array</strong></dt>
<dd><p class="first last">MethylationArray to evaluate.</p>
</dd>
<dt><strong>outcome_cols</strong></dt>
<dd><p class="first last">Outcome phenotype columns.</p>
</dd>
<dt><strong>metric</strong></dt>
<dd><p class="first last">Sklearn evaluation metric.</p>
</dd>
<dt><strong>run_bootstrap</strong></dt>
<dd><p class="first last">Make 95% CI from 1k bootstraps.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.general_machine_learning.MachineLearning.store_results">
<code class="descname">store_results</code><span class="sig-paren">(</span><em>self</em>, <em>output_pkl</em>, <em>results_dict={}</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/general_machine_learning.html#MachineLearning.store_results"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.general_machine_learning.MachineLearning.store_results" title="Permalink to this definition">¶</a></dt>
<dd><p>Store results in pickle file.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>output_pkl</strong></dt>
<dd><p class="first last">Output pickle to dump results to.</p>
</dd>
<dt><strong>results_dict</strong></dt>
<dd><p class="first last">Supply own results dict to be dumped.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.general_machine_learning.MachineLearning.transform">
<code class="descname">transform</code><span class="sig-paren">(</span><em>self</em>, <em>test_methyl_array</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/general_machine_learning.html#MachineLearning.transform"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.general_machine_learning.MachineLearning.transform" title="Permalink to this definition">¶</a></dt>
<dd><p>Transform test methylation array.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>test_methyl_array</strong></dt>
<dd><p class="first last">Testing MethylationArray.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymethylprocess.general_machine_learning.MachineLearning.transform_results_to_beta">
<code class="descname">transform_results_to_beta</code><span class="sig-paren">(</span><em>self</em>, <em>methyl_array</em>, <em>output_pkl</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymethylprocess/general_machine_learning.html#MachineLearning.transform_results_to_beta"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymethylprocess.general_machine_learning.MachineLearning.transform_results_to_beta" title="Permalink to this definition">¶</a></dt>
<dd><p>Transform beta matrix into reduced beta matrix and store.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>methyl_array</strong></dt>
<dd><p class="first last">MethylationArray.</p>
</dd>
<dt><strong>output_pkl</strong></dt>
<dd><p class="first last">Output pickle to dump MethylationArray to.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

</div>
<div class="section" id="pymethyl-install">
<h2>pymethyl-install<a class="headerlink" href="#pymethyl-install" title="Permalink to this headline">¶</a></h2>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-install <span class="o">[</span>OPTIONS<span class="o">]</span> COMMAND <span class="o">[</span>ARGS<span class="o">]</span>...
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-install-version">
<code class="descname">--version</code><code class="descclassname"></code><a class="headerlink" href="#cmdoption-pymethyl-install-version" title="Permalink to this definition">¶</a></dt>
<dd><p>Show the version and exit.</p>
</dd></dl>

<div class="section" id="pymethyl-install-change-gcc-path">
<h3>change_gcc_path<a class="headerlink" href="#pymethyl-install-change-gcc-path" title="Permalink to this headline">¶</a></h3>
<p>Change GCC and G++ paths if don’t have version 7.2.0. [Experimental]</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-install change_gcc_path <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
</div>
<div class="section" id="pymethyl-install-install-bioconductor">
<h3>install_bioconductor<a class="headerlink" href="#pymethyl-install-install-bioconductor" title="Permalink to this headline">¶</a></h3>
<p>Installs bioconductor.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-install install_bioconductor <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
</div>
<div class="section" id="pymethyl-install-install-custom">
<h3>install_custom<a class="headerlink" href="#pymethyl-install-install-custom" title="Permalink to this headline">¶</a></h3>
<p>Installs bioconductor packages.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-install install_custom <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-install-install-custom-p">
<code class="descname">-p</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--package</code><code class="descclassname"> &lt;package&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-install-install-custom-p" title="Permalink to this definition">¶</a></dt>
<dd><p>Custom packages.  [default: ENmix]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-install-install-custom-m">
<code class="descname">-m</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--manager</code><code class="descclassname"></code><a class="headerlink" href="#cmdoption-pymethyl-install-install-custom-m" title="Permalink to this definition">¶</a></dt>
<dd><p>Use BiocManager (recommended).</p>
</dd></dl>

</div>
<div class="section" id="pymethyl-install-install-meffil">
<h3>install_meffil<a class="headerlink" href="#pymethyl-install-install-meffil" title="Permalink to this headline">¶</a></h3>
<p>Installs meffil (update!).</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-install install_meffil <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
</div>
<div class="section" id="pymethyl-install-install-minfi-others">
<h3>install_minfi_others<a class="headerlink" href="#pymethyl-install-install-minfi-others" title="Permalink to this headline">¶</a></h3>
<p>Installs minfi and other dependencies.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-install install_minfi_others <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
</div>
<div class="section" id="pymethyl-install-install-r-packages">
<h3>install_r_packages<a class="headerlink" href="#pymethyl-install-install-r-packages" title="Permalink to this headline">¶</a></h3>
<p>Installs r packages.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-install install_r_packages <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-install-install-r-packages-p">
<code class="descname">-p</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--package</code><code class="descclassname"> &lt;package&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-install-install-r-packages-p" title="Permalink to this definition">¶</a></dt>
<dd><p>Custom packages.  [default: ]</p>
</dd></dl>

</div>
<div class="section" id="pymethyl-install-install-some-deps">
<h3>install_some_deps<a class="headerlink" href="#pymethyl-install-install-some-deps" title="Permalink to this headline">¶</a></h3>
<p>Installs bioconductor, minfi, enmix, tcga biolinks, and meffil.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-install install_some_deps <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
</div>
<div class="section" id="pymethyl-install-install-tcga-biolinks">
<h3>install_tcga_biolinks<a class="headerlink" href="#pymethyl-install-install-tcga-biolinks" title="Permalink to this headline">¶</a></h3>
<p>Installs tcga biolinks.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-install install_tcga_biolinks <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="pymethyl-visualize">
<h2>pymethyl-visualize<a class="headerlink" href="#pymethyl-visualize" title="Permalink to this headline">¶</a></h2>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-visualize <span class="o">[</span>OPTIONS<span class="o">]</span> COMMAND <span class="o">[</span>ARGS<span class="o">]</span>...
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-visualize-version">
<code class="descname">--version</code><code class="descclassname"></code><a class="headerlink" href="#cmdoption-pymethyl-visualize-version" title="Permalink to this definition">¶</a></dt>
<dd><p>Show the version and exit.</p>
</dd></dl>

<div class="section" id="pymethyl-visualize-plot-cell-type-results">
<h3>plot_cell_type_results<a class="headerlink" href="#pymethyl-visualize-plot-cell-type-results" title="Permalink to this headline">¶</a></h3>
<p>Plot csv containing cell type results into side by side boxplots.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-visualize plot_cell_type_results <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-visualize-plot-cell-type-results-i">
<code class="descname">-i</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--input_csv</code><code class="descclassname"> &lt;input_csv&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-visualize-plot-cell-type-results-i" title="Permalink to this definition">¶</a></dt>
<dd><p>Input csv.  [default: cell_type_estimates.csv]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-visualize-plot-cell-type-results-o">
<code class="descname">-o</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--outfilename</code><code class="descclassname"> &lt;outfilename&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-visualize-plot-cell-type-results-o" title="Permalink to this definition">¶</a></dt>
<dd><p>Output png.  [default: visualizations/cell_type_results.png]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-visualize-plot-cell-type-results-cols">
<code class="descname">-cols</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--plot_cols</code><code class="descclassname"> &lt;plot_cols&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-visualize-plot-cell-type-results-cols" title="Permalink to this definition">¶</a></dt>
<dd><p>Plot columns.  [default: Gran, CD4T, CD8T, Bcell, Mono, NK, gMDSC]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-visualize-plot-cell-type-results-fs">
<code class="descname">-fs</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--font_scale</code><code class="descclassname"> &lt;font_scale&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-visualize-plot-cell-type-results-fs" title="Permalink to this definition">¶</a></dt>
<dd><p>Font scaling  [default: 1.0]</p>
</dd></dl>

</div>
<div class="section" id="pymethyl-visualize-plot-heatmap">
<h3>plot_heatmap<a class="headerlink" href="#pymethyl-visualize-plot-heatmap" title="Permalink to this headline">¶</a></h3>
<p>Plot heatmap from CSV file.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-visualize plot_heatmap <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-visualize-plot-heatmap-i">
<code class="descname">-i</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--input_csv</code><code class="descclassname"> &lt;input_csv&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-visualize-plot-heatmap-i" title="Permalink to this definition">¶</a></dt>
<dd><p>Input csv.  [default: ]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-visualize-plot-heatmap-o">
<code class="descname">-o</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--outfilename</code><code class="descclassname"> &lt;outfilename&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-visualize-plot-heatmap-o" title="Permalink to this definition">¶</a></dt>
<dd><p>Output png.  [default: output.png]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-visualize-plot-heatmap-idx">
<code class="descname">-idx</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--index_col</code><code class="descclassname"> &lt;index_col&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-visualize-plot-heatmap-idx" title="Permalink to this definition">¶</a></dt>
<dd><p>Index load dataframe  [default: 0]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-visualize-plot-heatmap-fs">
<code class="descname">-fs</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--font_scale</code><code class="descclassname"> &lt;font_scale&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-visualize-plot-heatmap-fs" title="Permalink to this definition">¶</a></dt>
<dd><p>Font scaling  [default: 1.0]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-visualize-plot-heatmap-min">
<code class="descname">-min</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--min_val</code><code class="descclassname"> &lt;min_val&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-visualize-plot-heatmap-min" title="Permalink to this definition">¶</a></dt>
<dd><p>Min heat val  [default: 0.0]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-visualize-plot-heatmap-max">
<code class="descname">-max</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--max_val</code><code class="descclassname"> &lt;max_val&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-visualize-plot-heatmap-max" title="Permalink to this definition">¶</a></dt>
<dd><p>Max heat val, if -1, defaults to None  [default: 1.0]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-visualize-plot-heatmap-a">
<code class="descname">-a</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--annot</code><code class="descclassname"></code><a class="headerlink" href="#cmdoption-pymethyl-visualize-plot-heatmap-a" title="Permalink to this definition">¶</a></dt>
<dd><p>Annotate heatmap  [default: False]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-visualize-plot-heatmap-n">
<code class="descname">-n</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--norm</code><code class="descclassname"></code><a class="headerlink" href="#cmdoption-pymethyl-visualize-plot-heatmap-n" title="Permalink to this definition">¶</a></dt>
<dd><p>Normalize matrix data  [default: False]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-visualize-plot-heatmap-c">
<code class="descname">-c</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--cluster</code><code class="descclassname"></code><a class="headerlink" href="#cmdoption-pymethyl-visualize-plot-heatmap-c" title="Permalink to this definition">¶</a></dt>
<dd><p>Cluster matrix data  [default: False]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-visualize-plot-heatmap-m">
<code class="descname">-m</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--matrix_type</code><code class="descclassname"> &lt;matrix_type&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-visualize-plot-heatmap-m" title="Permalink to this definition">¶</a></dt>
<dd><p>Type of matrix supplied  [default: none]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-visualize-plot-heatmap-x">
<code class="descname">-x</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--xticks</code><code class="descclassname"></code><a class="headerlink" href="#cmdoption-pymethyl-visualize-plot-heatmap-x" title="Permalink to this definition">¶</a></dt>
<dd><p>Show x ticks  [default: False]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-visualize-plot-heatmap-y">
<code class="descname">-y</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--yticks</code><code class="descclassname"></code><a class="headerlink" href="#cmdoption-pymethyl-visualize-plot-heatmap-y" title="Permalink to this definition">¶</a></dt>
<dd><p>Show y ticks  [default: False]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-visualize-plot-heatmap-t">
<code class="descname">-t</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--transpose</code><code class="descclassname"></code><a class="headerlink" href="#cmdoption-pymethyl-visualize-plot-heatmap-t" title="Permalink to this definition">¶</a></dt>
<dd><p>Transpose matrix data  [default: False]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-visualize-plot-heatmap-col">
<code class="descname">-col</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--color_column</code><code class="descclassname"> &lt;color_column&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-visualize-plot-heatmap-col" title="Permalink to this definition">¶</a></dt>
<dd><p>Color column.  [default: color]</p>
</dd></dl>

</div>
<div class="section" id="pymethyl-visualize-transform-plot">
<h3>transform_plot<a class="headerlink" href="#pymethyl-visualize-transform-plot" title="Permalink to this headline">¶</a></h3>
<p>Dimensionality reduce VAE or original beta values using UMAP and plot using plotly.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-visualize transform_plot <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-visualize-transform-plot-i">
<code class="descname">-i</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--input_pkl</code><code class="descclassname"> &lt;input_pkl&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-visualize-transform-plot-i" title="Permalink to this definition">¶</a></dt>
<dd><p>Input database for beta and phenotype data.  [default: ./final_preprocessed/methyl_array.pkl]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-visualize-transform-plot-c">
<code class="descname">-c</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--column_of_interest</code><code class="descclassname"> &lt;column_of_interest&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-visualize-transform-plot-c" title="Permalink to this definition">¶</a></dt>
<dd><p>Column extract from phenotype data.  [default: disease]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-visualize-transform-plot-o">
<code class="descname">-o</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--output_file</code><code class="descclassname"> &lt;output_file&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-visualize-transform-plot-o" title="Permalink to this definition">¶</a></dt>
<dd><p>Output visualization.  [default: ./visualization.html]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-visualize-transform-plot-nn">
<code class="descname">-nn</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--n_neighbors</code><code class="descclassname"> &lt;n_neighbors&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-visualize-transform-plot-nn" title="Permalink to this definition">¶</a></dt>
<dd><p>Number of neighbors UMAP.  [default: 5]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-visualize-transform-plot-a">
<code class="descname">-a</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--axes_off</code><code class="descclassname"></code><a class="headerlink" href="#cmdoption-pymethyl-visualize-transform-plot-a" title="Permalink to this definition">¶</a></dt>
<dd><p>Whether to turn axes on or off.</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-visualize-transform-plot-s">
<code class="descname">-s</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--supervised</code><code class="descclassname"></code><a class="headerlink" href="#cmdoption-pymethyl-visualize-transform-plot-s" title="Permalink to this definition">¶</a></dt>
<dd><p>Supervise umap embedding.</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-visualize-transform-plot-d">
<code class="descname">-d</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--min_dist</code><code class="descclassname"> &lt;min_dist&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-visualize-transform-plot-d" title="Permalink to this definition">¶</a></dt>
<dd><p>UMAP min distance.  [default: 0.1]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-visualize-transform-plot-m">
<code class="descname">-m</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--metric</code><code class="descclassname"> &lt;metric&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-visualize-transform-plot-m" title="Permalink to this definition">¶</a></dt>
<dd><p>Reduction metric.  [default: euclidean]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-visualize-transform-plot-cc">
<code class="descname">-cc</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--case_control_override</code><code class="descclassname"></code><a class="headerlink" href="#cmdoption-pymethyl-visualize-transform-plot-cc" title="Permalink to this definition">¶</a></dt>
<dd><p>Add controls from case_control column and override current disease for classification tasks.  [default: False]</p>
</dd></dl>

</div>
</div>
<div class="section" id="pymethyl-preprocess">
<h2>pymethyl-preprocess<a class="headerlink" href="#pymethyl-preprocess" title="Permalink to this headline">¶</a></h2>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-preprocess <span class="o">[</span>OPTIONS<span class="o">]</span> COMMAND <span class="o">[</span>ARGS<span class="o">]</span>...
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-version">
<code class="descname">--version</code><code class="descclassname"></code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-version" title="Permalink to this definition">¶</a></dt>
<dd><p>Show the version and exit.</p>
</dd></dl>

<div class="section" id="pymethyl-preprocess-batch-deploy-preprocess">
<h3>batch_deploy_preprocess<a class="headerlink" href="#pymethyl-preprocess-batch-deploy-preprocess" title="Permalink to this headline">¶</a></h3>
<p>Deploy multiple preprocessing jobs in series or parallel.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-preprocess batch_deploy_preprocess <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-batch-deploy-preprocess-n">
<code class="descname">-n</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--n_cores</code><code class="descclassname"> &lt;n_cores&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-batch-deploy-preprocess-n" title="Permalink to this definition">¶</a></dt>
<dd><p>Number cores to use for preprocessing.  [default: 6]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-batch-deploy-preprocess-i">
<code class="descname">-i</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--subtype_output_dir</code><code class="descclassname"> &lt;subtype_output_dir&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-batch-deploy-preprocess-i" title="Permalink to this definition">¶</a></dt>
<dd><p>Output subtypes pheno csv.  [default: ./preprocess_outputs/]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-batch-deploy-preprocess-m">
<code class="descname">-m</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--meffil</code><code class="descclassname"></code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-batch-deploy-preprocess-m" title="Permalink to this definition">¶</a></dt>
<dd><p>Preprocess using meffil.</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-batch-deploy-preprocess-t">
<code class="descname">-t</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--torque</code><code class="descclassname"></code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-batch-deploy-preprocess-t" title="Permalink to this definition">¶</a></dt>
<dd><p>Job submission torque.</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-batch-deploy-preprocess-r">
<code class="descname">-r</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--run</code><code class="descclassname"></code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-batch-deploy-preprocess-r" title="Permalink to this definition">¶</a></dt>
<dd><p>Actually run local job or just print out command.</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-batch-deploy-preprocess-s">
<code class="descname">-s</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--series</code><code class="descclassname"></code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-batch-deploy-preprocess-s" title="Permalink to this definition">¶</a></dt>
<dd><p>Run commands in series.</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-batch-deploy-preprocess-p">
<code class="descname">-p</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--pc_qc_parameters_csv</code><code class="descclassname"> &lt;pc_qc_parameters_csv&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-batch-deploy-preprocess-p" title="Permalink to this definition">¶</a></dt>
<dd><p>For meffil, qc parameters and pcs for final qc and functional normalization.  [default: ./preprocess_outputs/pc_qc_parameters.csv]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-batch-deploy-preprocess-u">
<code class="descname">-u</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--use_cache</code><code class="descclassname"></code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-batch-deploy-preprocess-u" title="Permalink to this definition">¶</a></dt>
<dd><p>If this is selected, loads qc results rather than running qc again. Only works for meffil selection.</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-batch-deploy-preprocess-qc">
<code class="descname">-qc</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--qc_only</code><code class="descclassname"></code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-batch-deploy-preprocess-qc" title="Permalink to this definition">¶</a></dt>
<dd><p>Only perform QC for meffil pipeline, caches results into rds file for loading again, only works if use_cache is false.</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-batch-deploy-preprocess-c">
<code class="descname">-c</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--chunk_size</code><code class="descclassname"> &lt;chunk_size&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-batch-deploy-preprocess-c" title="Permalink to this definition">¶</a></dt>
<dd><p>If not series, chunk up and run these number of commands at once.. -1 means all commands at once.</p>
</dd></dl>

</div>
<div class="section" id="pymethyl-preprocess-combine-methylation-arrays">
<h3>combine_methylation_arrays<a class="headerlink" href="#pymethyl-preprocess-combine-methylation-arrays" title="Permalink to this headline">¶</a></h3>
<p>If split MethylationArrays by subtype for either preprocessing or imputation, can use to recombine data for downstream step.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-preprocess combine_methylation_arrays <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-combine-methylation-arrays-i">
<code class="descname">-i</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--input_pkls</code><code class="descclassname"> &lt;input_pkls&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-combine-methylation-arrays-i" title="Permalink to this definition">¶</a></dt>
<dd><p>Input pickles for beta and phenotype data.  [default: ./preprocess_outputs/methyl_array.pkl]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-combine-methylation-arrays-d">
<code class="descname">-d</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--optional_input_pkl_dir</code><code class="descclassname"> &lt;optional_input_pkl_dir&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-combine-methylation-arrays-d" title="Permalink to this definition">¶</a></dt>
<dd><p>Auto grab input pkls.  [default: ]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-combine-methylation-arrays-o">
<code class="descname">-o</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--output_pkl</code><code class="descclassname"> &lt;output_pkl&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-combine-methylation-arrays-o" title="Permalink to this definition">¶</a></dt>
<dd><p>Output database for beta and phenotype data.  [default: ./combined_outputs/methyl_array.pkl]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-combine-methylation-arrays-e">
<code class="descname">-e</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--exclude</code><code class="descclassname"> &lt;exclude&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-combine-methylation-arrays-e" title="Permalink to this definition">¶</a></dt>
<dd><p>If -d selected, these diseases will be excluded from study.  [default: ]</p>
</dd></dl>

</div>
<div class="section" id="pymethyl-preprocess-concat-sample-sheets">
<h3>concat_sample_sheets<a class="headerlink" href="#pymethyl-preprocess-concat-sample-sheets" title="Permalink to this headline">¶</a></h3>
<p>Concat two sample files for more fields for minfi+ input, adds more samples.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-preprocess concat_sample_sheets <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-concat-sample-sheets-s1">
<code class="descname">-s1</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--sample_sheet1</code><code class="descclassname"> &lt;sample_sheet1&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-concat-sample-sheets-s1" title="Permalink to this definition">¶</a></dt>
<dd><p>Clinical information downloaded from tcga/geo/custom, formatted using create_sample_sheet.  [default: ./tcga_idats/clinical_info1.csv]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-concat-sample-sheets-s2">
<code class="descname">-s2</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--sample_sheet2</code><code class="descclassname"> &lt;sample_sheet2&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-concat-sample-sheets-s2" title="Permalink to this definition">¶</a></dt>
<dd><p>Clinical information downloaded from tcga/geo/custom, formatted using create_sample_sheet.  [default: ./tcga_idats/clinical_info2.csv]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-concat-sample-sheets-os">
<code class="descname">-os</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--output_sample_sheet</code><code class="descclassname"> &lt;output_sample_sheet&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-concat-sample-sheets-os" title="Permalink to this definition">¶</a></dt>
<dd><p>CSV for minfi input.  [default: ./tcga_idats/minfiSheet.csv]</p>
</dd></dl>

</div>
<div class="section" id="pymethyl-preprocess-create-sample-sheet">
<h3>create_sample_sheet<a class="headerlink" href="#pymethyl-preprocess-create-sample-sheet" title="Permalink to this headline">¶</a></h3>
<p>Create sample sheet for input to minfi, meffil, or enmix.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-preprocess create_sample_sheet <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-create-sample-sheet-is">
<code class="descname">-is</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--input_sample_sheet</code><code class="descclassname"> &lt;input_sample_sheet&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-create-sample-sheet-is" title="Permalink to this definition">¶</a></dt>
<dd><p>Clinical information downloaded from tcga/geo/custom.  [default: ./tcga_idats/clinical_info.csv]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-create-sample-sheet-s">
<code class="descname">-s</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--source_type</code><code class="descclassname"> &lt;source_type&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-create-sample-sheet-s" title="Permalink to this definition">¶</a></dt>
<dd><p>Source type of data.  [default: tcga]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-create-sample-sheet-i">
<code class="descname">-i</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--idat_dir</code><code class="descclassname"> &lt;idat_dir&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-create-sample-sheet-i" title="Permalink to this definition">¶</a></dt>
<dd><p>Idat directory.  [default: ./tcga_idats/]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-create-sample-sheet-os">
<code class="descname">-os</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--output_sample_sheet</code><code class="descclassname"> &lt;output_sample_sheet&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-create-sample-sheet-os" title="Permalink to this definition">¶</a></dt>
<dd><p>CSV for minfi input.  [default: ./tcga_idats/minfiSheet.csv]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-create-sample-sheet-m">
<code class="descname">-m</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--mapping_file</code><code class="descclassname"> &lt;mapping_file&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-create-sample-sheet-m" title="Permalink to this definition">¶</a></dt>
<dd><p>Mapping file from uuid to TCGA barcode. Downloaded using download_tcga.  [default: ./idat_filename_case.txt]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-create-sample-sheet-l">
<code class="descname">-l</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--header_line</code><code class="descclassname"> &lt;header_line&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-create-sample-sheet-l" title="Permalink to this definition">¶</a></dt>
<dd><p>Line to begin reading csv/xlsx.  [default: 0]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-create-sample-sheet-d">
<code class="descname">-d</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--disease_class_column</code><code class="descclassname"> &lt;disease_class_column&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-create-sample-sheet-d" title="Permalink to this definition">¶</a></dt>
<dd><p>Disease classification column, for custom and geo datasets.  [default: methylation class:ch1]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-create-sample-sheet-b">
<code class="descname">-b</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--basename_col</code><code class="descclassname"> &lt;basename_col&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-create-sample-sheet-b" title="Permalink to this definition">¶</a></dt>
<dd><p>Basename classification column, for custom datasets.  [default: Sentrix ID (.idat)]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-create-sample-sheet-c">
<code class="descname">-c</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--include_columns_file</code><code class="descclassname"> &lt;include_columns_file&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-create-sample-sheet-c" title="Permalink to this definition">¶</a></dt>
<dd><p>Custom columns file containing columns to keep, separated by n. Add a tab for each line if you wish to rename columns: original_name t new_column_name  [default: ]</p>
</dd></dl>

</div>
<div class="section" id="pymethyl-preprocess-download-clinical">
<h3>download_clinical<a class="headerlink" href="#pymethyl-preprocess-download-clinical" title="Permalink to this headline">¶</a></h3>
<p>Download all TCGA 450k clinical info.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-preprocess download_clinical <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-download-clinical-o">
<code class="descname">-o</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--output_dir</code><code class="descclassname"> &lt;output_dir&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-download-clinical-o" title="Permalink to this definition">¶</a></dt>
<dd><p>Output directory for exported idats.  [default: ./tcga_idats/]</p>
</dd></dl>

</div>
<div class="section" id="pymethyl-preprocess-download-geo">
<h3>download_geo<a class="headerlink" href="#pymethyl-preprocess-download-geo" title="Permalink to this headline">¶</a></h3>
<p>Download geo methylation study idats and clinical info.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-preprocess download_geo <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-download-geo-g">
<code class="descname">-g</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--geo_query</code><code class="descclassname"> &lt;geo_query&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-download-geo-g" title="Permalink to this definition">¶</a></dt>
<dd><p>GEO study to query.  [default: ]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-download-geo-o">
<code class="descname">-o</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--output_dir</code><code class="descclassname"> &lt;output_dir&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-download-geo-o" title="Permalink to this definition">¶</a></dt>
<dd><p>Output directory for exported idats.  [default: ./geo_idats/]</p>
</dd></dl>

</div>
<div class="section" id="pymethyl-preprocess-download-tcga">
<h3>download_tcga<a class="headerlink" href="#pymethyl-preprocess-download-tcga" title="Permalink to this headline">¶</a></h3>
<p>Download all tcga 450k data.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-preprocess download_tcga <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-download-tcga-o">
<code class="descname">-o</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--output_dir</code><code class="descclassname"> &lt;output_dir&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-download-tcga-o" title="Permalink to this definition">¶</a></dt>
<dd><p>Output directory for exported idats.  [default: ./tcga_idats/]</p>
</dd></dl>

</div>
<div class="section" id="pymethyl-preprocess-feature-select">
<h3>feature_select<a class="headerlink" href="#pymethyl-preprocess-feature-select" title="Permalink to this headline">¶</a></h3>
<p>Filter CpGs by taking x top CpGs with highest mean absolute deviation scores or via spectral feature selection.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-preprocess feature_select <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-feature-select-i">
<code class="descname">-i</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--input_pkl</code><code class="descclassname"> &lt;input_pkl&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-feature-select-i" title="Permalink to this definition">¶</a></dt>
<dd><p>Input database for beta and phenotype data.  [default: ./imputed_outputs/methyl_array.pkl]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-feature-select-o">
<code class="descname">-o</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--output_pkl</code><code class="descclassname"> &lt;output_pkl&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-feature-select-o" title="Permalink to this definition">¶</a></dt>
<dd><p>Output database for beta and phenotype data.  [default: ./final_preprocessed/methyl_array.pkl]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-feature-select-n">
<code class="descname">-n</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--n_top_cpgs</code><code class="descclassname"> &lt;n_top_cpgs&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-feature-select-n" title="Permalink to this definition">¶</a></dt>
<dd><p>Number cpgs to include with highest variance across population.  [default: 300000]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-feature-select-f">
<code class="descname">-f</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--feature_selection_method</code><code class="descclassname"> &lt;feature_selection_method&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-feature-select-f" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-feature-select-mm">
<code class="descname">-mm</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--metric</code><code class="descclassname"> &lt;metric&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-feature-select-mm" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-feature-select-nn">
<code class="descname">-nn</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--n_neighbors</code><code class="descclassname"> &lt;n_neighbors&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-feature-select-nn" title="Permalink to this definition">¶</a></dt>
<dd><p>Number neighbors for feature selection, default enacts rbf kernel.  [default: 0]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-feature-select-m">
<code class="descname">-m</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--mad_top_cpgs</code><code class="descclassname"> &lt;mad_top_cpgs&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-feature-select-m" title="Permalink to this definition">¶</a></dt>
<dd><p>Number cpgs to apply mad filtering first before more sophisticated feature selection. If 0 or primary feature selection is mad, no mad pre-filtering.  [default: 0]</p>
</dd></dl>

</div>
<div class="section" id="pymethyl-preprocess-get-categorical-distribution">
<h3>get_categorical_distribution<a class="headerlink" href="#pymethyl-preprocess-get-categorical-distribution" title="Permalink to this headline">¶</a></h3>
<p>Get categorical distribution of columns of sample sheet.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-preprocess get_categorical_distribution <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-get-categorical-distribution-is">
<code class="descname">-is</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--formatted_sample_sheet</code><code class="descclassname"> &lt;formatted_sample_sheet&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-get-categorical-distribution-is" title="Permalink to this definition">¶</a></dt>
<dd><p>Clinical information downloaded from tcga/geo/custom, formatted using create_sample_sheet.  [default: ./tcga_idats/minfiSheet.csv]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-get-categorical-distribution-k">
<code class="descname">-k</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--key</code><code class="descclassname"> &lt;key&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-get-categorical-distribution-k" title="Permalink to this definition">¶</a></dt>
<dd><p>Column of csv to print statistics for.  [default: disease]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-get-categorical-distribution-d">
<code class="descname">-d</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--disease_only</code><code class="descclassname"></code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-get-categorical-distribution-d" title="Permalink to this definition">¶</a></dt>
<dd><p>Only look at disease, or text before subtype_delimiter.</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-get-categorical-distribution-sd">
<code class="descname">-sd</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--subtype_delimiter</code><code class="descclassname"> &lt;subtype_delimiter&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-get-categorical-distribution-sd" title="Permalink to this definition">¶</a></dt>
<dd><p>Delimiter for disease extraction.  [default: ,]</p>
</dd></dl>

</div>
<div class="section" id="pymethyl-preprocess-imputation-pipeline">
<h3>imputation_pipeline<a class="headerlink" href="#pymethyl-preprocess-imputation-pipeline" title="Permalink to this headline">¶</a></h3>
<p>Imputation of subtype or no subtype using various imputation methods.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-preprocess imputation_pipeline <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-imputation-pipeline-i">
<code class="descname">-i</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--input_pkl</code><code class="descclassname"> &lt;input_pkl&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-imputation-pipeline-i" title="Permalink to this definition">¶</a></dt>
<dd><p>Input database for beta and phenotype data.  [default: ./combined_outputs/methyl_array.pkl]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-imputation-pipeline-ss">
<code class="descname">-ss</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--split_by_subtype</code><code class="descclassname"></code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-imputation-pipeline-ss" title="Permalink to this definition">¶</a></dt>
<dd><p>Imputes CpGs by subtype before combining again.</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-imputation-pipeline-m">
<code class="descname">-m</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--method</code><code class="descclassname"> &lt;method&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-imputation-pipeline-m" title="Permalink to this definition">¶</a></dt>
<dd><p>Method of imputation.  [default: KNN]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-imputation-pipeline-s">
<code class="descname">-s</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--solver</code><code class="descclassname"> &lt;solver&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-imputation-pipeline-s" title="Permalink to this definition">¶</a></dt>
<dd><p>Imputation library.  [default: fancyimpute]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-imputation-pipeline-k">
<code class="descname">-k</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--n_neighbors</code><code class="descclassname"> &lt;n_neighbors&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-imputation-pipeline-k" title="Permalink to this definition">¶</a></dt>
<dd><p>Number neighbors for imputation if using KNN.  [default: 5]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-imputation-pipeline-r">
<code class="descname">-r</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--orientation</code><code class="descclassname"> &lt;orientation&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-imputation-pipeline-r" title="Permalink to this definition">¶</a></dt>
<dd><p>Impute CpGs or samples.  [default: Samples]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-imputation-pipeline-o">
<code class="descname">-o</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--output_pkl</code><code class="descclassname"> &lt;output_pkl&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-imputation-pipeline-o" title="Permalink to this definition">¶</a></dt>
<dd><p>Output database for beta and phenotype data.  [default: ./imputed_outputs/methyl_array.pkl]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-imputation-pipeline-n">
<code class="descname">-n</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--n_top_cpgs</code><code class="descclassname"> &lt;n_top_cpgs&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-imputation-pipeline-n" title="Permalink to this definition">¶</a></dt>
<dd><p>Number cpgs to include with highest variance across population. Greater than 0 allows for mad filtering during imputation to skip mad step.  [default: 0]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-imputation-pipeline-f">
<code class="descname">-f</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--feature_selection_method</code><code class="descclassname"> &lt;feature_selection_method&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-imputation-pipeline-f" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-imputation-pipeline-mm">
<code class="descname">-mm</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--metric</code><code class="descclassname"> &lt;metric&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-imputation-pipeline-mm" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-imputation-pipeline-nfs">
<code class="descname">-nfs</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--n_neighbors_fs</code><code class="descclassname"> &lt;n_neighbors_fs&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-imputation-pipeline-nfs" title="Permalink to this definition">¶</a></dt>
<dd><p>Number neighbors for feature selection, default enacts rbf kernel.  [default: 0]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-imputation-pipeline-d">
<code class="descname">-d</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--disease_only</code><code class="descclassname"></code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-imputation-pipeline-d" title="Permalink to this definition">¶</a></dt>
<dd><p>Only look at disease, or text before subtype_delimiter.</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-imputation-pipeline-sd">
<code class="descname">-sd</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--subtype_delimiter</code><code class="descclassname"> &lt;subtype_delimiter&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-imputation-pipeline-sd" title="Permalink to this definition">¶</a></dt>
<dd><p>Delimiter for disease extraction.  [default: ,]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-imputation-pipeline-st">
<code class="descname">-st</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--sample_threshold</code><code class="descclassname"> &lt;sample_threshold&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-imputation-pipeline-st" title="Permalink to this definition">¶</a></dt>
<dd><p>Value between 0 and 1 for NaN removal. If samples has sample_threshold proportion of cpgs missing, then remove sample. Set to -1 to not remove samples.  [default: -1.0]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-imputation-pipeline-ct">
<code class="descname">-ct</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--cpg_threshold</code><code class="descclassname"> &lt;cpg_threshold&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-imputation-pipeline-ct" title="Permalink to this definition">¶</a></dt>
<dd><p>Value between 0 and 1 for NaN removal. If cpgs has cpg_threshold proportion of samples missing, then remove cpg. Set to -1 to not remove samples.  [default: -1.0]</p>
</dd></dl>

</div>
<div class="section" id="pymethyl-preprocess-meffil-encode">
<h3>meffil_encode<a class="headerlink" href="#pymethyl-preprocess-meffil-encode" title="Permalink to this headline">¶</a></h3>
<p>Reformat file for meffil input.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-preprocess meffil_encode <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-meffil-encode-is">
<code class="descname">-is</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--input_sample_sheet</code><code class="descclassname"> &lt;input_sample_sheet&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-meffil-encode-is" title="Permalink to this definition">¶</a></dt>
<dd><p>CSV for minfi input.  [default: ./tcga_idats/minfiSheet.csv]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-meffil-encode-os">
<code class="descname">-os</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--output_sample_sheet</code><code class="descclassname"> &lt;output_sample_sheet&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-meffil-encode-os" title="Permalink to this definition">¶</a></dt>
<dd><p>CSV for minfi input.  [default: ./tcga_idats/minfiSheet.csv]</p>
</dd></dl>

</div>
<div class="section" id="pymethyl-preprocess-merge-sample-sheets">
<h3>merge_sample_sheets<a class="headerlink" href="#pymethyl-preprocess-merge-sample-sheets" title="Permalink to this headline">¶</a></h3>
<p>Merge two sample files for more fields for minfi+ input.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-preprocess merge_sample_sheets <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-merge-sample-sheets-s1">
<code class="descname">-s1</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--sample_sheet1</code><code class="descclassname"> &lt;sample_sheet1&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-merge-sample-sheets-s1" title="Permalink to this definition">¶</a></dt>
<dd><p>Clinical information downloaded from tcga/geo/custom, formatted using create_sample_sheet.  [default: ./tcga_idats/clinical_info1.csv]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-merge-sample-sheets-s2">
<code class="descname">-s2</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--sample_sheet2</code><code class="descclassname"> &lt;sample_sheet2&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-merge-sample-sheets-s2" title="Permalink to this definition">¶</a></dt>
<dd><p>Clinical information downloaded from tcga/geo/custom, formatted using create_sample_sheet.  [default: ./tcga_idats/clinical_info2.csv]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-merge-sample-sheets-os">
<code class="descname">-os</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--output_sample_sheet</code><code class="descclassname"> &lt;output_sample_sheet&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-merge-sample-sheets-os" title="Permalink to this definition">¶</a></dt>
<dd><p>CSV for minfi input.  [default: ./tcga_idats/minfiSheet.csv]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-merge-sample-sheets-d">
<code class="descname">-d</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--second_sheet_disease</code><code class="descclassname"></code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-merge-sample-sheets-d" title="Permalink to this definition">¶</a></dt>
<dd><p>Use second sheet’s disease column.</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-merge-sample-sheets-nd">
<code class="descname">-nd</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--no_disease_merge</code><code class="descclassname"></code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-merge-sample-sheets-nd" title="Permalink to this definition">¶</a></dt>
<dd><p>Don’t merge disease columns.</p>
</dd></dl>

</div>
<div class="section" id="pymethyl-preprocess-na-report">
<h3>na_report<a class="headerlink" href="#pymethyl-preprocess-na-report" title="Permalink to this headline">¶</a></h3>
<p>Print proportion of missing values throughout dataset.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-preprocess na_report <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-na-report-i">
<code class="descname">-i</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--input_pkl</code><code class="descclassname"> &lt;input_pkl&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-na-report-i" title="Permalink to this definition">¶</a></dt>
<dd><p>Input database for beta and phenotype data.  [default: ./preprocess_outputs/methyl_array.pkl]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-na-report-o">
<code class="descname">-o</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--output_dir</code><code class="descclassname"> &lt;output_dir&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-na-report-o" title="Permalink to this definition">¶</a></dt>
<dd><p>Output database for na report.  [default: ./na_report/]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-na-report-r">
<code class="descname">-r</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--head_directory</code><code class="descclassname"></code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-na-report-r" title="Permalink to this definition">¶</a></dt>
<dd><p>-i option becomes directory, and searches there for multiple input pickles.</p>
</dd></dl>

</div>
<div class="section" id="pymethyl-preprocess-preprocess-pipeline">
<h3>preprocess_pipeline<a class="headerlink" href="#pymethyl-preprocess-preprocess-pipeline" title="Permalink to this headline">¶</a></h3>
<p>Perform preprocessing of idats using enmix or meffil.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-preprocess preprocess_pipeline <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-preprocess-pipeline-i">
<code class="descname">-i</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--idat_dir</code><code class="descclassname"> &lt;idat_dir&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-preprocess-pipeline-i" title="Permalink to this definition">¶</a></dt>
<dd><p>Idat dir for one sample sheet, alternatively can be your phenotype sample sheet.  [default: ./tcga_idats/]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-preprocess-pipeline-n">
<code class="descname">-n</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--n_cores</code><code class="descclassname"> &lt;n_cores&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-preprocess-pipeline-n" title="Permalink to this definition">¶</a></dt>
<dd><p>Number cores to use for preprocessing.  [default: 6]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-preprocess-pipeline-o">
<code class="descname">-o</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--output_pkl</code><code class="descclassname"> &lt;output_pkl&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-preprocess-pipeline-o" title="Permalink to this definition">¶</a></dt>
<dd><p>Output database for beta and phenotype data.  [default: ./preprocess_outputs/methyl_array.pkl]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-preprocess-pipeline-m">
<code class="descname">-m</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--meffil</code><code class="descclassname"></code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-preprocess-pipeline-m" title="Permalink to this definition">¶</a></dt>
<dd><p>Preprocess using meffil.</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-preprocess-pipeline-pc">
<code class="descname">-pc</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--n_pcs</code><code class="descclassname"> &lt;n_pcs&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-preprocess-pipeline-pc" title="Permalink to this definition">¶</a></dt>
<dd><p>For meffil, number of principal components for functional normalization. If set to -1, then PCs are selected using elbow method.  [default: -1]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-preprocess-pipeline-p">
<code class="descname">-p</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--pipeline</code><code class="descclassname"> &lt;pipeline&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-preprocess-pipeline-p" title="Permalink to this definition">¶</a></dt>
<dd><p>If not meffil, preprocess using minfi or enmix.  [default: enmix]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-preprocess-pipeline-noob">
<code class="descname">-noob</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--noob_norm</code><code class="descclassname"></code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-preprocess-pipeline-noob" title="Permalink to this definition">¶</a></dt>
<dd><p>Run noob normalization of minfi selected.</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-preprocess-pipeline-u">
<code class="descname">-u</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--use_cache</code><code class="descclassname"></code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-preprocess-pipeline-u" title="Permalink to this definition">¶</a></dt>
<dd><p>If this is selected, loads qc results rather than running qc again and update with new qc parameters. Only works for meffil selection. Minfi and enmix just loads RG Set.</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-preprocess-pipeline-qc">
<code class="descname">-qc</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--qc_only</code><code class="descclassname"></code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-preprocess-pipeline-qc" title="Permalink to this definition">¶</a></dt>
<dd><p>Only perform QC for meffil pipeline, caches results into rds file for loading again, only works if use_cache is false. Minfi and enmix just saves the RGSet before preprocessing.</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-preprocess-pipeline-bns">
<code class="descname">-bns</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--p_beadnum_samples</code><code class="descclassname"> &lt;p_beadnum_samples&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-preprocess-pipeline-bns" title="Permalink to this definition">¶</a></dt>
<dd><p>From meffil documentation, “fraction of probes that failed the threshold of 3 beads”.  [default: 0.05]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-preprocess-pipeline-pds">
<code class="descname">-pds</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--p_detection_samples</code><code class="descclassname"> &lt;p_detection_samples&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-preprocess-pipeline-pds" title="Permalink to this definition">¶</a></dt>
<dd><p>From meffil documentation, “fraction of probes that failed a detection.pvalue threshold of 0.01”.  [default: 0.05]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-preprocess-pipeline-bnc">
<code class="descname">-bnc</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--p_beadnum_cpgs</code><code class="descclassname"> &lt;p_beadnum_cpgs&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-preprocess-pipeline-bnc" title="Permalink to this definition">¶</a></dt>
<dd><p>From meffil documentation, “fraction of samples that failed the threshold of 3 beads”.  [default: 0.05]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-preprocess-pipeline-pdc">
<code class="descname">-pdc</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--p_detection_cpgs</code><code class="descclassname"> &lt;p_detection_cpgs&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-preprocess-pipeline-pdc" title="Permalink to this definition">¶</a></dt>
<dd><p>From meffil documentation, “fraction of samples that failed a detection.pvalue threshold of 0.01”.  [default: 0.05]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-preprocess-pipeline-sc">
<code class="descname">-sc</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--sex_cutoff</code><code class="descclassname"> &lt;sex_cutoff&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-preprocess-pipeline-sc" title="Permalink to this definition">¶</a></dt>
<dd><p>From meffil documentation, “difference of total median intensity for Y chromosome probes and X chromosome probes”.  [default: -2]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-preprocess-pipeline-sd">
<code class="descname">-sd</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--sex_sd</code><code class="descclassname"> &lt;sex_sd&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-preprocess-pipeline-sd" title="Permalink to this definition">¶</a></dt>
<dd><p>From meffil documentation, “sex detection outliers if outside this range”.  [default: 5]</p>
</dd></dl>

</div>
<div class="section" id="pymethyl-preprocess-remove-diseases">
<h3>remove_diseases<a class="headerlink" href="#pymethyl-preprocess-remove-diseases" title="Permalink to this headline">¶</a></h3>
<p>Exclude diseases from study by count number or exclusion list.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-preprocess remove_diseases <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-remove-diseases-is">
<code class="descname">-is</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--formatted_sample_sheet</code><code class="descclassname"> &lt;formatted_sample_sheet&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-remove-diseases-is" title="Permalink to this definition">¶</a></dt>
<dd><p>Clinical information downloaded from tcga/geo/custom, formatted using create_sample_sheet.  [default: ./tcga_idats/clinical_info.csv]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-remove-diseases-e">
<code class="descname">-e</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--exclude_disease_list</code><code class="descclassname"> &lt;exclude_disease_list&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-remove-diseases-e" title="Permalink to this definition">¶</a></dt>
<dd><p>List of conditions to exclude, from disease column, comma delimited.  [default: ]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-remove-diseases-os">
<code class="descname">-os</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--output_sheet_name</code><code class="descclassname"> &lt;output_sheet_name&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-remove-diseases-os" title="Permalink to this definition">¶</a></dt>
<dd><p>CSV for minfi input.  [default: ./tcga_idats/minfiSheet.csv]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-remove-diseases-l">
<code class="descname">-l</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--low_count</code><code class="descclassname"> &lt;low_count&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-remove-diseases-l" title="Permalink to this definition">¶</a></dt>
<dd><p>Remove diseases if they are below a certain count, default this is not used.  [default: 0]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-remove-diseases-d">
<code class="descname">-d</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--disease_only</code><code class="descclassname"></code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-remove-diseases-d" title="Permalink to this definition">¶</a></dt>
<dd><p>Only look at disease, or text before subtype_delimiter.</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-remove-diseases-sd">
<code class="descname">-sd</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--subtype_delimiter</code><code class="descclassname"> &lt;subtype_delimiter&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-remove-diseases-sd" title="Permalink to this definition">¶</a></dt>
<dd><p>Delimiter for disease extraction.  [default: ,]</p>
</dd></dl>

</div>
<div class="section" id="pymethyl-preprocess-split-preprocess-input-by-subtype">
<h3>split_preprocess_input_by_subtype<a class="headerlink" href="#pymethyl-preprocess-split-preprocess-input-by-subtype" title="Permalink to this headline">¶</a></h3>
<p>Split preprocess input samplesheet by disease subtype.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-preprocess split_preprocess_input_by_subtype <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-split-preprocess-input-by-subtype-i">
<code class="descname">-i</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--idat_csv</code><code class="descclassname"> &lt;idat_csv&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-split-preprocess-input-by-subtype-i" title="Permalink to this definition">¶</a></dt>
<dd><p>Idat csv for one sample sheet, alternatively can be your phenotype sample sheet.  [default: ./tcga_idats/minfiSheet.csv]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-split-preprocess-input-by-subtype-d">
<code class="descname">-d</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--disease_only</code><code class="descclassname"></code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-split-preprocess-input-by-subtype-d" title="Permalink to this definition">¶</a></dt>
<dd><p>Only look at disease, or text before subtype_delimiter.</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-split-preprocess-input-by-subtype-sd">
<code class="descname">-sd</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--subtype_delimiter</code><code class="descclassname"> &lt;subtype_delimiter&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-split-preprocess-input-by-subtype-sd" title="Permalink to this definition">¶</a></dt>
<dd><p>Delimiter for disease extraction.  [default: ,]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-preprocess-split-preprocess-input-by-subtype-o">
<code class="descname">-o</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--subtype_output_dir</code><code class="descclassname"> &lt;subtype_output_dir&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-preprocess-split-preprocess-input-by-subtype-o" title="Permalink to this definition">¶</a></dt>
<dd><p>Output subtypes pheno csv.  [default: ./preprocess_outputs/]</p>
</dd></dl>

</div>
</div>
<div class="section" id="pymethyl-utils">
<h2>pymethyl-utils<a class="headerlink" href="#pymethyl-utils" title="Permalink to this headline">¶</a></h2>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-utils <span class="o">[</span>OPTIONS<span class="o">]</span> COMMAND <span class="o">[</span>ARGS<span class="o">]</span>...
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-utils-version">
<code class="descname">--version</code><code class="descclassname"></code><a class="headerlink" href="#cmdoption-pymethyl-utils-version" title="Permalink to this definition">¶</a></dt>
<dd><p>Show the version and exit.</p>
</dd></dl>

<div class="section" id="pymethyl-utils-backup-pkl">
<h3>backup_pkl<a class="headerlink" href="#pymethyl-utils-backup-pkl" title="Permalink to this headline">¶</a></h3>
<p>Copy methylarray pickle to new location to backup.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-utils backup_pkl <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-utils-backup-pkl-i">
<code class="descname">-i</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--input_pkl</code><code class="descclassname"> &lt;input_pkl&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-backup-pkl-i" title="Permalink to this definition">¶</a></dt>
<dd><p>Input database for beta and phenotype data.  [default: ./final_preprocessed/methyl_array.pkl]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-backup-pkl-o">
<code class="descname">-o</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--output_pkl</code><code class="descclassname"> &lt;output_pkl&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-backup-pkl-o" title="Permalink to this definition">¶</a></dt>
<dd><p>Output database for beta and phenotype data.  [default: ./backup/methyl_array.pkl]</p>
</dd></dl>

</div>
<div class="section" id="pymethyl-utils-bin-column">
<h3>bin_column<a class="headerlink" href="#pymethyl-utils-bin-column" title="Permalink to this headline">¶</a></h3>
<p>Convert continuous phenotype column into categorical by binning.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-utils bin_column <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-utils-bin-column-t">
<code class="descname">-t</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--test_pkl</code><code class="descclassname"> &lt;test_pkl&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-bin-column-t" title="Permalink to this definition">¶</a></dt>
<dd><p>Pickle containing testing set.  [default: ./train_val_test_sets/test_methyl_array.pkl]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-bin-column-c">
<code class="descname">-c</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--col</code><code class="descclassname"> &lt;col&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-bin-column-c" title="Permalink to this definition">¶</a></dt>
<dd><p>Column to turn into bins.  [default: age]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-bin-column-n">
<code class="descname">-n</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--n_bins</code><code class="descclassname"> &lt;n_bins&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-bin-column-n" title="Permalink to this definition">¶</a></dt>
<dd><p>Number of bins.  [default: 10]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-bin-column-ot">
<code class="descname">-ot</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--output_test_pkl</code><code class="descclassname"> &lt;output_test_pkl&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-bin-column-ot" title="Permalink to this definition">¶</a></dt>
<dd><p>Binned shap pickle for further testing.  [default: ./train_val_test_sets/test_methyl_array_shap_binned.pkl]</p>
</dd></dl>

</div>
<div class="section" id="pymethyl-utils-concat-csv">
<h3>concat_csv<a class="headerlink" href="#pymethyl-utils-concat-csv" title="Permalink to this headline">¶</a></h3>
<p>Concatenate two csv files together.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-utils concat_csv <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-utils-concat-csv-i1">
<code class="descname">-i1</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--input_csv</code><code class="descclassname"> &lt;input_csv&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-concat-csv-i1" title="Permalink to this definition">¶</a></dt>
<dd><p>Beta csv.  [default: ./beta1.csv]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-concat-csv-i2">
<code class="descname">-i2</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--input_csv2</code><code class="descclassname"> &lt;input_csv2&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-concat-csv-i2" title="Permalink to this definition">¶</a></dt>
<dd><p>Beta/other csv 2.  [default: ./cell_estimates.csv]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-concat-csv-o">
<code class="descname">-o</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--output_csv</code><code class="descclassname"> &lt;output_csv&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-concat-csv-o" title="Permalink to this definition">¶</a></dt>
<dd><p>Output csv.  [default: ./beta.concat.csv]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-concat-csv-a">
<code class="descname">-a</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--axis</code><code class="descclassname"> &lt;axis&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-concat-csv-a" title="Permalink to this definition">¶</a></dt>
<dd><p>Axis to merge on. Columns are 0, rows are 1.  [default: 1]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-concat-csv-i">
<code class="descname">-i</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--index_col</code><code class="descclassname"> &lt;index_col&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-concat-csv-i" title="Permalink to this definition">¶</a></dt>
<dd><p>Index Column.  [default: 0]</p>
</dd></dl>

</div>
<div class="section" id="pymethyl-utils-counts">
<h3>counts<a class="headerlink" href="#pymethyl-utils-counts" title="Permalink to this headline">¶</a></h3>
<p>Return categorical breakdown of phenotype column.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-utils counts <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-utils-counts-i">
<code class="descname">-i</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--input_pkl</code><code class="descclassname"> &lt;input_pkl&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-counts-i" title="Permalink to this definition">¶</a></dt>
<dd><p>Input database for beta and phenotype data.  [default: ./final_preprocessed/methyl_array.pkl]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-counts-k">
<code class="descname">-k</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--key</code><code class="descclassname"> &lt;key&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-counts-k" title="Permalink to this definition">¶</a></dt>
<dd><p>Key to split on.  [default: disease]</p>
</dd></dl>

</div>
<div class="section" id="pymethyl-utils-create-external-validation-set">
<h3>create_external_validation_set<a class="headerlink" href="#pymethyl-utils-create-external-validation-set" title="Permalink to this headline">¶</a></h3>
<p>Create external validation set containing same CpGs as training set.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-utils create_external_validation_set <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-utils-create-external-validation-set-t">
<code class="descname">-t</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--train_pkl</code><code class="descclassname"> &lt;train_pkl&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-create-external-validation-set-t" title="Permalink to this definition">¶</a></dt>
<dd><p>Input methyl array.  [default: ./train_val_test_sets/train_methyl_array.pkl]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-create-external-validation-set-q">
<code class="descname">-q</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--query_pkl</code><code class="descclassname"> &lt;query_pkl&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-create-external-validation-set-q" title="Permalink to this definition">¶</a></dt>
<dd><p>Input methylation array to add/subtract cpgs to.  [default: ./final_preprocessed/methyl_array.pkl]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-create-external-validation-set-o">
<code class="descname">-o</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--output_pkl</code><code class="descclassname"> &lt;output_pkl&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-create-external-validation-set-o" title="Permalink to this definition">¶</a></dt>
<dd><p>Output methyl array external validation.  [default: ./external_validation/methyl_array.pkl]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-create-external-validation-set-c">
<code class="descname">-c</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--cpg_replace_method</code><code class="descclassname"> &lt;cpg_replace_method&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-create-external-validation-set-c" title="Permalink to this definition">¶</a></dt>
<dd><p>What to do for missing CpGs.  [default: mid]</p>
</dd></dl>

</div>
<div class="section" id="pymethyl-utils-est-age">
<h3>est_age<a class="headerlink" href="#pymethyl-utils-est-age" title="Permalink to this headline">¶</a></h3>
<p>Estimate age using cgAgeR</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-utils est_age <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-utils-est-age-i">
<code class="descname">-i</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--input_pkl</code><code class="descclassname"> &lt;input_pkl&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-est-age-i" title="Permalink to this definition">¶</a></dt>
<dd><p>Input methyl array.  [default: ./train_val_test_sets/test_methyl_array.pkl]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-est-age-ac">
<code class="descname">-ac</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--age_column</code><code class="descclassname"> &lt;age_column&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-est-age-ac" title="Permalink to this definition">¶</a></dt>
<dd><p>Age column of Methylation Array. Leave blank for no age  [default: ]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-est-age-a">
<code class="descname">-a</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--analyses</code><code class="descclassname"> &lt;analyses&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-est-age-a" title="Permalink to this definition">¶</a></dt>
<dd><p>Analyses to run</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-est-age-o">
<code class="descname">-o</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--output_csv</code><code class="descclassname"> &lt;output_csv&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-est-age-o" title="Permalink to this definition">¶</a></dt>
<dd><p>Output csv  [default: age_estimation/output_age_estimations.csv]</p>
</dd></dl>

</div>
<div class="section" id="pymethyl-utils-feature-select-train-val-test">
<h3>feature_select_train_val_test<a class="headerlink" href="#pymethyl-utils-feature-select-train-val-test" title="Permalink to this headline">¶</a></h3>
<p>Filter CpGs by taking x top CpGs with highest mean absolute deviation scores or via spectral feature selection.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-utils feature_select_train_val_test <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-utils-feature-select-train-val-test-i">
<code class="descname">-i</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--input_pkl_dir</code><code class="descclassname"> &lt;input_pkl_dir&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-feature-select-train-val-test-i" title="Permalink to this definition">¶</a></dt>
<dd><p>Input database for beta and phenotype data.  [default: ./train_val_test_sets/]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-feature-select-train-val-test-o">
<code class="descname">-o</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--output_dir</code><code class="descclassname"> &lt;output_dir&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-feature-select-train-val-test-o" title="Permalink to this definition">¶</a></dt>
<dd><p>Output database for beta and phenotype data.  [default: ./train_val_test_sets_fs/]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-feature-select-train-val-test-n">
<code class="descname">-n</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--n_top_cpgs</code><code class="descclassname"> &lt;n_top_cpgs&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-feature-select-train-val-test-n" title="Permalink to this definition">¶</a></dt>
<dd><p>Number cpgs to include with highest variance across population.  [default: 300000]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-feature-select-train-val-test-f">
<code class="descname">-f</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--feature_selection_method</code><code class="descclassname"> &lt;feature_selection_method&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-feature-select-train-val-test-f" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-feature-select-train-val-test-mm">
<code class="descname">-mm</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--metric</code><code class="descclassname"> &lt;metric&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-feature-select-train-val-test-mm" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-feature-select-train-val-test-nn">
<code class="descname">-nn</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--n_neighbors</code><code class="descclassname"> &lt;n_neighbors&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-feature-select-train-val-test-nn" title="Permalink to this definition">¶</a></dt>
<dd><p>Number neighbors for feature selection, default enacts rbf kernel.  [default: 0]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-feature-select-train-val-test-m">
<code class="descname">-m</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--mad_top_cpgs</code><code class="descclassname"> &lt;mad_top_cpgs&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-feature-select-train-val-test-m" title="Permalink to this definition">¶</a></dt>
<dd><p>Number cpgs to apply mad filtering first before more sophisticated feature selection. If 0 or primary feature selection is mad, no mad pre-filtering.  [default: 0]</p>
</dd></dl>

</div>
<div class="section" id="pymethyl-utils-fix-key">
<h3>fix_key<a class="headerlink" href="#pymethyl-utils-fix-key" title="Permalink to this headline">¶</a></h3>
<p>Format certain column of phenotype array in MethylationArray.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-utils fix_key <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-utils-fix-key-i">
<code class="descname">-i</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--input_pkl</code><code class="descclassname"> &lt;input_pkl&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-fix-key-i" title="Permalink to this definition">¶</a></dt>
<dd><p>Input database for beta and phenotype data.  [default: ./final_preprocessed/methyl_array.pkl]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-fix-key-k">
<code class="descname">-k</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--key</code><code class="descclassname"> &lt;key&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-fix-key-k" title="Permalink to this definition">¶</a></dt>
<dd><p>Key to split on.  [default: disease]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-fix-key-d">
<code class="descname">-d</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--disease_only</code><code class="descclassname"></code><a class="headerlink" href="#cmdoption-pymethyl-utils-fix-key-d" title="Permalink to this definition">¶</a></dt>
<dd><p>Only look at disease, or text before subtype_delimiter.</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-fix-key-sd">
<code class="descname">-sd</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--subtype_delimiter</code><code class="descclassname"> &lt;subtype_delimiter&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-fix-key-sd" title="Permalink to this definition">¶</a></dt>
<dd><p>Delimiter for disease extraction.  [default: ,]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-fix-key-o">
<code class="descname">-o</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--output_pkl</code><code class="descclassname"> &lt;output_pkl&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-fix-key-o" title="Permalink to this definition">¶</a></dt>
<dd><p>Input database for beta and phenotype data.  [default: ./fixed_preprocessed/methyl_array.pkl]</p>
</dd></dl>

</div>
<div class="section" id="pymethyl-utils-modify-pheno-data">
<h3>modify_pheno_data<a class="headerlink" href="#pymethyl-utils-modify-pheno-data" title="Permalink to this headline">¶</a></h3>
<p>Use another spreadsheet to add more descriptive data to methylarray.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-utils modify_pheno_data <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-utils-modify-pheno-data-i">
<code class="descname">-i</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--input_pkl</code><code class="descclassname"> &lt;input_pkl&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-modify-pheno-data-i" title="Permalink to this definition">¶</a></dt>
<dd><p>Input database for beta and phenotype data.  [default: ./final_preprocessed/methyl_array.pkl]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-modify-pheno-data-is">
<code class="descname">-is</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--input_formatted_sample_sheet</code><code class="descclassname"> &lt;input_formatted_sample_sheet&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-modify-pheno-data-is" title="Permalink to this definition">¶</a></dt>
<dd><p>Information passed through function create_sample_sheet, has Basename and disease fields.  [default: ./tcga_idats/minfi_sheet.csv]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-modify-pheno-data-o">
<code class="descname">-o</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--output_pkl</code><code class="descclassname"> &lt;output_pkl&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-modify-pheno-data-o" title="Permalink to this definition">¶</a></dt>
<dd><p>Output database for beta and phenotype data.  [default: ./modified_processed/methyl_array.pkl]</p>
</dd></dl>

</div>
<div class="section" id="pymethyl-utils-move-jpg">
<h3>move_jpg<a class="headerlink" href="#pymethyl-utils-move-jpg" title="Permalink to this headline">¶</a></h3>
<p>Move preprocessing jpegs to preprocessing output directory.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-utils move_jpg <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-utils-move-jpg-i">
<code class="descname">-i</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--input_dir</code><code class="descclassname"> &lt;input_dir&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-move-jpg-i" title="Permalink to this definition">¶</a></dt>
<dd><p>Directory containing jpg.  [default: ./]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-move-jpg-o">
<code class="descname">-o</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--output_dir</code><code class="descclassname"> &lt;output_dir&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-move-jpg-o" title="Permalink to this definition">¶</a></dt>
<dd><p>Output directory for images.  [default: ./preprocess_output_images/]</p>
</dd></dl>

</div>
<div class="section" id="pymethyl-utils-overwrite-pheno-data">
<h3>overwrite_pheno_data<a class="headerlink" href="#pymethyl-utils-overwrite-pheno-data" title="Permalink to this headline">¶</a></h3>
<p>Use another spreadsheet to add more descriptive data to methylarray.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-utils overwrite_pheno_data <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-utils-overwrite-pheno-data-i">
<code class="descname">-i</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--input_pkl</code><code class="descclassname"> &lt;input_pkl&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-overwrite-pheno-data-i" title="Permalink to this definition">¶</a></dt>
<dd><p>Input database for beta and phenotype data.  [default: ./final_preprocessed/methyl_array.pkl]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-overwrite-pheno-data-is">
<code class="descname">-is</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--input_formatted_sample_sheet</code><code class="descclassname"> &lt;input_formatted_sample_sheet&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-overwrite-pheno-data-is" title="Permalink to this definition">¶</a></dt>
<dd><p>Information passed through function create_sample_sheet, has Basename and disease fields.  [default: ./tcga_idats/minfi_sheet.csv]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-overwrite-pheno-data-o">
<code class="descname">-o</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--output_pkl</code><code class="descclassname"> &lt;output_pkl&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-overwrite-pheno-data-o" title="Permalink to this definition">¶</a></dt>
<dd><p>Output database for beta and phenotype data.  [default: ./modified_processed/methyl_array.pkl]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-overwrite-pheno-data-c">
<code class="descname">-c</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--index_col</code><code class="descclassname"> &lt;index_col&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-overwrite-pheno-data-c" title="Permalink to this definition">¶</a></dt>
<dd><p>Index col when reading csv.  [default: 0]</p>
</dd></dl>

</div>
<div class="section" id="pymethyl-utils-pkl-to-csv">
<h3>pkl_to_csv<a class="headerlink" href="#pymethyl-utils-pkl-to-csv" title="Permalink to this headline">¶</a></h3>
<p>Output methylarray pickle to csv.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-utils pkl_to_csv <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-utils-pkl-to-csv-i">
<code class="descname">-i</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--input_pkl</code><code class="descclassname"> &lt;input_pkl&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-pkl-to-csv-i" title="Permalink to this definition">¶</a></dt>
<dd><p>Input database for beta and phenotype data.  [default: ./final_preprocessed/methyl_array.pkl]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-pkl-to-csv-o">
<code class="descname">-o</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--output_dir</code><code class="descclassname"> &lt;output_dir&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-pkl-to-csv-o" title="Permalink to this definition">¶</a></dt>
<dd><p>Input database for beta and phenotype data.  [default: ./final_preprocessed/]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-pkl-to-csv-c">
<code class="descname">-c</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--col</code><code class="descclassname"> &lt;col&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-pkl-to-csv-c" title="Permalink to this definition">¶</a></dt>
<dd><p>Column to color.  [default: ]</p>
</dd></dl>

</div>
<div class="section" id="pymethyl-utils-print-number-sex-cpgs">
<h3>print_number_sex_cpgs<a class="headerlink" href="#pymethyl-utils-print-number-sex-cpgs" title="Permalink to this headline">¶</a></h3>
<p>Print number of non-autosomal CpGs.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-utils print_number_sex_cpgs <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-utils-print-number-sex-cpgs-i">
<code class="descname">-i</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--input_pkl</code><code class="descclassname"> &lt;input_pkl&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-print-number-sex-cpgs-i" title="Permalink to this definition">¶</a></dt>
<dd><p>Input database for beta and phenotype data.  [default: ./final_preprocessed/methyl_array.pkl]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-print-number-sex-cpgs-a">
<code class="descname">-a</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--array_type</code><code class="descclassname"> &lt;array_type&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-print-number-sex-cpgs-a" title="Permalink to this definition">¶</a></dt>
<dd><p>Array Type.  [default: 450k]</p>
</dd></dl>

</div>
<div class="section" id="pymethyl-utils-print-shape">
<h3>print_shape<a class="headerlink" href="#pymethyl-utils-print-shape" title="Permalink to this headline">¶</a></h3>
<p>Print dimensions of beta matrix.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-utils print_shape <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-utils-print-shape-i">
<code class="descname">-i</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--input_pkl</code><code class="descclassname"> &lt;input_pkl&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-print-shape-i" title="Permalink to this definition">¶</a></dt>
<dd><p>Input database for beta and phenotype data.  [default: ./final_preprocessed/methyl_array.pkl]</p>
</dd></dl>

</div>
<div class="section" id="pymethyl-utils-rate-regression">
<h3>rate_regression<a class="headerlink" href="#pymethyl-utils-rate-regression" title="Permalink to this headline">¶</a></h3>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-utils rate_regression <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-utils-rate-regression-i">
<code class="descname">-i</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--input_csv</code><code class="descclassname"> &lt;input_csv&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-rate-regression-i" title="Permalink to this definition">¶</a></dt>
<dd><p>Results csv.  [default: ./results.csv]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-rate-regression-c1">
<code class="descname">-c1</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--pred_col</code><code class="descclassname"> &lt;pred_col&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-rate-regression-c1" title="Permalink to this definition">¶</a></dt>
<dd><p>Prediction column.  [default: y_pred]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-rate-regression-c2">
<code class="descname">-c2</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--true_col</code><code class="descclassname"> &lt;true_col&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-rate-regression-c2" title="Permalink to this definition">¶</a></dt>
<dd><p>True Column.  [default: y_true]</p>
</dd></dl>

</div>
<div class="section" id="pymethyl-utils-ref-estimate-cell-counts">
<h3>ref_estimate_cell_counts<a class="headerlink" href="#pymethyl-utils-ref-estimate-cell-counts" title="Permalink to this headline">¶</a></h3>
<p>Reference based cell type estimates.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-utils ref_estimate_cell_counts <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-utils-ref-estimate-cell-counts-ro">
<code class="descname">-ro</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--input_r_object_dir</code><code class="descclassname"> &lt;input_r_object_dir&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-ref-estimate-cell-counts-ro" title="Permalink to this definition">¶</a></dt>
<dd><p>Input directory containing qc data.  [default: ./preprocess_outputs/]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-ref-estimate-cell-counts-a">
<code class="descname">-a</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--algorithm</code><code class="descclassname"> &lt;algorithm&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-ref-estimate-cell-counts-a" title="Permalink to this definition">¶</a></dt>
<dd><p>Algorithm to run cell type.  [default: meffil]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-ref-estimate-cell-counts-ref">
<code class="descname">-ref</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--reference</code><code class="descclassname"> &lt;reference&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-ref-estimate-cell-counts-ref" title="Permalink to this definition">¶</a></dt>
<dd><p>Cell Type Reference.  [default: cord blood gse68456]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-ref-estimate-cell-counts-l">
<code class="descname">-l</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--library</code><code class="descclassname"> &lt;library&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-ref-estimate-cell-counts-l" title="Permalink to this definition">¶</a></dt>
<dd><p>IDOL Library.  [default: IDOLOptimizedCpGs450klegacy]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-ref-estimate-cell-counts-o">
<code class="descname">-o</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--output_csv</code><code class="descclassname"> &lt;output_csv&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-ref-estimate-cell-counts-o" title="Permalink to this definition">¶</a></dt>
<dd><p>Output cell type estimates.  [default: ./added_cell_counts/cell_type_estimates.csv]</p>
</dd></dl>

</div>
<div class="section" id="pymethyl-utils-ref-free-cell-deconv">
<h3>ref_free_cell_deconv<a class="headerlink" href="#pymethyl-utils-ref-free-cell-deconv" title="Permalink to this headline">¶</a></h3>
<p>Reference free cell type deconvolution</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-utils ref_free_cell_deconv <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-utils-ref-free-cell-deconv-tr">
<code class="descname">-tr</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--train_pkl</code><code class="descclassname"> &lt;train_pkl&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-ref-free-cell-deconv-tr" title="Permalink to this definition">¶</a></dt>
<dd><p>Input methyl array.  [default: ./train_val_test_sets/train_methyl_array.pkl]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-ref-free-cell-deconv-te">
<code class="descname">-te</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--test_pkl</code><code class="descclassname"> &lt;test_pkl&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-ref-free-cell-deconv-te" title="Permalink to this definition">¶</a></dt>
<dd><p>Input methyl array.  [default: ./train_val_test_sets/test_methyl_array.pkl]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-ref-free-cell-deconv-c">
<code class="descname">-c</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--cell_type_columns</code><code class="descclassname"> &lt;cell_type_columns&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-ref-free-cell-deconv-c" title="Permalink to this definition">¶</a></dt>
<dd><p>Cell type columns. Leave blank for auto selection, not incorporate reference information.  [default: ]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-ref-free-cell-deconv-k">
<code class="descname">-k</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--n_cell_types</code><code class="descclassname"> &lt;n_cell_types&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-ref-free-cell-deconv-k" title="Permalink to this definition">¶</a></dt>
<dd><p>Number of cell types.  [default: 7]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-ref-free-cell-deconv-a">
<code class="descname">-a</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--analysis</code><code class="descclassname"> &lt;analysis&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-ref-free-cell-deconv-a" title="Permalink to this definition">¶</a></dt>
<dd><p>Analyses to run  [default: reffreecellmix]</p>
</dd></dl>

</div>
<div class="section" id="pymethyl-utils-remove-sex">
<h3>remove_sex<a class="headerlink" href="#pymethyl-utils-remove-sex" title="Permalink to this headline">¶</a></h3>
<p>Remove non-autosomal CpGs.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-utils remove_sex <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-utils-remove-sex-i">
<code class="descname">-i</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--input_pkl</code><code class="descclassname"> &lt;input_pkl&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-remove-sex-i" title="Permalink to this definition">¶</a></dt>
<dd><p>Input database for beta and phenotype data.  [default: ./preprocess_outputs/methyl_array.pkl]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-remove-sex-o">
<code class="descname">-o</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--output_pkl</code><code class="descclassname"> &lt;output_pkl&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-remove-sex-o" title="Permalink to this definition">¶</a></dt>
<dd><p>Output methyl array autosomal.  [default: ./autosomal/methyl_array.pkl]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-remove-sex-a">
<code class="descname">-a</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--array_type</code><code class="descclassname"> &lt;array_type&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-remove-sex-a" title="Permalink to this definition">¶</a></dt>
<dd><p>Array Type.  [default: 450k]</p>
</dd></dl>

</div>
<div class="section" id="pymethyl-utils-remove-snps">
<h3>remove_snps<a class="headerlink" href="#pymethyl-utils-remove-snps" title="Permalink to this headline">¶</a></h3>
<p>Remove SNPs from methylation array.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-utils remove_snps <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-utils-remove-snps-i">
<code class="descname">-i</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--input_pkl</code><code class="descclassname"> &lt;input_pkl&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-remove-snps-i" title="Permalink to this definition">¶</a></dt>
<dd><p>Input database for beta and phenotype data.  [default: ./autosomal/methyl_array.pkl]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-remove-snps-o">
<code class="descname">-o</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--output_pkl</code><code class="descclassname"> &lt;output_pkl&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-remove-snps-o" title="Permalink to this definition">¶</a></dt>
<dd><p>Output methyl array autosomal.  [default: ./no_snp/methyl_array.pkl]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-remove-snps-a">
<code class="descname">-a</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--array_type</code><code class="descclassname"> &lt;array_type&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-remove-snps-a" title="Permalink to this definition">¶</a></dt>
<dd><p>Array Type.  [default: 450k]</p>
</dd></dl>

</div>
<div class="section" id="pymethyl-utils-set-part-array-background">
<h3>set_part_array_background<a class="headerlink" href="#pymethyl-utils-set-part-array-background" title="Permalink to this headline">¶</a></h3>
<p>Set subset of CpGs from beta matrix to background values.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-utils set_part_array_background <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-utils-set-part-array-background-i">
<code class="descname">-i</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--input_pkl</code><code class="descclassname"> &lt;input_pkl&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-set-part-array-background-i" title="Permalink to this definition">¶</a></dt>
<dd><p>Input methyl array.  [default: ./final_preprocessed/methyl_array.pkl]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-set-part-array-background-c">
<code class="descname">-c</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--cpg_pkl</code><code class="descclassname"> &lt;cpg_pkl&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-set-part-array-background-c" title="Permalink to this definition">¶</a></dt>
<dd><p>Pickled numpy array for subsetting.  [default: ./subset_cpgs.pkl]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-set-part-array-background-o">
<code class="descname">-o</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--output_pkl</code><code class="descclassname"> &lt;output_pkl&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-set-part-array-background-o" title="Permalink to this definition">¶</a></dt>
<dd><p>Output methyl array external validation.  [default: ./removal/methyl_array.pkl]</p>
</dd></dl>

</div>
<div class="section" id="pymethyl-utils-stratify">
<h3>stratify<a class="headerlink" href="#pymethyl-utils-stratify" title="Permalink to this headline">¶</a></h3>
<p>Split methylation array by key and store.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-utils stratify <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-utils-stratify-i">
<code class="descname">-i</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--input_pkl</code><code class="descclassname"> &lt;input_pkl&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-stratify-i" title="Permalink to this definition">¶</a></dt>
<dd><p>Input database for beta and phenotype data.  [default: ./final_preprocessed/methyl_array.pkl]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-stratify-k">
<code class="descname">-k</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--key</code><code class="descclassname"> &lt;key&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-stratify-k" title="Permalink to this definition">¶</a></dt>
<dd><p>Key to split on.  [default: disease]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-stratify-o">
<code class="descname">-o</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--output_dir</code><code class="descclassname"> &lt;output_dir&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-stratify-o" title="Permalink to this definition">¶</a></dt>
<dd><p>Output directory for stratified.  [default: ./stratified/]</p>
</dd></dl>

</div>
<div class="section" id="pymethyl-utils-subset-array">
<h3>subset_array<a class="headerlink" href="#pymethyl-utils-subset-array" title="Permalink to this headline">¶</a></h3>
<p>Only retain certain number of CpGs from methylation array.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-utils subset_array <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-utils-subset-array-i">
<code class="descname">-i</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--input_pkl</code><code class="descclassname"> &lt;input_pkl&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-subset-array-i" title="Permalink to this definition">¶</a></dt>
<dd><p>Input methyl array.  [default: ./final_preprocessed/methyl_array.pkl]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-subset-array-c">
<code class="descname">-c</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--cpg_pkl</code><code class="descclassname"> &lt;cpg_pkl&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-subset-array-c" title="Permalink to this definition">¶</a></dt>
<dd><p>Pickled numpy array for subsetting.  [default: ./subset_cpgs.pkl]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-subset-array-o">
<code class="descname">-o</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--output_pkl</code><code class="descclassname"> &lt;output_pkl&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-subset-array-o" title="Permalink to this definition">¶</a></dt>
<dd><p>Output methyl array external validation.  [default: ./subset/methyl_array.pkl]</p>
</dd></dl>

</div>
<div class="section" id="pymethyl-utils-train-test-val-split">
<h3>train_test_val_split<a class="headerlink" href="#pymethyl-utils-train-test-val-split" title="Permalink to this headline">¶</a></h3>
<p>Split methylation array into train, test, val.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-utils train_test_val_split <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-utils-train-test-val-split-i">
<code class="descname">-i</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--input_pkl</code><code class="descclassname"> &lt;input_pkl&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-train-test-val-split-i" title="Permalink to this definition">¶</a></dt>
<dd><p>Input database for beta and phenotype data.  [default: ./final_preprocessed/methyl_array.pkl]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-train-test-val-split-o">
<code class="descname">-o</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--output_dir</code><code class="descclassname"> &lt;output_dir&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-train-test-val-split-o" title="Permalink to this definition">¶</a></dt>
<dd><p>Output directory for training, testing, and validation sets.  [default: ./train_val_test_sets/]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-train-test-val-split-tp">
<code class="descname">-tp</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--train_percent</code><code class="descclassname"> &lt;train_percent&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-train-test-val-split-tp" title="Permalink to this definition">¶</a></dt>
<dd><p>Percent data training on.  [default: 0.8]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-train-test-val-split-vp">
<code class="descname">-vp</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--val_percent</code><code class="descclassname"> &lt;val_percent&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-train-test-val-split-vp" title="Permalink to this definition">¶</a></dt>
<dd><p>Percent of training data that comprises validation set.  [default: 0.1]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-train-test-val-split-cat">
<code class="descname">-cat</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--categorical</code><code class="descclassname"></code><a class="headerlink" href="#cmdoption-pymethyl-utils-train-test-val-split-cat" title="Permalink to this definition">¶</a></dt>
<dd><p>Multi-class prediction.  [default: False]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-train-test-val-split-do">
<code class="descname">-do</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--disease_only</code><code class="descclassname"></code><a class="headerlink" href="#cmdoption-pymethyl-utils-train-test-val-split-do" title="Permalink to this definition">¶</a></dt>
<dd><p>Only look at disease, or text before subtype_delimiter.</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-train-test-val-split-k">
<code class="descname">-k</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--key</code><code class="descclassname"> &lt;key&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-train-test-val-split-k" title="Permalink to this definition">¶</a></dt>
<dd><p>Key to split on.  [default: disease]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-train-test-val-split-sd">
<code class="descname">-sd</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--subtype_delimiter</code><code class="descclassname"> &lt;subtype_delimiter&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-train-test-val-split-sd" title="Permalink to this definition">¶</a></dt>
<dd><p>Delimiter for disease extraction.  [default: ,]</p>
</dd></dl>

</div>
<div class="section" id="pymethyl-utils-write-cpgs">
<h3>write_cpgs<a class="headerlink" href="#pymethyl-utils-write-cpgs" title="Permalink to this headline">¶</a></h3>
<p>Write CpGs in methylation array to file.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>pymethyl-utils write_cpgs <span class="o">[</span>OPTIONS<span class="o">]</span>
</pre></div>
</div>
<p class="rubric">Options</p>
<dl class="option">
<dt id="cmdoption-pymethyl-utils-write-cpgs-i">
<code class="descname">-i</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--input_pkl</code><code class="descclassname"> &lt;input_pkl&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-write-cpgs-i" title="Permalink to this definition">¶</a></dt>
<dd><p>Input methyl array.  [default: ./final_preprocessed/methyl_array.pkl]</p>
</dd></dl>

<dl class="option">
<dt id="cmdoption-pymethyl-utils-write-cpgs-c">
<code class="descname">-c</code><code class="descclassname"></code><code class="descclassname">, </code><code class="descname">--cpg_pkl</code><code class="descclassname"> &lt;cpg_pkl&gt;</code><a class="headerlink" href="#cmdoption-pymethyl-utils-write-cpgs-c" title="Permalink to this definition">¶</a></dt>
<dd><p>Pickled numpy array for subsetting.  [default: ./subset_cpgs.pkl]</p>
</dd></dl>

</div>
</div>
</div>
<div class="section" id="indices-and-tables">
<h1>Indices and tables<a class="headerlink" href="#indices-and-tables" title="Permalink to this headline">¶</a></h1>
<ul class="simple">
<li><a class="reference internal" href="genindex.html"><span class="std std-ref">Index</span></a></li>
<li><a class="reference internal" href="py-modindex.html"><span class="std std-ref">Module Index</span></a></li>
<li><a class="reference internal" href="search.html"><span class="std std-ref">Search Page</span></a></li>
</ul>
</div>


           </div>
           
          </div>
          <footer>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2019, Joshua Levy

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>